# MG US HLA imputation using CookHLA

**Start date:** 04-30-2021

**End date:** 04-30-2021

**Analysed by:** Ruth Chia

**Working directory:** `/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA`

___


<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#What-I-need-to-do-here:" data-toc-modified-id="What-I-need-to-do-here:-1">What I need to do here:</a></span></li><li><span><a href="#Code-to-use-for-HLA-imputation" data-toc-modified-id="Code-to-use-for-HLA-imputation-2">Code to use for HLA imputation</a></span></li><li><span><a href="#Download-code" data-toc-modified-id="Download-code-3">Download code</a></span><ul class="toc-item"><li><span><a href="#Download-additional-dependencies-as-instructed-on-github" data-toc-modified-id="Download-additional-dependencies-as-instructed-on-github-3.1">Download additional dependencies as instructed on github</a></span></li></ul></li><li><span><a href="#Generate-chr6-plink-file-for-HLA-imputation" data-toc-modified-id="Generate-chr6-plink-file-for-HLA-imputation-4">Generate chr6 plink file for HLA imputation</a></span></li><li><span><a href="#Generate-chr6-plink-file-for-HLA-imputation-(using-hg38-positions-of-genotyped-data)" data-toc-modified-id="Generate-chr6-plink-file-for-HLA-imputation-(using-hg38-positions-of-genotyped-data)-5">Generate chr6 plink file for HLA imputation (using hg38 positions of genotyped data)</a></span><ul class="toc-item"><li><span><a href="#Liftover-from-b37-to-b38" data-toc-modified-id="Liftover-from-b37-to-b38-5.1">Liftover from b37 to b38</a></span></li></ul></li><li><span><a href="#Run-imputation" data-toc-modified-id="Run-imputation-6">Run imputation</a></span></li><li><span><a href="#Download-code-for-HATK" data-toc-modified-id="Download-code-for-HATK-7">Download code for HATK</a></span></li><li><span><a href="#Run-HATK" data-toc-modified-id="Run-HATK-8">Run HATK</a></span><ul class="toc-item"><li><span><a href="#Imputation.1000G" data-toc-modified-id="Imputation.1000G-8.1">Imputation.1000G</a></span><ul class="toc-item"><li><span><a href="#Overall" data-toc-modified-id="Overall-8.1.1">Overall</a></span><ul class="toc-item"><li><span><a href="#plink-version-association---4field" data-toc-modified-id="plink-version-association---4field-8.1.1.1">plink version association - 4field</a></span></li><li><span><a href="#plink-version-conditional-association---4field" data-toc-modified-id="plink-version-conditional-association---4field-8.1.1.2">plink version conditional association - 4field</a></span></li></ul></li><li><span><a href="#EOMG" data-toc-modified-id="EOMG-8.1.2">EOMG</a></span><ul class="toc-item"><li><span><a href="#plink-version-association---4field" data-toc-modified-id="plink-version-association---4field-8.1.2.1">plink version association - 4field</a></span></li></ul></li><li><span><a href="#LOMG" data-toc-modified-id="LOMG-8.1.3">LOMG</a></span><ul class="toc-item"><li><span><a href="#plink-version-association---4field" data-toc-modified-id="plink-version-association---4field-8.1.3.1">plink version association - 4field</a></span></li><li><span><a href="#plink-version-conditional-association---4field" data-toc-modified-id="plink-version-conditional-association---4field-8.1.3.2">plink version conditional association - 4field</a></span></li></ul></li></ul></li></ul></li></ul></div>

## What I need to do here:
1. Plink1 files as input only for chr6 (can include related samples but no duplicates). To subset from:
    `/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/CLEAN.rawGenotype.keepRelated/FILTERED.US_mg_noDups.keepRelated.hwe1e-10.geno005`
    Need to remove 446 dbGAP samples: `/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/SamplesToRemove.dbGAP.FIDspaceIID.txt`
    
## Code to use for HLA imputation

Will use the latest improved tool to do this i.e. [CookHLA](https://github.com/WansonChoi/CookHLA)

Snippet from the website:

```
CookHLA is an accurate and efficient HLA imputation method.

Human leukocyte antigen (HLA) genes in the major histocompatibility complex (MHC) region influence many immune-related disease phenotypes. Discovering which HLA alleles or amino acids drive phenotypes often requires fine-mapping analysis in tens of thousands of samples, of which HLA typing can be prohibitively costly. Fortunately, the recent development of imputation methods enabled the prediction of HLA alleles using intergenic SNP data, allowing many studies to fine-map HLA for immune phenotypes. Nonetheless, existing imputation methods still have limitations in accuracy.

CookHLA substantially improved imputation accuracy over previous methods, including the predecessor SNP2HLA by implementing several changes. First, to increase accuracy and efficiency, we employed the recently released hidden Markov model (Beagle v4 and v5). Second, to further improve accuracy, we developed a procedure that can account for local variability in the exons of HLA genes. Highly polymorphic exons (exon 2/3/4 in Class I and 2/3 in Class II) are critical regions whose sequences can determine most of the 4-digit alleles. In our method, we repeat imputation by putting the marker set locally in each of the highly polymorphic exons, and use consensus posterior probabilities from the repeated analyses for final predictions(Figure B). Third, to increase accuracy even further, we adaptively learn the genetic map of MHC from the data. This map information allows us to account for the data-specific LD structure within MHC, which improves the imputation accuracy compared to the use of publicly available genetic maps(Figure C).
```
![image](https://github.com/WansonChoi/CookHLA/blob/master/img/Figure1.png?raw=true)


## Download code


```python
!git clone https://github.com/WansonChoi/CookHLA.git
```

    Cloning into 'CookHLA'...
    remote: Enumerating objects: 1369, done.[K
    remote: Counting objects: 100% (112/112), done.[K
    remote: Compressing objects: 100% (78/78), done.[K
    remote: Total 1369 (delta 65), reused 71 (delta 34), pack-reused 1257[K
    Receiving objects: 100% (1369/1369), 28.77 MiB | 20.77 MiB/s, done.
    Resolving deltas: 100% (869/869), done.
    Checking out files: 100% (113/113), done.


### Download additional dependencies as instructed on github


```python
# Run this in a sinteractive terminal on biowulf in the CookHLA directory
cd CookHLA/
source /data/chiarp/conda/etc/profile.d/conda.sh
conda activate base
conda env create -f CookHLA_LINUX.yml
conda activate CookHLA
```

## Generate chr6 plink file for HLA imputation

keep file small to contain only: chr6:25-38Mb (this is larger than the HLA/MHC region so that there is some buffer on either side).

The code does something weird at some point where it assumes that the FID is the same as IID. This is not the case. So, to keep things consistent with what the code expects, I will update the FID and IID so that the FID will be the same as IID


```bash
%%bash
module load plink/1.9.0-beta4.4

# create sample update name
awk '{print $1,$2,$2,$2}' /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/CLEAN.rawGenotype.keepRelated/FILTERED.US_mg_noDups.keepRelated.hwe1e-10.geno005.fam > SampleNameUpdate.txt

plink \
--bfile /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/CLEAN.rawGenotype.keepRelated/FILTERED.US_mg_noDups.keepRelated.hwe1e-10.geno005 \
--remove /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/SamplesToRemove.dbGAP.FIDspaceIID.txt \
--allow-no-sex \
--chr 6 \
--from-bp 25000000 \
--to-bp 38000000 \
--keep-allele-order \
--make-bed \
--out temp

plink \
--bfile temp \
--update-ids SampleNameUpdate.txt \
--allow-no-sex \
--keep-allele-order \
--make-bed \
--out USmerged.HLAregion.chr6

rm temp.*
```

    PLINK v1.90b4.4 64-bit (21 May 2017)           www.cog-genomics.org/plink/1.9/
    (C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3
    Logging to temp.log.
    Options in effect:
      --allow-no-sex
      --bfile /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/CLEAN.rawGenotype.keepRelated/FILTERED.US_mg_noDups.keepRelated.hwe1e-10.geno005
      --chr 6
      --from-bp 25000000
      --keep-allele-order
      --make-bed
      --out temp
      --remove /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/SamplesToRemove.dbGAP.FIDspaceIID.txt
      --to-bp 38000000
    
    257652 MB RAM detected; reserving 128826 MB for main workspace.
    5674 out of 520854 variants loaded from .bim file.
    40733 people (13086 males, 27647 females) loaded from .fam.
    40733 phenotype values loaded from .fam.
    --remove: 40269 people remaining.
    Using 1 thread (no multithreaded calculations invoked).
    Before main variant filters, 40269 founders and 0 nonfounders present.
    Calculating allele frequencies... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99% done.
    Total genotyping rate in remaining samples is 0.998533.
    5674 variants and 40269 people pass filters and QC.
    Among remaining phenotypes, 968 are cases and 39301 are controls.
    --make-bed to temp.bed + temp.bim + temp.fam ... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99%done.
    PLINK v1.90b4.4 64-bit (21 May 2017)           www.cog-genomics.org/plink/1.9/
    (C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3
    Logging to USmerged.HLAregion.chr6.log.
    Options in effect:
      --allow-no-sex
      --bfile temp
      --keep-allele-order
      --make-bed
      --out USmerged.HLAregion.chr6
      --update-ids SampleNameUpdate.txt
    
    257652 MB RAM detected; reserving 128826 MB for main workspace.
    5674 variants loaded from .bim file.
    40269 people (12919 males, 27350 females) loaded from .fam.
    40269 phenotype values loaded from .fam.
    --update-ids: 40269 people updated, 464 IDs not present.
    Using 1 thread (no multithreaded calculations invoked).
    Before main variant filters, 40269 founders and 0 nonfounders present.
    Calculating allele frequencies... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99% done.
    Total genotyping rate is 0.998533.
    5674 variants and 40269 people pass filters and QC.
    Among remaining phenotypes, 968 are cases and 39301 are controls.
    --make-bed to USmerged.HLAregion.chr6.bed + USmerged.HLAregion.chr6.bim +
    USmerged.HLAregion.chr6.fam ... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99%done.


    [-] Unloading plink  1.9.0-beta4.4  on cn3174 
    [+] Loading plink  1.9.0-beta4.4  on cn3174 


## Generate chr6 plink file for HLA imputation (using hg38 positions of genotyped data)

### Liftover from b37 to b38



```bash
%%bash
## create input file for liftover of SNPs from b36 to b37
module load R/4.0
R --vanilla --no-save

require("data.table")
bim <- fread("USmerged.HLAregion.chr6.bim", header=F)
bed <- bim[,c(1,4,2)]
bed$V3 <- bed$V4 + 1
bed <- bed[,c(1,2,4,3)]
bed$V1 <- paste("chr",bed$V1,sep="")
options(scipen=999, justified="none")
write.table(bed,"USmerged.HLAregion.b37.bed", col.names=F,row.names=F,quote=F,sep="\t")

```

    
    R version 4.0.5 (2021-03-31) -- "Shake and Throw"
    Copyright (C) 2021 The R Foundation for Statistical Computing
    Platform: x86_64-pc-linux-gnu (64-bit)
    
    R is free software and comes with ABSOLUTELY NO WARRANTY.
    You are welcome to redistribute it under certain conditions.
    Type 'license()' or 'licence()' for distribution details.
    
    R is a collaborative project with many contributors.
    Type 'contributors()' for more information and
    'citation()' on how to cite R or R packages in publications.
    
    Type 'demo()' for some demos, 'help()' for on-line help, or
    'help.start()' for an HTML browser interface to help.
    Type 'q()' to quit R.
    
    > 
    > require("data.table")
    > bim <- fread("USmerged.HLAregion.chr6.bim", header=F)
    > bed <- bim[,c(1,4,2)]
    > bed$V3 <- bed$V4 + 1
    > bed <- bed[,c(1,2,4,3)]
    > bed$V1 <- paste("chr",bed$V1,sep="")
    > options(scipen=999, justified="none")
    > write.table(bed,"USmerged.HLAregion.b37.bed", col.names=F,row.names=F,quote=F,sep="\t")
    > 


    [-] Unloading gcc  9.2.0  ... 
    [-] Unloading GSL 2.6 for GCC 9.2.0 ... 
    [-] Unloading openmpi 4.0.5  for GCC 9.2.0 
    [-] Unloading ImageMagick  7.0.8  on cn3174 
    [-] Unloading HDF5  1.10.4 
    [-] Unloading NetCDF 4.7.4_gcc9.2.0 
    [-] Unloading pandoc  2.13  on cn3174 
    [-] Unloading pcre2 10.21  ... 
    [-] Unloading R 4.0.5 
    [+] Loading gcc  9.2.0  ... 
    [+] Loading GSL 2.6 for GCC 9.2.0 ... 
    [-] Unloading gcc  9.2.0  ... 
    [+] Loading gcc  9.2.0  ... 
    [+] Loading openmpi 4.0.5  for GCC 9.2.0 
    [+] Loading ImageMagick  7.0.8  on cn3174 
    [+] Loading HDF5  1.10.4 
    [-] Unloading gcc  9.2.0  ... 
    [+] Loading gcc  9.2.0  ... 
    [+] Loading NetCDF 4.7.4_gcc9.2.0 
    [+] Loading pandoc  2.13  on cn3174 
    [+] Loading pcre2 10.21  ... 
    [+] Loading R 4.0.5 
    Loading required package: data.table



```bash
%%bash

## Download binary tools for liftover
mkdir LiftOver
wget http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/liftOver -O LiftOver/liftOver
wget --timestamping 'ftp://hgdownload.soe.ucsc.edu/goldenPath/hg19/liftOver/hg19ToHg38.over.chain.gz' -O LiftOver/hg19ToHg38.over.chain.gz

chmod u+x LiftOver/liftOver
LiftOver/liftOver USmerged.HLAregion.b37.bed LiftOver/hg19ToHg38.over.chain.gz USmerged.HLAregion.b38.bed unlifted.bed

grep -v "#Deleted" unlifted.bed | awk '{print $4}' > SNPs_not_in_b38.txt
awk '{print $4,$2}' USmerged.HLAregion.b38.bed > USmerged.HLAregion_update-map_to_b38.txt

# Update snp position
module load plink/1.9.0-beta4.4
plink \
--bfile USmerged.HLAregion.chr6 \
--update-map USmerged.HLAregion_update-map_to_b38.txt \
--exclude SNPs_not_in_b38.txt \
--make-bed \
--out USmerged.HLAregion.chr6.Liftover.hg38

```

    PLINK v1.90b4.4 64-bit (21 May 2017)           www.cog-genomics.org/plink/1.9/
    (C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3
    Logging to USmerged.HLAregion.chr6.Liftover.hg38.log.
    Options in effect:
      --bfile USmerged.HLAregion.chr6
      --exclude SNPs_not_in_b38.txt
      --make-bed
      --out USmerged.HLAregion.chr6.Liftover.hg38
      --update-map USmerged.HLAregion_update-map_to_b38.txt
    
    257652 MB RAM detected; reserving 128826 MB for main workspace.
    5674 variants loaded from .bim file.
    40269 people (12919 males, 27350 females) loaded from .fam.
    40269 phenotype values loaded from .fam.
    --update-map: 5674 values updated.
    --exclude: 5674 variants remaining.
    Using 1 thread (no multithreaded calculations invoked).
    Before main variant filters, 40269 founders and 0 nonfounders present.
    Calculating allele frequencies... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99% done.
    Total genotyping rate is 0.998533.
    5674 variants and 40269 people pass filters and QC.
    Among remaining phenotypes, 968 are cases and 39301 are controls.
    --make-bed to USmerged.HLAregion.chr6.Liftover.hg38.bed +
    USmerged.HLAregion.chr6.Liftover.hg38.bim +
    USmerged.HLAregion.chr6.Liftover.hg38.fam ... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99%done.


    --2021-05-03 23:30:56--  http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/liftOver
    Resolving dtn05-e0 (dtn05-e0)... 10.1.200.241
    Connecting to dtn05-e0 (dtn05-e0)|10.1.200.241|:3128... connected.
    Proxy request sent, awaiting response... 200 OK
    Length: 37475560 (36M)
    Saving to: 'LiftOver/liftOver'
    
         0K .......... .......... .......... .......... ..........  0%  322K 1m54s
        50K .......... .......... .......... .......... ..........  0%  734K 82s
       100K .......... .......... .......... .......... ..........  0% 58.1M 55s
       150K .......... .......... .......... .......... ..........  0% 49.0M 41s
       200K .......... .......... .......... .......... ..........  0%  760K 42s
       250K .......... .......... .......... .......... ..........  0% 59.6M 35s
       300K .......... .......... .......... .......... ..........  0%  752K 37s
       350K .......... .......... .......... .......... ..........  1% 56.5M 33s
       400K .......... .......... .......... .......... ..........  1% 54.8M 29s
       450K .......... .......... .......... .......... ..........  1% 49.4M 26s
       500K .......... .......... .......... .......... ..........  1% 39.9M 24s
       550K .......... .......... .......... .......... ..........  1% 58.2M 22s
       600K .......... .......... .......... .......... ..........  1%  798K 24s
       650K .......... .......... .......... .......... ..........  1% 56.7M 22s
       700K .......... .......... .......... .......... ..........  2% 51.8M 20s
       750K .......... .......... .......... .......... ..........  2% 59.2M 19s
       800K .......... .......... .......... .......... ..........  2% 42.1M 18s
       850K .......... .......... .......... .......... ..........  2%  782K 20s
       900K .......... .......... .......... .......... ..........  2% 28.9M 19s
       950K .......... .......... .......... .......... ..........  2% 56.0M 18s
      1000K .......... .......... .......... .......... ..........  2% 49.1M 17s
      1050K .......... .......... .......... .......... ..........  3% 80.9M 16s
      1100K .......... .......... .......... .......... ..........  3% 76.6M 15s
      1150K .......... .......... .......... .......... ..........  3% 78.8M 15s
      1200K .......... .......... .......... .......... ..........  3% 72.2M 14s
      1250K .......... .......... .......... .......... ..........  3% 51.5M 14s
      1300K .......... .......... .......... .......... ..........  3% 69.7M 13s
      1350K .......... .......... .......... .......... ..........  3% 44.2M 13s
      1400K .......... .......... .......... .......... ..........  3% 81.1M 12s
      1450K .......... .......... .......... .......... ..........  4%  146M 12s
      1500K .......... .......... .......... .......... ..........  4%  156M 11s
      1550K .......... .......... .......... .......... ..........  4%  179M 11s
      1600K .......... .......... .......... .......... ..........  4%  884K 12s
      1650K .......... .......... .......... .......... ..........  4% 78.1M 12s
      1700K .......... .......... .......... .......... ..........  4% 81.0M 11s
      1750K .......... .......... .......... .......... ..........  4% 92.4M 11s
      1800K .......... .......... .......... .......... ..........  5% 88.0M 11s
      1850K .......... .......... .......... .......... ..........  5% 97.0M 10s
      1900K .......... .......... .......... .......... ..........  5% 55.0M 10s
      1950K .......... .......... .......... .......... ..........  5%  117M 10s
      2000K .......... .......... .......... .......... ..........  5%  137M 10s
      2050K .......... .......... .......... .......... ..........  5%  119M 9s
      2100K .......... .......... .......... .......... ..........  5%  133M 9s
      2150K .......... .......... .......... .......... ..........  6%  140M 9s
      2200K .......... .......... .......... .......... ..........  6%  136M 9s
      2250K .......... .......... .......... .......... ..........  6% 72.4M 8s
      2300K .......... .......... .......... .......... ..........  6%  133M 8s
      2350K .......... .......... .......... .......... ..........  6% 69.2M 8s
      2400K .......... .......... .......... .......... ..........  6%  856K 9s
      2450K .......... .......... .......... .......... ..........  6% 55.2M 9s
      2500K .......... .......... .......... .......... ..........  6%  123M 8s
      2550K .......... .......... .......... .......... ..........  7%  113M 8s
      2600K .......... .......... .......... .......... ..........  7%  128M 8s
      2650K .......... .......... .......... .......... ..........  7%  119M 8s
      2700K .......... .......... .......... .......... ..........  7% 44.0M 8s
      2750K .......... .......... .......... .......... ..........  7% 88.1M 8s
      2800K .......... .......... .......... .......... ..........  7%  113M 7s
      2850K .......... .......... .......... .......... ..........  7% 57.3M 7s
      2900K .......... .......... .......... .......... ..........  8%  190M 7s
      2950K .......... .......... .......... .......... ..........  8%  155M 7s
      3000K .......... .......... .......... .......... ..........  8%  201M 7s
      3050K .......... .......... .......... .......... ..........  8%  212M 7s
      3100K .......... .......... .......... .......... ..........  8%  244M 7s
      3150K .......... .......... .......... .......... ..........  8%  211M 7s
      3200K .......... .......... .......... .......... ..........  8%  185M 7s
      3250K .......... .......... .......... .......... ..........  9%  176M 6s
      3300K .......... .......... .......... .......... ..........  9%  223M 6s
      3350K .......... .......... .......... .......... ..........  9%  185M 6s
      3400K .......... .......... .......... .......... ..........  9%  197M 6s
      3450K .......... .......... .......... .......... ..........  9%  171M 6s
      3500K .......... .......... .......... .......... ..........  9%  203M 6s
      3550K .......... .......... .......... .......... ..........  9%  204M 6s
      3600K .......... .......... .......... .......... ..........  9% 89.7M 6s
      3650K .......... .......... .......... .......... .......... 10%  177M 6s
      3700K .......... .......... .......... .......... .......... 10%  232M 6s
      3750K .......... .......... .......... .......... .......... 10%  216M 5s
      3800K .......... .......... .......... .......... .......... 10%  223M 5s
      3850K .......... .......... .......... .......... .......... 10%  237M 5s
      3900K .......... .......... .......... .......... .......... 10%  197M 5s
      3950K .......... .......... .......... .......... .......... 10%  150M 5s
      4000K .......... .......... .......... .......... .......... 11%  223M 5s
      4050K .......... .......... .......... .......... .......... 11%  109M 5s
      4100K .......... .......... .......... .......... .......... 11%  103M 5s
      4150K .......... .......... .......... .......... .......... 11% 76.7M 5s
      4200K .......... .......... .......... .......... .......... 11% 75.5M 5s
      4250K .......... .......... .......... .......... .......... 11%  927K 5s
      4300K .......... .......... .......... .......... .......... 11%  177M 5s
      4350K .......... .......... .......... .......... .......... 12%  132M 5s
      4400K .......... .......... .......... .......... .......... 12%  106M 5s
      4450K .......... .......... .......... .......... .......... 12%  136M 5s
      4500K .......... .......... .......... .......... .......... 12%  134M 5s
      4550K .......... .......... .......... .......... .......... 12%  132M 5s
      4600K .......... .......... .......... .......... .......... 12%  143M 5s
      4650K .......... .......... .......... .......... .......... 12%  138M 5s
      4700K .......... .......... .......... .......... .......... 12%  131M 5s
      4750K .......... .......... .......... .......... .......... 13% 69.4M 5s
      4800K .......... .......... .......... .......... .......... 13%  126M 5s
      4850K .......... .......... .......... .......... .......... 13%  126M 5s
      4900K .......... .......... .......... .......... .......... 13% 99.7M 4s
      4950K .......... .......... .......... .......... .......... 13%  136M 4s
      5000K .......... .......... .......... .......... .......... 13%  130M 4s
      5050K .......... .......... .......... .......... .......... 13%  147M 4s
      5100K .......... .......... .......... .......... .......... 14%  128M 4s
      5150K .......... .......... .......... .......... .......... 14% 89.5M 4s
      5200K .......... .......... .......... .......... .......... 14%  215M 4s
      5250K .......... .......... .......... .......... .......... 14%  154M 4s
      5300K .......... .......... .......... .......... .......... 14%  138M 4s
      5350K .......... .......... .......... .......... .......... 14% 66.2M 4s
      5400K .......... .......... .......... .......... .......... 14% 45.9M 4s
      5450K .......... .......... .......... .......... .......... 15% 99.0M 4s
      5500K .......... .......... .......... .......... .......... 15% 90.6M 4s
      5550K .......... .......... .......... .......... .......... 15%  100M 4s
      5600K .......... .......... .......... .......... .......... 15% 89.8M 4s
      5650K .......... .......... .......... .......... .......... 15%  100M 4s
      5700K .......... .......... .......... .......... .......... 15% 93.8M 4s
      5750K .......... .......... .......... .......... .......... 15% 99.4M 4s
      5800K .......... .......... .......... .......... .......... 15% 93.0M 4s
      5850K .......... .......... .......... .......... .......... 16% 97.3M 4s
      5900K .......... .......... .......... .......... .......... 16% 81.1M 4s
      5950K .......... .......... .......... .......... .......... 16%  101M 4s
      6000K .......... .......... .......... .......... .......... 16% 91.5M 4s
      6050K .......... .......... .......... .......... .......... 16% 97.8M 4s
      6100K .......... .......... .......... .......... .......... 16% 92.6M 4s
      6150K .......... .......... .......... .......... .......... 16% 49.1M 3s
      6200K .......... .......... .......... .......... .......... 17%  152M 3s
      6250K .......... .......... .......... .......... .......... 17% 1.00M 4s
      6300K .......... .......... .......... .......... .......... 17% 70.1M 4s
      6350K .......... .......... .......... .......... .......... 17% 80.4M 4s
      6400K .......... .......... .......... .......... .......... 17% 66.4M 4s
      6450K .......... .......... .......... .......... .......... 17% 83.4M 4s
      6500K .......... .......... .......... .......... .......... 17% 79.5M 4s
      6550K .......... .......... .......... .......... .......... 18% 82.2M 3s
      6600K .......... .......... .......... .......... .......... 18% 77.5M 3s
      6650K .......... .......... .......... .......... .......... 18% 85.6M 3s
      6700K .......... .......... .......... .......... .......... 18% 73.6M 3s
      6750K .......... .......... .......... .......... .......... 18% 53.9M 3s
      6800K .......... .......... .......... .......... .......... 18% 80.6M 3s
      6850K .......... .......... .......... .......... .......... 18% 86.4M 3s
      6900K .......... .......... .......... .......... .......... 18% 70.9M 3s
      6950K .......... .......... .......... .......... .......... 19% 91.6M 3s
      7000K .......... .......... .......... .......... .......... 19% 52.4M 3s
      7050K .......... .......... .......... .......... .......... 19% 62.7M 3s
      7100K .......... .......... .......... .......... .......... 19%  142M 3s
      7150K .......... .......... .......... .......... .......... 19% 80.4M 3s
      7200K .......... .......... .......... .......... .......... 19%  154M 3s
      7250K .......... .......... .......... .......... .......... 19% 77.7M 3s
      7300K .......... .......... .......... .......... .......... 20%  149M 3s
      7350K .......... .......... .......... .......... .......... 20%  163M 3s
      7400K .......... .......... .......... .......... .......... 20%  138M 3s
      7450K .......... .......... .......... .......... .......... 20%  168M 3s
      7500K .......... .......... .......... .......... .......... 20%  159M 3s
      7550K .......... .......... .......... .......... .......... 20%  171M 3s
      7600K .......... .......... .......... .......... .......... 20%  158M 3s
      7650K .......... .......... .......... .......... .......... 21%  151M 3s
      7700K .......... .......... .......... .......... .......... 21% 77.8M 3s
      7750K .......... .......... .......... .......... .......... 21%  159M 3s
      7800K .......... .......... .......... .......... .......... 21%  138M 3s
      7850K .......... .......... .......... .......... .......... 21%  157M 3s
      7900K .......... .......... .......... .......... .......... 21%  125M 3s
      7950K .......... .......... .......... .......... .......... 21%  168M 3s
      8000K .......... .......... .......... .......... .......... 21%  155M 3s
      8050K .......... .......... .......... .......... .......... 22%  171M 3s
      8100K .......... .......... .......... .......... .......... 22% 78.4M 3s
      8150K .......... .......... .......... .......... .......... 22% 53.5M 3s
      8200K .......... .......... .......... .......... .......... 22%  207M 3s
      8250K .......... .......... .......... .......... .......... 22% 51.9M 3s
      8300K .......... .......... .......... .......... .......... 22% 69.0M 3s
      8350K .......... .......... .......... .......... .......... 22% 68.7M 3s
      8400K .......... .......... .......... .......... .......... 23% 67.9M 3s
      8450K .......... .......... .......... .......... .......... 23% 84.8M 3s
      8500K .......... .......... .......... .......... .......... 23% 66.6M 3s
      8550K .......... .......... .......... .......... .......... 23% 57.3M 3s
      8600K .......... .......... .......... .......... .......... 23% 78.1M 3s
      8650K .......... .......... .......... .......... .......... 23% 51.8M 3s
      8700K .......... .......... .......... .......... .......... 23% 72.8M 3s
      8750K .......... .......... .......... .......... .......... 24% 89.4M 2s
      8800K .......... .......... .......... .......... .......... 24% 74.6M 2s
      8850K .......... .......... .......... .......... .......... 24% 85.1M 2s
      8900K .......... .......... .......... .......... .......... 24% 57.3M 2s
      8950K .......... .......... .......... .......... .......... 24% 57.4M 2s
      9000K .......... .......... .......... .......... .......... 24% 65.7M 2s
      9050K .......... .......... .......... .......... .......... 24% 69.8M 2s
      9100K .......... .......... .......... .......... .......... 25% 60.5M 2s
      9150K .......... .......... .......... .......... .......... 25% 1.20M 2s
      9200K .......... .......... .......... .......... .......... 25% 99.1M 2s
      9250K .......... .......... .......... .......... .......... 25%  109M 2s
      9300K .......... .......... .......... .......... .......... 25% 2.64M 2s
      9350K .......... .......... .......... .......... .......... 25%  110M 2s
      9400K .......... .......... .......... .......... .......... 25%  110M 2s
      9450K .......... .......... .......... .......... .......... 25%  122M 2s
      9500K .......... .......... .......... .......... .......... 26%  144M 2s
      9550K .......... .......... .......... .......... .......... 26%  136M 2s
      9600K .......... .......... .......... .......... .......... 26%  150M 2s
      9650K .......... .......... .......... .......... .......... 26%  138M 2s
      9700K .......... .......... .......... .......... .......... 26%  145M 2s
      9750K .......... .......... .......... .......... .......... 26% 60.3M 2s
      9800K .......... .......... .......... .......... .......... 26%  103M 2s
      9850K .......... .......... .......... .......... .......... 27% 77.5M 2s
      9900K .......... .......... .......... .......... .......... 27%  105M 2s
      9950K .......... .......... .......... .......... .......... 27%  107M 2s
     10000K .......... .......... .......... .......... .......... 27% 60.3M 2s
     10050K .......... .......... .......... .......... .......... 27% 58.8M 2s
     10100K .......... .......... .......... .......... .......... 27%  192M 2s
     10150K .......... .......... .......... .......... .......... 27%  200M 2s
     10200K .......... .......... .......... .......... .......... 28%  195M 2s
     10250K .......... .......... .......... .......... .......... 28%  178M 2s
     10300K .......... .......... .......... .......... .......... 28% 91.7M 2s
     10350K .......... .......... .......... .......... .......... 28% 82.2M 2s
     10400K .......... .......... .......... .......... .......... 28%  197M 2s
     10450K .......... .......... .......... .......... .......... 28%  199M 2s
     10500K .......... .......... .......... .......... .......... 28%  234M 2s
     10550K .......... .......... .......... .......... .......... 28%  195M 2s
     10600K .......... .......... .......... .......... .......... 29%  208M 2s
     10650K .......... .......... .......... .......... .......... 29%  191M 2s
     10700K .......... .......... .......... .......... .......... 29%  225M 2s
     10750K .......... .......... .......... .......... .......... 29%  247M 2s
     10800K .......... .......... .......... .......... .......... 29%  231M 2s
     10850K .......... .......... .......... .......... .......... 29%  172M 2s
     10900K .......... .......... .......... .......... .......... 29% 91.2M 2s
     10950K .......... .......... .......... .......... .......... 30% 1.47M 2s
     11000K .......... .......... .......... .......... .......... 30% 19.6M 2s
     11050K .......... .......... .......... .......... .......... 30%  131M 2s
     11100K .......... .......... .......... .......... .......... 30%  118M 2s
     11150K .......... .......... .......... .......... .......... 30%  131M 2s
     11200K .......... .......... .......... .......... .......... 30%  113M 2s
     11250K .......... .......... .......... .......... .......... 30%  125M 2s
     11300K .......... .......... .......... .......... .......... 31%  118M 2s
     11350K .......... .......... .......... .......... .......... 31%  111M 2s
     11400K .......... .......... .......... .......... .......... 31%  118M 2s
     11450K .......... .......... .......... .......... .......... 31%  132M 2s
     11500K .......... .......... .......... .......... .......... 31%  118M 2s
     11550K .......... .......... .......... .......... .......... 31%  127M 2s
     11600K .......... .......... .......... .......... .......... 31%  118M 2s
     11650K .......... .......... .......... .......... .......... 31%  127M 2s
     11700K .......... .......... .......... .......... .......... 32%  116M 2s
     11750K .......... .......... .......... .......... .......... 32%  127M 2s
     11800K .......... .......... .......... .......... .......... 32%  115M 2s
     11850K .......... .......... .......... .......... .......... 32%  127M 2s
     11900K .......... .......... .......... .......... .......... 32%  118M 2s
     11950K .......... .......... .......... .......... .......... 32%  131M 2s
     12000K .......... .......... .......... .......... .......... 32%  114M 2s
     12050K .......... .......... .......... .......... .......... 33%  128M 2s
     12100K .......... .......... .......... .......... .......... 33%  120M 2s
     12150K .......... .......... .......... .......... .......... 33% 70.0M 2s
     12200K .......... .......... .......... .......... .......... 33%  116M 2s
     12250K .......... .......... .......... .......... .......... 33% 83.2M 2s
     12300K .......... .......... .......... .......... .......... 33%  257M 2s
     12350K .......... .......... .......... .......... .......... 33% 74.8M 2s
     12400K .......... .......... .......... .......... .......... 34% 6.73M 2s
     12450K .......... .......... .......... .......... .......... 34%  177M 2s
     12500K .......... .......... .......... .......... .......... 34% 38.4M 2s
     12550K .......... .......... .......... .......... .......... 34%  124M 2s
     12600K .......... .......... .......... .......... .......... 34%  111M 2s
     12650K .......... .......... .......... .......... .......... 34%  114M 2s
     12700K .......... .......... .......... .......... .......... 34%  104M 2s
     12750K .......... .......... .......... .......... .......... 34%  119M 2s
     12800K .......... .......... .......... .......... .......... 35%  112M 2s
     12850K .......... .......... .......... .......... .......... 35% 98.4M 2s
     12900K .......... .......... .......... .......... .......... 35%  128M 2s
     12950K .......... .......... .......... .......... .......... 35%  136M 2s
     13000K .......... .......... .......... .......... .......... 35%  129M 2s
     13050K .......... .......... .......... .......... .......... 35%  146M 2s
     13100K .......... .......... .......... .......... .......... 35%  122M 2s
     13150K .......... .......... .......... .......... .......... 36%  136M 2s
     13200K .......... .......... .......... .......... .......... 36%  125M 2s
     13250K .......... .......... .......... .......... .......... 36%  134M 2s
     13300K .......... .......... .......... .......... .......... 36%  125M 2s
     13350K .......... .......... .......... .......... .......... 36%  113M 2s
     13400K .......... .......... .......... .......... .......... 36%  136M 2s
     13450K .......... .......... .......... .......... .......... 36%  129M 2s
     13500K .......... .......... .......... .......... .......... 37%  124M 2s
     13550K .......... .......... .......... .......... .......... 37%  143M 2s
     13600K .......... .......... .......... .......... .......... 37%  133M 2s
     13650K .......... .......... .......... .......... .......... 37% 56.1M 2s
     13700K .......... .......... .......... .......... .......... 37%  201M 2s
     13750K .......... .......... .......... .......... .......... 37%  210M 2s
     13800K .......... .......... .......... .......... .......... 37%  239M 2s
     13850K .......... .......... .......... .......... .......... 37%  196M 2s
     13900K .......... .......... .......... .......... .......... 38%  182M 2s
     13950K .......... .......... .......... .......... .......... 38%  200M 2s
     14000K .......... .......... .......... .......... .......... 38%  193M 1s
     14050K .......... .......... .......... .......... .......... 38%  199M 1s
     14100K .......... .......... .......... .......... .......... 38%  185M 1s
     14150K .......... .......... .......... .......... .......... 38%  201M 1s
     14200K .......... .......... .......... .......... .......... 38%  203M 1s
     14250K .......... .......... .......... .......... .......... 39%  264M 1s
     14300K .......... .......... .......... .......... .......... 39%  148M 1s
     14350K .......... .......... .......... .......... .......... 39%  203M 1s
     14400K .......... .......... .......... .......... .......... 39%  190M 1s
     14450K .......... .......... .......... .......... .......... 39%  210M 1s
     14500K .......... .......... .......... .......... .......... 39%  210M 1s
     14550K .......... .......... .......... .......... .......... 39%  240M 1s
     14600K .......... .......... .......... .......... .......... 40%  203M 1s
     14650K .......... .......... .......... .......... .......... 40%  210M 1s
     14700K .......... .......... .......... .......... .......... 40%  188M 1s
     14750K .......... .......... .......... .......... .......... 40% 1.37M 1s
     14800K .......... .......... .......... .......... .......... 40%  124M 1s
     14850K .......... .......... .......... .......... .......... 40%  119M 1s
     14900K .......... .......... .......... .......... .......... 40%  134M 1s
     14950K .......... .......... .......... .......... .......... 40%  109M 1s
     15000K .......... .......... .......... .......... .......... 41%  122M 1s
     15050K .......... .......... .......... .......... .......... 41%  111M 1s
     15100K .......... .......... .......... .......... .......... 41%  129M 1s
     15150K .......... .......... .......... .......... .......... 41%  121M 1s
     15200K .......... .......... .......... .......... .......... 41% 72.9M 1s
     15250K .......... .......... .......... .......... .......... 41%  122M 1s
     15300K .......... .......... .......... .......... .......... 41%  106M 1s
     15350K .......... .......... .......... .......... .......... 42%  119M 1s
     15400K .......... .......... .......... .......... .......... 42%  128M 1s
     15450K .......... .......... .......... .......... .......... 42%  114M 1s
     15500K .......... .......... .......... .......... .......... 42%  132M 1s
     15550K .......... .......... .......... .......... .......... 42%  112M 1s
     15600K .......... .......... .......... .......... .......... 42%  129M 1s
     15650K .......... .......... .......... .......... .......... 42%  117M 1s
     15700K .......... .......... .......... .......... .......... 43%  130M 1s
     15750K .......... .......... .......... .......... .......... 43%  122M 1s
     15800K .......... .......... .......... .......... .......... 43%  109M 1s
     15850K .......... .......... .......... .......... .......... 43%  118M 1s
     15900K .......... .......... .......... .......... .......... 43% 72.7M 1s
     15950K .......... .......... .......... .......... .......... 43%  119M 1s
     16000K .......... .......... .......... .......... .......... 43%  125M 1s
     16050K .......... .......... .......... .......... .......... 43%  114M 1s
     16100K .......... .......... .......... .......... .......... 44%  126M 1s
     16150K .......... .......... .......... .......... .......... 44%  115M 1s
     16200K .......... .......... .......... .......... .......... 44%  135M 1s
     16250K .......... .......... .......... .......... .......... 44% 66.3M 1s
     16300K .......... .......... .......... .......... .......... 44%  114M 1s
     16350K .......... .......... .......... .......... .......... 44% 60.2M 1s
     16400K .......... .......... .......... .......... .......... 44%  256M 1s
     16450K .......... .......... .......... .......... .......... 45% 4.94M 1s
     16500K .......... .......... .......... .......... .......... 45%  103M 1s
     16550K .......... .......... .......... .......... .......... 45%  103M 1s
     16600K .......... .......... .......... .......... .......... 45% 95.5M 1s
     16650K .......... .......... .......... .......... .......... 45%  107M 1s
     16700K .......... .......... .......... .......... .......... 45% 57.9M 1s
     16750K .......... .......... .......... .......... .......... 45% 62.0M 1s
     16800K .......... .......... .......... .......... .......... 46% 52.6M 1s
     16850K .......... .......... .......... .......... .......... 46%  124M 1s
     16900K .......... .......... .......... .......... .......... 46%  117M 1s
     16950K .......... .......... .......... .......... .......... 46%  123M 1s
     17000K .......... .......... .......... .......... .......... 46% 67.3M 1s
     17050K .......... .......... .......... .......... .......... 46%  130M 1s
     17100K .......... .......... .......... .......... .......... 46% 65.9M 1s
     17150K .......... .......... .......... .......... .......... 46%  130M 1s
     17200K .......... .......... .......... .......... .......... 47%  108M 1s
     17250K .......... .......... .......... .......... .......... 47%  128M 1s
     17300K .......... .......... .......... .......... .......... 47% 98.6M 1s
     17350K .......... .......... .......... .......... .......... 47%  118M 1s
     17400K .......... .......... .......... .......... .......... 47%  114M 1s
     17450K .......... .......... .......... .......... .......... 47%  132M 1s
     17500K .......... .......... .......... .......... .......... 47%  106M 1s
     17550K .......... .......... .......... .......... .......... 48% 74.4M 1s
     17600K .......... .......... .......... .......... .......... 48%  118M 1s
     17650K .......... .......... .......... .......... .......... 48% 70.1M 1s
     17700K .......... .......... .......... .......... .......... 48%  116M 1s
     17750K .......... .......... .......... .......... .......... 48%  117M 1s
     17800K .......... .......... .......... .......... .......... 48% 96.1M 1s
     17850K .......... .......... .......... .......... .......... 48% 54.0M 1s
     17900K .......... .......... .......... .......... .......... 49%  116M 1s
     17950K .......... .......... .......... .......... .......... 49%  130M 1s
     18000K .......... .......... .......... .......... .......... 49%  116M 1s
     18050K .......... .......... .......... .......... .......... 49%  123M 1s
     18100K .......... .......... .......... .......... .......... 49%  119M 1s
     18150K .......... .......... .......... .......... .......... 49%  124M 1s
     18200K .......... .......... .......... .......... .......... 49%  129M 1s
     18250K .......... .......... .......... .......... .......... 50%  120M 1s
     18300K .......... .......... .......... .......... .......... 50%  104M 1s
     18350K .......... .......... .......... .......... .......... 50%  128M 1s
     18400K .......... .......... .......... .......... .......... 50%  113M 1s
     18450K .......... .......... .......... .......... .......... 50%  131M 1s
     18500K .......... .......... .......... .......... .......... 50%  116M 1s
     18550K .......... .......... .......... .......... .......... 50%  126M 1s
     18600K .......... .......... .......... .......... .......... 50%  128M 1s
     18650K .......... .......... .......... .......... .......... 51% 69.6M 1s
     18700K .......... .......... .......... .......... .......... 51%  111M 1s
     18750K .......... .......... .......... .......... .......... 51%  128M 1s
     18800K .......... .......... .......... .......... .......... 51% 2.38M 1s
     18850K .......... .......... .......... .......... .......... 51% 73.0M 1s
     18900K .......... .......... .......... .......... .......... 51% 85.1M 1s
     18950K .......... .......... .......... .......... .......... 51% 81.5M 1s
     19000K .......... .......... .......... .......... .......... 52% 86.6M 1s
     19050K .......... .......... .......... .......... .......... 52% 42.6M 1s
     19100K .......... .......... .......... .......... .......... 52%  132M 1s
     19150K .......... .......... .......... .......... .......... 52%  117M 1s
     19200K .......... .......... .......... .......... .......... 52%  131M 1s
     19250K .......... .......... .......... .......... .......... 52%  118M 1s
     19300K .......... .......... .......... .......... .......... 52%  116M 1s
     19350K .......... .......... .......... .......... .......... 53%  131M 1s
     19400K .......... .......... .......... .......... .......... 53%  140M 1s
     19450K .......... .......... .......... .......... .......... 53%  132M 1s
     19500K .......... .......... .......... .......... .......... 53%  142M 1s
     19550K .......... .......... .......... .......... .......... 53%  129M 1s
     19600K .......... .......... .......... .......... .......... 53%  139M 1s
     19650K .......... .......... .......... .......... .......... 53%  122M 1s
     19700K .......... .......... .......... .......... .......... 53%  139M 1s
     19750K .......... .......... .......... .......... .......... 54%  123M 1s
     19800K .......... .......... .......... .......... .......... 54% 64.3M 1s
     19850K .......... .......... .......... .......... .......... 54%  141M 1s
     19900K .......... .......... .......... .......... .......... 54%  131M 1s
     19950K .......... .......... .......... .......... .......... 54%  128M 1s
     20000K .......... .......... .......... .......... .......... 54%  141M 1s
     20050K .......... .......... .......... .......... .......... 54%  129M 1s
     20100K .......... .......... .......... .......... .......... 55%  141M 1s
     20150K .......... .......... .......... .......... .......... 55% 50.6M 1s
     20200K .......... .......... .......... .......... .......... 55%  107M 1s
     20250K .......... .......... .......... .......... .......... 55%  210M 1s
     20300K .......... .......... .......... .......... .......... 55%  236M 1s
     20350K .......... .......... .......... .......... .......... 55%  241M 1s
     20400K .......... .......... .......... .......... .......... 55%  208M 1s
     20450K .......... .......... .......... .......... .......... 56% 95.5M 1s
     20500K .......... .......... .......... .......... .......... 56%  268M 1s
     20550K .......... .......... .......... .......... .......... 56% 4.86M 1s
     20600K .......... .......... .......... .......... .......... 56% 99.6M 1s
     20650K .......... .......... .......... .......... .......... 56%  116M 1s
     20700K .......... .......... .......... .......... .......... 56%  108M 1s
     20750K .......... .......... .......... .......... .......... 56%  116M 1s
     20800K .......... .......... .......... .......... .......... 56% 90.5M 1s
     20850K .......... .......... .......... .......... .......... 57%  127M 1s
     20900K .......... .......... .......... .......... .......... 57%  114M 1s
     20950K .......... .......... .......... .......... .......... 57% 68.1M 1s
     21000K .......... .......... .......... .......... .......... 57%  112M 1s
     21050K .......... .......... .......... .......... .......... 57%  138M 1s
     21100K .......... .......... .......... .......... .......... 57%  129M 1s
     21150K .......... .......... .......... .......... .......... 57%  119M 1s
     21200K .......... .......... .......... .......... .......... 58%  123M 1s
     21250K .......... .......... .......... .......... .......... 58% 61.6M 1s
     21300K .......... .......... .......... .......... .......... 58%  174M 1s
     21350K .......... .......... .......... .......... .......... 58%  253M 1s
     21400K .......... .......... .......... .......... .......... 58% 83.9M 1s
     21450K .......... .......... .......... .......... .......... 58%  224M 1s
     21500K .......... .......... .......... .......... .......... 58%  177M 1s
     21550K .......... .......... .......... .......... .......... 59%  233M 1s
     21600K .......... .......... .......... .......... .......... 59%  191M 1s
     21650K .......... .......... .......... .......... .......... 59%  195M 1s
     21700K .......... .......... .......... .......... .......... 59%  213M 1s
     21750K .......... .......... .......... .......... .......... 59%  160M 1s
     21800K .......... .......... .......... .......... .......... 59%  197M 1s
     21850K .......... .......... .......... .......... .......... 59%  228M 1s
     21900K .......... .......... .......... .......... .......... 59%  201M 1s
     21950K .......... .......... .......... .......... .......... 60% 94.1M 1s
     22000K .......... .......... .......... .......... .......... 60%  198M 1s
     22050K .......... .......... .......... .......... .......... 60%  196M 1s
     22100K .......... .......... .......... .......... .......... 60%  178M 1s
     22150K .......... .......... .......... .......... .......... 60%  194M 1s
     22200K .......... .......... .......... .......... .......... 60%  194M 1s
     22250K .......... .......... .......... .......... .......... 60%  170M 1s
     22300K .......... .......... .......... .......... .......... 61%  199M 1s
     22350K .......... .......... .......... .......... .......... 61% 90.8M 1s
     22400K .......... .......... .......... .......... .......... 61%  193M 1s
     22450K .......... .......... .......... .......... .......... 61%  195M 1s
     22500K .......... .......... .......... .......... .......... 61%  182M 1s
     22550K .......... .......... .......... .......... .......... 61%  243M 1s
     22600K .......... .......... .......... .......... .......... 61%  193M 1s
     22650K .......... .......... .......... .......... .......... 62%  199M 1s
     22700K .......... .......... .......... .......... .......... 62%  210M 1s
     22750K .......... .......... .......... .......... .......... 62%  160M 1s
     22800K .......... .......... .......... .......... .......... 62%  201M 1s
     22850K .......... .......... .......... .......... .......... 62%  215M 1s
     22900K .......... .......... .......... .......... .......... 62% 1.61M 1s
     22950K .......... .......... .......... .......... .......... 62% 54.8M 1s
     23000K .......... .......... .......... .......... .......... 62%  185M 1s
     23050K .......... .......... .......... .......... .......... 63%  170M 1s
     23100K .......... .......... .......... .......... .......... 63%  187M 1s
     23150K .......... .......... .......... .......... .......... 63% 87.7M 1s
     23200K .......... .......... .......... .......... .......... 63% 97.0M 1s
     23250K .......... .......... .......... .......... .......... 63%  144M 1s
     23300K .......... .......... .......... .......... .......... 63%  211M 1s
     23350K .......... .......... .......... .......... .......... 63%  207M 1s
     23400K .......... .......... .......... .......... .......... 64%  221M 1s
     23450K .......... .......... .......... .......... .......... 64%  209M 1s
     23500K .......... .......... .......... .......... .......... 64%  222M 1s
     23550K .......... .......... .......... .......... .......... 64%  208M 1s
     23600K .......... .......... .......... .......... .......... 64%  223M 1s
     23650K .......... .......... .......... .......... .......... 64%  219M 1s
     23700K .......... .......... .......... .......... .......... 64%  207M 1s
     23750K .......... .......... .......... .......... .......... 65%  150M 1s
     23800K .......... .......... .......... .......... .......... 65%  207M 1s
     23850K .......... .......... .......... .......... .......... 65%  195M 1s
     23900K .......... .......... .......... .......... .......... 65%  251M 1s
     23950K .......... .......... .......... .......... .......... 65%  203M 1s
     24000K .......... .......... .......... .......... .......... 65%  198M 1s
     24050K .......... .......... .......... .......... .......... 65%  184M 1s
     24100K .......... .......... .......... .......... .......... 65%  225M 1s
     24150K .......... .......... .......... .......... .......... 66%  202M 1s
     24200K .......... .......... .......... .......... .......... 66%  253M 1s
     24250K .......... .......... .......... .......... .......... 66%  155M 1s
     24300K .......... .......... .......... .......... .......... 66%  203M 1s
     24350K .......... .......... .......... .......... .......... 66% 85.1M 1s
     24400K .......... .......... .......... .......... .......... 66%  213M 1s
     24450K .......... .......... .......... .......... .......... 66% 2.84M 1s
     24500K .......... .......... .......... .......... .......... 67% 12.7M 1s
     24550K .......... .......... .......... .......... .......... 67% 87.7M 1s
     24600K .......... .......... .......... .......... .......... 67% 78.3M 1s
     24650K .......... .......... .......... .......... .......... 67% 95.2M 1s
     24700K .......... .......... .......... .......... .......... 67%  104M 1s
     24750K .......... .......... .......... .......... .......... 67% 83.7M 1s
     24800K .......... .......... .......... .......... .......... 67%  111M 1s
     24850K .......... .......... .......... .......... .......... 68%  100M 1s
     24900K .......... .......... .......... .......... .......... 68%  107M 1s
     24950K .......... .......... .......... .......... .......... 68% 57.0M 1s
     25000K .......... .......... .......... .......... .......... 68%  110M 1s
     25050K .......... .......... .......... .......... .......... 68% 95.2M 1s
     25100K .......... .......... .......... .......... .......... 68%  106M 1s
     25150K .......... .......... .......... .......... .......... 68% 97.7M 1s
     25200K .......... .......... .......... .......... .......... 68%  102M 1s
     25250K .......... .......... .......... .......... .......... 69% 83.5M 1s
     25300K .......... .......... .......... .......... .......... 69%  106M 1s
     25350K .......... .......... .......... .......... .......... 69% 90.9M 1s
     25400K .......... .......... .......... .......... .......... 69%  109M 1s
     25450K .......... .......... .......... .......... .......... 69% 98.2M 1s
     25500K .......... .......... .......... .......... .......... 69%  102M 0s
     25550K .......... .......... .......... .......... .......... 69% 62.9M 0s
     25600K .......... .......... .......... .......... .......... 70%  124M 0s
     25650K .......... .......... .......... .......... .......... 70%  120M 0s
     25700K .......... .......... .......... .......... .......... 70%  132M 0s
     25750K .......... .......... .......... .......... .......... 70%  100M 0s
     25800K .......... .......... .......... .......... .......... 70%  126M 0s
     25850K .......... .......... .......... .......... .......... 70%  120M 0s
     25900K .......... .......... .......... .......... .......... 70%  127M 0s
     25950K .......... .......... .......... .......... .......... 71%  116M 0s
     26000K .......... .......... .......... .......... .......... 71% 71.2M 0s
     26050K .......... .......... .......... .......... .......... 71%  119M 0s
     26100K .......... .......... .......... .......... .......... 71%  123M 0s
     26150K .......... .......... .......... .......... .......... 71%  116M 0s
     26200K .......... .......... .......... .......... .......... 71%  126M 0s
     26250K .......... .......... .......... .......... .......... 71%  104M 0s
     26300K .......... .......... .......... .......... .......... 71% 72.1M 0s
     26350K .......... .......... .......... .......... .......... 72%  122M 0s
     26400K .......... .......... .......... .......... .......... 72%  126M 0s
     26450K .......... .......... .......... .......... .......... 72%  115M 0s
     26500K .......... .......... .......... .......... .......... 72%  127M 0s
     26550K .......... .......... .......... .......... .......... 72%  114M 0s
     26600K .......... .......... .......... .......... .......... 72%  133M 0s
     26650K .......... .......... .......... .......... .......... 72%  112M 0s
     26700K .......... .......... .......... .......... .......... 73%  128M 0s
     26750K .......... .......... .......... .......... .......... 73%  112M 0s
     26800K .......... .......... .......... .......... .......... 73%  129M 0s
     26850K .......... .......... .......... .......... .......... 73%  122M 0s
     26900K .......... .......... .......... .......... .......... 73% 2.81M 0s
     26950K .......... .......... .......... .......... .......... 73% 94.4M 0s
     27000K .......... .......... .......... .......... .......... 73% 61.1M 0s
     27050K .......... .......... .......... .......... .......... 74% 84.4M 0s
     27100K .......... .......... .......... .......... .......... 74%  166M 0s
     27150K .......... .......... .......... .......... .......... 74%  198M 0s
     27200K .......... .......... .......... .......... .......... 74%  192M 0s
     27250K .......... .......... .......... .......... .......... 74% 82.2M 0s
     27300K .......... .......... .......... .......... .......... 74%  173M 0s
     27350K .......... .......... .......... .......... .......... 74%  219M 0s
     27400K .......... .......... .......... .......... .......... 75%  207M 0s
     27450K .......... .......... .......... .......... .......... 75%  229M 0s
     27500K .......... .......... .......... .......... .......... 75%  202M 0s
     27550K .......... .......... .......... .......... .......... 75%  204M 0s
     27600K .......... .......... .......... .......... .......... 75%  198M 0s
     27650K .......... .......... .......... .......... .......... 75%  212M 0s
     27700K .......... .......... .......... .......... .......... 75%  151M 0s
     27750K .......... .......... .......... .......... .......... 75%  199M 0s
     27800K .......... .......... .......... .......... .......... 76%  186M 0s
     27850K .......... .......... .......... .......... .......... 76%  201M 0s
     27900K .......... .......... .......... .......... .......... 76%  194M 0s
     27950K .......... .......... .......... .......... .......... 76%  196M 0s
     28000K .......... .......... .......... .......... .......... 76%  208M 0s
     28050K .......... .......... .......... .......... .......... 76%  198M 0s
     28100K .......... .......... .......... .......... .......... 76%  175M 0s
     28150K .......... .......... .......... .......... .......... 77%  222M 0s
     28200K .......... .......... .......... .......... .......... 77%  198M 0s
     28250K .......... .......... .......... .......... .......... 77%  204M 0s
     28300K .......... .......... .......... .......... .......... 77%  195M 0s
     28350K .......... .......... .......... .......... .......... 77%  247M 0s
     28400K .......... .......... .......... .......... .......... 77%  246M 0s
     28450K .......... .......... .......... .......... .......... 77%  250M 0s
     28500K .......... .......... .......... .......... .......... 78% 2.09M 0s
     28550K .......... .......... .......... .......... .......... 78% 50.9M 0s
     28600K .......... .......... .......... .......... .......... 78% 83.5M 0s
     28650K .......... .......... .......... .......... .......... 78% 72.6M 0s
     28700K .......... .......... .......... .......... .......... 78% 89.1M 0s
     28750K .......... .......... .......... .......... .......... 78% 85.5M 0s
     28800K .......... .......... .......... .......... .......... 78% 55.1M 0s
     28850K .......... .......... .......... .......... .......... 78% 75.3M 0s
     28900K .......... .......... .......... .......... .......... 79% 84.4M 0s
     28950K .......... .......... .......... .......... .......... 79% 79.3M 0s
     29000K .......... .......... .......... .......... .......... 79% 83.0M 0s
     29050K .......... .......... .......... .......... .......... 79% 73.0M 0s
     29100K .......... .......... .......... .......... .......... 79% 52.9M 0s
     29150K .......... .......... .......... .......... .......... 79% 62.3M 0s
     29200K .......... .......... .......... .......... .......... 79%  103M 0s
     29250K .......... .......... .......... .......... .......... 80%  122M 0s
     29300K .......... .......... .......... .......... .......... 80%  125M 0s
     29350K .......... .......... .......... .......... .......... 80%  118M 0s
     29400K .......... .......... .......... .......... .......... 80%  131M 0s
     29450K .......... .......... .......... .......... .......... 80%  117M 0s
     29500K .......... .......... .......... .......... .......... 80%  125M 0s
     29550K .......... .......... .......... .......... .......... 80%  120M 0s
     29600K .......... .......... .......... .......... .......... 81%  129M 0s
     29650K .......... .......... .......... .......... .......... 81%  119M 0s
     29700K .......... .......... .......... .......... .......... 81%  109M 0s
     29750K .......... .......... .......... .......... .......... 81%  125M 0s
     29800K .......... .......... .......... .......... .......... 81%  128M 0s
     29850K .......... .......... .......... .......... .......... 81%  114M 0s
     29900K .......... .......... .......... .......... .......... 81%  133M 0s
     29950K .......... .......... .......... .......... .......... 81%  118M 0s
     30000K .......... .......... .......... .......... .......... 82%  128M 0s
     30050K .......... .......... .......... .......... .......... 82%  121M 0s
     30100K .......... .......... .......... .......... .......... 82%  120M 0s
     30150K .......... .......... .......... .......... .......... 82%  118M 0s
     30200K .......... .......... .......... .......... .......... 82%  111M 0s
     30250K .......... .......... .......... .......... .......... 82%  116M 0s
     30300K .......... .......... .......... .......... .......... 82%  123M 0s
     30350K .......... .......... .......... .......... .......... 83%  124M 0s
     30400K .......... .......... .......... .......... .......... 83%  132M 0s
     30450K .......... .......... .......... .......... .......... 83%  118M 0s
     30500K .......... .......... .......... .......... .......... 83%  139M 0s
     30550K .......... .......... .......... .......... .......... 83%  120M 0s
     30600K .......... .......... .......... .......... .......... 83% 69.5M 0s
     30650K .......... .......... .......... .......... .......... 83%  108M 0s
     30700K .......... .......... .......... .......... .......... 84%  114M 0s
     30750K .......... .......... .......... .......... .......... 84%  126M 0s
     30800K .......... .......... .......... .......... .......... 84%  129M 0s
     30850K .......... .......... .......... .......... .......... 84%  117M 0s
     30900K .......... .......... .......... .......... .......... 84%  129M 0s
     30950K .......... .......... .......... .......... .......... 84% 2.99M 0s
     31000K .......... .......... .......... .......... .......... 84% 70.9M 0s
     31050K .......... .......... .......... .......... .......... 84% 65.0M 0s
     31100K .......... .......... .......... .......... .......... 85% 72.2M 0s
     31150K .......... .......... .......... .......... .......... 85% 68.2M 0s
     31200K .......... .......... .......... .......... .......... 85% 44.7M 0s
     31250K .......... .......... .......... .......... .......... 85% 68.9M 0s
     31300K .......... .......... .......... .......... .......... 85% 75.7M 0s
     31350K .......... .......... .......... .......... .......... 85% 71.4M 0s
     31400K .......... .......... .......... .......... .......... 85% 49.5M 0s
     31450K .......... .......... .......... .......... .......... 86% 64.1M 0s
     31500K .......... .......... .......... .......... .......... 86% 74.9M 0s
     31550K .......... .......... .......... .......... .......... 86% 58.4M 0s
     31600K .......... .......... .......... .......... .......... 86%  125M 0s
     31650K .......... .......... .......... .......... .......... 86%  118M 0s
     31700K .......... .......... .......... .......... .......... 86%  112M 0s
     31750K .......... .......... .......... .......... .......... 86%  117M 0s
     31800K .......... .......... .......... .......... .......... 87%  128M 0s
     31850K .......... .......... .......... .......... .......... 87%  116M 0s
     31900K .......... .......... .......... .......... .......... 87%  129M 0s
     31950K .......... .......... .......... .......... .......... 87%  112M 0s
     32000K .......... .......... .......... .......... .......... 87% 71.7M 0s
     32050K .......... .......... .......... .......... .......... 87%  111M 0s
     32100K .......... .......... .......... .......... .......... 87%  126M 0s
     32150K .......... .......... .......... .......... .......... 87%  120M 0s
     32200K .......... .......... .......... .......... .......... 88%  115M 0s
     32250K .......... .......... .......... .......... .......... 88%  117M 0s
     32300K .......... .......... .......... .......... .......... 88% 68.9M 0s
     32350K .......... .......... .......... .......... .......... 88%  115M 0s
     32400K .......... .......... .......... .......... .......... 88%  131M 0s
     32450K .......... .......... .......... .......... .......... 88%  129M 0s
     32500K .......... .......... .......... .......... .......... 88% 3.20M 0s
     32550K .......... .......... .......... .......... .......... 89% 66.0M 0s
     32600K .......... .......... .......... .......... .......... 89% 72.0M 0s
     32650K .......... .......... .......... .......... .......... 89% 42.9M 0s
     32700K .......... .......... .......... .......... .......... 89% 72.5M 0s
     32750K .......... .......... .......... .......... .......... 89% 72.7M 0s
     32800K .......... .......... .......... .......... .......... 89% 89.4M 0s
     32850K .......... .......... .......... .......... .......... 89% 90.0M 0s
     32900K .......... .......... .......... .......... .......... 90% 78.7M 0s
     32950K .......... .......... .......... .......... .......... 90% 38.9M 0s
     33000K .......... .......... .......... .......... .......... 90%  117M 0s
     33050K .......... .......... .......... .......... .......... 90%  116M 0s
     33100K .......... .......... .......... .......... .......... 90%  128M 0s
     33150K .......... .......... .......... .......... .......... 90% 99.4M 0s
     33200K .......... .......... .......... .......... .......... 90%  131M 0s
     33250K .......... .......... .......... .......... .......... 90% 67.1M 0s
     33300K .......... .......... .......... .......... .......... 91%  133M 0s
     33350K .......... .......... .......... .......... .......... 91%  114M 0s
     33400K .......... .......... .......... .......... .......... 91%  126M 0s
     33450K .......... .......... .......... .......... .......... 91%  115M 0s
     33500K .......... .......... .......... .......... .......... 91%  136M 0s
     33550K .......... .......... .......... .......... .......... 91%  123M 0s
     33600K .......... .......... .......... .......... .......... 91% 68.2M 0s
     33650K .......... .......... .......... .......... .......... 92% 99.8M 0s
     33700K .......... .......... .......... .......... .......... 92%  128M 0s
     33750K .......... .......... .......... .......... .......... 92%  124M 0s
     33800K .......... .......... .......... .......... .......... 92%  122M 0s
     33850K .......... .......... .......... .......... .......... 92%  125M 0s
     33900K .......... .......... .......... .......... .......... 92%  136M 0s
     33950K .......... .......... .......... .......... .......... 92% 66.2M 0s
     34000K .......... .......... .......... .......... .......... 93%  133M 0s
     34050K .......... .......... .......... .......... .......... 93%  113M 0s
     34100K .......... .......... .......... .......... .......... 93%  133M 0s
     34150K .......... .......... .......... .......... .......... 93%  105M 0s
     34200K .......... .......... .......... .......... .......... 93%  125M 0s
     34250K .......... .......... .......... .......... .......... 93% 53.6M 0s
     34300K .......... .......... .......... .......... .......... 93%  127M 0s
     34350K .......... .......... .......... .......... .......... 93%  124M 0s
     34400K .......... .......... .......... .......... .......... 94%  137M 0s
     34450K .......... .......... .......... .......... .......... 94%  115M 0s
     34500K .......... .......... .......... .......... .......... 94%  134M 0s
     34550K .......... .......... .......... .......... .......... 94%  124M 0s
     34600K .......... .......... .......... .......... .......... 94% 70.6M 0s
     34650K .......... .......... .......... .......... .......... 94%  116M 0s
     34700K .......... .......... .......... .......... .......... 94%  135M 0s
     34750K .......... .......... .......... .......... .......... 95%  128M 0s
     34800K .......... .......... .......... .......... .......... 95%  121M 0s
     34850K .......... .......... .......... .......... .......... 95%  120M 0s
     34900K .......... .......... .......... .......... .......... 95%  131M 0s
     34950K .......... .......... .......... .......... .......... 95%  122M 0s
     35000K .......... .......... .......... .......... .......... 95% 4.41M 0s
     35050K .......... .......... .......... .......... .......... 95% 18.7M 0s
     35100K .......... .......... .......... .......... .......... 96% 59.5M 0s
     35150K .......... .......... .......... .......... .......... 96% 60.5M 0s
     35200K .......... .......... .......... .......... .......... 96% 64.8M 0s
     35250K .......... .......... .......... .......... .......... 96% 69.6M 0s
     35300K .......... .......... .......... .......... .......... 96% 64.6M 0s
     35350K .......... .......... .......... .......... .......... 96% 47.1M 0s
     35400K .......... .......... .......... .......... .......... 96% 64.7M 0s
     35450K .......... .......... .......... .......... .......... 97% 69.5M 0s
     35500K .......... .......... .......... .......... .......... 97% 42.9M 0s
     35550K .......... .......... .......... .......... .......... 97% 51.8M 0s
     35600K .......... .......... .......... .......... .......... 97%  114M 0s
     35650K .......... .......... .......... .......... .......... 97%  111M 0s
     35700K .......... .......... .......... .......... .......... 97%  122M 0s
     35750K .......... .......... .......... .......... .......... 97%  123M 0s
     35800K .......... .......... .......... .......... .......... 97%  121M 0s
     35850K .......... .......... .......... .......... .......... 98%  133M 0s
     35900K .......... .......... .......... .......... .......... 98%  118M 0s
     35950K .......... .......... .......... .......... .......... 98%  129M 0s
     36000K .......... .......... .......... .......... .......... 98%  113M 0s
     36050K .......... .......... .......... .......... .......... 98%  122M 0s
     36100K .......... .......... .......... .......... .......... 98%  118M 0s
     36150K .......... .......... .......... .......... .......... 98% 67.8M 0s
     36200K .......... .......... .......... .......... .......... 99%  121M 0s
     36250K .......... .......... .......... .......... .......... 99%  133M 0s
     36300K .......... .......... .......... .......... .......... 99%  127M 0s
     36350K .......... .......... .......... .......... .......... 99%  120M 0s
     36400K .......... .......... .......... .......... .......... 99%  121M 0s
     36450K .......... .......... .......... .......... .......... 99%  120M 0s
     36500K .......... .......... .......... .......... .......... 99%  120M 0s
     36550K .......... .......... .......... .......... .......   100% 3.77M=1.4s
    
    2021-05-03 23:30:58 (26.4 MB/s) - 'LiftOver/liftOver' saved [37475560/37475560]
    
    WARNING: timestamping does nothing in combination with -O. See the manual
    for details.
    
    --2021-05-03 23:30:58--  ftp://hgdownload.soe.ucsc.edu/goldenPath/hg19/liftOver/hg19ToHg38.over.chain.gz
    Resolving dtn05-e0 (dtn05-e0)... 10.1.200.241
    Connecting to dtn05-e0 (dtn05-e0)|10.1.200.241|:3128... connected.
    Proxy request sent, awaiting response... 200 Gatewaying
    Length: 227698 (222K) [text/plain]
    Saving to: 'LiftOver/hg19ToHg38.over.chain.gz'
    
         0K .......... .......... .......... .......... .......... 22%  333K 1s
        50K .......... .......... .......... .......... .......... 44%  734K 0s
       100K .......... .......... .......... .......... .......... 67% 53.4M 0s
       150K .......... .......... .......... .......... .......... 89% 62.3M 0s
       200K .......... .......... ..                              100%  344K=0.3s
    
    2021-05-03 23:30:59 (780 KB/s) - 'LiftOver/hg19ToHg38.over.chain.gz' saved [227698/227698]
    
    Reading liftover chains
    Mapping coordinates
    [-] Unloading plink  1.9.0-beta4.4  on cn3174 
    [+] Loading plink  1.9.0-beta4.4  on cn3174 



```bash
%%bash
# number of variants lifted/unlifted
wc -l USmerged.HLAregion_update-map_to_b38.txt
wc -l SNPs_not_in_b38.txt
```

    5674 USmerged.HLAregion_update-map_to_b38.txt
    0 SNPs_not_in_b38.txt


## Run imputation

script to run imputation: `/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/CookHLA/impute.1000G.EUR.hg38.sh`

```
#!/bin/bash

source /data/chiarp/conda/etc/profile.d/conda.sh
conda activate CookHLA

DIR="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA"

mkdir /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G

python $DIR/CookHLA/CookHLA.py \
    -i $DIR/USmerged.HLAregion.chr6.Liftover.hg38 \
    -hg 38 \
    -o $DIR/Imputation.1000G/USmerged+1000G_REF.EUR \
    -ref $DIR/CookHLA/1000G_REF/1000G_REF.EUR.chr6.hg18.29mb-34mb.inT1DGC \
    -mem 240g \
    -mp 8
```


```bash
%%bash
cd CookHLA/
echo "sh impute.1000G.EUR.hg38.sh" > impute2.swarm
swarm --file impute2.swarm --logdir swarmOE_impute -g 240 -t 8 --time 12:00:00
```

    14237378


## Download code for HATK

HATK(HLA Analysis Tool-Kit) is a collection of tools and modules to perform HLA fine-mapping analysis, which is to identify which HLA allele or amino acid position of the HLA gene is driving the disease


```python
!git clone https://github.com/WansonChoi/HATK.git
```

    Cloning into 'HATK'...
    remote: Enumerating objects: 2246, done.[K
    remote: Counting objects: 100% (206/206), done.[K
    remote: Compressing objects: 100% (135/135), done.[K
    remote: Total 2246 (delta 92), reused 163 (delta 66), pack-reused 2040[K
    Receiving objects: 100% (2246/2246), 169.10 MiB | 10.57 MiB/s, done.
    Resolving deltas: 100% (1089/1089), done.
    Checking out files: 100% (718/718), done.


## Run HATK


```bash
%%bash
# Update covariate file so that FID = IID and it matches up with the original subsetted plink file used as input for imputation

module load R/3.5.2
R --vanilla --no-save

require(data.table)
require(tidyverse)

data <- fread("/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/COVARIATES.US_mg_noDups.UNRELATED.txt",header=T) %>%
        mutate(FID = IID)
head(data)
write.table(data,"COVARIATES.US_mg_noDups.UNRELATED.SampleNameUpdated.txt",quote=F,sep="\t",row.names=F,col.names=T)

```

    
    R version 3.5.2 (2018-12-20) -- "Eggshell Igloo"
    Copyright (C) 2018 The R Foundation for Statistical Computing
    Platform: x86_64-pc-linux-gnu (64-bit)
    
    R is free software and comes with ABSOLUTELY NO WARRANTY.
    You are welcome to redistribute it under certain conditions.
    Type 'license()' or 'licence()' for distribution details.
    
    R is a collaborative project with many contributors.
    Type 'contributors()' for more information and
    'citation()' on how to cite R or R packages in publications.
    
    Type 'demo()' for some demos, 'help()' for on-line help, or
    'help.start()' for an HTML browser interface to help.
    Type 'q()' to quit R.
    
    > 
    > require(data.table)
    > require(tidyverse)
    > 
    > data <- fread("/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/COVARIATES.US_mg_noDups.UNRELATED.txt",header=T) %>%
    +         mutate(FID = IID)
    > head(data)
          FID     IID MAT PAT SEX PHENO diagnosis genderONFILE age_at_onset
    1 BRC1058 BRC1058   0   0   1     1   control            0           70
    2  BRC618  BRC618   0   0   1     1   control            0           21
    3  BRC706  BRC706   0   0   2     1   control            1           56
    4  BRC710  BRC710   0   0   1     1   control            0           62
    5  BRC905  BRC905   0   0   1     1   control            0           72
    6  BRC991  BRC991   0   0   2     1   control            1           35
               PC1          PC2           PC3          PC4          PC5
    1 -0.008753894 -0.008942471 -0.0005265611 -0.005780903 -0.004512923
    2  0.004683084 -0.001838741  0.0181311300  0.005839165 -0.014278690
    3  0.017012450 -0.005869365 -0.0076155020 -0.002113765 -0.001482595
    4  0.007728595  0.013705830 -0.0070478520 -0.012808200 -0.007522605
    5 -0.175320200  0.016046630  0.0763218500 -0.004372520 -0.007606517
    6 -0.011514650  0.049522450 -0.0133695200  0.001673454  0.004551437
                PC6           PC7          PC8          PC9         PC10 GENDER
    1  0.0116358100  0.0032973190 -0.012263640  0.003986731 -0.015868050      0
    2  0.0312385300  0.0035149230 -0.011598180  0.006574303  0.013241020      0
    3  0.0134148800 -0.0088170730  0.005291236 -0.022391430 -0.006921480      1
    4  0.0119805500  0.0113048200 -0.017552920 -0.009450363 -0.004895789      0
    5  0.0000122211  0.0209567700 -0.028167390  0.006911984  0.004399125      0
    6 -0.0013035370  0.0007596483 -0.014509600 -0.005563398 -0.013608260      1
              FID_IID
    1 BRC1058_BRC1058
    2   BRC618_BRC618
    3   BRC706_BRC706
    4   BRC710_BRC710
    5   BRC905_BRC905
    6   BRC991_BRC991
    > write.table(data,"COVARIATES.US_mg_noDups.UNRELATED.SampleNameUpdated.txt",quote=F,sep="\t",row.names=F,col.names=T)
    > 


    [-] Unloading gcc  9.2.0  ... 
    [-] Unloading GSL 2.6 for GCC 9.2.0 ... 
    [-] Unloading openmpi 4.0.5  for GCC 9.2.0 
    [-] Unloading ImageMagick  7.0.8  on cn3174 
    [-] Unloading HDF5  1.10.4 
    [-] Unloading NetCDF 4.7.4_gcc9.2.0 
    [-] Unloading pandoc  2.13  on cn3174 
    [-] Unloading pcre2 10.21  ... 
    [-] Unloading R 4.0.5 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading GSL 2.4 for GCC 7.2.0 ... 
    [-] Unloading gcc  7.3.0  ... 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading openmpi 3.0.2  for GCC 7.3.0 
    [+] Loading ImageMagick  7.0.8  on cn3174 
    [+] Loading HDF5  1.10.4 
    [+] Loading pandoc  2.13  on cn3174 
    [+] Loading R 3.5.2 
    
    The following have been reloaded with a version change:
      1) GSL/2.6_gcc-9.2.0 => GSL/2.4_gcc-7.2.0     3) gcc/9.2.0 => gcc/7.3.0
      2) R/4.0 => R/3.5.2
    
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()


### Imputation.1000G

#### Overall

script to run imputation: `/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/HATK/hatk.1000G.4field.sh`

```
#!/bin/bash

module load plink/1.9.0-beta4.4
module load R/4.0

DIR="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA"

out="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G/4field"

python3 HATK.py \
    --variants $DIR/USmerged.HLAregion.chr6.Liftover.hg38 \
    --hped $DIR/Imputation.1000G/USmerged+1000G_REF.EUR.MHC.HLA_IMPUTATION_OUT.hped \
    --4field \
    --pheno $DIR/COVARIATES.US_mg_noDups.UNRELATED.SampleNameUpdated.txt \
    --pheno-name PHENO \
    --covar $DIR/COVARIATES.US_mg_noDups.UNRELATED.SampleNameUpdated.txt \
    --covar-name GENDER,age_at_onset,PC1,PC2,PC4,PC5,PC6,PC8,PC9,PC10 \
    --out $out/RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38 \
    --imgt 3320 \
    --hg 38 \
    --imgt-dir $DIR/HATK/example/IMGTHLA3320 \
    --multiprocess 8
```


```bash
%%bash
mkdir Imputation.1000G/4field
```


```bash
%%bash
cd HATK

echo "sh hatk.1000G.4field.sh" >> hatk.1000G.swarm
swarm --file hatk.1000G.swarm --logdir swarmOE_hatk -g 120 -t 8 
```

    14258806


##### plink version association - 4field


```bash
%%bash

module load plink/2.0-dev-20191128

DIR="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA"
out="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G/4field"
mkdir $out/GLM

plink \
--bfile $out/RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38 \
--pheno $DIR/COVARIATES.US_mg_noDups.UNRELATED.SampleNameUpdated.txt \
--pheno-name PHENO \
--covar $DIR/COVARIATES.US_mg_noDups.UNRELATED.SampleNameUpdated.txt \
--covar-name GENDER,age_at_onset,PC1,PC2,PC4,PC5,PC6,PC8,PC9,PC10 \
--glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err \
--out $out/GLM/RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38 \
--covar-variance-standardize

```

    PLINK v2.00a2LM 64-bit Intel (28 Nov 2019)     www.cog-genomics.org/plink/2.0/
    (C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
    Logging to /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G/4field/GLM/RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.log.
    Options in effect:
      --bfile /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G/4field/RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38
      --covar /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/COVARIATES.US_mg_noDups.UNRELATED.SampleNameUpdated.txt
      --covar-name GENDER,age_at_onset,PC1,PC2,PC4,PC5,PC6,PC8,PC9,PC10
      --covar-variance-standardize
      --glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err
      --out /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G/4field/GLM/RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38
      --pheno /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/COVARIATES.US_mg_noDups.UNRELATED.SampleNameUpdated.txt
      --pheno-name PHENO
    
    Start time: Wed May  5 00:19:46 2021
    257652 MiB RAM detected; reserving 128826 MiB for main workspace.
    Using up to 56 threads (change this with --threads).
    40269 samples (27350 females, 12919 males; 40269 founders) loaded from
    /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G/4field/RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.fam.
    6759 variants loaded from
    /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G/4field/RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.bim.
    1 binary phenotype loaded (964 cases, 35124 controls).
    10 covariates loaded from /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/COVARIATES.US_mg_noDups.UNRELATED.SampleNameUpdated.txt.
    --covar-variance-standardize: 10 covariates transformed.
    Calculating allele frequencies... 0%done.
    --glm logistic-Firth hybrid regression on phenotype 'PHENO': 0%done.
    Results written to /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G/4field/GLM/RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.PHENO.glm.logistic.hybrid .
    End time: Wed May  5 00:20:42 2021


    [-] Unloading plink  1.9.0-beta4.4  on cn3451 
    [+] Loading plink  2.0-dev-20191128 
    
    The following have been reloaded with a version change:
      1) plink/1.9.0-beta4.4 => plink/2.0-dev-20191128
    


##### plink version conditional association - 4field


```bash
%%bash

module load plink/2.0-dev-20191128

DIR="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA"
out="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G/4field"
mkdir $out/GLM.conditional

plink \
--bfile $out/RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38 \
--condition HLA_DRB1*13:01:02 \
--pheno $DIR/COVARIATES.US_mg_noDups.UNRELATED.SampleNameUpdated.txt \
--pheno-name PHENO \
--covar $DIR/COVARIATES.US_mg_noDups.UNRELATED.SampleNameUpdated.txt \
--covar-name GENDER,age_at_onset,PC1,PC2,PC4,PC5,PC6,PC8,PC9,PC10 \
--glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err \
--out $out/GLM.conditional/RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.HLA_DRB1 \
--covar-variance-standardize

plink \
--bfile $out/RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38 \
--condition AA_B_45_31356825_exon2_E \
--pheno $DIR/COVARIATES.US_mg_noDups.UNRELATED.SampleNameUpdated.txt \
--pheno-name PHENO \
--covar $DIR/COVARIATES.US_mg_noDups.UNRELATED.SampleNameUpdated.txt \
--covar-name GENDER,age_at_onset,PC1,PC2,PC4,PC5,PC6,PC8,PC9,PC10 \
--glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err \
--out $out/GLM.conditional/RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.AA.HLA_B \
--covar-variance-standardize
```

    PLINK v2.00a2LM 64-bit Intel (28 Nov 2019)     www.cog-genomics.org/plink/2.0/
    (C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
    Logging to /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G/4field/GLM.conditional/RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.HLA_DRB1.log.
    Options in effect:
      --bfile /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G/4field/RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38
      --condition HLA_DRB1*13:01:02
      --covar /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/COVARIATES.US_mg_noDups.UNRELATED.SampleNameUpdated.txt
      --covar-name GENDER,age_at_onset,PC1,PC2,PC4,PC5,PC6,PC8,PC9,PC10
      --covar-variance-standardize
      --glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err
      --out /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G/4field/GLM.conditional/RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.HLA_DRB1
      --pheno /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/COVARIATES.US_mg_noDups.UNRELATED.SampleNameUpdated.txt
      --pheno-name PHENO
    
    Start time: Thu May  6 03:59:43 2021
    257652 MiB RAM detected; reserving 128826 MiB for main workspace.
    Using up to 56 threads (change this with --threads).
    40269 samples (27350 females, 12919 males; 40269 founders) loaded from
    /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G/4field/RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.fam.
    6759 variants loaded from
    /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G/4field/RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.bim.
    1 binary phenotype loaded (964 cases, 35124 controls).
    10 covariates loaded from /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/COVARIATES.US_mg_noDups.UNRELATED.SampleNameUpdated.txt.
    --covar-variance-standardize: 10 covariates transformed.
    Calculating allele frequencies... 0%done.
    --glm logistic-Firth hybrid regression on phenotype 'PHENO': 0%done.
    Results written to /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G/4field/GLM.conditional/RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.HLA_DRB1.PHENO.glm.logistic.hybrid .
    End time: Thu May  6 04:00:47 2021
    PLINK v2.00a2LM 64-bit Intel (28 Nov 2019)     www.cog-genomics.org/plink/2.0/
    (C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
    Logging to /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G/4field/GLM.conditional/RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.AA.HLA_B.log.
    Options in effect:
      --bfile /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G/4field/RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38
      --condition AA_B_45_31356825_exon2_E
      --covar /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/COVARIATES.US_mg_noDups.UNRELATED.SampleNameUpdated.txt
      --covar-name GENDER,age_at_onset,PC1,PC2,PC4,PC5,PC6,PC8,PC9,PC10
      --covar-variance-standardize
      --glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err
      --out /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G/4field/GLM.conditional/RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.AA.HLA_B
      --pheno /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/COVARIATES.US_mg_noDups.UNRELATED.SampleNameUpdated.txt
      --pheno-name PHENO
    
    Start time: Thu May  6 04:00:47 2021
    257652 MiB RAM detected; reserving 128826 MiB for main workspace.
    Using up to 56 threads (change this with --threads).
    40269 samples (27350 females, 12919 males; 40269 founders) loaded from
    /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G/4field/RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.fam.
    6759 variants loaded from
    /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G/4field/RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.bim.
    1 binary phenotype loaded (964 cases, 35124 controls).
    10 covariates loaded from /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/COVARIATES.US_mg_noDups.UNRELATED.SampleNameUpdated.txt.
    --covar-variance-standardize: 10 covariates transformed.
    Calculating allele frequencies... 0%done.
    --glm logistic-Firth hybrid regression on phenotype 'PHENO': 0%done.
    Results written to /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G/4field/GLM.conditional/RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.AA.HLA_B.PHENO.glm.logistic.hybrid .
    End time: Thu May  6 04:02:01 2021


    [-] Unloading plink  1.9.0-beta4.4  on cn3451 
    [+] Loading plink  2.0-dev-20191128 
    
    The following have been reloaded with a version change:
      1) plink/1.9.0-beta4.4 => plink/2.0-dev-20191128
    
    mkdir: cannot create directory '/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G/4field/GLM.conditional': File exists



```bash
%%bash
# reformat file for meta analysis
cp /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/toMetaprep.glm.updated.v2.HLA.R .
cat toMetaprep.glm.updated.v2.HLA.R
```

    #!/usr/bin/env Rscript
    
    args <- commandArgs(trailingOnly=TRUE)
    if (length(args) != 2) {
      stop("USAGE: Rscript toMetaprep.glm.updated.v2.R args[1] args[2]
      where args[1] = glm_results
            args[2] = OutputFileName")
    }
    
    require(data.table)
    require(tidyverse)
    data <- fread(args[1],header=T)
    
    # Reformat
    data$A2 <- ifelse(data$A1 == data$ALT, data$REF, data$ALT)
    data$EffectAllele <- ifelse(data$A1_FREQ < 0.5, data$A1, data$A2)
    data$OtherAllele <- ifelse(data$A1_FREQ < 0.5, data$A2, data$A1)
    data$BETA <- ifelse(data$EffectAllele == data$A1, log(data$OR), log(data$OR)*-1)
    data$SE <- data$`LOG(OR)_SE`
    
    ## freq
    data$Freq_EffectAllele <- ifelse(data$EffectAllele == data$A1, data$A1_FREQ, 1-data$A1_FREQ)
    data$Freq_EffectAllele.CASE <- ifelse(data$EffectAllele == data$A1, data$A1_CASE_FREQ, 1-data$A1_CASE_FREQ)
    data$Freq_EffectAllele.CTRL <- ifelse(data$EffectAllele == data$A1, data$A1_CTRL_FREQ, 1-data$A1_CTRL_FREQ)
    
    ## counts
    data$EffectAllele_CT <- ifelse(data$EffectAllele == data$A1, data$A1_CT, data$ALLELE_CT-data$A1_CT)
    data$EffectAllele_CT.CASE <- ifelse(data$EffectAllele == data$A1, data$A1_CASE_CT, data$CASE_ALLELE_CT-data$A1_CASE_CT)
    data$EffectAllele_CT.CTRL <- ifelse(data$EffectAllele == data$A1, data$A1_CTRL_CT, data$CTRL_ALLELE_CT-data$A1_CTRL_CT)
    
    data$marker <- paste(data$ID,data$`#CHROM`,data$POS,data$OtherAllele,data$EffectAllele,sep=":")
    data1 <- data %>% rename(CHROM = `#CHROM`) %>% 
             select(CHROM,POS,ID,marker,EffectAllele,OtherAllele,
                    OR,BETA,SE,P,Freq_EffectAllele,Freq_EffectAllele.CASE,Freq_EffectAllele.CTRL,
                    ALLELE_CT,CASE_ALLELE_CT,CTRL_ALLELE_CT,
                    EffectAllele_CT,EffectAllele_CT.CASE,EffectAllele_CT.CTRL) %>%
             filter(BETA > -5 & BETA < 5) %>%
             filter(!is.na(BETA))
             
    dups <- data1 %>% select(marker,P) %>% filter(marker == duplicated(marker))
    temp1 <- subset(data1, data1$marker %in% dups$marker) %>%
             group_by(marker) %>%
             filter(P == min(P)) %>%
             data.frame()
    temp2 <- subset(data1, !(data1$marker %in% temp1$marker))
    data2 <- rbind(temp1,temp2) %>% arrange(CHROM,POS)
    
    write.table(data2, file=paste("toMeta.",args[2],".tab", sep=""), quote = F, sep = "\t", row.names = F)
    



```bash
%%bash
out="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G/4field"
cd $out/GLM

module load R/3.5.2
chmod u+x ../../../toMetaprep.glm.updated.v2.HLA.R
../../../toMetaprep.glm.updated.v2.HLA.R RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.PHENO.glm.logistic.hybrid RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38

cd $out/GLM.conditional
../../../toMetaprep.glm.updated.v2.HLA.R RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.HLA_DRB1.PHENO.glm.logistic.hybrid RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.HLA_DRB1
../../../toMetaprep.glm.updated.v2.HLA.R RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.AA.HLA_B.PHENO.glm.logistic.hybrid RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.AA.HLA_B
```

    [-] Unloading gcc  9.2.0  ... 
    [-] Unloading GSL 2.6 for GCC 9.2.0 ... 
    [-] Unloading openmpi 4.0.5  for GCC 9.2.0 
    [-] Unloading ImageMagick  7.0.8  on cn3451 
    [-] Unloading HDF5  1.10.4 
    [-] Unloading NetCDF 4.7.4_gcc9.2.0 
    [-] Unloading pandoc  2.13  on cn3451 
    [-] Unloading pcre2 10.21  ... 
    [-] Unloading R 4.0.5 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading GSL 2.4 for GCC 7.2.0 ... 
    [-] Unloading gcc  7.3.0  ... 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading openmpi 3.0.2  for GCC 7.3.0 
    [+] Loading ImageMagick  7.0.8  on cn3451 
    [+] Loading HDF5  1.10.4 
    [+] Loading pandoc  2.13  on cn3451 
    [+] Loading R 3.5.2 
    
    The following have been reloaded with a version change:
      1) GSL/2.6_gcc-9.2.0 => GSL/2.4_gcc-7.2.0     3) gcc/9.2.0 => gcc/7.3.0
      2) R/4.0 => R/3.5.2
    
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()



```python
import pandas as pd
pd.read_csv("Imputation.1000G/4field/GLM/toMeta.RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.tab",sep="\t").head()
```




<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>CHROM</th>
      <th>POS</th>
      <th>ID</th>
      <th>marker</th>
      <th>EffectAllele</th>
      <th>OtherAllele</th>
      <th>OR</th>
      <th>BETA</th>
      <th>SE</th>
      <th>P</th>
      <th>Freq_EffectAllele</th>
      <th>Freq_EffectAllele.CASE</th>
      <th>Freq_EffectAllele.CTRL</th>
      <th>ALLELE_CT</th>
      <th>CASE_ALLELE_CT</th>
      <th>CTRL_ALLELE_CT</th>
      <th>EffectAllele_CT</th>
      <th>EffectAllele_CT.CASE</th>
      <th>EffectAllele_CT.CTRL</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>6</td>
      <td>25006557</td>
      <td>rs1410689</td>
      <td>rs1410689:6:25006557:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>0.992779</td>
      <td>-0.007247</td>
      <td>0.047887</td>
      <td>0.879709</td>
      <td>0.423092</td>
      <td>0.417012</td>
      <td>0.423265</td>
      <td>69772</td>
      <td>1928</td>
      <td>67844</td>
      <td>29520</td>
      <td>804</td>
      <td>28716</td>
    </tr>
    <tr>
      <th>1</th>
      <td>6</td>
      <td>25010771</td>
      <td>rs10946741</td>
      <td>rs10946741:6:25010771:A:G</td>
      <td>G</td>
      <td>A</td>
      <td>1.001270</td>
      <td>0.001269</td>
      <td>0.047421</td>
      <td>0.978594</td>
      <td>0.445887</td>
      <td>0.442368</td>
      <td>0.445987</td>
      <td>69762</td>
      <td>1926</td>
      <td>67836</td>
      <td>31106</td>
      <td>852</td>
      <td>30254</td>
    </tr>
    <tr>
      <th>2</th>
      <td>6</td>
      <td>25011939</td>
      <td>rs2166681</td>
      <td>rs2166681:6:25011939:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>1.085830</td>
      <td>0.082345</td>
      <td>0.079349</td>
      <td>0.299372</td>
      <td>0.093557</td>
      <td>0.100836</td>
      <td>0.093351</td>
      <td>69380</td>
      <td>1914</td>
      <td>67466</td>
      <td>6491</td>
      <td>193</td>
      <td>6298</td>
    </tr>
    <tr>
      <th>3</th>
      <td>6</td>
      <td>25013892</td>
      <td>rs9467377</td>
      <td>rs9467377:6:25013892:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>0.758851</td>
      <td>-0.275950</td>
      <td>0.146816</td>
      <td>0.060167</td>
      <td>0.034008</td>
      <td>0.025961</td>
      <td>0.034237</td>
      <td>69806</td>
      <td>1926</td>
      <td>67880</td>
      <td>2374</td>
      <td>50</td>
      <td>2324</td>
    </tr>
    <tr>
      <th>4</th>
      <td>6</td>
      <td>25020792</td>
      <td>rs34632338</td>
      <td>rs34632338:6:25020792:G:T</td>
      <td>T</td>
      <td>G</td>
      <td>1.030680</td>
      <td>0.030219</td>
      <td>0.114503</td>
      <td>0.791857</td>
      <td>0.045105</td>
      <td>0.044087</td>
      <td>0.045134</td>
      <td>69748</td>
      <td>1928</td>
      <td>67820</td>
      <td>3146</td>
      <td>85</td>
      <td>3061</td>
    </tr>
  </tbody>
</table>
</div>




```python
import pandas as pd
pd.read_csv("Imputation.1000G/4field/GLM.conditional/toMeta.RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.HLA_DRB1.tab",sep="\t").head()
```




<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>CHROM</th>
      <th>POS</th>
      <th>ID</th>
      <th>marker</th>
      <th>EffectAllele</th>
      <th>OtherAllele</th>
      <th>OR</th>
      <th>BETA</th>
      <th>SE</th>
      <th>P</th>
      <th>Freq_EffectAllele</th>
      <th>Freq_EffectAllele.CASE</th>
      <th>Freq_EffectAllele.CTRL</th>
      <th>ALLELE_CT</th>
      <th>CASE_ALLELE_CT</th>
      <th>CTRL_ALLELE_CT</th>
      <th>EffectAllele_CT</th>
      <th>EffectAllele_CT.CASE</th>
      <th>EffectAllele_CT.CTRL</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>6</td>
      <td>25006557</td>
      <td>rs1410689</td>
      <td>rs1410689:6:25006557:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>0.993869</td>
      <td>-0.006150</td>
      <td>0.047946</td>
      <td>0.897939</td>
      <td>0.423092</td>
      <td>0.417012</td>
      <td>0.423265</td>
      <td>69772</td>
      <td>1928</td>
      <td>67844</td>
      <td>29520</td>
      <td>804</td>
      <td>28716</td>
    </tr>
    <tr>
      <th>1</th>
      <td>6</td>
      <td>25010771</td>
      <td>rs10946741</td>
      <td>rs10946741:6:25010771:A:G</td>
      <td>G</td>
      <td>A</td>
      <td>1.001840</td>
      <td>0.001838</td>
      <td>0.047457</td>
      <td>0.969020</td>
      <td>0.445887</td>
      <td>0.442368</td>
      <td>0.445987</td>
      <td>69762</td>
      <td>1926</td>
      <td>67836</td>
      <td>31106</td>
      <td>852</td>
      <td>30254</td>
    </tr>
    <tr>
      <th>2</th>
      <td>6</td>
      <td>25011939</td>
      <td>rs2166681</td>
      <td>rs2166681:6:25011939:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>1.084710</td>
      <td>0.081313</td>
      <td>0.079339</td>
      <td>0.305437</td>
      <td>0.093557</td>
      <td>0.100836</td>
      <td>0.093351</td>
      <td>69380</td>
      <td>1914</td>
      <td>67466</td>
      <td>6491</td>
      <td>193</td>
      <td>6298</td>
    </tr>
    <tr>
      <th>3</th>
      <td>6</td>
      <td>25013892</td>
      <td>rs9467377</td>
      <td>rs9467377:6:25013892:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>0.766988</td>
      <td>-0.265284</td>
      <td>0.146963</td>
      <td>0.071058</td>
      <td>0.034008</td>
      <td>0.025961</td>
      <td>0.034237</td>
      <td>69806</td>
      <td>1926</td>
      <td>67880</td>
      <td>2374</td>
      <td>50</td>
      <td>2324</td>
    </tr>
    <tr>
      <th>4</th>
      <td>6</td>
      <td>25020792</td>
      <td>rs34632338</td>
      <td>rs34632338:6:25020792:G:T</td>
      <td>T</td>
      <td>G</td>
      <td>1.033850</td>
      <td>0.033290</td>
      <td>0.114597</td>
      <td>0.771412</td>
      <td>0.045105</td>
      <td>0.044087</td>
      <td>0.045134</td>
      <td>69748</td>
      <td>1928</td>
      <td>67820</td>
      <td>3146</td>
      <td>85</td>
      <td>3061</td>
    </tr>
  </tbody>
</table>
</div>




```python
import pandas as pd
pd.read_csv("Imputation.1000G/4field/GLM.conditional/toMeta.RESULT_MG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.AA.HLA_B.tab",sep="\t").head()
```




<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>CHROM</th>
      <th>POS</th>
      <th>ID</th>
      <th>marker</th>
      <th>EffectAllele</th>
      <th>OtherAllele</th>
      <th>OR</th>
      <th>BETA</th>
      <th>SE</th>
      <th>P</th>
      <th>Freq_EffectAllele</th>
      <th>Freq_EffectAllele.CASE</th>
      <th>Freq_EffectAllele.CTRL</th>
      <th>ALLELE_CT</th>
      <th>CASE_ALLELE_CT</th>
      <th>CTRL_ALLELE_CT</th>
      <th>EffectAllele_CT</th>
      <th>EffectAllele_CT.CASE</th>
      <th>EffectAllele_CT.CTRL</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>6</td>
      <td>25006557</td>
      <td>rs1410689</td>
      <td>rs1410689:6:25006557:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>0.993523</td>
      <td>-0.006498</td>
      <td>0.047868</td>
      <td>0.892017</td>
      <td>0.423092</td>
      <td>0.417012</td>
      <td>0.423265</td>
      <td>69772</td>
      <td>1928</td>
      <td>67844</td>
      <td>29520</td>
      <td>804</td>
      <td>28716</td>
    </tr>
    <tr>
      <th>1</th>
      <td>6</td>
      <td>25010771</td>
      <td>rs10946741</td>
      <td>rs10946741:6:25010771:A:G</td>
      <td>G</td>
      <td>A</td>
      <td>1.002830</td>
      <td>0.002826</td>
      <td>0.047407</td>
      <td>0.952525</td>
      <td>0.445887</td>
      <td>0.442368</td>
      <td>0.445987</td>
      <td>69762</td>
      <td>1926</td>
      <td>67836</td>
      <td>31106</td>
      <td>852</td>
      <td>30254</td>
    </tr>
    <tr>
      <th>2</th>
      <td>6</td>
      <td>25011939</td>
      <td>rs2166681</td>
      <td>rs2166681:6:25011939:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>1.093900</td>
      <td>0.089749</td>
      <td>0.079405</td>
      <td>0.258344</td>
      <td>0.093557</td>
      <td>0.100836</td>
      <td>0.093351</td>
      <td>69380</td>
      <td>1914</td>
      <td>67466</td>
      <td>6491</td>
      <td>193</td>
      <td>6298</td>
    </tr>
    <tr>
      <th>3</th>
      <td>6</td>
      <td>25013892</td>
      <td>rs9467377</td>
      <td>rs9467377:6:25013892:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>0.753124</td>
      <td>-0.283525</td>
      <td>0.146892</td>
      <td>0.053587</td>
      <td>0.034008</td>
      <td>0.025961</td>
      <td>0.034237</td>
      <td>69806</td>
      <td>1926</td>
      <td>67880</td>
      <td>2374</td>
      <td>50</td>
      <td>2324</td>
    </tr>
    <tr>
      <th>4</th>
      <td>6</td>
      <td>25020792</td>
      <td>rs34632338</td>
      <td>rs34632338:6:25020792:G:T</td>
      <td>T</td>
      <td>G</td>
      <td>1.028250</td>
      <td>0.027858</td>
      <td>0.114540</td>
      <td>0.807827</td>
      <td>0.045105</td>
      <td>0.044087</td>
      <td>0.045134</td>
      <td>69748</td>
      <td>1928</td>
      <td>67820</td>
      <td>3146</td>
      <td>85</td>
      <td>3061</td>
    </tr>
  </tbody>
</table>
</div>



#### EOMG


```bash
%%bash
# Create updated sample same covariate file for EOMG i.e. FID = IID

module load R/3.5.2
R --vanilla --no-save

require(data.table)
require(tidyverse)

data <- fread("/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Analysis.GLM.Onset.hg38/US.JointPostImputation/EOMG/COVARIATES.USmerged.EOMG.txt",header=T) %>%
        mutate(FID = IID)
head(data)
write.table(data,"COVARIATES.USmerged.EOMG.SampleNameUpdated.txt",quote=F,sep="\t",col.names=T,row.names=F)

```

    
    R version 3.5.2 (2018-12-20) -- "Eggshell Igloo"
    Copyright (C) 2018 The R Foundation for Statistical Computing
    Platform: x86_64-pc-linux-gnu (64-bit)
    
    R is free software and comes with ABSOLUTELY NO WARRANTY.
    You are welcome to redistribute it under certain conditions.
    Type 'license()' or 'licence()' for distribution details.
    
    R is a collaborative project with many contributors.
    Type 'contributors()' for more information and
    'citation()' on how to cite R or R packages in publications.
    
    Type 'demo()' for some demos, 'help()' for on-line help, or
    'help.start()' for an HTML browser interface to help.
    Type 'q()' to quit R.
    
    > 
    > require(data.table)
    > require(tidyverse)
    > 
    > data <- fread("/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Analysis.GLM.Onset.hg38/US.JointPostImputation/EOMG/COVARIATES.USmerged.EOMG.txt",header=T) %>%
    +         mutate(FID = IID)
    > head(data)
             FID        IID PAT MAT SEX PHENO diagnosis genderONFILE age_at_onset
    1     BRC618     BRC618   0   0   1     1   control            0           21
    2     BRC991     BRC991   0   0   2     1   control            1           35
    3     NYT319     NYT319   0   0   2     1   control            1           28
    4 UMARY-1209 UMARY-1209   0   0   2     1   control            1           39
    5  UMARY-142  UMARY-142   0   0   1     1   control            0           16
    6 UMARY-1535 UMARY-1535   0   0   1     1   control            0           34
      GENDER               FID_IID           PC1          PC2          PC3
    1      0         BRC618_BRC618  1.343813e-05 -0.033372280  0.024885510
    2      1         BRC991_BRC991  1.218568e-02  0.056565740 -0.006907660
    3      1         NYT319_NYT319 -2.703337e-02 -0.030471890 -0.003027102
    4      1 UMARY-1209_UMARY-1209  2.580313e-02 -0.022267050 -0.020335480
    5      0   UMARY-142_UMARY-142  5.335392e-02 -0.006374142  0.005949997
    6      0 UMARY-1535_UMARY-1535 -1.004264e-02 -0.031583640  0.024475180
               PC4          PC5          PC6          PC7         PC8         PC9
    1 -0.008516317  0.023408330  0.021640320 -0.005154528 -0.02215929  0.01742999
    2  0.033340760 -0.019569620 -0.059399560 -0.003975566 -0.03795998 -0.03626275
    3  0.022571160 -0.027054250  0.035824770  0.067054670 -0.02136425  0.02390296
    4 -0.006097497 -0.002747690  0.015083010  0.028302310 -0.05787386  0.01191399
    5  0.027527880  0.013941180 -0.004079655 -0.022588600  0.02350963  0.04162686
    6  0.018383570 -0.007623544  0.035583260  0.060033610  0.01201529 -0.00505930
              PC10
    1 -0.048708240
    2 -0.047662190
    3 -0.046356800
    4 -0.037911510
    5  0.009828542
    6  0.022971790
    > write.table(data,"COVARIATES.USmerged.EOMG.SampleNameUpdated.txt",quote=F,sep="\t",col.names=T,row.names=F)
    > 


    [-] Unloading gcc  9.2.0  ... 
    [-] Unloading GSL 2.6 for GCC 9.2.0 ... 
    [-] Unloading openmpi 4.0.5  for GCC 9.2.0 
    [-] Unloading ImageMagick  7.0.8  on cn3174 
    [-] Unloading HDF5  1.10.4 
    [-] Unloading NetCDF 4.7.4_gcc9.2.0 
    [-] Unloading pandoc  2.13  on cn3174 
    [-] Unloading pcre2 10.21  ... 
    [-] Unloading R 4.0.5 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading GSL 2.4 for GCC 7.2.0 ... 
    [-] Unloading gcc  7.3.0  ... 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading openmpi 3.0.2  for GCC 7.3.0 
    [+] Loading ImageMagick  7.0.8  on cn3174 
    [+] Loading HDF5  1.10.4 
    [+] Loading pandoc  2.13  on cn3174 
    [+] Loading R 3.5.2 
    
    The following have been reloaded with a version change:
      1) GSL/2.6_gcc-9.2.0 => GSL/2.4_gcc-7.2.0     3) gcc/9.2.0 => gcc/7.3.0
      2) R/4.0 => R/3.5.2
    
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()



```bash
%%bash
mkdir Imputation.1000G.EOMG
mkdir Imputation.1000G.EOMG/4field
```

script to run imputation: `/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/HATK/hatk.1000G.4field.EOMG.sh`

```
#!/bin/bash

module load plink/1.9.0-beta4.4
module load R/4.0

DIR="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA"

out="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.EOMG/4field"

python3 HATK.py \
    --variants $DIR/USmerged.HLAregion.chr6.Liftover.hg38 \
    --hped $DIR/Imputation.1000G/USmerged+1000G_REF.EUR.MHC.HLA_IMPUTATION_OUT.hped \
    --4field \
    --pheno $DIR/COVARIATES.USmerged.EOMG.SampleNameUpdated.txt \
    --pheno-name PHENO \
    --covar $DIR/COVARIATES.USmerged.EOMG.SampleNameUpdated.txt \
    --covar-name age_at_onset,PC1,PC2,PC5 \
    --out $out/RESULT_EOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38 \
    --imgt 3320 \
    --hg 38 \
    --imgt-dir $DIR/HATK/example/IMGTHLA3320 \
    --multiprocess 8
```


```bash
%%bash
cd HATK

echo "sh hatk.1000G.4field.EOMG.sh" >> hatk.1000G.EOMG.swarm
swarm --file hatk.1000G.EOMG.swarm --logdir swarmOE_hatk -g 120 -t 8 
```

    14305492


##### plink version association - 4field


```bash
%%bash

module load plink/2.0-dev-20191128

DIR="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA"
out="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.EOMG/4field"
mkdir $out/GLM

plink \
--bfile $out/RESULT_EOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38 \
--pheno $DIR/COVARIATES.USmerged.EOMG.SampleNameUpdated.txt \
--pheno-name PHENO \
--covar $DIR/COVARIATES.USmerged.EOMG.SampleNameUpdated.txt \
--covar-name age_at_onset,PC1,PC2,PC5 \
--glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err \
--out $out/GLM/RESULT_EOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38 \
--covar-variance-standardize
```

    PLINK v2.00a2LM 64-bit Intel (28 Nov 2019)     www.cog-genomics.org/plink/2.0/
    (C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
    Logging to /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.EOMG/4field/GLM/RESULT_EOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.log.
    Options in effect:
      --bfile /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.EOMG/4field/RESULT_EOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38
      --covar /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/COVARIATES.USmerged.EOMG.SampleNameUpdated.txt
      --covar-name age_at_onset,PC1,PC2,PC5
      --covar-variance-standardize
      --glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err
      --out /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.EOMG/4field/GLM/RESULT_EOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38
      --pheno /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/COVARIATES.USmerged.EOMG.SampleNameUpdated.txt
      --pheno-name PHENO
    
    Start time: Wed May  5 00:35:04 2021
    257652 MiB RAM detected; reserving 128826 MiB for main workspace.
    Using up to 56 threads (change this with --threads).
    40269 samples (27350 females, 12919 males; 40269 founders) loaded from
    /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.EOMG/4field/RESULT_EOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.fam.
    6759 variants loaded from
    /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.EOMG/4field/RESULT_EOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.bim.
    1 binary phenotype loaded (234 cases, 2052 controls).
    4 covariates loaded from /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/COVARIATES.USmerged.EOMG.SampleNameUpdated.txt.
    --covar-variance-standardize: 4 covariates transformed.
    Calculating allele frequencies... 0%done.
    --glm logistic-Firth hybrid regression on phenotype 'PHENO': 0%done.
    Results written to /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.EOMG/4field/GLM/RESULT_EOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.PHENO.glm.logistic.hybrid .
    End time: Wed May  5 00:35:05 2021


    [-] Unloading plink  1.9.0-beta4.4  on cn3451 
    [+] Loading plink  2.0-dev-20191128 
    
    The following have been reloaded with a version change:
      1) plink/1.9.0-beta4.4 => plink/2.0-dev-20191128
    



```bash
%%bash
out="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.EOMG/4field"
cd $out/GLM

module load R/3.5.2
../../../toMetaprep.glm.updated.v2.HLA.R RESULT_EOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.PHENO.glm.logistic.hybrid RESULT_EOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38
```

    [-] Unloading gcc  9.2.0  ... 
    [-] Unloading GSL 2.6 for GCC 9.2.0 ... 
    [-] Unloading openmpi 4.0.5  for GCC 9.2.0 
    [-] Unloading ImageMagick  7.0.8  on cn3451 
    [-] Unloading HDF5  1.10.4 
    [-] Unloading NetCDF 4.7.4_gcc9.2.0 
    [-] Unloading pandoc  2.13  on cn3451 
    [-] Unloading pcre2 10.21  ... 
    [-] Unloading R 4.0.5 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading GSL 2.4 for GCC 7.2.0 ... 
    [-] Unloading gcc  7.3.0  ... 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading openmpi 3.0.2  for GCC 7.3.0 
    [+] Loading ImageMagick  7.0.8  on cn3451 
    [+] Loading HDF5  1.10.4 
    [+] Loading pandoc  2.13  on cn3451 
    [+] Loading R 3.5.2 
    
    The following have been reloaded with a version change:
      1) GSL/2.6_gcc-9.2.0 => GSL/2.4_gcc-7.2.0     3) gcc/9.2.0 => gcc/7.3.0
      2) R/4.0 => R/3.5.2
    
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()



```python
import pandas as pd
pd.read_csv("Imputation.1000G.EOMG/4field/GLM/toMeta.RESULT_EOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.tab",sep="\t").head()
```




<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>CHROM</th>
      <th>POS</th>
      <th>ID</th>
      <th>marker</th>
      <th>EffectAllele</th>
      <th>OtherAllele</th>
      <th>OR</th>
      <th>BETA</th>
      <th>SE</th>
      <th>P</th>
      <th>Freq_EffectAllele</th>
      <th>Freq_EffectAllele.CASE</th>
      <th>Freq_EffectAllele.CTRL</th>
      <th>ALLELE_CT</th>
      <th>CASE_ALLELE_CT</th>
      <th>CTRL_ALLELE_CT</th>
      <th>EffectAllele_CT</th>
      <th>EffectAllele_CT.CASE</th>
      <th>EffectAllele_CT.CTRL</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>6</td>
      <td>25006557</td>
      <td>rs1410689</td>
      <td>rs1410689:6:25006557:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>1.015270</td>
      <td>0.015155</td>
      <td>0.104714</td>
      <td>0.884929</td>
      <td>0.432604</td>
      <td>0.423077</td>
      <td>0.433691</td>
      <td>4570</td>
      <td>468</td>
      <td>4102</td>
      <td>1977</td>
      <td>198</td>
      <td>1779</td>
    </tr>
    <tr>
      <th>1</th>
      <td>6</td>
      <td>25010771</td>
      <td>rs10946741</td>
      <td>rs10946741:6:25010771:A:G</td>
      <td>G</td>
      <td>A</td>
      <td>1.010780</td>
      <td>0.010722</td>
      <td>0.102983</td>
      <td>0.917106</td>
      <td>0.455799</td>
      <td>0.444444</td>
      <td>0.457094</td>
      <td>4570</td>
      <td>468</td>
      <td>4102</td>
      <td>2083</td>
      <td>208</td>
      <td>1875</td>
    </tr>
    <tr>
      <th>2</th>
      <td>6</td>
      <td>25011939</td>
      <td>rs2166681</td>
      <td>rs2166681:6:25011939:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>0.872207</td>
      <td>-0.136728</td>
      <td>0.187790</td>
      <td>0.466558</td>
      <td>0.091568</td>
      <td>0.086580</td>
      <td>0.092131</td>
      <td>4554</td>
      <td>462</td>
      <td>4092</td>
      <td>417</td>
      <td>40</td>
      <td>377</td>
    </tr>
    <tr>
      <th>3</th>
      <td>6</td>
      <td>25013892</td>
      <td>rs9467377</td>
      <td>rs9467377:6:25013892:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>0.931154</td>
      <td>-0.071331</td>
      <td>0.278291</td>
      <td>0.797707</td>
      <td>0.032385</td>
      <td>0.032051</td>
      <td>0.032423</td>
      <td>4570</td>
      <td>468</td>
      <td>4102</td>
      <td>148</td>
      <td>15</td>
      <td>133</td>
    </tr>
    <tr>
      <th>4</th>
      <td>6</td>
      <td>25020792</td>
      <td>rs34632338</td>
      <td>rs34632338:6:25020792:G:T</td>
      <td>T</td>
      <td>G</td>
      <td>0.975173</td>
      <td>-0.025140</td>
      <td>0.252191</td>
      <td>0.920592</td>
      <td>0.046211</td>
      <td>0.040598</td>
      <td>0.046852</td>
      <td>4566</td>
      <td>468</td>
      <td>4098</td>
      <td>211</td>
      <td>19</td>
      <td>192</td>
    </tr>
  </tbody>
</table>
</div>



#### LOMG


```bash
%%bash
# Create updated sample same covariate file for LOMG i.e. FID = IID

module load R/3.5.2
R --vanilla --no-save

require(data.table)
require(tidyverse)

data <- fread("/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Analysis.GLM.Onset.hg38/US.JointPostImputation/LOMG/COVARIATES.USmerged.LOMG.txt",header=T) %>%
        mutate(FID = IID)
head(data)
write.table(data,"COVARIATES.USmerged.LOMG.SampleNameUpdated.txt",quote=F,sep="\t",col.names=T,row.names=F)

```

    
    R version 3.5.2 (2018-12-20) -- "Eggshell Igloo"
    Copyright (C) 2018 The R Foundation for Statistical Computing
    Platform: x86_64-pc-linux-gnu (64-bit)
    
    R is free software and comes with ABSOLUTELY NO WARRANTY.
    You are welcome to redistribute it under certain conditions.
    Type 'license()' or 'licence()' for distribution details.
    
    R is a collaborative project with many contributors.
    Type 'contributors()' for more information and
    'citation()' on how to cite R or R packages in publications.
    
    Type 'demo()' for some demos, 'help()' for on-line help, or
    'help.start()' for an HTML browser interface to help.
    Type 'q()' to quit R.
    
    > 
    > require(data.table)
    > require(tidyverse)
    > 
    > data <- fread("/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Analysis.GLM.Onset.hg38/US.JointPostImputation/LOMG/COVARIATES.USmerged.LOMG.txt",header=T) %>%
    +         mutate(FID = IID)
    > head(data)
          FID     IID PAT MAT SEX PHENO diagnosis genderONFILE age_at_onset GENDER
    1 BRC1058 BRC1058   0   0   1     1   control            0           70      0
    2  BRC706  BRC706   0   0   2     1   control            1           56      1
    3  BRC710  BRC710   0   0   1     1   control            0           62      0
    4  BRC905  BRC905   0   0   1     1   control            0           72      0
    5  BRC993  BRC993   0   0   1     1   control            0           66      0
    6  BRC994  BRC994   0   0   1     1   control            0           49      0
              FID_IID          PC1          PC2          PC3          PC4
    1 BRC1058_BRC1058  0.007482243 -0.010550130  0.002877840  0.002250730
    2   BRC706_BRC706 -0.012950120 -0.006236746 -0.002705750  0.002569052
    3   BRC710_BRC710 -0.011608890  0.013236820 -0.007219775  0.007140881
    4   BRC905_BRC905  0.174476400  0.018098780  0.073299900 -0.001204830
    5   BRC993_BRC993 -0.011612770 -0.005820990  0.003480047  0.010897770
    6   BRC994_BRC994 -0.014772670  0.002438232 -0.002749577 -0.008577396
               PC5           PC6          PC7           PC8          PC9
    1 -0.003045962  0.0080432830 -0.018158320  0.0195862300  0.022433030
    2  0.003131390 -0.0011734790 -0.007943211  0.0026158130  0.007089923
    3  0.004070663  0.0001551576 -0.013914010  0.0002502791  0.022166010
    4  0.009610681  0.0157004700  0.014872410  0.0099963200  0.006619532
    5  0.010489900  0.0013667130  0.007157154 -0.0016602770  0.002149683
    6  0.002600974  0.0198046700 -0.009023734 -0.0138707300 -0.012088390
               PC10
    1  0.0129857800
    2  0.0244002000
    3 -0.0004060526
    4 -0.0095526830
    5  0.0123991600
    6 -0.0021346570
    > write.table(data,"COVARIATES.USmerged.LOMG.SampleNameUpdated.txt",quote=F,sep="\t",col.names=T,row.names=F)
    > 


    [-] Unloading gcc  9.2.0  ... 
    [-] Unloading GSL 2.6 for GCC 9.2.0 ... 
    [-] Unloading openmpi 4.0.5  for GCC 9.2.0 
    [-] Unloading ImageMagick  7.0.8  on cn3174 
    [-] Unloading HDF5  1.10.4 
    [-] Unloading NetCDF 4.7.4_gcc9.2.0 
    [-] Unloading pandoc  2.13  on cn3174 
    [-] Unloading pcre2 10.21  ... 
    [-] Unloading R 4.0.5 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading GSL 2.4 for GCC 7.2.0 ... 
    [-] Unloading gcc  7.3.0  ... 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading openmpi 3.0.2  for GCC 7.3.0 
    [+] Loading ImageMagick  7.0.8  on cn3174 
    [+] Loading HDF5  1.10.4 
    [+] Loading pandoc  2.13  on cn3174 
    [+] Loading R 3.5.2 
    
    The following have been reloaded with a version change:
      1) GSL/2.6_gcc-9.2.0 => GSL/2.4_gcc-7.2.0     3) gcc/9.2.0 => gcc/7.3.0
      2) R/4.0 => R/3.5.2
    
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()



```bash
%%bash
mkdir Imputation.1000G.LOMG
mkdir Imputation.1000G.LOMG/4field
```

script to run imputation: `/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/HATK/hatk.1000G.4field.LOMG.sh`

```
#!/bin/bash

module load plink/1.9.0-beta4.4
module load R/4.0

DIR="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA"

out="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.LOMG/4field"

python3 HATK.py \
    --variants $DIR/USmerged.HLAregion.chr6.Liftover.hg38 \
    --hped $DIR/Imputation.1000G/USmerged+1000G_REF.EUR.MHC.HLA_IMPUTATION_OUT.hped \
    --4field \
    --pheno $DIR/COVARIATES.USmerged.LOMG.SampleNameUpdated.txt \
    --pheno-name PHENO \
    --covar $DIR/COVARIATES.USmerged.LOMG.SampleNameUpdated.txt \
    --covar-name GENDER,age_at_onset,PC1,PC2,PC3,PC4,PC5,PC7 \
    --out $out/RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38 \
    --imgt 3320 \
    --hg 38 \
    --imgt-dir $DIR/HATK/example/IMGTHLA3320 \
    --multiprocess 8
```


```bash
%%bash
cd HATK

echo "sh hatk.1000G.4field.LOMG.sh" >> hatk.1000G.LOMG.swarm
swarm --file hatk.1000G.LOMG.swarm --logdir swarmOE_hatk -g 120 -t 8 
```

    14305646


##### plink version association - 4field


```bash
%%bash

module load plink/2.0-dev-20191128

DIR="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA"
out="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.LOMG/4field"
mkdir $out/GLM

plink \
--bfile $out/RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38 \
--pheno $DIR/COVARIATES.USmerged.LOMG.SampleNameUpdated.txt \
--pheno-name PHENO \
--covar $DIR/COVARIATES.USmerged.LOMG.SampleNameUpdated.txt \
--covar-name GENDER,age_at_onset,PC1,PC2,PC3,PC4,PC5,PC7 \
--glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err \
--out $out/GLM/RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38 \
--covar-variance-standardize
```

    PLINK v2.00a2LM 64-bit Intel (28 Nov 2019)     www.cog-genomics.org/plink/2.0/
    (C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
    Logging to /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.LOMG/4field/GLM/RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.log.
    Options in effect:
      --bfile /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.LOMG/4field/RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38
      --covar /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/COVARIATES.USmerged.LOMG.SampleNameUpdated.txt
      --covar-name GENDER,age_at_onset,PC1,PC2,PC3,PC4,PC5,PC7
      --covar-variance-standardize
      --glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err
      --out /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.LOMG/4field/GLM/RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38
      --pheno /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/COVARIATES.USmerged.LOMG.SampleNameUpdated.txt
      --pheno-name PHENO
    
    Start time: Wed May  5 00:38:38 2021
    257652 MiB RAM detected; reserving 128826 MiB for main workspace.
    Using up to 56 threads (change this with --threads).
    40269 samples (27350 females, 12919 males; 40269 founders) loaded from
    /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.LOMG/4field/RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.fam.
    6759 variants loaded from
    /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.LOMG/4field/RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.bim.
    1 binary phenotype loaded (730 cases, 33072 controls).
    8 covariates loaded from /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/COVARIATES.USmerged.LOMG.SampleNameUpdated.txt.
    --covar-variance-standardize: 8 covariates transformed.
    Calculating allele frequencies... 0%done.
    --glm logistic-Firth hybrid regression on phenotype 'PHENO': 0%done.
    Results written to /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.LOMG/4field/GLM/RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.PHENO.glm.logistic.hybrid .
    End time: Wed May  5 00:39:19 2021


    [-] Unloading plink  1.9.0-beta4.4  on cn3451 
    [+] Loading plink  2.0-dev-20191128 
    
    The following have been reloaded with a version change:
      1) plink/1.9.0-beta4.4 => plink/2.0-dev-20191128
    


##### plink version conditional association - 4field


```bash
%%bash

module load plink/2.0-dev-20191128

DIR="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA"
out="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.LOMG/4field"
#mkdir $out/GLM.conditional

plink \
--bfile $out/RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38 \
--condition AA_DRB1_13_32584346_exon2_S \
--pheno $DIR/COVARIATES.USmerged.LOMG.SampleNameUpdated.txt \
--pheno-name PHENO \
--covar $DIR/COVARIATES.USmerged.LOMG.SampleNameUpdated.txt \
--covar-name GENDER,age_at_onset,PC1,PC2,PC3,PC4,PC5,PC7 \
--glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err \
--out $out/GLM.conditional/RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.AA.HLA_DRB1 \
--covar-variance-standardize

plink \
--bfile $out/RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38 \
--condition HLA_DQB1*06:03:02 \
--pheno $DIR/COVARIATES.USmerged.LOMG.SampleNameUpdated.txt \
--pheno-name PHENO \
--covar $DIR/COVARIATES.USmerged.LOMG.SampleNameUpdated.txt \
--covar-name GENDER,age_at_onset,PC1,PC2,PC3,PC4,PC5,PC7 \
--glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err \
--out $out/GLM.conditional/RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.HLA_DQB1 \
--covar-variance-standardize
```

    PLINK v2.00a2LM 64-bit Intel (28 Nov 2019)     www.cog-genomics.org/plink/2.0/
    (C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
    Logging to /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.LOMG/4field/GLM.conditional/RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.AA.HLA_DRB1.log.
    Options in effect:
      --bfile /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.LOMG/4field/RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38
      --condition AA_DRB1_13_32584346_exon2_S
      --covar /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/COVARIATES.USmerged.LOMG.SampleNameUpdated.txt
      --covar-name GENDER,age_at_onset,PC1,PC2,PC3,PC4,PC5,PC7
      --covar-variance-standardize
      --glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err
      --out /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.LOMG/4field/GLM.conditional/RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.AA.HLA_DRB1
      --pheno /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/COVARIATES.USmerged.LOMG.SampleNameUpdated.txt
      --pheno-name PHENO
    
    Start time: Thu May  6 10:18:54 2021
    257652 MiB RAM detected; reserving 128826 MiB for main workspace.
    Using up to 56 threads (change this with --threads).
    40269 samples (27350 females, 12919 males; 40269 founders) loaded from
    /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.LOMG/4field/RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.fam.
    6759 variants loaded from
    /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.LOMG/4field/RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.bim.
    1 binary phenotype loaded (730 cases, 33072 controls).
    8 covariates loaded from /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/COVARIATES.USmerged.LOMG.SampleNameUpdated.txt.
    --covar-variance-standardize: 8 covariates transformed.
    Calculating allele frequencies... 0%done.
    --glm logistic-Firth hybrid regression on phenotype 'PHENO': 0%done.
    Results written to /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.LOMG/4field/GLM.conditional/RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.AA.HLA_DRB1.PHENO.glm.logistic.hybrid .
    End time: Thu May  6 10:19:40 2021
    PLINK v2.00a2LM 64-bit Intel (28 Nov 2019)     www.cog-genomics.org/plink/2.0/
    (C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
    Logging to /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.LOMG/4field/GLM.conditional/RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.HLA_DQB1.log.
    Options in effect:
      --bfile /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.LOMG/4field/RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38
      --condition HLA_DQB1*06:03:02
      --covar /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/COVARIATES.USmerged.LOMG.SampleNameUpdated.txt
      --covar-name GENDER,age_at_onset,PC1,PC2,PC3,PC4,PC5,PC7
      --covar-variance-standardize
      --glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err
      --out /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.LOMG/4field/GLM.conditional/RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.HLA_DQB1
      --pheno /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/COVARIATES.USmerged.LOMG.SampleNameUpdated.txt
      --pheno-name PHENO
    
    Start time: Thu May  6 10:19:40 2021
    257652 MiB RAM detected; reserving 128826 MiB for main workspace.
    Using up to 56 threads (change this with --threads).
    40269 samples (27350 females, 12919 males; 40269 founders) loaded from
    /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.LOMG/4field/RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.fam.
    6759 variants loaded from
    /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.LOMG/4field/RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.bim.
    1 binary phenotype loaded (730 cases, 33072 controls).
    8 covariates loaded from /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/COVARIATES.USmerged.LOMG.SampleNameUpdated.txt.
    --covar-variance-standardize: 8 covariates transformed.
    Calculating allele frequencies... 0%done.
    --glm logistic-Firth hybrid regression on phenotype 'PHENO': 0%done.
    Results written to /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.LOMG/4field/GLM.conditional/RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.HLA_DQB1.PHENO.glm.logistic.hybrid .
    End time: Thu May  6 10:20:26 2021


    [-] Unloading plink  1.9.0-beta4.4  on cn3451 
    [+] Loading plink  2.0-dev-20191128 
    
    The following have been reloaded with a version change:
      1) plink/1.9.0-beta4.4 => plink/2.0-dev-20191128
    



```bash
%%bash
out="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Imputation.HLA/Imputation.1000G.LOMG/4field"
cd $out/GLM

module load R/3.5.2
../../../toMetaprep.glm.updated.v2.HLA.R RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.PHENO.glm.logistic.hybrid RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38

cd $out/GLM.conditional
../../../toMetaprep.glm.updated.v2.HLA.R RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.AA.HLA_DRB1.PHENO.glm.logistic.hybrid RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.AA.HLA_DRB1
../../../toMetaprep.glm.updated.v2.HLA.R RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.HLA_DQB1.PHENO.glm.logistic.hybrid RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.HLA_DQB1
```

    [-] Unloading gcc  9.2.0  ... 
    [-] Unloading GSL 2.6 for GCC 9.2.0 ... 
    [-] Unloading openmpi 4.0.5  for GCC 9.2.0 
    [-] Unloading ImageMagick  7.0.8  on cn3451 
    [-] Unloading HDF5  1.10.4 
    [-] Unloading NetCDF 4.7.4_gcc9.2.0 
    [-] Unloading pandoc  2.13  on cn3451 
    [-] Unloading pcre2 10.21  ... 
    [-] Unloading R 4.0.5 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading GSL 2.4 for GCC 7.2.0 ... 
    [-] Unloading gcc  7.3.0  ... 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading openmpi 3.0.2  for GCC 7.3.0 
    [+] Loading ImageMagick  7.0.8  on cn3451 
    [+] Loading HDF5  1.10.4 
    [+] Loading pandoc  2.13  on cn3451 
    [+] Loading R 3.5.2 
    
    The following have been reloaded with a version change:
      1) GSL/2.6_gcc-9.2.0 => GSL/2.4_gcc-7.2.0     3) gcc/9.2.0 => gcc/7.3.0
      2) R/4.0 => R/3.5.2
    
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()



```python
import pandas as pd
pd.read_csv("Imputation.1000G.LOMG/4field/GLM/toMeta.RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.tab",sep="\t").head()
```




<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>CHROM</th>
      <th>POS</th>
      <th>ID</th>
      <th>marker</th>
      <th>EffectAllele</th>
      <th>OtherAllele</th>
      <th>OR</th>
      <th>BETA</th>
      <th>SE</th>
      <th>P</th>
      <th>Freq_EffectAllele</th>
      <th>Freq_EffectAllele.CASE</th>
      <th>Freq_EffectAllele.CTRL</th>
      <th>ALLELE_CT</th>
      <th>CASE_ALLELE_CT</th>
      <th>CTRL_ALLELE_CT</th>
      <th>EffectAllele_CT</th>
      <th>EffectAllele_CT.CASE</th>
      <th>EffectAllele_CT.CTRL</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>6</td>
      <td>25006557</td>
      <td>rs1410689</td>
      <td>rs1410689:6:25006557:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>0.985222</td>
      <td>-0.014888</td>
      <td>0.054231</td>
      <td>0.783668</td>
      <td>0.422426</td>
      <td>0.415068</td>
      <td>0.422594</td>
      <td>65202</td>
      <td>1460</td>
      <td>63742</td>
      <td>27543</td>
      <td>606</td>
      <td>26937</td>
    </tr>
    <tr>
      <th>1</th>
      <td>6</td>
      <td>25010771</td>
      <td>rs10946741</td>
      <td>rs10946741:6:25010771:A:G</td>
      <td>G</td>
      <td>A</td>
      <td>0.995367</td>
      <td>-0.004644</td>
      <td>0.053732</td>
      <td>0.931124</td>
      <td>0.445193</td>
      <td>0.441701</td>
      <td>0.445273</td>
      <td>65192</td>
      <td>1458</td>
      <td>63734</td>
      <td>29023</td>
      <td>644</td>
      <td>28379</td>
    </tr>
    <tr>
      <th>2</th>
      <td>6</td>
      <td>25011939</td>
      <td>rs2166681</td>
      <td>rs2166681:6:25011939:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>1.135290</td>
      <td>0.126888</td>
      <td>0.087667</td>
      <td>0.147778</td>
      <td>0.093697</td>
      <td>0.105372</td>
      <td>0.093429</td>
      <td>64826</td>
      <td>1452</td>
      <td>63374</td>
      <td>6074</td>
      <td>153</td>
      <td>5921</td>
    </tr>
    <tr>
      <th>3</th>
      <td>6</td>
      <td>25013892</td>
      <td>rs9467377</td>
      <td>rs9467377:6:25013892:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>0.694350</td>
      <td>-0.364779</td>
      <td>0.174457</td>
      <td>0.036533</td>
      <td>0.034122</td>
      <td>0.024005</td>
      <td>0.034354</td>
      <td>65236</td>
      <td>1458</td>
      <td>63778</td>
      <td>2226</td>
      <td>35</td>
      <td>2191</td>
    </tr>
    <tr>
      <th>4</th>
      <td>6</td>
      <td>25020792</td>
      <td>rs34632338</td>
      <td>rs34632338:6:25020792:G:T</td>
      <td>T</td>
      <td>G</td>
      <td>1.035260</td>
      <td>0.034653</td>
      <td>0.128526</td>
      <td>0.787452</td>
      <td>0.045028</td>
      <td>0.045206</td>
      <td>0.045024</td>
      <td>65182</td>
      <td>1460</td>
      <td>63722</td>
      <td>2935</td>
      <td>66</td>
      <td>2869</td>
    </tr>
  </tbody>
</table>
</div>




```python
import pandas as pd
pd.read_csv("Imputation.1000G.LOMG/4field/GLM.conditional/toMeta.RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.AA.HLA_DRB1.tab",sep="\t").head()
```




<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>CHROM</th>
      <th>POS</th>
      <th>ID</th>
      <th>marker</th>
      <th>EffectAllele</th>
      <th>OtherAllele</th>
      <th>OR</th>
      <th>BETA</th>
      <th>SE</th>
      <th>P</th>
      <th>Freq_EffectAllele</th>
      <th>Freq_EffectAllele.CASE</th>
      <th>Freq_EffectAllele.CTRL</th>
      <th>ALLELE_CT</th>
      <th>CASE_ALLELE_CT</th>
      <th>CTRL_ALLELE_CT</th>
      <th>EffectAllele_CT</th>
      <th>EffectAllele_CT.CASE</th>
      <th>EffectAllele_CT.CTRL</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>6</td>
      <td>25006557</td>
      <td>rs1410689</td>
      <td>rs1410689:6:25006557:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>1.003330</td>
      <td>0.003324</td>
      <td>0.054305</td>
      <td>0.951214</td>
      <td>0.422426</td>
      <td>0.415068</td>
      <td>0.422594</td>
      <td>65202</td>
      <td>1460</td>
      <td>63742</td>
      <td>27543</td>
      <td>606</td>
      <td>26937</td>
    </tr>
    <tr>
      <th>1</th>
      <td>6</td>
      <td>25010771</td>
      <td>rs10946741</td>
      <td>rs10946741:6:25010771:A:G</td>
      <td>G</td>
      <td>A</td>
      <td>1.011390</td>
      <td>0.011326</td>
      <td>0.053775</td>
      <td>0.833125</td>
      <td>0.445193</td>
      <td>0.441701</td>
      <td>0.445273</td>
      <td>65192</td>
      <td>1458</td>
      <td>63734</td>
      <td>29023</td>
      <td>644</td>
      <td>28379</td>
    </tr>
    <tr>
      <th>2</th>
      <td>6</td>
      <td>25011939</td>
      <td>rs2166681</td>
      <td>rs2166681:6:25011939:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>1.135330</td>
      <td>0.126923</td>
      <td>0.087925</td>
      <td>0.148867</td>
      <td>0.093697</td>
      <td>0.105372</td>
      <td>0.093429</td>
      <td>64826</td>
      <td>1452</td>
      <td>63374</td>
      <td>6074</td>
      <td>153</td>
      <td>5921</td>
    </tr>
    <tr>
      <th>3</th>
      <td>6</td>
      <td>25013892</td>
      <td>rs9467377</td>
      <td>rs9467377:6:25013892:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>0.698338</td>
      <td>-0.359052</td>
      <td>0.174673</td>
      <td>0.039824</td>
      <td>0.034122</td>
      <td>0.024005</td>
      <td>0.034354</td>
      <td>65236</td>
      <td>1458</td>
      <td>63778</td>
      <td>2226</td>
      <td>35</td>
      <td>2191</td>
    </tr>
    <tr>
      <th>4</th>
      <td>6</td>
      <td>25020792</td>
      <td>rs34632338</td>
      <td>rs34632338:6:25020792:G:T</td>
      <td>T</td>
      <td>G</td>
      <td>1.048890</td>
      <td>0.047732</td>
      <td>0.129217</td>
      <td>0.711849</td>
      <td>0.045028</td>
      <td>0.045206</td>
      <td>0.045024</td>
      <td>65182</td>
      <td>1460</td>
      <td>63722</td>
      <td>2935</td>
      <td>66</td>
      <td>2869</td>
    </tr>
  </tbody>
</table>
</div>




```python
import pandas as pd
pd.read_csv("Imputation.1000G.LOMG/4field/GLM.conditional/toMeta.RESULT_LOMG.USmerged.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.HLA_DQB1.tab",sep="\t").head()
```




<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>CHROM</th>
      <th>POS</th>
      <th>ID</th>
      <th>marker</th>
      <th>EffectAllele</th>
      <th>OtherAllele</th>
      <th>OR</th>
      <th>BETA</th>
      <th>SE</th>
      <th>P</th>
      <th>Freq_EffectAllele</th>
      <th>Freq_EffectAllele.CASE</th>
      <th>Freq_EffectAllele.CTRL</th>
      <th>ALLELE_CT</th>
      <th>CASE_ALLELE_CT</th>
      <th>CTRL_ALLELE_CT</th>
      <th>EffectAllele_CT</th>
      <th>EffectAllele_CT.CASE</th>
      <th>EffectAllele_CT.CTRL</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>6</td>
      <td>25006557</td>
      <td>rs1410689</td>
      <td>rs1410689:6:25006557:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>0.986060</td>
      <td>-0.014038</td>
      <td>0.054270</td>
      <td>0.795891</td>
      <td>0.422426</td>
      <td>0.415068</td>
      <td>0.422594</td>
      <td>65202</td>
      <td>1460</td>
      <td>63742</td>
      <td>27543</td>
      <td>606</td>
      <td>26937</td>
    </tr>
    <tr>
      <th>1</th>
      <td>6</td>
      <td>25010771</td>
      <td>rs10946741</td>
      <td>rs10946741:6:25010771:A:G</td>
      <td>G</td>
      <td>A</td>
      <td>0.996072</td>
      <td>-0.003936</td>
      <td>0.053743</td>
      <td>0.941627</td>
      <td>0.445193</td>
      <td>0.441701</td>
      <td>0.445273</td>
      <td>65192</td>
      <td>1458</td>
      <td>63734</td>
      <td>29023</td>
      <td>644</td>
      <td>28379</td>
    </tr>
    <tr>
      <th>2</th>
      <td>6</td>
      <td>25011939</td>
      <td>rs2166681</td>
      <td>rs2166681:6:25011939:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>1.137460</td>
      <td>0.128798</td>
      <td>0.087634</td>
      <td>0.141644</td>
      <td>0.093697</td>
      <td>0.105372</td>
      <td>0.093429</td>
      <td>64826</td>
      <td>1452</td>
      <td>63374</td>
      <td>6074</td>
      <td>153</td>
      <td>5921</td>
    </tr>
    <tr>
      <th>3</th>
      <td>6</td>
      <td>25013892</td>
      <td>rs9467377</td>
      <td>rs9467377:6:25013892:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>0.698654</td>
      <td>-0.358600</td>
      <td>0.174685</td>
      <td>0.040088</td>
      <td>0.034122</td>
      <td>0.024005</td>
      <td>0.034354</td>
      <td>65236</td>
      <td>1458</td>
      <td>63778</td>
      <td>2226</td>
      <td>35</td>
      <td>2191</td>
    </tr>
    <tr>
      <th>4</th>
      <td>6</td>
      <td>25020792</td>
      <td>rs34632338</td>
      <td>rs34632338:6:25020792:G:T</td>
      <td>T</td>
      <td>G</td>
      <td>1.038150</td>
      <td>0.037440</td>
      <td>0.128558</td>
      <td>0.770852</td>
      <td>0.045028</td>
      <td>0.045206</td>
      <td>0.045024</td>
      <td>65182</td>
      <td>1460</td>
      <td>63722</td>
      <td>2935</td>
      <td>66</td>
      <td>2869</td>
    </tr>
  </tbody>
</table>
</div>


