# MG Itals HLA imputation using CookHLA

**Start date:** 04-30-2021

**End date:** 04-30-2021

**Analysed by:** Ruth Chia

**Working directory:** `/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA`

___


<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#What-I-need-to-do-here:" data-toc-modified-id="What-I-need-to-do-here:-1">What I need to do here:</a></span></li><li><span><a href="#Code-to-use-for-HLA-imputation" data-toc-modified-id="Code-to-use-for-HLA-imputation-2">Code to use for HLA imputation</a></span></li><li><span><a href="#Download-code-for-CookHLA" data-toc-modified-id="Download-code-for-CookHLA-3">Download code for CookHLA</a></span></li><li><span><a href="#Generate-chr6-plink-file-for-HLA-imputation" data-toc-modified-id="Generate-chr6-plink-file-for-HLA-imputation-4">Generate chr6 plink file for HLA imputation</a></span></li><li><span><a href="#Generate-chr6-plink-file-for-HLA-imputation-(using-hg38-positions-of-genotyped-data)" data-toc-modified-id="Generate-chr6-plink-file-for-HLA-imputation-(using-hg38-positions-of-genotyped-data)-5">Generate chr6 plink file for HLA imputation (using hg38 positions of genotyped data)</a></span><ul class="toc-item"><li><span><a href="#Liftover-from-b37-to-b38" data-toc-modified-id="Liftover-from-b37-to-b38-5.1">Liftover from b37 to b38</a></span></li></ul></li><li><span><a href="#Run-imputation" data-toc-modified-id="Run-imputation-6">Run imputation</a></span></li><li><span><a href="#Download-code-for-HATK" data-toc-modified-id="Download-code-for-HATK-7">Download code for HATK</a></span></li><li><span><a href="#Run-HATK" data-toc-modified-id="Run-HATK-8">Run HATK</a></span><ul class="toc-item"><li><span><a href="#Imputation.1000G" data-toc-modified-id="Imputation.1000G-8.1">Imputation.1000G</a></span><ul class="toc-item"><li><span><a href="#Overall" data-toc-modified-id="Overall-8.1.1">Overall</a></span><ul class="toc-item"><li><span><a href="#plink-version-association---4field" data-toc-modified-id="plink-version-association---4field-8.1.1.1">plink version association - 4field</a></span></li><li><span><a href="#plink-version-conditional-association---4field" data-toc-modified-id="plink-version-conditional-association---4field-8.1.1.2">plink version conditional association - 4field</a></span></li></ul></li><li><span><a href="#EOMG" data-toc-modified-id="EOMG-8.1.2">EOMG</a></span><ul class="toc-item"><li><span><a href="#plink-version-association---4field" data-toc-modified-id="plink-version-association---4field-8.1.2.1">plink version association - 4field</a></span></li></ul></li><li><span><a href="#LOMG" data-toc-modified-id="LOMG-8.1.3">LOMG</a></span><ul class="toc-item"><li><span><a href="#plink-version-association---4field" data-toc-modified-id="plink-version-association---4field-8.1.3.1">plink version association - 4field</a></span></li><li><span><a href="#plink-version-conditional-association---4field" data-toc-modified-id="plink-version-conditional-association---4field-8.1.3.2">plink version conditional association - 4field</a></span></li></ul></li></ul></li></ul></li></ul></div>

## What I need to do here:
1. Plink1 files as input only for chr6 (can include related samples but no duplicates). To subset from:
    `/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/CLEAN.rawGenotype.keepRelated/FILTERED.Itals_mg_noDups.keepRelated.hwe1e-10.geno005`
    
## Code to use for HLA imputation

Will use the latest improved tool to do this i.e. [CookHLA](https://github.com/WansonChoi/CookHLA)

Snippet from the website:

```
CookHLA is an accurate and efficient HLA imputation method.

Human leukocyte antigen (HLA) genes in the major histocompatibility complex (MHC) region influence many immune-related disease phenotypes. Discovering which HLA alleles or amino acids drive phenotypes often requires fine-mapping analysis in tens of thousands of samples, of which HLA typing can be prohibitively costly. Fortunately, the recent development of imputation methods enabled the prediction of HLA alleles using intergenic SNP data, allowing many studies to fine-map HLA for immune phenotypes. Nonetheless, existing imputation methods still have limitations in accuracy.

CookHLA substantially improved imputation accuracy over previous methods, including the predecessor SNP2HLA by implementing several changes. First, to increase accuracy and efficiency, we employed the recently released hidden Markov model (Beagle v4 and v5). Second, to further improve accuracy, we developed a procedure that can account for local variability in the exons of HLA genes. Highly polymorphic exons (exon 2/3/4 in Class I and 2/3 in Class II) are critical regions whose sequences can determine most of the 4-digit alleles. In our method, we repeat imputation by putting the marker set locally in each of the highly polymorphic exons, and use consensus posterior probabilities from the repeated analyses for final predictions(Figure B). Third, to increase accuracy even further, we adaptively learn the genetic map of MHC from the data. This map information allows us to account for the data-specific LD structure within MHC, which improves the imputation accuracy compared to the use of publicly available genetic maps(Figure C).
```
![image](https://github.com/WansonChoi/CookHLA/blob/master/img/Figure1.png?raw=true)


## Download code for CookHLA


```python
!git clone https://github.com/WansonChoi/CookHLA.git
```

    Cloning into 'CookHLA'...
    remote: Enumerating objects: 1369, done.[K
    remote: Counting objects: 100% (112/112), done.[K
    remote: Compressing objects: 100% (78/78), done.[K
    remote: Total 1369 (delta 65), reused 71 (delta 34), pack-reused 1257[K
    Receiving objects: 100% (1369/1369), 28.77 MiB | 14.31 MiB/s, done.
    Resolving deltas: 100% (869/869), done.
    Checking out files: 100% (113/113), done.



```bash
%%bash
cd CookHLA/
source /data/chiarp/conda/etc/profile.d/conda.sh
#conda env create -f CookHLA_LINUX.yml
conda activate CookHLA
```

## Generate chr6 plink file for HLA imputation

keep file small to contain only: chr6:25-38Mb (this is larger than the HLA/MHC region so that there is some buffer on either side).

The code does something weird at some point where it assumes that the FID is the same as IID. This is not the case. So, to keep things consistent with what the code expects, I will update the FID and IID so that the FID will be the same as IID


```bash
%%bash
module load plink/1.9.0-beta4.4

# create sample update name
awk '{print $1,$2,$2,$2}' /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/CLEAN.rawGenotype.keepRelated/FILTERED.Itals_mg_noDups.keepRelated.hwe1e-10.geno005.fam > SampleNameUpdate.txt

plink \
--bfile /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/CLEAN.rawGenotype.keepRelated/FILTERED.Itals_mg_noDups.keepRelated.hwe1e-10.geno005 \
--update-ids SampleNameUpdate.txt \
--allow-no-sex \
--chr 6 \
--from-bp 25000000 \
--to-bp 38000000 \
--keep-allele-order \
--make-bed \
--out Itals.HLAregion.chr6

```

    PLINK v1.90b4.4 64-bit (21 May 2017)           www.cog-genomics.org/plink/1.9/
    (C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3
    Logging to Itals.HLAregion.chr6.log.
    Options in effect:
      --allow-no-sex
      --bfile /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/CLEAN.rawGenotype.keepRelated/FILTERED.Itals_mg_noDups.keepRelated.hwe1e-10.geno005
      --chr 6
      --from-bp 25000000
      --keep-allele-order
      --make-bed
      --out Itals.HLAregion.chr6
      --to-bp 38000000
      --update-ids SampleNameUpdate.txt
    
    257652 MB RAM detected; reserving 128826 MB for main workspace.
    6293 out of 523703 variants loaded from .bim file.
    3497 people (1914 males, 1583 females) loaded from .fam.
    3497 phenotype values loaded from .fam.
    --update-ids: 3497 people updated.
    Using 1 thread (no multithreaded calculations invoked).
    Before main variant filters, 3497 founders and 0 nonfounders present.
    Calculating allele frequencies... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99% done.
    Total genotyping rate is 0.998061.
    6293 variants and 3497 people pass filters and QC.
    Among remaining phenotypes, 924 are cases and 2573 are controls.
    --make-bed to Itals.HLAregion.chr6.bed + Itals.HLAregion.chr6.bim +
    Itals.HLAregion.chr6.fam ... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99%done.


    [-] Unloading plink  1.9.0-beta4.4  on cn3174 
    [+] Loading plink  1.9.0-beta4.4  on cn3174 


## Generate chr6 plink file for HLA imputation (using hg38 positions of genotyped data)

### Liftover from b37 to b38



```bash
%%bash
## create input file for liftover of SNPs from b36 to b37
module load R/4.0
R --vanilla --no-save

require("data.table")
bim <- fread("Itals.HLAregion.chr6.bim", header=F)
bed <- bim[,c(1,4,2)]
bed$V3 <- bed$V4 + 1
bed <- bed[,c(1,2,4,3)]
bed$V1 <- paste("chr",bed$V1,sep="")
options(scipen=999, justified="none")
write.table(bed,"Itals.HLAregion.b37.bed", col.names=F,row.names=F,quote=F,sep="\t")

```

    
    R version 4.0.5 (2021-03-31) -- "Shake and Throw"
    Copyright (C) 2021 The R Foundation for Statistical Computing
    Platform: x86_64-pc-linux-gnu (64-bit)
    
    R is free software and comes with ABSOLUTELY NO WARRANTY.
    You are welcome to redistribute it under certain conditions.
    Type 'license()' or 'licence()' for distribution details.
    
    R is a collaborative project with many contributors.
    Type 'contributors()' for more information and
    'citation()' on how to cite R or R packages in publications.
    
    Type 'demo()' for some demos, 'help()' for on-line help, or
    'help.start()' for an HTML browser interface to help.
    Type 'q()' to quit R.
    
    > 
    > require("data.table")
    > bim <- fread("Itals.HLAregion.chr6.bim", header=F)
    > bed <- bim[,c(1,4,2)]
    > bed$V3 <- bed$V4 + 1
    > bed <- bed[,c(1,2,4,3)]
    > bed$V1 <- paste("chr",bed$V1,sep="")
    > options(scipen=999, justified="none")
    > write.table(bed,"Itals.HLAregion.b37.bed", col.names=F,row.names=F,quote=F,sep="\t")
    > 


    [-] Unloading gcc  9.2.0  ... 
    [-] Unloading GSL 2.6 for GCC 9.2.0 ... 
    [-] Unloading openmpi 4.0.5  for GCC 9.2.0 
    [-] Unloading ImageMagick  7.0.8  on cn3174 
    [-] Unloading HDF5  1.10.4 
    [-] Unloading NetCDF 4.7.4_gcc9.2.0 
    [-] Unloading pandoc  2.13  on cn3174 
    [-] Unloading pcre2 10.21  ... 
    [-] Unloading R 4.0.5 
    [+] Loading gcc  9.2.0  ... 
    [+] Loading GSL 2.6 for GCC 9.2.0 ... 
    [-] Unloading gcc  9.2.0  ... 
    [+] Loading gcc  9.2.0  ... 
    [+] Loading openmpi 4.0.5  for GCC 9.2.0 
    [+] Loading ImageMagick  7.0.8  on cn3174 
    [+] Loading HDF5  1.10.4 
    [-] Unloading gcc  9.2.0  ... 
    [+] Loading gcc  9.2.0  ... 
    [+] Loading NetCDF 4.7.4_gcc9.2.0 
    [+] Loading pandoc  2.13  on cn3174 
    [+] Loading pcre2 10.21  ... 
    [+] Loading R 4.0.5 
    Loading required package: data.table



```bash
%%bash

## Download binary tools for liftover
mkdir LiftOver
wget http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/liftOver -O LiftOver/liftOver
wget --timestamping 'ftp://hgdownload.soe.ucsc.edu/goldenPath/hg19/liftOver/hg19ToHg38.over.chain.gz' -O LiftOver/hg19ToHg38.over.chain.gz

chmod u+x LiftOver/liftOver
LiftOver/liftOver Itals.HLAregion.b37.bed LiftOver/hg19ToHg38.over.chain.gz Itals.HLAregion.b38.bed unlifted.bed

grep -v "#Deleted" unlifted.bed | awk '{print $4}' > SNPs_not_in_b38.txt
awk '{print $4,$2}' Itals.HLAregion.b38.bed > Itals.HLAregion_update-map_to_b38.txt

# Update snp position
module load plink/1.9.0-beta4.4
plink \
--bfile Itals.HLAregion.chr6 \
--update-map Itals.HLAregion_update-map_to_b38.txt \
--exclude SNPs_not_in_b38.txt \
--make-bed \
--out Itals.HLAregion.chr6.Liftover.hg38

```

    PLINK v1.90b4.4 64-bit (21 May 2017)           www.cog-genomics.org/plink/1.9/
    (C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3
    Logging to Itals.HLAregion.chr6.Liftover.hg38.log.
    Options in effect:
      --bfile Itals.HLAregion.chr6
      --exclude SNPs_not_in_b38.txt
      --make-bed
      --out Itals.HLAregion.chr6.Liftover.hg38
      --update-map Itals.HLAregion_update-map_to_b38.txt
    
    257652 MB RAM detected; reserving 128826 MB for main workspace.
    6293 variants loaded from .bim file.
    3497 people (1914 males, 1583 females) loaded from .fam.
    3497 phenotype values loaded from .fam.
    --update-map: 6293 values updated.
    --exclude: 6293 variants remaining.
    Using 1 thread (no multithreaded calculations invoked).
    Before main variant filters, 3497 founders and 0 nonfounders present.
    Calculating allele frequencies... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99% done.
    Total genotyping rate is 0.998061.
    6293 variants and 3497 people pass filters and QC.
    Among remaining phenotypes, 924 are cases and 2573 are controls.
    --make-bed to Itals.HLAregion.chr6.Liftover.hg38.bed +
    Itals.HLAregion.chr6.Liftover.hg38.bim + Itals.HLAregion.chr6.Liftover.hg38.fam
    ... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99%done.


    --2021-05-03 23:13:47--  http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/liftOver
    Resolving dtn05-e0 (dtn05-e0)... 10.1.200.241
    Connecting to dtn05-e0 (dtn05-e0)|10.1.200.241|:3128... connected.
    Proxy request sent, awaiting response... 200 OK
    Length: 37475560 (36M)
    Saving to: 'LiftOver/liftOver'
    
         0K .......... .......... .......... .......... ..........  0%  322K 1m54s
        50K .......... .......... .......... .......... ..........  0%  738K 81s
       100K .......... .......... .......... .......... ..........  0% 79.4M 54s
       150K .......... .......... .......... .......... ..........  0% 73.9M 41s
       200K .......... .......... .......... .......... ..........  0%  758K 42s
       250K .......... .......... .......... .......... ..........  0% 94.0M 35s
       300K .......... .......... .......... .......... ..........  0%  759K 37s
       350K .......... .......... .......... .......... ..........  1%  113M 32s
       400K .......... .......... .......... .......... ..........  1%  141M 29s
       450K .......... .......... .......... .......... ..........  1%  115M 26s
       500K .......... .......... .......... .......... ..........  1%  176M 23s
       550K .......... .......... .......... .......... ..........  1%  759K 25s
       600K .......... .......... .......... .......... ..........  1%  191M 23s
       650K .......... .......... .......... .......... ..........  1%  229M 22s
       700K .......... .......... .......... .......... ..........  2%  190M 20s
       750K .......... .......... .......... .......... ..........  2%  222M 19s
       800K .......... .......... .......... .......... ..........  2%  186M 18s
       850K .......... .......... .......... .......... ..........  2%  234M 17s
       900K .......... .......... .......... .......... ..........  2%  184M 16s
       950K .......... .......... .......... .......... ..........  2%  186M 15s
      1000K .......... .......... .......... .......... ..........  2%  203M 14s
      1050K .......... .......... .......... .......... ..........  3%  231M 14s
      1100K .......... .......... .......... .......... ..........  3%  785K 15s
      1150K .......... .......... .......... .......... ..........  3% 35.3M 14s
      1200K .......... .......... .......... .......... ..........  3% 56.6M 14s
      1250K .......... .......... .......... .......... ..........  3% 81.0M 13s
      1300K .......... .......... .......... .......... ..........  3% 88.3M 13s
      1350K .......... .......... .......... .......... ..........  3% 81.8M 12s
      1400K .......... .......... .......... .......... ..........  3% 92.1M 12s
      1450K .......... .......... .......... .......... ..........  4% 73.9M 12s
      1500K .......... .......... .......... .......... ..........  4% 92.1M 11s
      1550K .......... .......... .......... .......... ..........  4%  815K 12s
      1600K .......... .......... .......... .......... ..........  4% 29.1M 12s
      1650K .......... .......... .......... .......... ..........  4% 81.7M 11s
      1700K .......... .......... .......... .......... ..........  4%  173M 11s
      1750K .......... .......... .......... .......... ..........  4%  215M 11s
      1800K .......... .......... .......... .......... ..........  5%  183M 11s
      1850K .......... .......... .......... .......... ..........  5%  210M 10s
      1900K .......... .......... .......... .......... ..........  5%  189M 10s
      1950K .......... .......... .......... .......... ..........  5%  191M 10s
      2000K .......... .......... .......... .......... ..........  5%  208M 9s
      2050K .......... .......... .......... .......... ..........  5%  198M 9s
      2100K .......... .......... .......... .......... ..........  5%  211M 9s
      2150K .......... .......... .......... .......... ..........  6%  204M 9s
      2200K .......... .......... .......... .......... ..........  6%  232M 9s
      2250K .......... .......... .......... .......... ..........  6%  193M 8s
      2300K .......... .......... .......... .......... ..........  6%  158M 8s
      2350K .......... .......... .......... .......... ..........  6%  222M 8s
      2400K .......... .......... .......... .......... ..........  6%  134M 8s
      2450K .......... .......... .......... .......... ..........  6%  127M 8s
      2500K .......... .......... .......... .......... ..........  6%  198M 8s
      2550K .......... .......... .......... .......... ..........  7% 85.4M 7s
      2600K .......... .......... .......... .......... ..........  7%  116M 7s
      2650K .......... .......... .......... .......... ..........  7%  127M 7s
      2700K .......... .......... .......... .......... ..........  7%  121M 7s
      2750K .......... .......... .......... .......... ..........  7%  121M 7s
      2800K .......... .......... .......... .......... ..........  7%  106M 7s
      2850K .......... .......... .......... .......... ..........  7% 80.5M 7s
      2900K .......... .......... .......... .......... ..........  8%  122M 6s
      2950K .......... .......... .......... .......... ..........  8% 39.0M 6s
      3000K .......... .......... .......... .......... ..........  8%  115M 6s
      3050K .......... .......... .......... .......... ..........  8%  128M 6s
      3100K .......... .......... .......... .......... ..........  8%  897K 7s
      3150K .......... .......... .......... .......... ..........  8%  123M 7s
      3200K .......... .......... .......... .......... ..........  8% 32.3M 6s
      3250K .......... .......... .......... .......... ..........  9%  118M 6s
      3300K .......... .......... .......... .......... ..........  9% 67.4M 6s
      3350K .......... .......... .......... .......... ..........  9%  221M 6s
      3400K .......... .......... .......... .......... ..........  9% 81.9M 6s
      3450K .......... .......... .......... .......... ..........  9%  206M 6s
      3500K .......... .......... .......... .......... ..........  9%  220M 6s
      3550K .......... .......... .......... .......... ..........  9%  217M 6s
      3600K .......... .......... .......... .......... ..........  9%  214M 6s
      3650K .......... .......... .......... .......... .......... 10%  212M 6s
      3700K .......... .......... .......... .......... .......... 10%  199M 6s
      3750K .......... .......... .......... .......... .......... 10%  223M 5s
      3800K .......... .......... .......... .......... .......... 10%  199M 5s
      3850K .......... .......... .......... .......... .......... 10%  228M 5s
      3900K .......... .......... .......... .......... .......... 10%  189M 5s
      3950K .......... .......... .......... .......... .......... 10%  188M 5s
      4000K .......... .......... .......... .......... .......... 11% 98.2M 5s
      4050K .......... .......... .......... .......... .......... 11%  213M 5s
      4100K .......... .......... .......... .......... .......... 11%  118M 5s
      4150K .......... .......... .......... .......... .......... 11% 74.7M 5s
      4200K .......... .......... .......... .......... .......... 11% 58.8M 5s
      4250K .......... .......... .......... .......... .......... 11% 55.2M 5s
      4300K .......... .......... .......... .......... .......... 11%  113M 5s
      4350K .......... .......... .......... .......... .......... 12% 71.8M 5s
      4400K .......... .......... .......... .......... .......... 12% 54.9M 5s
      4450K .......... .......... .......... .......... .......... 12% 41.5M 5s
      4500K .......... .......... .......... .......... .......... 12%  928K 5s
      4550K .......... .......... .......... .......... .......... 12% 54.6M 5s
      4600K .......... .......... .......... .......... .......... 12% 69.2M 5s
      4650K .......... .......... .......... .......... .......... 12% 71.1M 5s
      4700K .......... .......... .......... .......... .......... 12% 48.8M 5s
      4750K .......... .......... .......... .......... .......... 13% 66.2M 5s
      4800K .......... .......... .......... .......... .......... 13% 76.8M 5s
      4850K .......... .......... .......... .......... .......... 13% 73.1M 5s
      4900K .......... .......... .......... .......... .......... 13% 44.6M 4s
      4950K .......... .......... .......... .......... .......... 13%  137M 4s
      5000K .......... .......... .......... .......... .......... 13%  149M 4s
      5050K .......... .......... .......... .......... .......... 13%  142M 4s
      5100K .......... .......... .......... .......... .......... 14% 75.7M 4s
      5150K .......... .......... .......... .......... .......... 14%  149M 4s
      5200K .......... .......... .......... .......... .......... 14%  156M 4s
      5250K .......... .......... .......... .......... .......... 14%  141M 4s
      5300K .......... .......... .......... .......... .......... 14%  150M 4s
      5350K .......... .......... .......... .......... .......... 14%  138M 4s
      5400K .......... .......... .......... .......... .......... 14%  137M 4s
      5450K .......... .......... .......... .......... .......... 15%  157M 4s
      5500K .......... .......... .......... .......... .......... 15%  164M 4s
      5550K .......... .......... .......... .......... .......... 15%  141M 4s
      5600K .......... .......... .......... .......... .......... 15%  165M 4s
      5650K .......... .......... .......... .......... .......... 15%  148M 4s
      5700K .......... .......... .......... .......... .......... 15%  158M 4s
      5750K .......... .......... .......... .......... .......... 15%  138M 4s
      5800K .......... .......... .......... .......... .......... 15%  153M 4s
      5850K .......... .......... .......... .......... .......... 16%  161M 4s
      5900K .......... .......... .......... .......... .......... 16%  128M 4s
      5950K .......... .......... .......... .......... .......... 16%  143M 4s
      6000K .......... .......... .......... .......... .......... 16%  159M 4s
      6050K .......... .......... .......... .......... .......... 16%  145M 4s
      6100K .......... .......... .......... .......... .......... 16%  149M 4s
      6150K .......... .......... .......... .......... .......... 16%  148M 3s
      6200K .......... .......... .......... .......... .......... 17%  144M 3s
      6250K .......... .......... .......... .......... .......... 17%  139M 3s
      6300K .......... .......... .......... .......... .......... 17%  146M 3s
      6350K .......... .......... .......... .......... .......... 17%  150M 3s
      6400K .......... .......... .......... .......... .......... 17%  130M 3s
      6450K .......... .......... .......... .......... .......... 17%  155M 3s
      6500K .......... .......... .......... .......... .......... 17%  164M 3s
      6550K .......... .......... .......... .......... .......... 18% 77.3M 3s
      6600K .......... .......... .......... .......... .......... 18%  163M 3s
      6650K .......... .......... .......... .......... .......... 18%  133M 3s
      6700K .......... .......... .......... .......... .......... 18%  161M 3s
      6750K .......... .......... .......... .......... .......... 18%  141M 3s
      6800K .......... .......... .......... .......... .......... 18% 1.05M 3s
      6850K .......... .......... .......... .......... .......... 18% 48.0M 3s
      6900K .......... .......... .......... .......... .......... 18% 51.6M 3s
      6950K .......... .......... .......... .......... .......... 19% 93.1M 3s
      7000K .......... .......... .......... .......... .......... 19% 54.7M 3s
      7050K .......... .......... .......... .......... .......... 19% 88.2M 3s
      7100K .......... .......... .......... .......... .......... 19% 85.7M 3s
      7150K .......... .......... .......... .......... .......... 19% 87.7M 3s
      7200K .......... .......... .......... .......... .......... 19% 84.3M 3s
      7250K .......... .......... .......... .......... .......... 19% 92.8M 3s
      7300K .......... .......... .......... .......... .......... 20% 81.3M 3s
      7350K .......... .......... .......... .......... .......... 20% 81.6M 3s
      7400K .......... .......... .......... .......... .......... 20% 86.9M 3s
      7450K .......... .......... .......... .......... .......... 20% 97.3M 3s
      7500K .......... .......... .......... .......... .......... 20% 89.2M 3s
      7550K .......... .......... .......... .......... .......... 20% 95.4M 3s
      7600K .......... .......... .......... .......... .......... 20% 65.1M 3s
      7650K .......... .......... .......... .......... .......... 21%  138M 3s
      7700K .......... .......... .......... .......... .......... 21%  124M 3s
      7750K .......... .......... .......... .......... .......... 21%  153M 3s
      7800K .......... .......... .......... .......... .......... 21% 72.9M 3s
      7850K .......... .......... .......... .......... .......... 21%  124M 3s
      7900K .......... .......... .......... .......... .......... 21%  139M 3s
      7950K .......... .......... .......... .......... .......... 21%  166M 3s
      8000K .......... .......... .......... .......... .......... 21%  150M 3s
      8050K .......... .......... .......... .......... .......... 22%  152M 3s
      8100K .......... .......... .......... .......... .......... 22%  137M 3s
      8150K .......... .......... .......... .......... .......... 22%  100M 3s
      8200K .......... .......... .......... .......... .......... 22%  105M 3s
      8250K .......... .......... .......... .......... .......... 22%  152M 3s
      8300K .......... .......... .......... .......... .......... 22%  140M 3s
      8350K .......... .......... .......... .......... .......... 22%  124M 3s
      8400K .......... .......... .......... .......... .......... 23%  146M 3s
      8450K .......... .......... .......... .......... .......... 23%  176M 3s
      8500K .......... .......... .......... .......... .......... 23%  143M 3s
      8550K .......... .......... .......... .......... .......... 23%  157M 3s
      8600K .......... .......... .......... .......... .......... 23%  142M 3s
      8650K .......... .......... .......... .......... .......... 23% 78.7M 2s
      8700K .......... .......... .......... .......... .......... 23%  139M 2s
      8750K .......... .......... .......... .......... .......... 24%  155M 2s
      8800K .......... .......... .......... .......... .......... 24%  139M 2s
      8850K .......... .......... .......... .......... .......... 24% 71.0M 2s
      8900K .......... .......... .......... .......... .......... 24%  134M 2s
      8950K .......... .......... .......... .......... .......... 24% 54.5M 2s
      9000K .......... .......... .......... .......... .......... 24%  139M 2s
      9050K .......... .......... .......... .......... .......... 24%  146M 2s
      9100K .......... .......... .......... .......... .......... 25%  147M 2s
      9150K .......... .......... .......... .......... .......... 25%  155M 2s
      9200K .......... .......... .......... .......... .......... 25%  137M 2s
      9250K .......... .......... .......... .......... .......... 25%  141M 2s
      9300K .......... .......... .......... .......... .......... 25%  134M 2s
      9350K .......... .......... .......... .......... .......... 25%  126M 2s
      9400K .......... .......... .......... .......... .......... 25%  150M 2s
      9450K .......... .......... .......... .......... .......... 25%  146M 2s
      9500K .......... .......... .......... .......... .......... 26% 78.0M 2s
      9550K .......... .......... .......... .......... .......... 26%  158M 2s
      9600K .......... .......... .......... .......... .......... 26%  144M 2s
      9650K .......... .......... .......... .......... .......... 26%  151M 2s
      9700K .......... .......... .......... .......... .......... 26%  132M 2s
      9750K .......... .......... .......... .......... .......... 26%  150M 2s
      9800K .......... .......... .......... .......... .......... 26%  136M 2s
      9850K .......... .......... .......... .......... .......... 27%  125M 2s
      9900K .......... .......... .......... .......... .......... 27%  147M 2s
      9950K .......... .......... .......... .......... .......... 27% 1.21M 2s
     10000K .......... .......... .......... .......... .......... 27% 75.4M 2s
     10050K .......... .......... .......... .......... .......... 27% 78.9M 2s
     10100K .......... .......... .......... .......... .......... 27% 65.8M 2s
     10150K .......... .......... .......... .......... .......... 27%  131M 2s
     10200K .......... .......... .......... .......... .......... 28%  123M 2s
     10250K .......... .......... .......... .......... .......... 28%  138M 2s
     10300K .......... .......... .......... .......... .......... 28% 72.4M 2s
     10350K .......... .......... .......... .......... .......... 28%  121M 2s
     10400K .......... .......... .......... .......... .......... 28%  138M 2s
     10450K .......... .......... .......... .......... .......... 28%  130M 2s
     10500K .......... .......... .......... .......... .......... 28%  142M 2s
     10550K .......... .......... .......... .......... .......... 28% 72.2M 2s
     10600K .......... .......... .......... .......... .......... 29%  152M 2s
     10650K .......... .......... .......... .......... .......... 29%  133M 2s
     10700K .......... .......... .......... .......... .......... 29%  156M 2s
     10750K .......... .......... .......... .......... .......... 29%  138M 2s
     10800K .......... .......... .......... .......... .......... 29%  144M 2s
     10850K .......... .......... .......... .......... .......... 29%  125M 2s
     10900K .......... .......... .......... .......... .......... 29%  151M 2s
     10950K .......... .......... .......... .......... .......... 30%  158M 2s
     11000K .......... .......... .......... .......... .......... 30%  156M 2s
     11050K .......... .......... .......... .......... .......... 30%  129M 2s
     11100K .......... .......... .......... .......... .......... 30%  149M 2s
     11150K .......... .......... .......... .......... .......... 30%  141M 2s
     11200K .......... .......... .......... .......... .......... 30%  149M 2s
     11250K .......... .......... .......... .......... .......... 30%  130M 2s
     11300K .......... .......... .......... .......... .......... 31%  150M 2s
     11350K .......... .......... .......... .......... .......... 31% 67.9M 2s
     11400K .......... .......... .......... .......... .......... 31%  123M 2s
     11450K .......... .......... .......... .......... .......... 31%  114M 2s
     11500K .......... .......... .......... .......... .......... 31% 91.2M 2s
     11550K .......... .......... .......... .......... .......... 31%  134M 2s
     11600K .......... .......... .......... .......... .......... 31%  196M 2s
     11650K .......... .......... .......... .......... .......... 31%  242M 2s
     11700K .......... .......... .......... .......... .......... 32%  259M 2s
     11750K .......... .......... .......... .......... .......... 32%  248M 2s
     11800K .......... .......... .......... .......... .......... 32%  163M 2s
     11850K .......... .......... .......... .......... .......... 32%  102M 2s
     11900K .......... .......... .......... .......... .......... 32%  126M 2s
     11950K .......... .......... .......... .......... .......... 32%  116M 2s
     12000K .......... .......... .......... .......... .......... 32%  127M 2s
     12050K .......... .......... .......... .......... .......... 33%  118M 2s
     12100K .......... .......... .......... .......... .......... 33%  132M 2s
     12150K .......... .......... .......... .......... .......... 33%  117M 2s
     12200K .......... .......... .......... .......... .......... 33%  118M 2s
     12250K .......... .......... .......... .......... .......... 33%  248K 2s
     12300K .......... .......... .......... .......... .......... 33%  275M 2s
     12350K .......... .......... .......... .......... .......... 33%  243M 2s
     12400K .......... .......... .......... .......... .......... 34%  222M 2s
     12450K .......... .......... .......... .......... .......... 34%  222M 2s
     12500K .......... .......... .......... .......... .......... 34%  269M 2s
     12550K .......... .......... .......... .......... .......... 34%  285M 2s
     12600K .......... .......... .......... .......... .......... 34%  268M 2s
     12650K .......... .......... .......... .......... .......... 34%  240M 2s
     12700K .......... .......... .......... .......... .......... 34%  288M 2s
     12750K .......... .......... .......... .......... .......... 34% 55.1M 2s
     12800K .......... .......... .......... .......... .......... 35% 43.3M 2s
     12850K .......... .......... .......... .......... .......... 35% 65.1M 2s
     12900K .......... .......... .......... .......... .......... 35% 70.5M 2s
     12950K .......... .......... .......... .......... .......... 35% 43.7M 2s
     13000K .......... .......... .......... .......... .......... 35% 85.9M 2s
     13050K .......... .......... .......... .......... .......... 35% 65.5M 2s
     13100K .......... .......... .......... .......... .......... 35% 70.3M 2s
     13150K .......... .......... .......... .......... .......... 36% 65.7M 2s
     13200K .......... .......... .......... .......... .......... 36% 73.9M 2s
     13250K .......... .......... .......... .......... .......... 36% 66.6M 2s
     13300K .......... .......... .......... .......... .......... 36% 60.5M 2s
     13350K .......... .......... .......... .......... .......... 36% 38.2M 2s
     13400K .......... .......... .......... .......... .......... 36%  123M 2s
     13450K .......... .......... .......... .......... .......... 36%  110M 2s
     13500K .......... .......... .......... .......... .......... 37%  130M 2s
     13550K .......... .......... .......... .......... .......... 37%  112M 2s
     13600K .......... .......... .......... .......... .......... 37%  139M 2s
     13650K .......... .......... .......... .......... .......... 37%  114M 2s
     13700K .......... .......... .......... .......... .......... 37%  126M 2s
     13750K .......... .......... .......... .......... .......... 37%  122M 2s
     13800K .......... .......... .......... .......... .......... 37%  113M 2s
     13850K .......... .......... .......... .......... .......... 37%  112M 2s
     13900K .......... .......... .......... .......... .......... 38%  130M 2s
     13950K .......... .......... .......... .......... .......... 38%  118M 2s
     14000K .......... .......... .......... .......... .......... 38%  134M 2s
     14050K .......... .......... .......... .......... .......... 38%  121M 2s
     14100K .......... .......... .......... .......... .......... 38%  131M 2s
     14150K .......... .......... .......... .......... .......... 38%  120M 2s
     14200K .......... .......... .......... .......... .......... 38%  132M 2s
     14250K .......... .......... .......... .......... .......... 39%  109M 2s
     14300K .......... .......... .......... .......... .......... 39%  113M 2s
     14350K .......... .......... .......... .......... .......... 39%  116M 2s
     14400K .......... .......... .......... .......... .......... 39%  134M 2s
     14450K .......... .......... .......... .......... .......... 39%  122M 2s
     14500K .......... .......... .......... .......... .......... 39%  131M 2s
     14550K .......... .......... .......... .......... .......... 39% 67.8M 2s
     14600K .......... .......... .......... .......... .......... 40%  131M 2s
     14650K .......... .......... .......... .......... .......... 40% 66.9M 2s
     14700K .......... .......... .......... .......... .......... 40%  124M 2s
     14750K .......... .......... .......... .......... .......... 40%  115M 2s
     14800K .......... .......... .......... .......... .......... 40%  113M 2s
     14850K .......... .......... .......... .......... .......... 40%  124M 2s
     14900K .......... .......... .......... .......... .......... 40%  130M 2s
     14950K .......... .......... .......... .......... .......... 40% 69.7M 2s
     15000K .......... .......... .......... .......... .......... 41%  126M 2s
     15050K .......... .......... .......... .......... .......... 41%  100M 2s
     15100K .......... .......... .......... .......... .......... 41%  119M 2s
     15150K .......... .......... .......... .......... .......... 41%  111M 2s
     15200K .......... .......... .......... .......... .......... 41%  119M 2s
     15250K .......... .......... .......... .......... .......... 41%  109M 1s
     15300K .......... .......... .......... .......... .......... 41%  103M 1s
     15350K .......... .......... .......... .......... .......... 42%  115M 1s
     15400K .......... .......... .......... .......... .......... 42%  120M 1s
     15450K .......... .......... .......... .......... .......... 42%  102M 1s
     15500K .......... .......... .......... .......... .......... 42%  122M 1s
     15550K .......... .......... .......... .......... .......... 42%  114M 1s
     15600K .......... .......... .......... .......... .......... 42% 67.9M 1s
     15650K .......... .......... .......... .......... .......... 42%  116M 1s
     15700K .......... .......... .......... .......... .......... 43%  124M 1s
     15750K .......... .......... .......... .......... .......... 43%  119M 1s
     15800K .......... .......... .......... .......... .......... 43%  108M 1s
     15850K .......... .......... .......... .......... .......... 43%  113M 1s
     15900K .......... .......... .......... .......... .......... 43%  131M 1s
     15950K .......... .......... .......... .......... .......... 43% 68.6M 1s
     16000K .......... .......... .......... .......... .......... 43%  129M 1s
     16050K .......... .......... .......... .......... .......... 43%  116M 1s
     16100K .......... .......... .......... .......... .......... 44%  124M 1s
     16150K .......... .......... .......... .......... .......... 44%  117M 1s
     16200K .......... .......... .......... .......... .......... 44%  126M 1s
     16250K .......... .......... .......... .......... .......... 44%  105M 1s
     16300K .......... .......... .......... .......... .......... 44%  112M 1s
     16350K .......... .......... .......... .......... .......... 44% 99.2M 1s
     16400K .......... .......... .......... .......... .......... 44%  166M 1s
     16450K .......... .......... .......... .......... .......... 45%  123M 1s
     16500K .......... .......... .......... .......... .......... 45%  132M 1s
     16550K .......... .......... .......... .......... .......... 45%  116M 1s
     16600K .......... .......... .......... .......... .......... 45%  133M 1s
     16650K .......... .......... .......... .......... .......... 45% 63.6M 1s
     16700K .......... .......... .......... .......... .......... 45%  126M 1s
     16750K .......... .......... .......... .......... .......... 45% 64.6M 1s
     16800K .......... .......... .......... .......... .......... 46%  112M 1s
     16850K .......... .......... .......... .......... .......... 46%  118M 1s
     16900K .......... .......... .......... .......... .......... 46%  134M 1s
     16950K .......... .......... .......... .......... .......... 46% 64.9M 1s
     17000K .......... .......... .......... .......... .......... 46%  124M 1s
     17050K .......... .......... .......... .......... .......... 46%  110M 1s
     17100K .......... .......... .......... .......... .......... 46%  128M 1s
     17150K .......... .......... .......... .......... .......... 46% 67.1M 1s
     17200K .......... .......... .......... .......... .......... 47%  132M 1s
     17250K .......... .......... .......... .......... .......... 47%  113M 1s
     17300K .......... .......... .......... .......... .......... 47%  111M 1s
     17350K .......... .......... .......... .......... .......... 47%  131M 1s
     17400K .......... .......... .......... .......... .......... 47%  124M 1s
     17450K .......... .......... .......... .......... .......... 47%  114M 1s
     17500K .......... .......... .......... .......... .......... 47%  132M 1s
     17550K .......... .......... .......... .......... .......... 48% 68.0M 1s
     17600K .......... .......... .......... .......... .......... 48%  123M 1s
     17650K .......... .......... .......... .......... .......... 48%  122M 1s
     17700K .......... .......... .......... .......... .......... 48%  125M 1s
     17750K .......... .......... .......... .......... .......... 48%  121M 1s
     17800K .......... .......... .......... .......... .......... 48%  108M 1s
     17850K .......... .......... .......... .......... .......... 48%  112M 1s
     17900K .......... .......... .......... .......... .......... 49%  129M 1s
     17950K .......... .......... .......... .......... .......... 49%  124M 1s
     18000K .......... .......... .......... .......... .......... 49%  131M 1s
     18050K .......... .......... .......... .......... .......... 49% 68.4M 1s
     18100K .......... .......... .......... .......... .......... 49%  129M 1s
     18150K .......... .......... .......... .......... .......... 49%  118M 1s
     18200K .......... .......... .......... .......... .......... 49%  127M 1s
     18250K .......... .......... .......... .......... .......... 50%  101M 1s
     18300K .......... .......... .......... .......... .......... 50%  132M 1s
     18350K .......... .......... .......... .......... .......... 50%  125M 1s
     18400K .......... .......... .......... .......... .......... 50% 69.3M 1s
     18450K .......... .......... .......... .......... .......... 50%  124M 1s
     18500K .......... .......... .......... .......... .......... 50%  127M 1s
     18550K .......... .......... .......... .......... .......... 50%  122M 1s
     18600K .......... .......... .......... .......... .......... 50%  127M 1s
     18650K .......... .......... .......... .......... .......... 51%  110M 1s
     18700K .......... .......... .......... .......... .......... 51%  128M 1s
     18750K .......... .......... .......... .......... .......... 51% 61.4M 1s
     18800K .......... .......... .......... .......... .......... 51%  120M 1s
     18850K .......... .......... .......... .......... .......... 51%  116M 1s
     18900K .......... .......... .......... .......... .......... 51%  119M 1s
     18950K .......... .......... .......... .......... .......... 51%  118M 1s
     19000K .......... .......... .......... .......... .......... 52%  128M 1s
     19050K .......... .......... .......... .......... .......... 52% 66.0M 1s
     19100K .......... .......... .......... .......... .......... 52%  126M 1s
     19150K .......... .......... .......... .......... .......... 52%  121M 1s
     19200K .......... .......... .......... .......... .......... 52%  128M 1s
     19250K .......... .......... .......... .......... .......... 52% 96.7M 1s
     19300K .......... .......... .......... .......... .......... 52%  126M 1s
     19350K .......... .......... .......... .......... .......... 53% 70.2M 1s
     19400K .......... .......... .......... .......... .......... 53%  126M 1s
     19450K .......... .......... .......... .......... .......... 53%  114M 1s
     19500K .......... .......... .......... .......... .......... 53%  134M 1s
     19550K .......... .......... .......... .......... .......... 53%  120M 1s
     19600K .......... .......... .......... .......... .......... 53%  122M 1s
     19650K .......... .......... .......... .......... .......... 53%  120M 1s
     19700K .......... .......... .......... .......... .......... 53%  125M 1s
     19750K .......... .......... .......... .......... .......... 54%  101M 1s
     19800K .......... .......... .......... .......... .......... 54%  130M 1s
     19850K .......... .......... .......... .......... .......... 54%  106M 1s
     19900K .......... .......... .......... .......... .......... 54%  131M 1s
     19950K .......... .......... .......... .......... .......... 54%  126M 1s
     20000K .......... .......... .......... .......... .......... 54%  129M 1s
     20050K .......... .......... .......... .......... .......... 54%  120M 1s
     20100K .......... .......... .......... .......... .......... 55% 51.5M 1s
     20150K .......... .......... .......... .......... .......... 55%  117M 1s
     20200K .......... .......... .......... .......... .......... 55%  126M 1s
     20250K .......... .......... .......... .......... .......... 55% 97.6M 1s
     20300K .......... .......... .......... .......... .......... 55%  129M 1s
     20350K .......... .......... .......... .......... .......... 55%  123M 1s
     20400K .......... .......... .......... .......... .......... 55%  123M 1s
     20450K .......... .......... .......... .......... .......... 56% 68.2M 1s
     20500K .......... .......... .......... .......... .......... 56%  257M 1s
     20550K .......... .......... .......... .......... .......... 56%  159M 1s
     20600K .......... .......... .......... .......... .......... 56%  121M 1s
     20650K .......... .......... .......... .......... .......... 56%  115M 1s
     20700K .......... .......... .......... .......... .......... 56%  125M 1s
     20750K .......... .......... .......... .......... .......... 56% 97.9M 1s
     20800K .......... .......... .......... .......... .......... 56%  128M 1s
     20850K .......... .......... .......... .......... .......... 57%  121M 1s
     20900K .......... .......... .......... .......... .......... 57%  127M 1s
     20950K .......... .......... .......... .......... .......... 57%  126M 1s
     21000K .......... .......... .......... .......... .......... 57%  139M 1s
     21050K .......... .......... .......... .......... .......... 57%  115M 1s
     21100K .......... .......... .......... .......... .......... 57%  123M 1s
     21150K .......... .......... .......... .......... .......... 57%  114M 1s
     21200K .......... .......... .......... .......... .......... 58%  132M 1s
     21250K .......... .......... .......... .......... .......... 58%  100M 1s
     21300K .......... .......... .......... .......... .......... 58%  124M 1s
     21350K .......... .......... .......... .......... .......... 58%  127M 1s
     21400K .......... .......... .......... .......... .......... 58%  130M 1s
     21450K .......... .......... .......... .......... .......... 58%  115M 1s
     21500K .......... .......... .......... .......... .......... 58% 71.9M 1s
     21550K .......... .......... .......... .......... .......... 59% 68.3M 1s
     21600K .......... .......... .......... .......... .......... 59%  126M 1s
     21650K .......... .......... .......... .......... .......... 59%  117M 1s
     21700K .......... .......... .......... .......... .......... 59%  123M 1s
     21750K .......... .......... .......... .......... .......... 59%  107M 1s
     21800K .......... .......... .......... .......... .......... 59%  127M 1s
     21850K .......... .......... .......... .......... .......... 59%  112M 1s
     21900K .......... .......... .......... .......... .......... 59%  124M 1s
     21950K .......... .......... .......... .......... .......... 60%  123M 1s
     22000K .......... .......... .......... .......... .......... 60%  125M 1s
     22050K .......... .......... .......... .......... .......... 60%  128M 1s
     22100K .......... .......... .......... .......... .......... 60%  125M 1s
     22150K .......... .......... .......... .......... .......... 60%  120M 1s
     22200K .......... .......... .......... .......... .......... 60%  128M 1s
     22250K .......... .......... .......... .......... .......... 60% 62.7M 1s
     22300K .......... .......... .......... .......... .......... 61%  128M 1s
     22350K .......... .......... .......... .......... .......... 61%  118M 1s
     22400K .......... .......... .......... .......... .......... 61%  124M 1s
     22450K .......... .......... .......... .......... .......... 61%  123M 1s
     22500K .......... .......... .......... .......... .......... 61%  124M 1s
     22550K .......... .......... .......... .......... .......... 61%  116M 1s
     22600K .......... .......... .......... .......... .......... 61% 66.5M 1s
     22650K .......... .......... .......... .......... .......... 62% 69.6M 1s
     22700K .......... .......... .......... .......... .......... 62%  128M 1s
     22750K .......... .......... .......... .......... .......... 62%  108M 1s
     22800K .......... .......... .......... .......... .......... 62%  131M 1s
     22850K .......... .......... .......... .......... .......... 62%  124M 1s
     22900K .......... .......... .......... .......... .......... 62% 70.3M 1s
     22950K .......... .......... .......... .......... .......... 62%  118M 1s
     23000K .......... .......... .......... .......... .......... 62%  126M 1s
     23050K .......... .......... .......... .......... .......... 63% 98.8M 1s
     23100K .......... .......... .......... .......... .......... 63%  107M 1s
     23150K .......... .......... .......... .......... .......... 63%  108M 1s
     23200K .......... .......... .......... .......... .......... 63%  105M 1s
     23250K .......... .......... .......... .......... .......... 63% 95.9M 1s
     23300K .......... .......... .......... .......... .......... 63%  114M 1s
     23350K .......... .......... .......... .......... .......... 63%  110M 1s
     23400K .......... .......... .......... .......... .......... 64%  106M 1s
     23450K .......... .......... .......... .......... .......... 64% 59.8M 1s
     23500K .......... .......... .......... .......... .......... 64% 68.7M 1s
     23550K .......... .......... .......... .......... .......... 64%  105M 1s
     23600K .......... .......... .......... .......... .......... 64%  123M 1s
     23650K .......... .......... .......... .......... .......... 64%  116M 1s
     23700K .......... .......... .......... .......... .......... 64%  121M 1s
     23750K .......... .......... .......... .......... .......... 65%  112M 1s
     23800K .......... .......... .......... .......... .......... 65%  126M 1s
     23850K .......... .......... .......... .......... .......... 65%  111M 1s
     23900K .......... .......... .......... .......... .......... 65%  129M 1s
     23950K .......... .......... .......... .......... .......... 65%  118M 1s
     24000K .......... .......... .......... .......... .......... 65%  129M 1s
     24050K .......... .......... .......... .......... .......... 65%  120M 1s
     24100K .......... .......... .......... .......... .......... 65%  128M 1s
     24150K .......... .......... .......... .......... .......... 66%  114M 1s
     24200K .......... .......... .......... .......... .......... 66% 72.3M 1s
     24250K .......... .......... .......... .......... .......... 66% 63.3M 1s
     24300K .......... .......... .......... .......... .......... 66%  127M 1s
     24350K .......... .......... .......... .......... .......... 66%  120M 1s
     24400K .......... .......... .......... .......... .......... 66%  128M 1s
     24450K .......... .......... .......... .......... .......... 66%  117M 1s
     24500K .......... .......... .......... .......... .......... 67% 52.7M 1s
     24550K .......... .......... .......... .......... .......... 67%  430K 1s
     24600K .......... .......... .......... .......... .......... 67%  221M 1s
     24650K .......... .......... .......... .......... .......... 67%  215M 1s
     24700K .......... .......... .......... .......... .......... 67%  253M 1s
     24750K .......... .......... .......... .......... .......... 67%  264M 1s
     24800K .......... .......... .......... .......... .......... 67%  287M 1s
     24850K .......... .......... .......... .......... .......... 68%  257M 1s
     24900K .......... .......... .......... .......... .......... 68%  192M 1s
     24950K .......... .......... .......... .......... .......... 68% 52.1M 1s
     25000K .......... .......... .......... .......... .......... 68% 31.9M 1s
     25050K .......... .......... .......... .......... .......... 68% 99.1M 1s
     25100K .......... .......... .......... .......... .......... 68%  122M 1s
     25150K .......... .......... .......... .......... .......... 68%  106M 1s
     25200K .......... .......... .......... .......... .......... 68%  104M 1s
     25250K .......... .......... .......... .......... .......... 69%  123M 1s
     25300K .......... .......... .......... .......... .......... 69%  129M 1s
     25350K .......... .......... .......... .......... .......... 69%  123M 1s
     25400K .......... .......... .......... .......... .......... 69%  126M 1s
     25450K .......... .......... .......... .......... .......... 69%  114M 1s
     25500K .......... .......... .......... .......... .......... 69%  131M 1s
     25550K .......... .......... .......... .......... .......... 69% 66.0M 1s
     25600K .......... .......... .......... .......... .......... 70%  114M 1s
     25650K .......... .......... .......... .......... .......... 70%  114M 1s
     25700K .......... .......... .......... .......... .......... 70%  107M 1s
     25750K .......... .......... .......... .......... .......... 70%  112M 1s
     25800K .......... .......... .......... .......... .......... 70%  135M 1s
     25850K .......... .......... .......... .......... .......... 70%  111M 1s
     25900K .......... .......... .......... .......... .......... 70%  117M 1s
     25950K .......... .......... .......... .......... .......... 71%  123M 1s
     26000K .......... .......... .......... .......... .......... 71%  131M 1s
     26050K .......... .......... .......... .......... .......... 71% 69.6M 1s
     26100K .......... .......... .......... .......... .......... 71% 74.0M 1s
     26150K .......... .......... .......... .......... .......... 71%  116M 1s
     26200K .......... .......... .......... .......... .......... 71%  112M 1s
     26250K .......... .......... .......... .......... .......... 71%  112M 1s
     26300K .......... .......... .......... .......... .......... 71%  128M 1s
     26350K .......... .......... .......... .......... .......... 72%  115M 0s
     26400K .......... .......... .......... .......... .......... 72%  122M 0s
     26450K .......... .......... .......... .......... .......... 72%  118M 0s
     26500K .......... .......... .......... .......... .......... 72% 74.3M 0s
     26550K .......... .......... .......... .......... .......... 72%  124M 0s
     26600K .......... .......... .......... .......... .......... 72%  120M 0s
     26650K .......... .......... .......... .......... .......... 72%  110M 0s
     26700K .......... .......... .......... .......... .......... 73%  110M 0s
     26750K .......... .......... .......... .......... .......... 73%  115M 0s
     26800K .......... .......... .......... .......... .......... 73%  131M 0s
     26850K .......... .......... .......... .......... .......... 73%  144M 0s
     26900K .......... .......... .......... .......... .......... 73%  156M 0s
     26950K .......... .......... .......... .......... .......... 73%  151M 0s
     27000K .......... .......... .......... .......... .......... 73%  185M 0s
     27050K .......... .......... .......... .......... .......... 74%  130M 0s
     27100K .......... .......... .......... .......... .......... 74%  129M 0s
     27150K .......... .......... .......... .......... .......... 74%  113M 0s
     27200K .......... .......... .......... .......... .......... 74%  110M 0s
     27250K .......... .......... .......... .......... .......... 74%  125M 0s
     27300K .......... .......... .......... .......... .......... 74%  127M 0s
     27350K .......... .......... .......... .......... .......... 74% 51.9M 0s
     27400K .......... .......... .......... .......... .......... 75%  130M 0s
     27450K .......... .......... .......... .......... .......... 75%  110M 0s
     27500K .......... .......... .......... .......... .......... 75%  131M 0s
     27550K .......... .......... .......... .......... .......... 75% 68.5M 0s
     27600K .......... .......... .......... .......... .......... 75%  128M 0s
     27650K .......... .......... .......... .......... .......... 75%  116M 0s
     27700K .......... .......... .......... .......... .......... 75%  108M 0s
     27750K .......... .......... .......... .......... .......... 75%  114M 0s
     27800K .......... .......... .......... .......... .......... 76%  126M 0s
     27850K .......... .......... .......... .......... .......... 76%  110M 0s
     27900K .......... .......... .......... .......... .......... 76% 71.5M 0s
     27950K .......... .......... .......... .......... .......... 76%  116M 0s
     28000K .......... .......... .......... .......... .......... 76%  126M 0s
     28050K .......... .......... .......... .......... .......... 76%  121M 0s
     28100K .......... .......... .......... .......... .......... 76%  128M 0s
     28150K .......... .......... .......... .......... .......... 77%  118M 0s
     28200K .......... .......... .......... .......... .......... 77% 56.7M 0s
     28250K .......... .......... .......... .......... .......... 77% 43.5M 0s
     28300K .......... .......... .......... .......... .......... 77% 80.6M 0s
     28350K .......... .......... .......... .......... .......... 77%  110M 0s
     28400K .......... .......... .......... .......... .......... 77% 69.9M 0s
     28450K .......... .......... .......... .......... .......... 77%  114M 0s
     28500K .......... .......... .......... .......... .......... 78%  118M 0s
     28550K .......... .......... .......... .......... .......... 78%  109M 0s
     28600K .......... .......... .......... .......... .......... 78%  120M 0s
     28650K .......... .......... .......... .......... .......... 78% 93.0M 0s
     28700K .......... .......... .......... .......... .......... 78%  146M 0s
     28750K .......... .......... .......... .......... .......... 78%  119M 0s
     28800K .......... .......... .......... .......... .......... 78%  126M 0s
     28850K .......... .......... .......... .......... .......... 78%  120M 0s
     28900K .......... .......... .......... .......... .......... 79%  127M 0s
     28950K .......... .......... .......... .......... .......... 79%  125M 0s
     29000K .......... .......... .......... .......... .......... 79%  123M 0s
     29050K .......... .......... .......... .......... .......... 79%  112M 0s
     29100K .......... .......... .......... .......... .......... 79%  122M 0s
     29150K .......... .......... .......... .......... .......... 79%  120M 0s
     29200K .......... .......... .......... .......... .......... 79%  107M 0s
     29250K .......... .......... .......... .......... .......... 80%  133M 0s
     29300K .......... .......... .......... .......... .......... 80%  127M 0s
     29350K .......... .......... .......... .......... .......... 80% 65.5M 0s
     29400K .......... .......... .......... .......... .......... 80%  130M 0s
     29450K .......... .......... .......... .......... .......... 80% 66.2M 0s
     29500K .......... .......... .......... .......... .......... 80%  125M 0s
     29550K .......... .......... .......... .......... .......... 80%  116M 0s
     29600K .......... .......... .......... .......... .......... 81%  121M 0s
     29650K .......... .......... .......... .......... .......... 81%  119M 0s
     29700K .......... .......... .......... .......... .......... 81%  118M 0s
     29750K .......... .......... .......... .......... .......... 81% 65.7M 0s
     29800K .......... .......... .......... .......... .......... 81%  132M 0s
     29850K .......... .......... .......... .......... .......... 81%  109M 0s
     29900K .......... .......... .......... .......... .......... 81%  123M 0s
     29950K .......... .......... .......... .......... .......... 81%  114M 0s
     30000K .......... .......... .......... .......... .......... 82%  126M 0s
     30050K .......... .......... .......... .......... .......... 82%  125M 0s
     30100K .......... .......... .......... .......... .......... 82%  123M 0s
     30150K .......... .......... .......... .......... .......... 82%  118M 0s
     30200K .......... .......... .......... .......... .......... 82%  113M 0s
     30250K .......... .......... .......... .......... .......... 82%  116M 0s
     30300K .......... .......... .......... .......... .......... 82%  127M 0s
     30350K .......... .......... .......... .......... .......... 83%  118M 0s
     30400K .......... .......... .......... .......... .......... 83%  133M 0s
     30450K .......... .......... .......... .......... .......... 83% 68.5M 0s
     30500K .......... .......... .......... .......... .......... 83%  129M 0s
     30550K .......... .......... .......... .......... .......... 83%  109M 0s
     30600K .......... .......... .......... .......... .......... 83%  124M 0s
     30650K .......... .......... .......... .......... .......... 83% 96.9M 0s
     30700K .......... .......... .......... .......... .......... 84%  126M 0s
     30750K .......... .......... .......... .......... .......... 84%  121M 0s
     30800K .......... .......... .......... .......... .......... 84%  126M 0s
     30850K .......... .......... .......... .......... .......... 84%  123M 0s
     30900K .......... .......... .......... .......... .......... 84%  131M 0s
     30950K .......... .......... .......... .......... .......... 84%  119M 0s
     31000K .......... .......... .......... .......... .......... 84%  129M 0s
     31050K .......... .......... .......... .......... .......... 84%  107M 0s
     31100K .......... .......... .......... .......... .......... 85%  125M 0s
     31150K .......... .......... .......... .......... .......... 85%  103M 0s
     31200K .......... .......... .......... .......... .......... 85%  127M 0s
     31250K .......... .......... .......... .......... .......... 85%  124M 0s
     31300K .......... .......... .......... .......... .......... 85%  124M 0s
     31350K .......... .......... .......... .......... .......... 85%  123M 0s
     31400K .......... .......... .......... .......... .......... 85%  130M 0s
     31450K .......... .......... .......... .......... .......... 86%  112M 0s
     31500K .......... .......... .......... .......... .......... 86%  129M 0s
     31550K .......... .......... .......... .......... .......... 86%  118M 0s
     31600K .......... .......... .......... .......... .......... 86%  124M 0s
     31650K .......... .......... .......... .......... .......... 86%  105M 0s
     31700K .......... .......... .......... .......... .......... 86%  132M 0s
     31750K .......... .......... .......... .......... .......... 86%  116M 0s
     31800K .......... .......... .......... .......... .......... 87%  133M 0s
     31850K .......... .......... .......... .......... .......... 87% 66.0M 0s
     31900K .......... .......... .......... .......... .......... 87%  128M 0s
     31950K .......... .......... .......... .......... .......... 87%  115M 0s
     32000K .......... .......... .......... .......... .......... 87%  127M 0s
     32050K .......... .......... .......... .......... .......... 87%  124M 0s
     32100K .......... .......... .......... .......... .......... 87%  127M 0s
     32150K .......... .......... .......... .......... .......... 87%  100M 0s
     32200K .......... .......... .......... .......... .......... 88%  132M 0s
     32250K .......... .......... .......... .......... .......... 88%  114M 0s
     32300K .......... .......... .......... .......... .......... 88%  132M 0s
     32350K .......... .......... .......... .......... .......... 88%  120M 0s
     32400K .......... .......... .......... .......... .......... 88%  134M 0s
     32450K .......... .......... .......... .......... .......... 88%  118M 0s
     32500K .......... .......... .......... .......... .......... 88%  124M 0s
     32550K .......... .......... .......... .......... .......... 89%  121M 0s
     32600K .......... .......... .......... .......... .......... 89%  120M 0s
     32650K .......... .......... .......... .......... .......... 89% 96.0M 0s
     32700K .......... .......... .......... .......... .......... 89%  129M 0s
     32750K .......... .......... .......... .......... .......... 89%  111M 0s
     32800K .......... .......... .......... .......... .......... 89%  146M 0s
     32850K .......... .......... .......... .......... .......... 89% 69.9M 0s
     32900K .......... .......... .......... .......... .......... 90%  119M 0s
     32950K .......... .......... .......... .......... .......... 90% 68.0M 0s
     33000K .......... .......... .......... .......... .......... 90%  127M 0s
     33050K .......... .......... .......... .......... .......... 90%  112M 0s
     33100K .......... .......... .......... .......... .......... 90%  128M 0s
     33150K .......... .......... .......... .......... .......... 90% 64.5M 0s
     33200K .......... .......... .......... .......... .......... 90%  126M 0s
     33250K .......... .......... .......... .......... .......... 90%  106M 0s
     33300K .......... .......... .......... .......... .......... 91%  115M 0s
     33350K .......... .......... .......... .......... .......... 91%  102M 0s
     33400K .......... .......... .......... .......... .......... 91%  111M 0s
     33450K .......... .......... .......... .......... .......... 91%  101M 0s
     33500K .......... .......... .......... .......... .......... 91%  119M 0s
     33550K .......... .......... .......... .......... .......... 91%  109M 0s
     33600K .......... .......... .......... .......... .......... 91%  111M 0s
     33650K .......... .......... .......... .......... .......... 92% 63.0M 0s
     33700K .......... .......... .......... .......... .......... 92%  121M 0s
     33750K .......... .......... .......... .......... .......... 92%  118M 0s
     33800K .......... .......... .......... .......... .......... 92%  127M 0s
     33850K .......... .......... .......... .......... .......... 92%  117M 0s
     33900K .......... .......... .......... .......... .......... 92% 70.2M 0s
     33950K .......... .......... .......... .......... .......... 92%  113M 0s
     34000K .......... .......... .......... .......... .......... 93%  129M 0s
     34050K .......... .......... .......... .......... .......... 93%  121M 0s
     34100K .......... .......... .......... .......... .......... 93%  121M 0s
     34150K .......... .......... .......... .......... .......... 93%  106M 0s
     34200K .......... .......... .......... .......... .......... 93% 71.1M 0s
     34250K .......... .......... .......... .......... .......... 93%  114M 0s
     34300K .......... .......... .......... .......... .......... 93% 70.7M 0s
     34350K .......... .......... .......... .......... .......... 93%  120M 0s
     34400K .......... .......... .......... .......... .......... 94%  127M 0s
     34450K .......... .......... .......... .......... .......... 94%  122M 0s
     34500K .......... .......... .......... .......... .......... 94% 70.2M 0s
     34550K .......... .......... .......... .......... .......... 94%  119M 0s
     34600K .......... .......... .......... .......... .......... 94%  130M 0s
     34650K .......... .......... .......... .......... .......... 94% 95.5M 0s
     34700K .......... .......... .......... .......... .......... 94%  129M 0s
     34750K .......... .......... .......... .......... .......... 95%  118M 0s
     34800K .......... .......... .......... .......... .......... 95%  135M 0s
     34850K .......... .......... .......... .......... .......... 95% 70.8M 0s
     34900K .......... .......... .......... .......... .......... 95%  127M 0s
     34950K .......... .......... .......... .......... .......... 95%  116M 0s
     35000K .......... .......... .......... .......... .......... 95%  131M 0s
     35050K .......... .......... .......... .......... .......... 95%  108M 0s
     35100K .......... .......... .......... .......... .......... 96%  125M 0s
     35150K .......... .......... .......... .......... .......... 96% 63.5M 0s
     35200K .......... .......... .......... .......... .......... 96%  136M 0s
     35250K .......... .......... .......... .......... .......... 96%  129M 0s
     35300K .......... .......... .......... .......... .......... 96%  128M 0s
     35350K .......... .......... .......... .......... .......... 96%  122M 0s
     35400K .......... .......... .......... .......... .......... 96% 73.8M 0s
     35450K .......... .......... .......... .......... .......... 97%  111M 0s
     35500K .......... .......... .......... .......... .......... 97% 7.29M 0s
     35550K .......... .......... .......... .......... .......... 97%  167M 0s
     35600K .......... .......... .......... .......... .......... 97%  114M 0s
     35650K .......... .......... .......... .......... .......... 97%  141M 0s
     35700K .......... .......... .......... .......... .......... 97%  260M 0s
     35750K .......... .......... .......... .......... .......... 97%  249M 0s
     35800K .......... .......... .......... .......... .......... 97%  103M 0s
     35850K .......... .......... .......... .......... .......... 98%  144M 0s
     35900K .......... .......... .......... .......... .......... 98%  213M 0s
     35950K .......... .......... .......... .......... .......... 98%  239M 0s
     36000K .......... .......... .......... .......... .......... 98%  253M 0s
     36050K .......... .......... .......... .......... .......... 98%  193M 0s
     36100K .......... .......... .......... .......... .......... 98%  149M 0s
     36150K .......... .......... .......... .......... .......... 98%  186M 0s
     36200K .......... .......... .......... .......... .......... 99%  190M 0s
     36250K .......... .......... .......... .......... .......... 99%  235M 0s
     36300K .......... .......... .......... .......... .......... 99%  196M 0s
     36350K .......... .......... .......... .......... .......... 99%  229M 0s
     36400K .......... .......... .......... .......... .......... 99%  198M 0s
     36450K .......... .......... .......... .......... .......... 99%  237M 0s
     36500K .......... .......... .......... .......... .......... 99%  188M 0s
     36550K .......... .......... .......... .......... .......   100%  181M=1.4s
    
    2021-05-03 23:13:49 (25.8 MB/s) - 'LiftOver/liftOver' saved [37475560/37475560]
    
    WARNING: timestamping does nothing in combination with -O. See the manual
    for details.
    
    --2021-05-03 23:13:49--  ftp://hgdownload.soe.ucsc.edu/goldenPath/hg19/liftOver/hg19ToHg38.over.chain.gz
    Resolving dtn05-e0 (dtn05-e0)... 10.1.200.241
    Connecting to dtn05-e0 (dtn05-e0)|10.1.200.241|:3128... connected.
    Proxy request sent, awaiting response... 200 Gatewaying
    Length: 227698 (222K) [text/plain]
    Saving to: 'LiftOver/hg19ToHg38.over.chain.gz'
    
         0K .......... .......... .......... .......... .......... 22%  334K 1s
        50K .......... .......... .......... .......... .......... 44%  736K 0s
       100K .......... .......... .......... .......... .......... 67% 99.2M 0s
       150K .......... .......... .......... .......... .......... 89% 88.4M 0s
       200K .......... .......... ..                              100%  339K=0.3s
    
    2021-05-03 23:13:50 (781 KB/s) - 'LiftOver/hg19ToHg38.over.chain.gz' saved [227698/227698]
    
    Reading liftover chains
    Mapping coordinates
    [-] Unloading plink  1.9.0-beta4.4  on cn3174 
    [+] Loading plink  1.9.0-beta4.4  on cn3174 



```bash
%%bash
# number of variants lifted/unlifted
wc -l Itals.HLAregion_update-map_to_b38.txt
wc -l SNPs_not_in_b38.txt
```

    6293 Itals.HLAregion_update-map_to_b38.txt
    0 SNPs_not_in_b38.txt


## Run imputation

script to run imputation: `impute.1000G.EUR.hg38.sh`

```
#!/bin/bash

source /data/chiarp/conda/etc/profile.d/conda.sh
conda activate CookHLA

DIR="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA"

mkdir /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G

python $DIR/CookHLA/CookHLA.py \
    -i $DIR/Itals.HLAregion.chr6.Liftover.hg38 \
    -hg 38 \
    -o $DIR/Imputation.1000G/Itals+1000G_REF.EUR \
    -ref $DIR/CookHLA/1000G_REF/1000G_REF.EUR.chr6.hg18.29mb-34mb.inT1DGC \
    -mem 120g \
    -mp 8
```


```bash
%%bash
cd CookHLA/
echo "sh impute.1000G.EUR.hg38.sh" > impute2.swarm
swarm --file impute2.swarm --logdir swarmOE_impute -g 120 -t 8 
```

    14236957


## Download code for HATK

HATK(HLA Analysis Tool-Kit) is a collection of tools and modules to perform HLA fine-mapping analysis, which is to identify which HLA allele or amino acid position of the HLA gene is driving the disease


```python
!git clone https://github.com/WansonChoi/HATK.git
```

    Cloning into 'HATK'...
    remote: Enumerating objects: 2246, done.[K
    remote: Counting objects: 100% (206/206), done.[K
    remote: Compressing objects: 100% (135/135), done.[K
    remote: Total 2246 (delta 92), reused 163 (delta 66), pack-reused 2040[K
    Receiving objects: 100% (2246/2246), 169.10 MiB | 20.82 MiB/s, done.
    Resolving deltas: 100% (1089/1089), done.
    Checking out files: 100% (718/718), done.



```python
#%%bash
#cd HATK
#source /data/chiarp/conda/etc/profile.d/conda.sh
#conda activate base
#conda env create -f HATK_LINUX.yml
#conda activate HATK
```

## Run HATK


```bash
%%bash
# Update covariate file so that FID = IID and it matches up with the original subsetted plink file used as input for imputation

module load R/3.5.2
R --vanilla --no-save

require(data.table)
require(tidyverse)

data <- fread("/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/COVARIATES.Itals_mg_noDups.UNRELATED.txt",header=T) %>%
        mutate(FID = IID)
head(data)
write.table(data,"COVARIATES.Itals_mg_noDups.UNRELATED.SampleNameUpdated.txt",quote=F,sep="\t",row.names=F,col.names=T)

```

    
    R version 3.5.2 (2018-12-20) -- "Eggshell Igloo"
    Copyright (C) 2018 The R Foundation for Statistical Computing
    Platform: x86_64-pc-linux-gnu (64-bit)
    
    R is free software and comes with ABSOLUTELY NO WARRANTY.
    You are welcome to redistribute it under certain conditions.
    Type 'license()' or 'licence()' for distribution details.
    
    R is a collaborative project with many contributors.
    Type 'contributors()' for more information and
    'citation()' on how to cite R or R packages in publications.
    
    Type 'demo()' for some demos, 'help()' for on-line help, or
    'help.start()' for an HTML browser interface to help.
    Type 'q()' to quit R.
    
    > 
    > require(data.table)
    > require(tidyverse)
    > 
    > data <- fread("/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/COVARIATES.Itals_mg_noDups.UNRELATED.txt",header=T) %>%
    +         mutate(FID = IID)
    > head(data)
            FID       IID MAT PAT SEX PHENO genderONFILE age_at_onset diagnosis
    1 10039-DVG 10039-DVG   0   0   2     1            1           60   control
    2  10071-BR  10071-BR   0   0   2     1            1           54   control
    3  10081-BC  10081-BC   0   0   1     1            0           75   control
    4 10137-MMT 10137-MMT   0   0   2     1            1           63   control
    5  10147-OR  10147-OR   0   0   1     1            0           69   control
    6   1022-TS   1022-TS   0   0   1     1            0           55   control
               PC1         PC2          PC3          PC4          PC5          PC6
    1 -0.025312730  0.02164687  0.006001769  0.013145200 -0.025761550  0.012118420
    2 -0.020662860  0.02438660 -0.041687700  0.044762190  0.027410460  0.013000710
    3 -0.013087100 -0.08244215 -0.014605790 -0.008541102 -0.020152580 -0.008590607
    4 -0.026193970 -0.01373366  0.012337870 -0.003966964  0.006837094 -0.049749590
    5 -0.007320636 -0.04046117 -0.020739560 -0.010027520  0.019149870  0.026729550
    6 -0.042137960  0.02172641 -0.006119175 -0.027821780  0.015159360  0.011921870
               PC7         PC8          PC9          PC10 GENDER
    1  0.026402570 -0.03195026  0.016432260  0.0231902000      1
    2  0.017429430 -0.04442212  0.011062090  0.0091400870      1
    3 -0.006106247 -0.02398282  0.005996563  0.0022551240      0
    4 -0.039859960  0.05736454 -0.037509020  0.0080890020      1
    5  0.039740670 -0.02125327  0.020937230 -0.0004138881      0
    6 -0.014626500 -0.01505404  0.058594910  0.0331456400      0
                  FID_IID
    1 10039-DVG_10039-DVG
    2   10071-BR_10071-BR
    3   10081-BC_10081-BC
    4 10137-MMT_10137-MMT
    5   10147-OR_10147-OR
    6     1022-TS_1022-TS
    > write.table(data,"COVARIATES.Itals_mg_noDups.UNRELATED.SampleNameUpdated.txt",quote=F,sep="\t",row.names=F,col.names=T)
    > 


    [-] Unloading gcc  9.2.0  ... 
    [-] Unloading GSL 2.6 for GCC 9.2.0 ... 
    [-] Unloading openmpi 4.0.5  for GCC 9.2.0 
    [-] Unloading ImageMagick  7.0.8  on cn3174 
    [-] Unloading HDF5  1.10.4 
    [-] Unloading NetCDF 4.7.4_gcc9.2.0 
    [-] Unloading pandoc  2.13  on cn3174 
    [-] Unloading pcre2 10.21  ... 
    [-] Unloading R 4.0.5 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading GSL 2.4 for GCC 7.2.0 ... 
    [-] Unloading gcc  7.3.0  ... 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading openmpi 3.0.2  for GCC 7.3.0 
    [+] Loading ImageMagick  7.0.8  on cn3174 
    [+] Loading HDF5  1.10.4 
    [+] Loading pandoc  2.13  on cn3174 
    [+] Loading R 3.5.2 
    
    The following have been reloaded with a version change:
      1) GSL/2.6_gcc-9.2.0 => GSL/2.4_gcc-7.2.0     3) gcc/9.2.0 => gcc/7.3.0
      2) R/4.0 => R/3.5.2
    
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()


### Imputation.1000G
#### Overall

script to run imputation: `/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/HATK/hatk.1000G.4field.sh`

```
#!/bin/bash

module load plink/1.9.0-beta4.4
module load R/4.0

DIR="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA"

out="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G/4field"

python3 HATK.py \
    --variants $DIR/Itals.HLAregion.chr6.Liftover.hg38 \
    --hped $DIR/Imputation.1000G/Itals+1000G_REF.EUR.MHC.HLA_IMPUTATION_OUT.hped \
    --4field \
    --pheno $DIR/COVARIATES.Itals_mg_noDups.UNRELATED.SampleNameUpdated.txt \
    --pheno-name PHENO \
    --covar $DIR/COVARIATES.Itals_mg_noDups.UNRELATED.SampleNameUpdated.txt \
    --covar-name GENDER,age_at_onset,PC1,PC2,PC3,PC4,PC5,PC6,PC10 \
    --out $out/RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38 \
    --imgt 3320 \
    --hg 38 \
    --imgt-dir $DIR/HATK/example/IMGTHLA3320 \
    --multiprocess 8
```




```bash
%%bash
mkdir Imputation.1000G/4field
```


```bash
%%bash
cd HATK

echo "sh hatk.1000G.4field.sh" >> hatk.1000G.swarm
swarm --file hatk.1000G.swarm --logdir swarmOE_hatk -g 120 -t 8 
```

    14271625


##### plink version association - 4field


```bash
%%bash

module load plink/2.0-dev-20191128

DIR="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA"
out="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G/4field"
mkdir $out/GLM

plink \
--bfile $out/RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38 \
--pheno $DIR/COVARIATES.Itals_mg_noDups.UNRELATED.SampleNameUpdated.txt \
--pheno-name PHENO \
--covar $DIR/COVARIATES.Itals_mg_noDups.UNRELATED.SampleNameUpdated.txt \
--covar-name GENDER,age_at_onset,PC1,PC2,PC3,PC4,PC5,PC6,PC10 \
--glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err \
--out $out/GLM/RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38 \
--covar-variance-standardize

```

    PLINK v2.00a2LM 64-bit Intel (28 Nov 2019)     www.cog-genomics.org/plink/2.0/
    (C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
    Logging to /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G/4field/GLM/RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.log.
    Options in effect:
      --bfile /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G/4field/RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38
      --covar /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/COVARIATES.Itals_mg_noDups.UNRELATED.SampleNameUpdated.txt
      --covar-name GENDER,age_at_onset,PC1,PC2,PC3,PC4,PC5,PC6,PC10
      --covar-variance-standardize
      --glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err
      --out /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G/4field/GLM/RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38
      --pheno /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/COVARIATES.Itals_mg_noDups.UNRELATED.SampleNameUpdated.txt
      --pheno-name PHENO
    
    Start time: Wed May  5 00:50:40 2021
    257652 MiB RAM detected; reserving 128826 MiB for main workspace.
    Using up to 56 threads (change this with --threads).
    3497 samples (1583 females, 1914 males; 3497 founders) loaded from
    /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G/4field/RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.fam.
    10828 variants loaded from
    /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G/4field/RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.bim.
    1 binary phenotype loaded (919 cases, 2478 controls).
    9 covariates loaded from /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/COVARIATES.Itals_mg_noDups.UNRELATED.SampleNameUpdated.txt.
    --covar-variance-standardize: 9 covariates transformed.
    Calculating allele frequencies... 0%done.
    --glm logistic-Firth hybrid regression on phenotype 'PHENO': 0%done.
    Results written to /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G/4field/GLM/RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.PHENO.glm.logistic.hybrid .
    End time: Wed May  5 00:50:45 2021


    [-] Unloading plink  1.9.0-beta4.4  on cn3451 
    [+] Loading plink  2.0-dev-20191128 
    
    The following have been reloaded with a version change:
      1) plink/1.9.0-beta4.4 => plink/2.0-dev-20191128
    


##### plink version conditional association - 4field


```bash
%%bash

module load plink/2.0-dev-20191128

DIR="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA"
out="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G/4field"
mkdir $out/GLM.conditional

plink \
--bfile $out/RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38 \
--condition HLA_DRB1*13:01:02 \
--pheno $DIR/COVARIATES.Itals_mg_noDups.UNRELATED.SampleNameUpdated.txt \
--pheno-name PHENO \
--covar $DIR/COVARIATES.Itals_mg_noDups.UNRELATED.SampleNameUpdated.txt \
--covar-name GENDER,age_at_onset,PC1,PC2,PC3,PC4,PC5,PC6,PC10 \
--glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err \
--out $out/GLM.conditional/RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.HLA_DRB1 \
--covar-variance-standardize

plink \
--bfile $out/RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38 \
--condition AA_B_45_31356825_exon2_E \
--pheno $DIR/COVARIATES.Itals_mg_noDups.UNRELATED.SampleNameUpdated.txt \
--pheno-name PHENO \
--covar $DIR/COVARIATES.Itals_mg_noDups.UNRELATED.SampleNameUpdated.txt \
--covar-name GENDER,age_at_onset,PC1,PC2,PC3,PC4,PC5,PC6,PC10 \
--glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err \
--out $out/GLM.conditional/RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.AA.HLA_B \
--covar-variance-standardize

```

    PLINK v2.00a2LM 64-bit Intel (28 Nov 2019)     www.cog-genomics.org/plink/2.0/
    (C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
    Logging to /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G/4field/GLM.conditional/RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.HLA_DRB1.log.
    Options in effect:
      --bfile /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G/4field/RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38
      --condition HLA_DRB1*13:01:02
      --covar /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/COVARIATES.Itals_mg_noDups.UNRELATED.SampleNameUpdated.txt
      --covar-name GENDER,age_at_onset,PC1,PC2,PC3,PC4,PC5,PC6,PC10
      --covar-variance-standardize
      --glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err
      --out /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G/4field/GLM.conditional/RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.HLA_DRB1
      --pheno /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/COVARIATES.Itals_mg_noDups.UNRELATED.SampleNameUpdated.txt
      --pheno-name PHENO
    
    Start time: Thu May  6 04:01:17 2021
    257652 MiB RAM detected; reserving 128826 MiB for main workspace.
    Using up to 56 threads (change this with --threads).
    3497 samples (1583 females, 1914 males; 3497 founders) loaded from
    /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G/4field/RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.fam.
    10828 variants loaded from
    /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G/4field/RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.bim.
    1 binary phenotype loaded (919 cases, 2478 controls).
    9 covariates loaded from /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/COVARIATES.Itals_mg_noDups.UNRELATED.SampleNameUpdated.txt.
    --covar-variance-standardize: 9 covariates transformed.
    Calculating allele frequencies... 0%done.
    --glm logistic-Firth hybrid regression on phenotype 'PHENO': 0%done.
    Results written to /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G/4field/GLM.conditional/RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.HLA_DRB1.PHENO.glm.logistic.hybrid .
    End time: Thu May  6 04:01:30 2021
    PLINK v2.00a2LM 64-bit Intel (28 Nov 2019)     www.cog-genomics.org/plink/2.0/
    (C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
    Logging to /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G/4field/GLM.conditional/RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.AA.HLA_B.log.
    Options in effect:
      --bfile /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G/4field/RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38
      --condition AA_B_45_31356825_exon2_E
      --covar /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/COVARIATES.Itals_mg_noDups.UNRELATED.SampleNameUpdated.txt
      --covar-name GENDER,age_at_onset,PC1,PC2,PC3,PC4,PC5,PC6,PC10
      --covar-variance-standardize
      --glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err
      --out /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G/4field/GLM.conditional/RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.AA.HLA_B
      --pheno /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/COVARIATES.Itals_mg_noDups.UNRELATED.SampleNameUpdated.txt
      --pheno-name PHENO
    
    Start time: Thu May  6 04:01:30 2021
    257652 MiB RAM detected; reserving 128826 MiB for main workspace.
    Using up to 56 threads (change this with --threads).
    3497 samples (1583 females, 1914 males; 3497 founders) loaded from
    /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G/4field/RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.fam.
    10828 variants loaded from
    /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G/4field/RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.bim.
    1 binary phenotype loaded (919 cases, 2478 controls).
    9 covariates loaded from /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/COVARIATES.Itals_mg_noDups.UNRELATED.SampleNameUpdated.txt.
    --covar-variance-standardize: 9 covariates transformed.
    Calculating allele frequencies... 0%done.
    --glm logistic-Firth hybrid regression on phenotype 'PHENO': 0%done.
    Results written to /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G/4field/GLM.conditional/RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.AA.HLA_B.PHENO.glm.logistic.hybrid .
    End time: Thu May  6 04:01:45 2021


    [-] Unloading plink  1.9.0-beta4.4  on cn3451 
    [+] Loading plink  2.0-dev-20191128 
    
    The following have been reloaded with a version change:
      1) plink/1.9.0-beta4.4 => plink/2.0-dev-20191128
    



```bash
%%bash
# reformat file for meta analysis
cat toMetaprep.glm.updated.v2.HLA.R
```

    #!/usr/bin/env Rscript
    
    args <- commandArgs(trailingOnly=TRUE)
    if (length(args) != 2) {
      stop("USAGE: Rscript toMetaprep.glm.updated.v2.R args[1] args[2]
      where args[1] = glm_results
            args[2] = OutputFileName")
    }
    
    require(data.table)
    require(tidyverse)
    data <- fread(args[1],header=T)
    
    # Reformat
    data$A2 <- ifelse(data$A1 == data$ALT, data$REF, data$ALT)
    data$EffectAllele <- ifelse(data$A1_FREQ < 0.5, data$A1, data$A2)
    data$OtherAllele <- ifelse(data$A1_FREQ < 0.5, data$A2, data$A1)
    data$BETA <- ifelse(data$EffectAllele == data$A1, log(data$OR), log(data$OR)*-1)
    data$SE <- data$`LOG(OR)_SE`
    
    ## freq
    data$Freq_EffectAllele <- ifelse(data$EffectAllele == data$A1, data$A1_FREQ, 1-data$A1_FREQ)
    data$Freq_EffectAllele.CASE <- ifelse(data$EffectAllele == data$A1, data$A1_CASE_FREQ, 1-data$A1_CASE_FREQ)
    data$Freq_EffectAllele.CTRL <- ifelse(data$EffectAllele == data$A1, data$A1_CTRL_FREQ, 1-data$A1_CTRL_FREQ)
    
    ## counts
    data$EffectAllele_CT <- ifelse(data$EffectAllele == data$A1, data$A1_CT, data$ALLELE_CT-data$A1_CT)
    data$EffectAllele_CT.CASE <- ifelse(data$EffectAllele == data$A1, data$A1_CASE_CT, data$CASE_ALLELE_CT-data$A1_CASE_CT)
    data$EffectAllele_CT.CTRL <- ifelse(data$EffectAllele == data$A1, data$A1_CTRL_CT, data$CTRL_ALLELE_CT-data$A1_CTRL_CT)
    
    data$marker <- paste(data$ID,data$`#CHROM`,data$POS,data$OtherAllele,data$EffectAllele,sep=":")
    data1 <- data %>% rename(CHROM = `#CHROM`) %>% 
             select(CHROM,POS,ID,marker,EffectAllele,OtherAllele,
                    OR,BETA,SE,P,Freq_EffectAllele,Freq_EffectAllele.CASE,Freq_EffectAllele.CTRL,
                    ALLELE_CT,CASE_ALLELE_CT,CTRL_ALLELE_CT,
                    EffectAllele_CT,EffectAllele_CT.CASE,EffectAllele_CT.CTRL) %>%
             filter(BETA > -5 & BETA < 5) %>%
             filter(!is.na(BETA))
             
    dups <- data1 %>% select(marker,P) %>% filter(marker == duplicated(marker))
    temp1 <- subset(data1, data1$marker %in% dups$marker) %>%
             group_by(marker) %>%
             filter(P == min(P)) %>%
             data.frame()
    temp2 <- subset(data1, !(data1$marker %in% temp1$marker))
    data2 <- rbind(temp1,temp2) %>% arrange(CHROM,POS)
    
    write.table(data2, file=paste("toMeta.",args[2],".tab", sep=""), quote = F, sep = "\t", row.names = F)
    



```bash
%%bash
out="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G/4field"
cd $out/GLM

module load R/3.5.2
chmod u+x ../../../toMetaprep.glm.updated.v2.HLA.R
../../../toMetaprep.glm.updated.v2.HLA.R RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.PHENO.glm.logistic.hybrid RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38

cd $out/GLM.conditional
../../../toMetaprep.glm.updated.v2.HLA.R RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.HLA_DRB1.PHENO.glm.logistic.hybrid RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.HLA_DRB1
../../../toMetaprep.glm.updated.v2.HLA.R RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.AA.HLA_B.PHENO.glm.logistic.hybrid RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.AA.HLA_B

```

    [-] Unloading gcc  9.2.0  ... 
    [-] Unloading GSL 2.6 for GCC 9.2.0 ... 
    [-] Unloading openmpi 4.0.5  for GCC 9.2.0 
    [-] Unloading ImageMagick  7.0.8  on cn3451 
    [-] Unloading HDF5  1.10.4 
    [-] Unloading NetCDF 4.7.4_gcc9.2.0 
    [-] Unloading pandoc  2.13  on cn3451 
    [-] Unloading pcre2 10.21  ... 
    [-] Unloading R 4.0.5 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading GSL 2.4 for GCC 7.2.0 ... 
    [-] Unloading gcc  7.3.0  ... 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading openmpi 3.0.2  for GCC 7.3.0 
    [+] Loading ImageMagick  7.0.8  on cn3451 
    [+] Loading HDF5  1.10.4 
    [+] Loading pandoc  2.13  on cn3451 
    [+] Loading R 3.5.2 
    
    The following have been reloaded with a version change:
      1) GSL/2.6_gcc-9.2.0 => GSL/2.4_gcc-7.2.0     3) gcc/9.2.0 => gcc/7.3.0
      2) R/4.0 => R/3.5.2
    
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()



```python
import pandas as pd
pd.read_csv("Imputation.1000G/4field/GLM/toMeta.RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.tab",sep="\t").head()
```




<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>CHROM</th>
      <th>POS</th>
      <th>ID</th>
      <th>marker</th>
      <th>EffectAllele</th>
      <th>OtherAllele</th>
      <th>OR</th>
      <th>BETA</th>
      <th>SE</th>
      <th>P</th>
      <th>Freq_EffectAllele</th>
      <th>Freq_EffectAllele.CASE</th>
      <th>Freq_EffectAllele.CTRL</th>
      <th>ALLELE_CT</th>
      <th>CASE_ALLELE_CT</th>
      <th>CTRL_ALLELE_CT</th>
      <th>EffectAllele_CT</th>
      <th>EffectAllele_CT.CASE</th>
      <th>EffectAllele_CT.CTRL</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>6</td>
      <td>25006557</td>
      <td>rs1410689</td>
      <td>rs1410689:6:25006557:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>1.102910</td>
      <td>0.097952</td>
      <td>0.057720</td>
      <td>0.089686</td>
      <td>0.481013</td>
      <td>0.516520</td>
      <td>0.467635</td>
      <td>6636</td>
      <td>1816</td>
      <td>4820</td>
      <td>3192</td>
      <td>938</td>
      <td>2254</td>
    </tr>
    <tr>
      <th>1</th>
      <td>6</td>
      <td>25010771</td>
      <td>rs10946741</td>
      <td>rs10946741:6:25010771:A:G</td>
      <td>G</td>
      <td>A</td>
      <td>1.101830</td>
      <td>0.096972</td>
      <td>0.057475</td>
      <td>0.091567</td>
      <td>0.497289</td>
      <td>0.533553</td>
      <td>0.483617</td>
      <td>6640</td>
      <td>1818</td>
      <td>4822</td>
      <td>3302</td>
      <td>970</td>
      <td>2332</td>
    </tr>
    <tr>
      <th>2</th>
      <td>6</td>
      <td>25013892</td>
      <td>rs9467377</td>
      <td>rs9467377:6:25013892:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>1.033620</td>
      <td>0.033067</td>
      <td>0.147611</td>
      <td>0.822731</td>
      <td>0.037361</td>
      <td>0.039604</td>
      <td>0.036514</td>
      <td>6638</td>
      <td>1818</td>
      <td>4820</td>
      <td>248</td>
      <td>72</td>
      <td>176</td>
    </tr>
    <tr>
      <th>3</th>
      <td>6</td>
      <td>25015453</td>
      <td>rs2183067</td>
      <td>rs2183067:6:25015453:A:C</td>
      <td>C</td>
      <td>A</td>
      <td>0.919167</td>
      <td>-0.084287</td>
      <td>0.057558</td>
      <td>0.143084</td>
      <td>0.460193</td>
      <td>0.425028</td>
      <td>0.473433</td>
      <td>6632</td>
      <td>1814</td>
      <td>4818</td>
      <td>3052</td>
      <td>771</td>
      <td>2281</td>
    </tr>
    <tr>
      <th>4</th>
      <td>6</td>
      <td>25020792</td>
      <td>rs34632338</td>
      <td>rs34632338:6:25020792:C:A</td>
      <td>A</td>
      <td>C</td>
      <td>1.189810</td>
      <td>0.173794</td>
      <td>0.127172</td>
      <td>0.171755</td>
      <td>0.052568</td>
      <td>0.057947</td>
      <td>0.050541</td>
      <td>6620</td>
      <td>1812</td>
      <td>4808</td>
      <td>348</td>
      <td>105</td>
      <td>243</td>
    </tr>
  </tbody>
</table>
</div>




```python
import pandas as pd
pd.read_csv("Imputation.1000G/4field/GLM.conditional/toMeta.RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.HLA_DRB1.tab",sep="\t").head()
```




<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>CHROM</th>
      <th>POS</th>
      <th>ID</th>
      <th>marker</th>
      <th>EffectAllele</th>
      <th>OtherAllele</th>
      <th>OR</th>
      <th>BETA</th>
      <th>SE</th>
      <th>P</th>
      <th>Freq_EffectAllele</th>
      <th>Freq_EffectAllele.CASE</th>
      <th>Freq_EffectAllele.CTRL</th>
      <th>ALLELE_CT</th>
      <th>CASE_ALLELE_CT</th>
      <th>CTRL_ALLELE_CT</th>
      <th>EffectAllele_CT</th>
      <th>EffectAllele_CT.CASE</th>
      <th>EffectAllele_CT.CTRL</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>6</td>
      <td>25006557</td>
      <td>rs1410689</td>
      <td>rs1410689:6:25006557:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>1.100750</td>
      <td>0.095992</td>
      <td>0.058190</td>
      <td>0.099016</td>
      <td>0.481013</td>
      <td>0.516520</td>
      <td>0.467635</td>
      <td>6636</td>
      <td>1816</td>
      <td>4820</td>
      <td>3192</td>
      <td>938</td>
      <td>2254</td>
    </tr>
    <tr>
      <th>1</th>
      <td>6</td>
      <td>25010771</td>
      <td>rs10946741</td>
      <td>rs10946741:6:25010771:A:G</td>
      <td>G</td>
      <td>A</td>
      <td>1.103310</td>
      <td>0.098315</td>
      <td>0.057927</td>
      <td>0.089644</td>
      <td>0.497289</td>
      <td>0.533553</td>
      <td>0.483617</td>
      <td>6640</td>
      <td>1818</td>
      <td>4822</td>
      <td>3302</td>
      <td>970</td>
      <td>2332</td>
    </tr>
    <tr>
      <th>2</th>
      <td>6</td>
      <td>25013892</td>
      <td>rs9467377</td>
      <td>rs9467377:6:25013892:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>1.019710</td>
      <td>0.019518</td>
      <td>0.148366</td>
      <td>0.895324</td>
      <td>0.037361</td>
      <td>0.039604</td>
      <td>0.036514</td>
      <td>6638</td>
      <td>1818</td>
      <td>4820</td>
      <td>248</td>
      <td>72</td>
      <td>176</td>
    </tr>
    <tr>
      <th>3</th>
      <td>6</td>
      <td>25015453</td>
      <td>rs2183067</td>
      <td>rs2183067:6:25015453:A:C</td>
      <td>C</td>
      <td>A</td>
      <td>0.918365</td>
      <td>-0.085160</td>
      <td>0.057946</td>
      <td>0.141659</td>
      <td>0.460193</td>
      <td>0.425028</td>
      <td>0.473433</td>
      <td>6632</td>
      <td>1814</td>
      <td>4818</td>
      <td>3052</td>
      <td>771</td>
      <td>2281</td>
    </tr>
    <tr>
      <th>4</th>
      <td>6</td>
      <td>25020792</td>
      <td>rs34632338</td>
      <td>rs34632338:6:25020792:C:A</td>
      <td>A</td>
      <td>C</td>
      <td>1.203780</td>
      <td>0.185467</td>
      <td>0.127887</td>
      <td>0.146985</td>
      <td>0.052568</td>
      <td>0.057947</td>
      <td>0.050541</td>
      <td>6620</td>
      <td>1812</td>
      <td>4808</td>
      <td>348</td>
      <td>105</td>
      <td>243</td>
    </tr>
  </tbody>
</table>
</div>




```python
import pandas as pd
pd.read_csv("Imputation.1000G/4field/GLM.conditional/toMeta.RESULT_MG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.AA.HLA_B.tab",sep="\t").head()
```




<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>CHROM</th>
      <th>POS</th>
      <th>ID</th>
      <th>marker</th>
      <th>EffectAllele</th>
      <th>OtherAllele</th>
      <th>OR</th>
      <th>BETA</th>
      <th>SE</th>
      <th>P</th>
      <th>Freq_EffectAllele</th>
      <th>Freq_EffectAllele.CASE</th>
      <th>Freq_EffectAllele.CTRL</th>
      <th>ALLELE_CT</th>
      <th>CASE_ALLELE_CT</th>
      <th>CTRL_ALLELE_CT</th>
      <th>EffectAllele_CT</th>
      <th>EffectAllele_CT.CASE</th>
      <th>EffectAllele_CT.CTRL</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>6</td>
      <td>25006557</td>
      <td>rs1410689</td>
      <td>rs1410689:6:25006557:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>1.10842</td>
      <td>0.102936</td>
      <td>0.057966</td>
      <td>0.075760</td>
      <td>0.481013</td>
      <td>0.516520</td>
      <td>0.467635</td>
      <td>6636</td>
      <td>1816</td>
      <td>4820</td>
      <td>3192</td>
      <td>938</td>
      <td>2254</td>
    </tr>
    <tr>
      <th>1</th>
      <td>6</td>
      <td>25010771</td>
      <td>rs10946741</td>
      <td>rs10946741:6:25010771:A:G</td>
      <td>G</td>
      <td>A</td>
      <td>1.10718</td>
      <td>0.101816</td>
      <td>0.057711</td>
      <td>0.077689</td>
      <td>0.497289</td>
      <td>0.533553</td>
      <td>0.483617</td>
      <td>6640</td>
      <td>1818</td>
      <td>4822</td>
      <td>3302</td>
      <td>970</td>
      <td>2332</td>
    </tr>
    <tr>
      <th>2</th>
      <td>6</td>
      <td>25013892</td>
      <td>rs9467377</td>
      <td>rs9467377:6:25013892:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>1.04056</td>
      <td>0.039759</td>
      <td>0.147933</td>
      <td>0.788126</td>
      <td>0.037361</td>
      <td>0.039604</td>
      <td>0.036514</td>
      <td>6638</td>
      <td>1818</td>
      <td>4820</td>
      <td>248</td>
      <td>72</td>
      <td>176</td>
    </tr>
    <tr>
      <th>3</th>
      <td>6</td>
      <td>25015453</td>
      <td>rs2183067</td>
      <td>rs2183067:6:25015453:A:C</td>
      <td>C</td>
      <td>A</td>
      <td>0.91840</td>
      <td>-0.085122</td>
      <td>0.057796</td>
      <td>0.140799</td>
      <td>0.460193</td>
      <td>0.425028</td>
      <td>0.473433</td>
      <td>6632</td>
      <td>1814</td>
      <td>4818</td>
      <td>3052</td>
      <td>771</td>
      <td>2281</td>
    </tr>
    <tr>
      <th>4</th>
      <td>6</td>
      <td>25020792</td>
      <td>rs34632338</td>
      <td>rs34632338:6:25020792:C:A</td>
      <td>A</td>
      <td>C</td>
      <td>1.19952</td>
      <td>0.181921</td>
      <td>0.127423</td>
      <td>0.153382</td>
      <td>0.052568</td>
      <td>0.057947</td>
      <td>0.050541</td>
      <td>6620</td>
      <td>1812</td>
      <td>4808</td>
      <td>348</td>
      <td>105</td>
      <td>243</td>
    </tr>
  </tbody>
</table>
</div>



#### EOMG


```bash
%%bash
# Create updated sample same covariate file for EOMG i.e. FID = IID

module load R/3.5.2
R --vanilla --no-save

require(data.table)
require(tidyverse)

data <- fread("/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Analysis.GLM.Onset.hg38.rerun/EOMG/COVARIATES.Itals.EOMG.txt",header=T) %>%
        mutate(FID = IID)
head(data)
write.table(data,"COVARIATES.Itals.EOMG.SampleNameUpdated.txt",quote=F,sep="\t",col.names=T,row.names=F)

```

    
    R version 3.5.2 (2018-12-20) -- "Eggshell Igloo"
    Copyright (C) 2018 The R Foundation for Statistical Computing
    Platform: x86_64-pc-linux-gnu (64-bit)
    
    R is free software and comes with ABSOLUTELY NO WARRANTY.
    You are welcome to redistribute it under certain conditions.
    Type 'license()' or 'licence()' for distribution details.
    
    R is a collaborative project with many contributors.
    Type 'contributors()' for more information and
    'citation()' on how to cite R or R packages in publications.
    
    Type 'demo()' for some demos, 'help()' for on-line help, or
    'help.start()' for an HTML browser interface to help.
    Type 'q()' to quit R.
    
    > 
    > require(data.table)
    > require(tidyverse)
    > 
    > data <- fread("/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Analysis.GLM.Onset.hg38.rerun/EOMG/COVARIATES.Itals.EOMG.txt",header=T) %>%
    +         mutate(FID = IID)
    > head(data)
           FID      IID PAT MAT SEX PHENO genderONFILE age_at_onset diagnosis
    1  1456-BT  1456-BT   0   0   2     1            1           36   control
    2  1496-PD  1496-PD   0   0   2     1            1           20   control
    3  1713-SS  1713-SS   0   0   2     1            1           25   control
    4 2007-103 2007-103   0   0   1     1            0           25   control
    5  2007-11  2007-11   0   0   1     1            0           29   control
    6 2007-137 2007-137   0   0   1     1            0           31   control
      GENDER           FID_IID          PC1         PC2          PC3          PC4
    1      1   1456-BT_1456-BT -0.070285740 -0.06310271 -0.011045250 -0.001020783
    2      1   1496-PD_1496-PD -0.012274090 -0.02077367 -0.005937213 -0.024044650
    3      1   1713-SS_1713-SS  0.028196240 -0.01245390  0.011958100  0.062919500
    4      0 2007-103_2007-103  0.012393700 -0.01603721  0.078889650 -0.025766590
    5      0   2007-11_2007-11 -0.009087398 -0.03744468 -0.020056840 -0.016304210
    6      0 2007-137_2007-137  0.003682725 -0.01934224  0.042337270  0.025452550
              PC5         PC6         PC7           PC8           PC9          PC10
    1 -0.01888926 -0.06362445 -0.04055051  0.0515868800  0.0048434970  0.0326448800
    2  0.03030711  0.07841835  0.02716239 -0.0259659500 -0.0082065420 -0.0155713500
    3 -0.02962732  0.01473741  0.03008657  0.0009544024 -0.0019816930  0.0006110831
    4  0.01019803 -0.02692307  0.02860149  0.0364650100  0.0006222475 -0.0242537000
    5 -0.01562509 -0.02089466 -0.02027291 -0.0212793900  0.0257809500 -0.0381966700
    6 -0.09196004  0.01584275 -0.04940903  0.0305706500  0.0205519200  0.0110589700
    > write.table(data,"COVARIATES.Itals.EOMG.SampleNameUpdated.txt",quote=F,sep="\t",col.names=T,row.names=F)
    > 


    [-] Unloading gcc  9.2.0  ... 
    [-] Unloading GSL 2.6 for GCC 9.2.0 ... 
    [-] Unloading openmpi 4.0.5  for GCC 9.2.0 
    [-] Unloading ImageMagick  7.0.8  on cn3174 
    [-] Unloading HDF5  1.10.4 
    [-] Unloading NetCDF 4.7.4_gcc9.2.0 
    [-] Unloading pandoc  2.13  on cn3174 
    [-] Unloading pcre2 10.21  ... 
    [-] Unloading R 4.0.5 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading GSL 2.4 for GCC 7.2.0 ... 
    [-] Unloading gcc  7.3.0  ... 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading openmpi 3.0.2  for GCC 7.3.0 
    [+] Loading ImageMagick  7.0.8  on cn3174 
    [+] Loading HDF5  1.10.4 
    [+] Loading pandoc  2.13  on cn3174 
    [+] Loading R 3.5.2 
    
    The following have been reloaded with a version change:
      1) GSL/2.6_gcc-9.2.0 => GSL/2.4_gcc-7.2.0     3) gcc/9.2.0 => gcc/7.3.0
      2) R/4.0 => R/3.5.2
    
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()



```bash
%%bash
mkdir Imputation.1000G.EOMG
mkdir Imputation.1000G.EOMG/4field
```

script to run imputation: `/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/HATK/hatk.1000G.4field.EOMG.sh`

```
#!/bin/bash

module load plink/1.9.0-beta4.4
module load R/4.0

DIR="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA"

out="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.EOMG/4field"

python3 HATK.py \
    --variants $DIR/Itals.HLAregion.chr6.Liftover.hg38 \
    --hped $DIR/Imputation.1000G/Itals+1000G_REF.EUR.MHC.HLA_IMPUTATION_OUT.hped \
    --4field \
    --pheno $DIR/COVARIATES.Itals.EOMG.SampleNameUpdated.txt \
    --pheno-name PHENO \
    --covar $DIR/COVARIATES.Itals.EOMG.SampleNameUpdated.txt \
    --covar-name GENDER,age_at_onset,PC1,PC8 \
    --out $out/RESULT_EOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38 \
    --imgt 3320 \
    --hg 38 \
    --imgt-dir $DIR/HATK/example/IMGTHLA3320 \
    --multiprocess 8
```




```bash
%%bash
cd HATK

echo "sh hatk.1000G.4field.EOMG.sh" >> hatk.1000G.EOMG.swarm
swarm --file hatk.1000G.EOMG.swarm --logdir swarmOE_hatk -g 120 -t 8 
```

    14299083


##### plink version association - 4field


```bash
%%bash

module load plink/2.0-dev-20191128

DIR="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA"
out="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.EOMG/4field"
mkdir $out/GLM

plink \
--bfile $out/RESULT_EOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38 \
--pheno $DIR/COVARIATES.Itals.EOMG.SampleNameUpdated.txt \
--pheno-name PHENO \
--covar $DIR/COVARIATES.Itals.EOMG.SampleNameUpdated.txt \
--covar-name GENDER,age_at_onset,PC1,PC8 \
--glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err \
--out $out/GLM/RESULT_EOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38 \
--covar-variance-standardize

```

    PLINK v2.00a2LM 64-bit Intel (28 Nov 2019)     www.cog-genomics.org/plink/2.0/
    (C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
    Logging to /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.EOMG/4field/GLM/RESULT_EOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.log.
    Options in effect:
      --bfile /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.EOMG/4field/RESULT_EOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38
      --covar /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/COVARIATES.Itals.EOMG.SampleNameUpdated.txt
      --covar-name GENDER,age_at_onset,PC1,PC8
      --covar-variance-standardize
      --glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err
      --out /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.EOMG/4field/GLM/RESULT_EOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38
      --pheno /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/COVARIATES.Itals.EOMG.SampleNameUpdated.txt
      --pheno-name PHENO
    
    Start time: Wed May  5 00:54:08 2021
    257652 MiB RAM detected; reserving 128826 MiB for main workspace.
    Using up to 56 threads (change this with --threads).
    3497 samples (1583 females, 1914 males; 3497 founders) loaded from
    /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.EOMG/4field/RESULT_EOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.fam.
    10828 variants loaded from
    /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.EOMG/4field/RESULT_EOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.bim.
    1 binary phenotype loaded (361 cases, 666 controls).
    4 covariates loaded from /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/COVARIATES.Itals.EOMG.SampleNameUpdated.txt.
    --covar-variance-standardize: 4 covariates transformed.
    Calculating allele frequencies... 0%done.
    --glm logistic-Firth hybrid regression on phenotype 'PHENO': 0%done.
    Results written to /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.EOMG/4field/GLM/RESULT_EOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.PHENO.glm.logistic.hybrid .
    End time: Wed May  5 00:54:09 2021


    [-] Unloading plink  1.9.0-beta4.4  on cn3451 
    [+] Loading plink  2.0-dev-20191128 
    
    The following have been reloaded with a version change:
      1) plink/1.9.0-beta4.4 => plink/2.0-dev-20191128
    



```bash
%%bash
out="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.EOMG/4field"
cd $out/GLM

module load R/3.5.2
chmod u+x ../../../toMetaprep.glm.updated.v2.HLA.R
../../../toMetaprep.glm.updated.v2.HLA.R RESULT_EOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.PHENO.glm.logistic.hybrid RESULT_EOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38
```

    [-] Unloading gcc  9.2.0  ... 
    [-] Unloading GSL 2.6 for GCC 9.2.0 ... 
    [-] Unloading openmpi 4.0.5  for GCC 9.2.0 
    [-] Unloading ImageMagick  7.0.8  on cn3451 
    [-] Unloading HDF5  1.10.4 
    [-] Unloading NetCDF 4.7.4_gcc9.2.0 
    [-] Unloading pandoc  2.13  on cn3451 
    [-] Unloading pcre2 10.21  ... 
    [-] Unloading R 4.0.5 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading GSL 2.4 for GCC 7.2.0 ... 
    [-] Unloading gcc  7.3.0  ... 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading openmpi 3.0.2  for GCC 7.3.0 
    [+] Loading ImageMagick  7.0.8  on cn3451 
    [+] Loading HDF5  1.10.4 
    [+] Loading pandoc  2.13  on cn3451 
    [+] Loading R 3.5.2 
    
    The following have been reloaded with a version change:
      1) GSL/2.6_gcc-9.2.0 => GSL/2.4_gcc-7.2.0     3) gcc/9.2.0 => gcc/7.3.0
      2) R/4.0 => R/3.5.2
    
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()



```python
import pandas as pd
pd.read_csv("Imputation.1000G.EOMG/4field/GLM/toMeta.RESULT_EOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.tab",sep="\t").head()
```




<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>CHROM</th>
      <th>POS</th>
      <th>ID</th>
      <th>marker</th>
      <th>EffectAllele</th>
      <th>OtherAllele</th>
      <th>OR</th>
      <th>BETA</th>
      <th>SE</th>
      <th>P</th>
      <th>Freq_EffectAllele</th>
      <th>Freq_EffectAllele.CASE</th>
      <th>Freq_EffectAllele.CTRL</th>
      <th>ALLELE_CT</th>
      <th>CASE_ALLELE_CT</th>
      <th>CTRL_ALLELE_CT</th>
      <th>EffectAllele_CT</th>
      <th>EffectAllele_CT.CASE</th>
      <th>EffectAllele_CT.CTRL</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>6</td>
      <td>25006557</td>
      <td>rs1410689</td>
      <td>rs1410689:6:25006557:A:G</td>
      <td>G</td>
      <td>A</td>
      <td>1.125110</td>
      <td>-0.117881</td>
      <td>0.102546</td>
      <td>0.250321</td>
      <td>0.488791</td>
      <td>0.463989</td>
      <td>0.502256</td>
      <td>2052</td>
      <td>722</td>
      <td>1330</td>
      <td>1003</td>
      <td>335</td>
      <td>668</td>
    </tr>
    <tr>
      <th>1</th>
      <td>6</td>
      <td>25010771</td>
      <td>rs10946741</td>
      <td>rs10946741:6:25010771:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>1.077830</td>
      <td>-0.074950</td>
      <td>0.101787</td>
      <td>0.461543</td>
      <td>0.471220</td>
      <td>0.450139</td>
      <td>0.482681</td>
      <td>2050</td>
      <td>722</td>
      <td>1328</td>
      <td>966</td>
      <td>325</td>
      <td>641</td>
    </tr>
    <tr>
      <th>2</th>
      <td>6</td>
      <td>25013892</td>
      <td>rs9467377</td>
      <td>rs9467377:6:25013892:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>1.017250</td>
      <td>0.017103</td>
      <td>0.255881</td>
      <td>0.946723</td>
      <td>0.042885</td>
      <td>0.045706</td>
      <td>0.041353</td>
      <td>2052</td>
      <td>722</td>
      <td>1330</td>
      <td>88</td>
      <td>33</td>
      <td>55</td>
    </tr>
    <tr>
      <th>3</th>
      <td>6</td>
      <td>25015453</td>
      <td>rs2183067</td>
      <td>rs2183067:6:25015453:A:C</td>
      <td>C</td>
      <td>A</td>
      <td>0.917908</td>
      <td>-0.085658</td>
      <td>0.103226</td>
      <td>0.406646</td>
      <td>0.428780</td>
      <td>0.405556</td>
      <td>0.441353</td>
      <td>2050</td>
      <td>720</td>
      <td>1330</td>
      <td>879</td>
      <td>292</td>
      <td>587</td>
    </tr>
    <tr>
      <th>4</th>
      <td>6</td>
      <td>25020792</td>
      <td>rs34632338</td>
      <td>rs34632338:6:25020792:C:A</td>
      <td>A</td>
      <td>C</td>
      <td>1.009780</td>
      <td>0.009732</td>
      <td>0.219214</td>
      <td>0.964584</td>
      <td>0.058594</td>
      <td>0.061111</td>
      <td>0.057229</td>
      <td>2048</td>
      <td>720</td>
      <td>1328</td>
      <td>120</td>
      <td>44</td>
      <td>76</td>
    </tr>
  </tbody>
</table>
</div>



#### LOMG


```bash
%%bash
# Create updated sample same covariate file for EOMG i.e. FID = IID

module load R/3.5.2
R --vanilla --no-save

require(data.table)
require(tidyverse)

data <- fread("/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Analysis.GLM.Onset.hg38.rerun/LOMG/COVARIATES.Itals.LOMG.txt",header=T) %>%
        mutate(FID = IID)
head(data)
write.table(data,"COVARIATES.Itals.LOMG.SampleNameUpdated.txt",quote=F,sep="\t",col.names=T,row.names=F)

```

    
    R version 3.5.2 (2018-12-20) -- "Eggshell Igloo"
    Copyright (C) 2018 The R Foundation for Statistical Computing
    Platform: x86_64-pc-linux-gnu (64-bit)
    
    R is free software and comes with ABSOLUTELY NO WARRANTY.
    You are welcome to redistribute it under certain conditions.
    Type 'license()' or 'licence()' for distribution details.
    
    R is a collaborative project with many contributors.
    Type 'contributors()' for more information and
    'citation()' on how to cite R or R packages in publications.
    
    Type 'demo()' for some demos, 'help()' for on-line help, or
    'help.start()' for an HTML browser interface to help.
    Type 'q()' to quit R.
    
    > 
    > require(data.table)
    > require(tidyverse)
    > 
    > data <- fread("/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Analysis.GLM.Onset.hg38.rerun/LOMG/COVARIATES.Itals.LOMG.txt",header=T) %>%
    +         mutate(FID = IID)
    > head(data)
            FID       IID PAT MAT SEX PHENO genderONFILE age_at_onset diagnosis
    1 10039-DVG 10039-DVG   0   0   2     1            1           60   control
    2  10071-BR  10071-BR   0   0   2     1            1           54   control
    3  10081-BC  10081-BC   0   0   1     1            0           75   control
    4 10137-MMT 10137-MMT   0   0   2     1            1           63   control
    5  10147-OR  10147-OR   0   0   1     1            0           69   control
    6   1022-TS   1022-TS   0   0   1     1            0           55   control
      GENDER             FID_IID         PC1          PC2          PC3
    1      1 10039-DVG_10039-DVG -0.03592598  0.019366710  0.023100850
    2      1   10071-BR_10071-BR -0.01835936  0.007710356  0.030716180
    3      0   10081-BC_10081-BC -0.02419094 -0.077981520  0.014007530
    4      1 10137-MMT_10137-MMT -0.03343514 -0.004523363 -0.006637570
    5      0   10147-OR_10147-OR -0.01370738 -0.045348130  0.006170238
    6      0     1022-TS_1022-TS -0.04447609  0.016539540  0.025123870
                PC4           PC5           PC6          PC7          PC8
    1  0.0030573820  0.0256172700  0.0269780000  0.008886617  0.003452797
    2  0.0272226300  0.0036744140  0.0063453110  0.031166250 -0.041638030
    3  0.0205659000  0.0008613125  0.0426906700 -0.001773494 -0.029734640
    4 -0.0129351200  0.0302583600  0.0550394500 -0.012845950  0.003522238
    5 -0.0006929465 -0.0298753700 -0.0007072009 -0.061430140  0.024932720
    6 -0.0005637007 -0.0043654680 -0.0623927000  0.019055970  0.061366270
               PC9         PC10
    1  0.007054026 -0.025260800
    2 -0.035830180 -0.007057510
    3  0.011656180 -0.023752770
    4  0.038912320  0.043950700
    5 -0.018167600 -0.006129554
    6  0.002510806 -0.069960410
    > write.table(data,"COVARIATES.Itals.LOMG.SampleNameUpdated.txt",quote=F,sep="\t",col.names=T,row.names=F)
    > 


    [-] Unloading gcc  9.2.0  ... 
    [-] Unloading GSL 2.6 for GCC 9.2.0 ... 
    [-] Unloading openmpi 4.0.5  for GCC 9.2.0 
    [-] Unloading ImageMagick  7.0.8  on cn3174 
    [-] Unloading HDF5  1.10.4 
    [-] Unloading NetCDF 4.7.4_gcc9.2.0 
    [-] Unloading pandoc  2.13  on cn3174 
    [-] Unloading pcre2 10.21  ... 
    [-] Unloading R 4.0.5 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading GSL 2.4 for GCC 7.2.0 ... 
    [-] Unloading gcc  7.3.0  ... 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading openmpi 3.0.2  for GCC 7.3.0 
    [+] Loading ImageMagick  7.0.8  on cn3174 
    [+] Loading HDF5  1.10.4 
    [+] Loading pandoc  2.13  on cn3174 
    [+] Loading R 3.5.2 
    
    The following have been reloaded with a version change:
      1) GSL/2.6_gcc-9.2.0 => GSL/2.4_gcc-7.2.0     3) gcc/9.2.0 => gcc/7.3.0
      2) R/4.0 => R/3.5.2
    
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()



```bash
%%bash
mkdir Imputation.1000G.LOMG
mkdir Imputation.1000G.LOMG/4field
```

script to run imputation: `/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/HATK/hatk.1000G.4field.LOMG.sh`

```
#!/bin/bash

module load plink/1.9.0-beta4.4
module load R/4.0

DIR="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA"

out="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.LOMG/4field"

python3 HATK.py \
    --variants $DIR/Itals.HLAregion.chr6.Liftover.hg38 \
    --hped $DIR/Imputation.1000G/Itals+1000G_REF.EUR.MHC.HLA_IMPUTATION_OUT.hped \
    --4field \
    --pheno $DIR/COVARIATES.Itals.LOMG.SampleNameUpdated.txt \
    --pheno-name PHENO \
    --covar $DIR/COVARIATES.Itals.LOMG.SampleNameUpdated.txt \
    --covar-name GENDER,age_at_onset,PC1,PC2,PC3,PC4,PC5,PC7,PC8,PC10 \
    --out $out/RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38 \
    --imgt 3320 \
    --hg 38 \
    --imgt-dir $DIR/HATK/example/IMGTHLA3320 \
    --multiprocess 8
```




```bash
%%bash
cd HATK

echo "sh hatk.1000G.4field.LOMG.sh" >> hatk.1000G.LOMG.swarm
swarm --file hatk.1000G.LOMG.swarm --logdir swarmOE_hatk -g 120 -t 8 
```

    14299242


##### plink version association - 4field


```bash
%%bash

module load plink/2.0-dev-20191128

DIR="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA"
out="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.LOMG/4field"
mkdir $out/GLM

plink \
--bfile $out/RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38 \
--pheno $DIR/COVARIATES.Itals.LOMG.SampleNameUpdated.txt \
--pheno-name PHENO \
--covar $DIR/COVARIATES.Itals.LOMG.SampleNameUpdated.txt \
--covar-name GENDER,age_at_onset,PC1,PC2,PC3,PC4,PC5,PC7,PC8,PC10 \
--glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err \
--out $out/GLM/RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38 \
--covar-variance-standardize

```

    PLINK v2.00a2LM 64-bit Intel (28 Nov 2019)     www.cog-genomics.org/plink/2.0/
    (C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
    Logging to /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.LOMG/4field/GLM/RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.log.
    Options in effect:
      --bfile /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.LOMG/4field/RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38
      --covar /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/COVARIATES.Itals.LOMG.SampleNameUpdated.txt
      --covar-name GENDER,age_at_onset,PC1,PC2,PC3,PC4,PC5,PC7,PC8,PC10
      --covar-variance-standardize
      --glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err
      --out /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.LOMG/4field/GLM/RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38
      --pheno /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/COVARIATES.Itals.LOMG.SampleNameUpdated.txt
      --pheno-name PHENO
    
    Start time: Wed May  5 00:59:31 2021
    257652 MiB RAM detected; reserving 128826 MiB for main workspace.
    Using up to 56 threads (change this with --threads).
    3497 samples (1583 females, 1914 males; 3497 founders) loaded from
    /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.LOMG/4field/RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.fam.
    10828 variants loaded from
    /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.LOMG/4field/RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.bim.
    1 binary phenotype loaded (558 cases, 1812 controls).
    10 covariates loaded from /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/COVARIATES.Itals.LOMG.SampleNameUpdated.txt.
    --covar-variance-standardize: 10 covariates transformed.
    Calculating allele frequencies... 0%done.
    --glm logistic-Firth hybrid regression on phenotype 'PHENO': 0%done.
    Results written to /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.LOMG/4field/GLM/RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.PHENO.glm.logistic.hybrid .
    End time: Wed May  5 00:59:35 2021


    [-] Unloading plink  1.9.0-beta4.4  on cn3451 
    [+] Loading plink  2.0-dev-20191128 
    
    The following have been reloaded with a version change:
      1) plink/1.9.0-beta4.4 => plink/2.0-dev-20191128
    


##### plink version conditional association - 4field


```bash
%%bash

module load plink/2.0-dev-20191128

DIR="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA"
out="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.LOMG/4field"
#mkdir $out/GLM.conditional

plink \
--bfile $out/RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38 \
--condition AA_DRB1_13_32584346_exon2_S \
--pheno $DIR/COVARIATES.Itals.LOMG.SampleNameUpdated.txt \
--pheno-name PHENO \
--covar $DIR/COVARIATES.Itals.LOMG.SampleNameUpdated.txt \
--covar-name GENDER,age_at_onset,PC1,PC2,PC3,PC4,PC5,PC7,PC8,PC10 \
--glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err \
--out $out/GLM.conditional/RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.AA.HLA_DRB1 \
--covar-variance-standardize

plink \
--bfile $out/RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38 \
--condition HLA_DQB1*06:03:02 \
--pheno $DIR/COVARIATES.Itals.LOMG.SampleNameUpdated.txt \
--pheno-name PHENO \
--covar $DIR/COVARIATES.Itals.LOMG.SampleNameUpdated.txt \
--covar-name GENDER,age_at_onset,PC1,PC2,PC3,PC4,PC5,PC7,PC8,PC10 \
--glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err \
--out $out/GLM.conditional/RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.HLA_DQB1 \
--covar-variance-standardize

```

    PLINK v2.00a2LM 64-bit Intel (28 Nov 2019)     www.cog-genomics.org/plink/2.0/
    (C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
    Logging to /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.LOMG/4field/GLM.conditional/RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.AA.HLA_DRB1.log.
    Options in effect:
      --bfile /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.LOMG/4field/RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38
      --condition AA_DRB1_13_32584346_exon2_S
      --covar /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/COVARIATES.Itals.LOMG.SampleNameUpdated.txt
      --covar-name GENDER,age_at_onset,PC1,PC2,PC3,PC4,PC5,PC7,PC8,PC10
      --covar-variance-standardize
      --glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err
      --out /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.LOMG/4field/GLM.conditional/RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.AA.HLA_DRB1
      --pheno /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/COVARIATES.Itals.LOMG.SampleNameUpdated.txt
      --pheno-name PHENO
    
    Start time: Thu May  6 10:20:27 2021
    257652 MiB RAM detected; reserving 128826 MiB for main workspace.
    Using up to 56 threads (change this with --threads).
    3497 samples (1583 females, 1914 males; 3497 founders) loaded from
    /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.LOMG/4field/RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.fam.
    10828 variants loaded from
    /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.LOMG/4field/RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.bim.
    1 binary phenotype loaded (558 cases, 1812 controls).
    10 covariates loaded from /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/COVARIATES.Itals.LOMG.SampleNameUpdated.txt.
    --covar-variance-standardize: 10 covariates transformed.
    Calculating allele frequencies... 0%done.
    --glm logistic-Firth hybrid regression on phenotype 'PHENO': 0%done.
    Results written to /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.LOMG/4field/GLM.conditional/RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.AA.HLA_DRB1.PHENO.glm.logistic.hybrid .
    End time: Thu May  6 10:20:32 2021
    PLINK v2.00a2LM 64-bit Intel (28 Nov 2019)     www.cog-genomics.org/plink/2.0/
    (C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
    Logging to /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.LOMG/4field/GLM.conditional/RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.HLA_DQB1.log.
    Options in effect:
      --bfile /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.LOMG/4field/RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38
      --condition HLA_DQB1*06:03:02
      --covar /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/COVARIATES.Itals.LOMG.SampleNameUpdated.txt
      --covar-name GENDER,age_at_onset,PC1,PC2,PC3,PC4,PC5,PC7,PC8,PC10
      --covar-variance-standardize
      --glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err
      --out /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.LOMG/4field/GLM.conditional/RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.HLA_DQB1
      --pheno /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/COVARIATES.Itals.LOMG.SampleNameUpdated.txt
      --pheno-name PHENO
    
    Start time: Thu May  6 10:20:32 2021
    257652 MiB RAM detected; reserving 128826 MiB for main workspace.
    Using up to 56 threads (change this with --threads).
    3497 samples (1583 females, 1914 males; 3497 founders) loaded from
    /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.LOMG/4field/RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.fam.
    10828 variants loaded from
    /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.LOMG/4field/RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.bim.
    1 binary phenotype loaded (558 cases, 1812 controls).
    10 covariates loaded from /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/COVARIATES.Itals.LOMG.SampleNameUpdated.txt.
    --covar-variance-standardize: 10 covariates transformed.
    Calculating allele frequencies... 0%done.
    --glm logistic-Firth hybrid regression on phenotype 'PHENO': 0%done.
    Results written to /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.LOMG/4field/GLM.conditional/RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.HLA_DQB1.PHENO.glm.logistic.hybrid .
    End time: Thu May  6 10:20:36 2021


    [-] Unloading plink  1.9.0-beta4.4  on cn3451 
    [+] Loading plink  2.0-dev-20191128 
    
    The following have been reloaded with a version change:
      1) plink/1.9.0-beta4.4 => plink/2.0-dev-20191128
    



```bash
%%bash
out="/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Imputation.HLA/Imputation.1000G.LOMG/4field"
cd $out/GLM

module load R/3.5.2
../../../toMetaprep.glm.updated.v2.HLA.R RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.PHENO.glm.logistic.hybrid RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38

cd $out/GLM.conditional
../../../toMetaprep.glm.updated.v2.HLA.R RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.AA.HLA_DRB1.PHENO.glm.logistic.hybrid RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.AA.HLA_DRB1
../../../toMetaprep.glm.updated.v2.HLA.R RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.HLA_DQB1.PHENO.glm.logistic.hybrid RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.HLA_DQB1
```

    [-] Unloading gcc  9.2.0  ... 
    [-] Unloading GSL 2.6 for GCC 9.2.0 ... 
    [-] Unloading openmpi 4.0.5  for GCC 9.2.0 
    [-] Unloading ImageMagick  7.0.8  on cn3451 
    [-] Unloading HDF5  1.10.4 
    [-] Unloading NetCDF 4.7.4_gcc9.2.0 
    [-] Unloading pandoc  2.13  on cn3451 
    [-] Unloading pcre2 10.21  ... 
    [-] Unloading R 4.0.5 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading GSL 2.4 for GCC 7.2.0 ... 
    [-] Unloading gcc  7.3.0  ... 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading openmpi 3.0.2  for GCC 7.3.0 
    [+] Loading ImageMagick  7.0.8  on cn3451 
    [+] Loading HDF5  1.10.4 
    [+] Loading pandoc  2.13  on cn3451 
    [+] Loading R 3.5.2 
    
    The following have been reloaded with a version change:
      1) GSL/2.6_gcc-9.2.0 => GSL/2.4_gcc-7.2.0     3) gcc/9.2.0 => gcc/7.3.0
      2) R/4.0 => R/3.5.2
    
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()



```python
import pandas as pd
pd.read_csv("Imputation.1000G.LOMG/4field/GLM/toMeta.RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.tab",sep="\t").head()
```




<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>CHROM</th>
      <th>POS</th>
      <th>ID</th>
      <th>marker</th>
      <th>EffectAllele</th>
      <th>OtherAllele</th>
      <th>OR</th>
      <th>BETA</th>
      <th>SE</th>
      <th>P</th>
      <th>Freq_EffectAllele</th>
      <th>Freq_EffectAllele.CASE</th>
      <th>Freq_EffectAllele.CTRL</th>
      <th>ALLELE_CT</th>
      <th>CASE_ALLELE_CT</th>
      <th>CTRL_ALLELE_CT</th>
      <th>EffectAllele_CT</th>
      <th>EffectAllele_CT.CASE</th>
      <th>EffectAllele_CT.CTRL</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>6</td>
      <td>25006557</td>
      <td>rs1410689</td>
      <td>rs1410689:6:25006557:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>1.069470</td>
      <td>0.067163</td>
      <td>0.073472</td>
      <td>0.360623</td>
      <td>0.467496</td>
      <td>0.503656</td>
      <td>0.456160</td>
      <td>4584</td>
      <td>1094</td>
      <td>3490</td>
      <td>2143</td>
      <td>551</td>
      <td>1592</td>
    </tr>
    <tr>
      <th>1</th>
      <td>6</td>
      <td>25010771</td>
      <td>rs10946741</td>
      <td>rs10946741:6:25010771:A:G</td>
      <td>G</td>
      <td>A</td>
      <td>1.088620</td>
      <td>0.084911</td>
      <td>0.073230</td>
      <td>0.246248</td>
      <td>0.483224</td>
      <td>0.522810</td>
      <td>0.470807</td>
      <td>4590</td>
      <td>1096</td>
      <td>3494</td>
      <td>2218</td>
      <td>573</td>
      <td>1645</td>
    </tr>
    <tr>
      <th>2</th>
      <td>6</td>
      <td>25013892</td>
      <td>rs9467377</td>
      <td>rs9467377:6:25013892:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>1.020330</td>
      <td>0.020126</td>
      <td>0.194924</td>
      <td>0.917745</td>
      <td>0.034889</td>
      <td>0.035584</td>
      <td>0.034670</td>
      <td>4586</td>
      <td>1096</td>
      <td>3490</td>
      <td>160</td>
      <td>39</td>
      <td>121</td>
    </tr>
    <tr>
      <th>3</th>
      <td>6</td>
      <td>25015453</td>
      <td>rs2183067</td>
      <td>rs2183067:6:25015453:A:C</td>
      <td>C</td>
      <td>A</td>
      <td>0.940021</td>
      <td>-0.061853</td>
      <td>0.073089</td>
      <td>0.397407</td>
      <td>0.474247</td>
      <td>0.437843</td>
      <td>0.485665</td>
      <td>4582</td>
      <td>1094</td>
      <td>3488</td>
      <td>2173</td>
      <td>479</td>
      <td>1694</td>
    </tr>
    <tr>
      <th>4</th>
      <td>6</td>
      <td>25020792</td>
      <td>rs34632338</td>
      <td>rs34632338:6:25020792:C:A</td>
      <td>A</td>
      <td>C</td>
      <td>1.242900</td>
      <td>0.217447</td>
      <td>0.163601</td>
      <td>0.183806</td>
      <td>0.049869</td>
      <td>0.055861</td>
      <td>0.047988</td>
      <td>4572</td>
      <td>1092</td>
      <td>3480</td>
      <td>228</td>
      <td>61</td>
      <td>167</td>
    </tr>
  </tbody>
</table>
</div>




```python
import pandas as pd
pd.read_csv("Imputation.1000G.LOMG/4field/GLM.conditional/toMeta.RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.AA.HLA_DRB1.tab",sep="\t").head()
```




<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>CHROM</th>
      <th>POS</th>
      <th>ID</th>
      <th>marker</th>
      <th>EffectAllele</th>
      <th>OtherAllele</th>
      <th>OR</th>
      <th>BETA</th>
      <th>SE</th>
      <th>P</th>
      <th>Freq_EffectAllele</th>
      <th>Freq_EffectAllele.CASE</th>
      <th>Freq_EffectAllele.CTRL</th>
      <th>ALLELE_CT</th>
      <th>CASE_ALLELE_CT</th>
      <th>CTRL_ALLELE_CT</th>
      <th>EffectAllele_CT</th>
      <th>EffectAllele_CT.CASE</th>
      <th>EffectAllele_CT.CTRL</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>6</td>
      <td>25006557</td>
      <td>rs1410689</td>
      <td>rs1410689:6:25006557:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>1.070750</td>
      <td>0.068359</td>
      <td>0.074010</td>
      <td>0.355669</td>
      <td>0.467496</td>
      <td>0.503656</td>
      <td>0.456160</td>
      <td>4584</td>
      <td>1094</td>
      <td>3490</td>
      <td>2143</td>
      <td>551</td>
      <td>1592</td>
    </tr>
    <tr>
      <th>1</th>
      <td>6</td>
      <td>25010771</td>
      <td>rs10946741</td>
      <td>rs10946741:6:25010771:A:G</td>
      <td>G</td>
      <td>A</td>
      <td>1.093390</td>
      <td>0.089283</td>
      <td>0.073750</td>
      <td>0.226042</td>
      <td>0.483224</td>
      <td>0.522810</td>
      <td>0.470807</td>
      <td>4590</td>
      <td>1096</td>
      <td>3494</td>
      <td>2218</td>
      <td>573</td>
      <td>1645</td>
    </tr>
    <tr>
      <th>2</th>
      <td>6</td>
      <td>25013892</td>
      <td>rs9467377</td>
      <td>rs9467377:6:25013892:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>1.011460</td>
      <td>0.011395</td>
      <td>0.196528</td>
      <td>0.953745</td>
      <td>0.034889</td>
      <td>0.035584</td>
      <td>0.034670</td>
      <td>4586</td>
      <td>1096</td>
      <td>3490</td>
      <td>160</td>
      <td>39</td>
      <td>121</td>
    </tr>
    <tr>
      <th>3</th>
      <td>6</td>
      <td>25015453</td>
      <td>rs2183067</td>
      <td>rs2183067:6:25015453:A:C</td>
      <td>C</td>
      <td>A</td>
      <td>0.934447</td>
      <td>-0.067800</td>
      <td>0.073588</td>
      <td>0.356865</td>
      <td>0.474247</td>
      <td>0.437843</td>
      <td>0.485665</td>
      <td>4582</td>
      <td>1094</td>
      <td>3488</td>
      <td>2173</td>
      <td>479</td>
      <td>1694</td>
    </tr>
    <tr>
      <th>4</th>
      <td>6</td>
      <td>25020792</td>
      <td>rs34632338</td>
      <td>rs34632338:6:25020792:C:A</td>
      <td>A</td>
      <td>C</td>
      <td>1.226730</td>
      <td>0.204352</td>
      <td>0.165053</td>
      <td>0.215682</td>
      <td>0.049869</td>
      <td>0.055861</td>
      <td>0.047988</td>
      <td>4572</td>
      <td>1092</td>
      <td>3480</td>
      <td>228</td>
      <td>61</td>
      <td>167</td>
    </tr>
  </tbody>
</table>
</div>




```python
import pandas as pd
pd.read_csv("Imputation.1000G.LOMG/4field/GLM.conditional/toMeta.RESULT_LOMG.Itals.1000G_REF.EUR.hatk.chr6.4field.hg38.conditioned.HLA_DQB1.tab",sep="\t").head()
```




<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>CHROM</th>
      <th>POS</th>
      <th>ID</th>
      <th>marker</th>
      <th>EffectAllele</th>
      <th>OtherAllele</th>
      <th>OR</th>
      <th>BETA</th>
      <th>SE</th>
      <th>P</th>
      <th>Freq_EffectAllele</th>
      <th>Freq_EffectAllele.CASE</th>
      <th>Freq_EffectAllele.CTRL</th>
      <th>ALLELE_CT</th>
      <th>CASE_ALLELE_CT</th>
      <th>CTRL_ALLELE_CT</th>
      <th>EffectAllele_CT</th>
      <th>EffectAllele_CT.CASE</th>
      <th>EffectAllele_CT.CTRL</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>6</td>
      <td>25006557</td>
      <td>rs1410689</td>
      <td>rs1410689:6:25006557:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>1.072590</td>
      <td>0.070076</td>
      <td>0.074039</td>
      <td>0.343923</td>
      <td>0.467496</td>
      <td>0.503656</td>
      <td>0.456160</td>
      <td>4584</td>
      <td>1094</td>
      <td>3490</td>
      <td>2143</td>
      <td>551</td>
      <td>1592</td>
    </tr>
    <tr>
      <th>1</th>
      <td>6</td>
      <td>25010771</td>
      <td>rs10946741</td>
      <td>rs10946741:6:25010771:A:G</td>
      <td>G</td>
      <td>A</td>
      <td>1.093580</td>
      <td>0.089457</td>
      <td>0.073761</td>
      <td>0.225196</td>
      <td>0.483224</td>
      <td>0.522810</td>
      <td>0.470807</td>
      <td>4590</td>
      <td>1096</td>
      <td>3494</td>
      <td>2218</td>
      <td>573</td>
      <td>1645</td>
    </tr>
    <tr>
      <th>2</th>
      <td>6</td>
      <td>25013892</td>
      <td>rs9467377</td>
      <td>rs9467377:6:25013892:G:A</td>
      <td>A</td>
      <td>G</td>
      <td>1.020230</td>
      <td>0.020028</td>
      <td>0.195910</td>
      <td>0.918580</td>
      <td>0.034889</td>
      <td>0.035584</td>
      <td>0.034670</td>
      <td>4586</td>
      <td>1096</td>
      <td>3490</td>
      <td>160</td>
      <td>39</td>
      <td>121</td>
    </tr>
    <tr>
      <th>3</th>
      <td>6</td>
      <td>25015453</td>
      <td>rs2183067</td>
      <td>rs2183067:6:25015453:A:C</td>
      <td>C</td>
      <td>A</td>
      <td>0.938762</td>
      <td>-0.063193</td>
      <td>0.073511</td>
      <td>0.389980</td>
      <td>0.474247</td>
      <td>0.437843</td>
      <td>0.485665</td>
      <td>4582</td>
      <td>1094</td>
      <td>3488</td>
      <td>2173</td>
      <td>479</td>
      <td>1694</td>
    </tr>
    <tr>
      <th>4</th>
      <td>6</td>
      <td>25020792</td>
      <td>rs34632338</td>
      <td>rs34632338:6:25020792:C:A</td>
      <td>A</td>
      <td>C</td>
      <td>1.247380</td>
      <td>0.221045</td>
      <td>0.164538</td>
      <td>0.179129</td>
      <td>0.049869</td>
      <td>0.055861</td>
      <td>0.047988</td>
      <td>4572</td>
      <td>1092</td>
      <td>3480</td>
      <td>228</td>
      <td>61</td>
      <td>167</td>
    </tr>
  </tbody>
</table>
</div>


