{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "powerful-genre",
   "metadata": {},
   "source": [
    "# Genetic correlation using LDSC between MG and other dementia-related disorders\n",
    "\n",
    "**Start date:** 03-19-2021\n",
    "\n",
    "**Updated date:** 03-19-2021\n",
    "\n",
    "**Analysed by:** Ruth Chia\n",
    "\n",
    "**Working directory:** `/data/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/LDSC`\n",
    "___\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "still-snowboard",
   "metadata": {
    "toc": true
   },
   "source": [
    "<h1>Table of Contents<span class=\"tocSkip\"></span></h1>\n",
    "<div class=\"toc\"><ul class=\"toc-item\"><li><span><a href=\"#Rationale/background\" data-toc-modified-id=\"Rationale/background-1\">Rationale/background</a></span></li><li><span><a href=\"#What-to-do-here\" data-toc-modified-id=\"What-to-do-here-2\">What to do here</a></span></li><li><span><a href=\"#Files-needed-here\" data-toc-modified-id=\"Files-needed-here-3\">Files needed here</a></span></li><li><span><a href=\"#Clone-LDSC-github-repository-to-here\" data-toc-modified-id=\"Clone-LDSC-github-repository-to-here-4\">Clone LDSC github repository to here</a></span></li><li><span><a href=\"#Liftover-hg19-summ-stats-to-hg38\" data-toc-modified-id=\"Liftover-hg19-summ-stats-to-hg38-5\">Liftover hg19 summ stats to hg38</a></span><ul class=\"toc-item\"><li><span><a href=\"#Run-liftover\" data-toc-modified-id=\"Run-liftover-5.1\">Run liftover</a></span></li><li><span><a href=\"#Update-summ-stats-to-hg38\" data-toc-modified-id=\"Update-summ-stats-to-hg38-5.2\">Update summ stats to hg38</a></span></li></ul></li><li><span><a href=\"#Merge-summary-stats-with-hg38-rsid-file-to-get-rsids\" data-toc-modified-id=\"Merge-summary-stats-with-hg38-rsid-file-to-get-rsids-6\">Merge summary stats with hg38 rsid file to get rsids</a></span></li><li><span><a href=\"#Reformat-summ-stats\" data-toc-modified-id=\"Reformat-summ-stats-7\">Reformat summ stats</a></span></li><li><span><a href=\"#Compute-heritability-and-the-LD-Score-regression-intercept-for-MG\" data-toc-modified-id=\"Compute-heritability-and-the-LD-Score-regression-intercept-for-MG-8\">Compute heritability and the LD Score regression intercept for MG</a></span></li><li><span><a href=\"#Run-LDSC-and-genetic-correlation-on-MG-vs.-neurological-diseases\" data-toc-modified-id=\"Run-LDSC-and-genetic-correlation-on-MG-vs.-neurological-diseases-9\">Run LDSC and genetic correlation on MG vs. neurological diseases</a></span><ul class=\"toc-item\"><li><span><a href=\"#Estimating-Heritability,-Genetic-Correlation-and-the-LD-Score-Regression-Intercept\" data-toc-modified-id=\"Estimating-Heritability,-Genetic-Correlation-and-the-LD-Score-Regression-Intercept-9.1\">Estimating Heritability, Genetic Correlation and the LD Score Regression Intercept</a></span></li></ul></li><li><span><a href=\"#Run-LDSC-and-genetic-correlation-on-autoimmune-diseases\" data-toc-modified-id=\"Run-LDSC-and-genetic-correlation-on-autoimmune-diseases-10\">Run LDSC and genetic correlation on autoimmune diseases</a></span><ul class=\"toc-item\"><li><span><a href=\"#Reformat-autoimmune-summ-stats\" data-toc-modified-id=\"Reformat-autoimmune-summ-stats-10.1\">Reformat autoimmune summ stats</a></span></li><li><span><a href=\"#Estimating-Heritability,-Genetic-Correlation-with-autoimmune-diseases\" data-toc-modified-id=\"Estimating-Heritability,-Genetic-Correlation-with-autoimmune-diseases-10.2\">Estimating Heritability, Genetic Correlation with autoimmune diseases</a></span></li><li><span><a href=\"#Estimating-Heritability,-Genetic-Correlation-with-neurodisease-and-autoimmune-diseases\" data-toc-modified-id=\"Estimating-Heritability,-Genetic-Correlation-with-neurodisease-and-autoimmune-diseases-10.3\">Estimating Heritability, Genetic Correlation with neurodisease and autoimmune diseases</a></span></li><li><span><a href=\"#Create-pretty-genetic-correlation-table\" data-toc-modified-id=\"Create-pretty-genetic-correlation-table-10.4\">Create pretty genetic correlation table</a></span></li></ul></li></ul></div>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fantastic-measure",
   "metadata": {},
   "source": [
    "## Rationale/background\n",
    "Run LDSC to determine the genetic correlation between MG and:\n",
    "1. Neurological diseases - AD,PD,LBD and ALS/FTD (disorders with a component of dementia as a disease phenotype)\n",
    "2. Autoimmune diseases\n",
    "Question to answer: ***Is there a correlation?***\n",
    "\n",
    "The reason for doing this is because there is some evidence that there is an association between MG and cognitive function:\n",
    "1. [Association between myasthenia gravis and cognitive function: A systematic review and meta-analysis](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4445185/)\n",
    "\n",
    "2. [Assessment of cognitive function in patients with myasthenia gravis](https://nnjournal.net/article/view/90)\n",
    "\n",
    "3. [Cognitive performance in patients with Myasthenia Gravis: an association with glucocorticosteroid use and depression](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7500821/)\n",
    "\n",
    "4. [Cognition in myasthenia](https://n.neurology.org/content/neurology/39/7/1002.3.full.pdf)\n",
    "\n",
    "\n",
    "## What to do here\n",
    "1. Format summ stats file \n",
    "2. Run LDSC (Wiki how-to to run [Heritability and Genetic Correlation](https://github.com/bulik/ldsc/wiki/Heritability-and-Genetic-Correlation))\n",
    "3. Plot\n",
    "\n",
    "\n",
    "## Files needed here\n",
    "Will need summary stats file for AD (Jansen et al), PD (Nalls et al), LBD (Chia et al) and ALS/FTD (Nicolas et al)\n",
    "\n",
    "1. AD: `/data/ALS_50k/DementiaSeq_LBD/GWAS_LBD.v2/FIGURES_PUBLICATION/AdditionalInfo/LBDoverlap.AD.PD.GWAS/AD_sumstats_Jansenetal.txt`. Note that this based on hg19 and has rsid. Will need to update to hg38..\n",
    "\n",
    "\n",
    "2. PD: `/data/ALS_50k/DementiaSeq_LBD/GWAS_LBD.v2/FIGURES_PUBLICATION/AdditionalInfo/LBDoverlap.AD.PD.GWAS/nallsEtAl2019_excluding23andMe_allVariants.tab`. Note that this based on hg19 and not does not have rsid. Will need to update to hg38 and include rsid.\n",
    "\n",
    "\n",
    "3. LBD: `/data/ALS_50k/DementiaSeq_LBD/GWAS_LBD.v2/LBDpath.LBDclin.ALLcontrols4K.unrelated.glm.noSpanningDels.snvCOVS.hwe1e-6.maf001cases.excludeCentromereFlank10kb.exclVDJ.GnomadTopmedOutliersRemoved.withValidation.FINAL.TIDY.rsid.txt`. This is based on hg38 and has rsid.\n",
    "\n",
    "\n",
    "4. ALS/FTD : as originally located `/Volumes/labchip/ALS_Project_3rd_June_2007/CURRENT_PROJECTS/mega_GWAS/summaryStats/fromSaraB_4sept2019/MODIFIED.CLEAN.MetaAmericans4PCs10BIT3PCsBFr3PCsUK2PCswheaderRsq0.3B5NoMAFwoVRSE2.TBL.tab`. Made a copy to this working directory on biowulf `/data/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/LDSC/MODIFIED.CLEAN.MetaAmericans4PCs10BIT3PCsBFr3PCsUK2PCswheaderRsq0.3B5NoMAFwoVRSE2.TBL.tab`. Note that this summ stats is in hg19 and has no rsid. Will need to update to hg38 and include rsid.\n",
    "\n",
    "\n",
    "5. Summary stats from GWAS catalog for autoimmune diseases.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "acoustic-prairie",
   "metadata": {},
   "source": [
    "## Clone LDSC github repository to here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "decent-brunei",
   "metadata": {},
   "outputs": [],
   "source": [
    "!git clone https://github.com/bulik/ldsc.git"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "entire-immigration",
   "metadata": {},
   "outputs": [],
   "source": [
    "!wget https://data.broadinstitute.org/alkesgroup/LDSCORE/eur_w_ld_chr.tar.bz2\n",
    "!tar -jxvf eur_w_ld_chr.tar.bz2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "respective-winter",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2021-03-19 14:48:57--  https://data.broadinstitute.org/alkesgroup/LDSCORE/w_hm3.snplist.bz2\n",
      "Resolving dtn08-e0 (dtn08-e0)... 10.1.200.244\n",
      "Connecting to dtn08-e0 (dtn08-e0)|10.1.200.244|:3128... connected.\n",
      "Proxy request sent, awaiting response... 301 Moved Permanently\n",
      "Location: https://alkesgroup.broadinstitute.org/LDSCORE/w_hm3.snplist.bz2 [following]\n",
      "--2021-03-19 14:48:57--  https://alkesgroup.broadinstitute.org/LDSCORE/w_hm3.snplist.bz2\n",
      "Connecting to dtn08-e0 (dtn08-e0)|10.1.200.244|:3128... connected.\n",
      "Proxy request sent, awaiting response... 200 OK\n",
      "Length: 4320662 (4.1M) [application/octet-stream]\n",
      "Saving to: 'w_hm3.snplist.bz2'\n",
      "\n",
      "100%[======================================>] 4,320,662   --.-K/s   in 0.05s   \n",
      "\n",
      "2021-03-19 14:48:57 (88.4 MB/s) - 'w_hm3.snplist.bz2' saved [4320662/4320662]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget https://data.broadinstitute.org/alkesgroup/LDSCORE/w_hm3.snplist.bz2\n",
    "!bunzip2 w_hm3.snplist.bz2"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "reported-questionnaire",
   "metadata": {},
   "source": [
    "## Liftover hg19 summ stats to hg38"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "urban-sacrifice",
   "metadata": {},
   "outputs": [],
   "source": [
    "!cp /data/ALS_50k/DementiaSeq_LBD/GWAS_LBD.v2/FIGURES_PUBLICATION/AdditionalInfo/LBDoverlap.AD.PD.GWAS/AD_sumstats_Jansenetal.txt .\n",
    "!cp /data/ALS_50k/DementiaSeq_LBD/GWAS_LBD.v2/FIGURES_PUBLICATION/AdditionalInfo/LBDoverlap.AD.PD.GWAS/nallsEtAl2019_excluding23andMe_allVariants.tab .\n",
    "!cp /data/ALS_50k/DementiaSeq_LBD/GWAS_LBD.v2/LBDpath.LBDclin.ALLcontrols4K.unrelated.glm.noSpanningDels.snvCOVS.hwe1e-6.maf001cases.excludeCentromereFlank10kb.exclVDJ.GnomadTopmedOutliersRemoved.withValidation.FINAL.TIDY.rsid.txt ."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "norman-amber",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "R version 3.5.2 (2018-12-20) -- \"Eggshell Igloo\"\n",
      "Copyright (C) 2018 The R Foundation for Statistical Computing\n",
      "Platform: x86_64-pc-linux-gnu (64-bit)\n",
      "\n",
      "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
      "You are welcome to redistribute it under certain conditions.\n",
      "Type 'license()' or 'licence()' for distribution details.\n",
      "\n",
      "R is a collaborative project with many contributors.\n",
      "Type 'contributors()' for more information and\n",
      "'citation()' on how to cite R or R packages in publications.\n",
      "\n",
      "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
      "'help.start()' for an HTML browser interface to help.\n",
      "Type 'q()' to quit R.\n",
      "\n",
      "> \n",
      "> require(data.table)\n",
      "> require(tidyverse)\n",
      "> \n",
      "> # create function to run prep file for liftover hg19 to hg38\n",
      "> \n",
      "> ForLiftover.hg19.to.hg38 <- function(data,outName) {\n",
      "+     data <- data %>% rename(CHROM_hg19 = CHROM, POS_hg19 = POS) %>%\n",
      "+             mutate(chrom = paste(\"chr\",CHROM_hg19,sep=\"\"))\n",
      "+     bed <- data[,c(\"chrom\",\"POS_hg19\",\"MarkerName\")]\n",
      "+     bed$`POS_hg19_1` <- as.numeric(bed$POS_hg19) + 1\n",
      "+     bed <- bed[,c(1,2,4,3)]\n",
      "+     options(scipen=999, justified=\"none\")\n",
      "+     write.table(bed, paste(outName,\".hg19.ForLiftOver.bed\",sep=\"\"), col.names=F,row.names=F,quote=F,sep=\"\\t\")\n",
      "+ }\n",
      "> \n",
      "> # Read in files and rename columne names so that there is a CHROM and POS column\n",
      "> \n",
      "> ## AD\n",
      "> AD <- fread(\"AD_sumstats_Jansenetal.txt\",header=F)\n",
      "> colnames(AD) <- c(\"MarkerName\",\"CHROM\",\"POS\",\"A1\",\"A2\",\"SNP\",\"Z\",\"P\",\"Nsum\",\"Neff\",\"dir\",\"MAF\",\"BETA\",\"SE\")\n",
      "> \n",
      "> ## PD\n",
      "> PD <- fread(\"/data/ALS_50k/DementiaSeq_LBD/GWAS_LBD.v2/FIGURES_PUBLICATION/AdditionalInfo/LBDoverlap.AD.PD.GWAS/nallsEtAl2019_excluding23andMe_allVariants.tab\",header=T)\n",
      "> PD <- PD %>%\n",
      "+       mutate(snp = SNP) %>%\n",
      "+       separate(snp, c(\"CHROM\",\"POS\"),sep=\":\") %>%\n",
      "+       rename(MarkerName = SNP)\n",
      "> PD$CHROM <- gsub(\"chr\",\"\",PD$CHROM)\n",
      "> \n",
      "> ## ALS/FTD\n",
      "> ALS <- fread(\"MODIFIED.CLEAN.MetaAmericans4PCs10BIT3PCsBFr3PCsUK2PCswheaderRsq0.3B5NoMAFwoVRSE2.TBL.tab\",header=T)\n",
      "> ALS <- ALS %>%\n",
      "+        mutate(snp = MarkerName) %>%\n",
      "+        separate(snp, c(\"CHROM\",\"POS\"),sep=\":\")\n",
      "> \n",
      "> AD.1 <- ForLiftover.hg19.to.hg38(AD,\"AD.Jansen\")\n",
      "> PD.1 <- ForLiftover.hg19.to.hg38(PD,\"PD.Nalls\")\n",
      "> ALS.1 <- ForLiftover.hg19.to.hg38(ALS,\"ALS.Nicolas\")\n",
      "> \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[+] Loading gcc  7.3.0  ... \n",
      "[+] Loading GSL 2.4 for GCC 7.2.0 ... \n",
      "[-] Unloading gcc  7.3.0  ... \n",
      "[+] Loading gcc  7.3.0  ... \n",
      "[+] Loading openmpi 3.0.2  for GCC 7.3.0 \n",
      "[+] Loading ImageMagick  7.0.8  on cn2074 \n",
      "[+] Loading HDF5  1.10.4 \n",
      "[+] Loading pandoc  2.11.4  on cn2074 \n",
      "[+] Loading R 3.5.2 \n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "module load R/3.5.2\n",
    "R --vanilla --no-save\n",
    "\n",
    "require(data.table)\n",
    "require(tidyverse)\n",
    "\n",
    "# create function to run prep file for liftover hg19 to hg38\n",
    "\n",
    "ForLiftover.hg19.to.hg38 <- function(data,outName) {\n",
    "    data <- data %>% rename(CHROM_hg19 = CHROM, POS_hg19 = POS) %>%\n",
    "            mutate(chrom = paste(\"chr\",CHROM_hg19,sep=\"\"))\n",
    "    bed <- data[,c(\"chrom\",\"POS_hg19\",\"MarkerName\")]\n",
    "    bed$`POS_hg19_1` <- as.numeric(bed$POS_hg19) + 1\n",
    "    bed <- bed[,c(1,2,4,3)]\n",
    "    options(scipen=999, justified=\"none\")\n",
    "    write.table(bed, paste(outName,\".hg19.ForLiftOver.bed\",sep=\"\"), col.names=F,row.names=F,quote=F,sep=\"\\t\")\n",
    "}\n",
    "\n",
    "# Read in files and rename columne names so that there is a CHROM and POS column\n",
    "\n",
    "## AD\n",
    "AD <- fread(\"AD_sumstats_Jansenetal.txt\",header=F)\n",
    "colnames(AD) <- c(\"MarkerName\",\"CHROM\",\"POS\",\"A1\",\"A2\",\"SNP\",\"Z\",\"P\",\"Nsum\",\"Neff\",\"dir\",\"MAF\",\"BETA\",\"SE\")\n",
    "\n",
    "## PD\n",
    "PD <- fread(\"/data/ALS_50k/DementiaSeq_LBD/GWAS_LBD.v2/FIGURES_PUBLICATION/AdditionalInfo/LBDoverlap.AD.PD.GWAS/nallsEtAl2019_excluding23andMe_allVariants.tab\",header=T)\n",
    "PD <- PD %>%\n",
    "      mutate(snp = SNP) %>%\n",
    "      separate(snp, c(\"CHROM\",\"POS\"),sep=\":\") %>%\n",
    "      rename(MarkerName = SNP)\n",
    "PD$CHROM <- gsub(\"chr\",\"\",PD$CHROM)\n",
    "\n",
    "## ALS/FTD\n",
    "ALS <- fread(\"MODIFIED.CLEAN.MetaAmericans4PCs10BIT3PCsBFr3PCsUK2PCswheaderRsq0.3B5NoMAFwoVRSE2.TBL.tab\",header=T)\n",
    "ALS <- ALS %>%\n",
    "       mutate(snp = MarkerName) %>%\n",
    "       separate(snp, c(\"CHROM\",\"POS\"),sep=\":\")\n",
    "\n",
    "AD.1 <- ForLiftover.hg19.to.hg38(AD,\"AD.Jansen\")\n",
    "PD.1 <- ForLiftover.hg19.to.hg38(PD,\"PD.Nalls\")\n",
    "ALS.1 <- ForLiftover.hg19.to.hg38(ALS,\"ALS.Nicolas\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "japanese-teddy",
   "metadata": {},
   "source": [
    "### Run liftover"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "noted-jesus",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading liftover chains\n",
      "Mapping coordinates\n",
      "Reading liftover chains\n",
      "Mapping coordinates\n",
      "Reading liftover chains\n",
      "Mapping coordinates\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "mkdir LiftOver\n",
    "wget http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/liftOver -O LiftOver/liftOver\n",
    "wget --timestamping 'ftp://hgdownload.soe.ucsc.edu/goldenPath/hg19/liftOver/hg19ToHg38.over.chain.gz' -O LiftOver/hg19ToHg38.over.chain.gz\n",
    "chmod u+x LiftOver/liftOver\n",
    "\n",
    "for cohort in {AD.Jansen,PD.Nalls,ALS.Nicolas}\n",
    "do\n",
    "LiftOver/liftOver ${cohort}.hg19.ForLiftOver.bed LiftOver/hg19ToHg38.over.chain.gz ${cohort}.hg38.ForLiftOver.bed unlifted.${cohort}.bed\n",
    "grep -v \"#Deleted\" unlifted.${cohort}.bed | awk '{print $4}' > unlifted.${cohort}_not_in_b38.txt\n",
    "awk '{print $4,$2}' ${cohort}.hg38.ForLiftOver.bed > ${cohort}.hg19.ForLiftOver.bed_update-map_to_b38.txt\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "sudden-tractor",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7036 unlifted.AD.Jansen.bed\n",
      "13363781 AD.Jansen.hg19.ForLiftOver.bed_update-map_to_b38.txt\n",
      "2910 unlifted.PD.Nalls.bed\n",
      "17509162 PD.Nalls.hg19.ForLiftOver.bed_update-map_to_b38.txt\n",
      "5744 unlifted.ALS.Nicolas.bed\n",
      "10479018 ALS.Nicolas.hg19.ForLiftOver.bed_update-map_to_b38.txt\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "for cohort in {AD.Jansen,PD.Nalls,ALS.Nicolas}\n",
    "do\n",
    "wc -l unlifted.${cohort}.bed\n",
    "wc -l ${cohort}.hg19.ForLiftOver.bed_update-map_to_b38.txt\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "stuck-valley",
   "metadata": {},
   "source": [
    "### Update summ stats to hg38"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "enhanced-roommate",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "R version 3.5.2 (2018-12-20) -- \"Eggshell Igloo\"\n",
      "Copyright (C) 2018 The R Foundation for Statistical Computing\n",
      "Platform: x86_64-pc-linux-gnu (64-bit)\n",
      "\n",
      "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
      "You are welcome to redistribute it under certain conditions.\n",
      "Type 'license()' or 'licence()' for distribution details.\n",
      "\n",
      "R is a collaborative project with many contributors.\n",
      "Type 'contributors()' for more information and\n",
      "'citation()' on how to cite R or R packages in publications.\n",
      "\n",
      "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
      "'help.start()' for an HTML browser interface to help.\n",
      "Type 'q()' to quit R.\n",
      "\n",
      "> \n",
      "> require(data.table)\n",
      "> require(tidyverse)\n",
      "> \n",
      "> # AD\n",
      "> data <- fread(\"AD_sumstats_Jansenetal.txt\",header=F)\n",
      "> colnames(data) <- c(\"MarkerName\",\"CHROM\",\"POS\",\"A1\",\"A2\",\"SNP\",\"Z\",\"P\",\"Nsum\",\"Neff\",\"dir\",\"MAF\",\"BETA\",\"SE\")\n",
      "> map  <- fread(\"AD.Jansen.hg19.ForLiftOver.bed_update-map_to_b38.txt\",header=F) %>%\n",
      "+         rename(MarkerName = V1, POS_hg38 = V2)\n",
      "> \n",
      "> ## Merge to update positions\n",
      "> data1 <- merge(data,map,by=\"MarkerName\")\n",
      "> data2 <- data1 %>% \n",
      "+          filter(MAF >= 0.01 & MAF <= 0.99) %>%\n",
      "+          select(CHROM,POS_hg38,MarkerName,SNP,A1,A2,MAF,Neff,BETA,SE,P) %>%\n",
      "+          rename(A1_Freq = MAF, N_total = Neff, rsid = SNP)\n",
      "> write.table(data2,\"AD.Jansen.hg38.TIDY.txt\",quote=F,sep=\"\\t\",row.names=F,col.names=T)\n",
      "> \n",
      "> \n",
      "> \n",
      "> # PD\n",
      "> data <- fread(\"/data/ALS_50k/DementiaSeq_LBD/GWAS_LBD.v2/FIGURES_PUBLICATION/AdditionalInfo/LBDoverlap.AD.PD.GWAS/nallsEtAl2019_excluding23andMe_allVariants.tab\",header=T)\n",
      "> data <- data %>%\n",
      "+         mutate(snp = SNP) %>%\n",
      "+         separate(snp, c(\"CHROM\",\"POS\"),sep=\":\") %>%\n",
      "+         mutate(CHROM = str_replace(CHROM, \"chr\",\"\"))\n",
      "> map  <- fread(\"PD.Nalls.hg19.ForLiftOver.bed_update-map_to_b38.txt\",header=F) %>%\n",
      "+         rename(MarkerName = V1, POS_hg38 = V2)\n",
      "> \n",
      "> ## Merge to update positions\n",
      "> data1 <- merge(data,map,by.x=\"SNP\", by.y=\"MarkerName\")\n",
      "> data2 <- data1 %>%\n",
      "+          filter(freq >= 0.01 & freq <= 0.99) %>%\n",
      "+          mutate(N_total = N_cases + N_controls) %>%\n",
      "+          rename(MarkerName = SNP, BETA = b, SE = se, P = p, A1_Freq = freq) %>%\n",
      "+          mutate(rsid = \"NA\") %>%\n",
      "+          select(CHROM,POS_hg38,MarkerName,rsid,A1,A2,A1_Freq,N_total,BETA,SE,P)\n",
      "> write.table(data2,\"PD.Nalls.hg38.TIDY.txt\",quote=F,sep=\"\\t\",row.names=F,col.names=T)\n",
      "> \n",
      "> \n",
      "> ## ALS/FTD\n",
      "> data <- fread(\"MODIFIED.CLEAN.MetaAmericans4PCs10BIT3PCsBFr3PCsUK2PCswheaderRsq0.3B5NoMAFwoVRSE2.TBL.tab\",header=T)\n",
      "> data <- data %>%\n",
      "+        mutate(snp = MarkerName) %>%\n",
      "+        separate(snp, c(\"CHROM\",\"POS\"),sep=\":\")\n",
      "> map  <- fread(\"ALS.Nicolas.hg19.ForLiftOver.bed_update-map_to_b38.txt\",header=F) %>%\n",
      "+         rename(MarkerName = V1, POS_hg38 = V2)\n",
      "> \n",
      "> ## Merge to update positions\n",
      "> data1 <- merge(data,map,by.x=\"MarkerName\", by.y=\"MarkerName\")\n",
      "> data2 <- data1[!grepl(\"\\\\?\",data1$Direction),]\n",
      "> data2$N_cases <- 20806\n",
      "> data2$N_controls <- 59804\n",
      "> data2$N_total <- data2$N_cases + data2$N_controls\n",
      "> data3 <- data2 %>%\n",
      "+          filter(effectAlleleFreq >= 0.01 & effectAlleleFreq <= 0.99) %>%    \n",
      "+          rename(A1 = Allele1, A2 = Allele2, BETA = Effect, SE = StdErr, P = `P-value`, A1_Freq = effectAlleleFreq) %>%\n",
      "+          mutate(rsid = \"NA\") %>%\n",
      "+          select(CHROM,POS_hg38,MarkerName,rsid,A1,A2,A1_Freq,N_total,BETA,SE,P)\n",
      "> write.table(data3,\"ALS.Nicolas.hg38.TIDY.txt\",quote=F,sep=\"\\t\",row.names=F,col.names=T)\n",
      ">          \n",
      "> \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[+] Loading gcc  7.3.0  ... \n",
      "[+] Loading GSL 2.4 for GCC 7.2.0 ... \n",
      "[-] Unloading gcc  7.3.0  ... \n",
      "[+] Loading gcc  7.3.0  ... \n",
      "[+] Loading openmpi 3.0.2  for GCC 7.3.0 \n",
      "[+] Loading ImageMagick  7.0.8  on cn1820 \n",
      "[+] Loading HDF5  1.10.4 \n",
      "[+] Loading pandoc  2.11.4  on cn1820 \n",
      "[+] Loading R 3.5.2 \n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "module load R/3.5.2\n",
    "R --vanilla --no-save\n",
    "\n",
    "require(data.table)\n",
    "require(tidyverse)\n",
    "\n",
    "# AD\n",
    "data <- fread(\"AD_sumstats_Jansenetal.txt\",header=F)\n",
    "colnames(data) <- c(\"MarkerName\",\"CHROM\",\"POS\",\"A1\",\"A2\",\"SNP\",\"Z\",\"P\",\"Nsum\",\"Neff\",\"dir\",\"MAF\",\"BETA\",\"SE\")\n",
    "map  <- fread(\"AD.Jansen.hg19.ForLiftOver.bed_update-map_to_b38.txt\",header=F) %>%\n",
    "        rename(MarkerName = V1, POS_hg38 = V2)\n",
    "\n",
    "## Merge to update positions\n",
    "data1 <- merge(data,map,by=\"MarkerName\")\n",
    "data2 <- data1 %>% \n",
    "         filter(MAF >= 0.01 & MAF <= 0.99) %>%\n",
    "         select(CHROM,POS_hg38,MarkerName,SNP,A1,A2,MAF,Neff,BETA,SE,P) %>%\n",
    "         rename(A1_Freq = MAF, N_total = Neff, rsid = SNP)\n",
    "write.table(data2,\"AD.Jansen.hg38.TIDY.txt\",quote=F,sep=\"\\t\",row.names=F,col.names=T)\n",
    "\n",
    "\n",
    "\n",
    "# PD\n",
    "data <- fread(\"/data/ALS_50k/DementiaSeq_LBD/GWAS_LBD.v2/FIGURES_PUBLICATION/AdditionalInfo/LBDoverlap.AD.PD.GWAS/nallsEtAl2019_excluding23andMe_allVariants.tab\",header=T)\n",
    "data <- data %>%\n",
    "        mutate(snp = SNP) %>%\n",
    "        separate(snp, c(\"CHROM\",\"POS\"),sep=\":\") %>%\n",
    "        mutate(CHROM = str_replace(CHROM, \"chr\",\"\"))\n",
    "map  <- fread(\"PD.Nalls.hg19.ForLiftOver.bed_update-map_to_b38.txt\",header=F) %>%\n",
    "        rename(MarkerName = V1, POS_hg38 = V2)\n",
    "\n",
    "## Merge to update positions\n",
    "data1 <- merge(data,map,by.x=\"SNP\", by.y=\"MarkerName\")\n",
    "data2 <- data1 %>%\n",
    "         filter(freq >= 0.01 & freq <= 0.99) %>%\n",
    "         mutate(N_total = N_cases + N_controls) %>%\n",
    "         rename(MarkerName = SNP, BETA = b, SE = se, P = p, A1_Freq = freq) %>%\n",
    "         mutate(rsid = \"NA\") %>%\n",
    "         select(CHROM,POS_hg38,MarkerName,rsid,A1,A2,A1_Freq,N_total,BETA,SE,P)\n",
    "write.table(data2,\"PD.Nalls.hg38.TIDY.txt\",quote=F,sep=\"\\t\",row.names=F,col.names=T)\n",
    "\n",
    "\n",
    "## ALS/FTD\n",
    "data <- fread(\"MODIFIED.CLEAN.MetaAmericans4PCs10BIT3PCsBFr3PCsUK2PCswheaderRsq0.3B5NoMAFwoVRSE2.TBL.tab\",header=T)\n",
    "data <- data %>%\n",
    "       mutate(snp = MarkerName) %>%\n",
    "       separate(snp, c(\"CHROM\",\"POS\"),sep=\":\")\n",
    "map  <- fread(\"ALS.Nicolas.hg19.ForLiftOver.bed_update-map_to_b38.txt\",header=F) %>%\n",
    "        rename(MarkerName = V1, POS_hg38 = V2)\n",
    "\n",
    "## Merge to update positions\n",
    "data1 <- merge(data,map,by.x=\"MarkerName\", by.y=\"MarkerName\")\n",
    "data2 <- data1[!grepl(\"\\\\?\",data1$Direction),]\n",
    "data2$N_cases <- 20806\n",
    "data2$N_controls <- 59804\n",
    "data2$N_total <- data2$N_cases + data2$N_controls\n",
    "data3 <- data2 %>%\n",
    "         filter(effectAlleleFreq >= 0.01 & effectAlleleFreq <= 0.99) %>%    \n",
    "         rename(A1 = Allele1, A2 = Allele2, BETA = Effect, SE = StdErr, P = `P-value`, A1_Freq = effectAlleleFreq) %>%\n",
    "         mutate(rsid = \"NA\") %>%\n",
    "         select(CHROM,POS_hg38,MarkerName,rsid,A1,A2,A1_Freq,N_total,BETA,SE,P)\n",
    "write.table(data3,\"ALS.Nicolas.hg38.TIDY.txt\",quote=F,sep=\"\\t\",row.names=F,col.names=T)\n",
    "         "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "conditional-toolbox",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "R version 3.5.2 (2018-12-20) -- \"Eggshell Igloo\"\n",
      "Copyright (C) 2018 The R Foundation for Statistical Computing\n",
      "Platform: x86_64-pc-linux-gnu (64-bit)\n",
      "\n",
      "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
      "You are welcome to redistribute it under certain conditions.\n",
      "Type 'license()' or 'licence()' for distribution details.\n",
      "\n",
      "R is a collaborative project with many contributors.\n",
      "Type 'contributors()' for more information and\n",
      "'citation()' on how to cite R or R packages in publications.\n",
      "\n",
      "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
      "'help.start()' for an HTML browser interface to help.\n",
      "Type 'q()' to quit R.\n",
      "\n",
      "> \n",
      "> require(data.table)\n",
      "> require(tidyverse)\n",
      "> \n",
      "> data <- fread(\"/data/NDRS_LNG/ALS_pathway_plink_files_UPDATED.March2020/Analysis.GLM.hg19/MetaAnalysis.VanRheenen/VanRheenen.meta_freqsfixed.tab\",header=T)\n",
      "> data1 <- data %>%\n",
      "+         rename(SNP = snp, A1 = a1, A2 = a2, BETA = b, P = p, MAF = effectAlleleFreq) %>%\n",
      "+         mutate(N = cases + controls) %>%\n",
      "+         filter(MAF >= 0.01 & MAF <= 0.99) %>%\n",
      "+         select(SNP,A1,A2,N,P,BETA)\n",
      "> write.table(data1, \"ALS.VanRheenen.hg38.rsid.forLDSC.txt\",quote=F,sep=\"\\t\",row.names=F,col.names=T)\n",
      "> \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[+] Loading gcc  7.3.0  ... \n",
      "[+] Loading GSL 2.4 for GCC 7.2.0 ... \n",
      "[-] Unloading gcc  7.3.0  ... \n",
      "[+] Loading gcc  7.3.0  ... \n",
      "[+] Loading openmpi 3.0.2  for GCC 7.3.0 \n",
      "[+] Loading ImageMagick  7.0.8  on cn1820 \n",
      "[+] Loading HDF5  1.10.4 \n",
      "[+] Loading pandoc  2.11.4  on cn1820 \n",
      "[+] Loading R 3.5.2 \n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Prep ALS GWAS from Van Rheenen - already has rsid. No need to liftover to hg38\n",
    "# file: `/data/NDRS_LNG/ALS_pathway_plink_files_UPDATED.March2020/Analysis.GLM.hg19/MetaAnalysis.VanRheenen/VanRheenen.meta_freqsfixed.tab`\n",
    "\n",
    "module load R/3.5.2\n",
    "R --vanilla --no-save\n",
    "\n",
    "require(data.table)\n",
    "require(tidyverse)\n",
    "\n",
    "data <- fread(\"/data/NDRS_LNG/ALS_pathway_plink_files_UPDATED.March2020/Analysis.GLM.hg19/MetaAnalysis.VanRheenen/VanRheenen.meta_freqsfixed.tab\",header=T)\n",
    "data1 <- data %>%\n",
    "        rename(SNP = snp, A1 = a1, A2 = a2, BETA = b, P = p, MAF = effectAlleleFreq) %>%\n",
    "        mutate(N = cases + controls) %>%\n",
    "        filter(MAF >= 0.01 & MAF <= 0.99) %>%\n",
    "        select(SNP,A1,A2,N,P,BETA)\n",
    "write.table(data1, \"ALS.VanRheenen.hg38.rsid.forLDSC.txt\",quote=F,sep=\"\\t\",row.names=F,col.names=T)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "applied-timing",
   "metadata": {},
   "source": [
    "## Merge summary stats with hg38 rsid file to get rsids\n",
    "\n",
    "**Notes:**\n",
    "\n",
    "Location of rsid hg38 file: \n",
    "- master file: `/data/ALS_50k/DementiaSeq_LBD/GWAS_LBD.v2/scripts/hg38_rsID_AUG2019_goodFormat.ALLFINAL.txt`\n",
    "- chunked per chr: `/data/ALS_50k/DementiaSeq_LBD/GWAS_LBD.v2/scripts/rsid_perChr/rsid.chr*.txt`\n",
    "\n",
    "Script to merge by chromosome: `getRsid.summStats.R`\n",
    "\n",
    "```bash\n",
    "#!/usr/bin/env Rscript\n",
    "\n",
    "# read command line\n",
    "args <- commandArgs(trailingOnly=TRUE)\n",
    "if (length(args) != 3) {\n",
    "stop(\"USAGE: Rscript getRsid.summStats.R inputSummStatsFile CHNUM OutputFileName\")\n",
    "}\n",
    "\n",
    "# Load required libraries\n",
    "require(data.table)\n",
    "require(tidyverse)\n",
    "\n",
    "# Read in files\n",
    "data <- fread(args[1], header=T)\n",
    "data1 <- subset(data, data$CHROM == args[2])\n",
    "data1$A1 <- toupper(data1$A1)\n",
    "data1$A2 <- toupper(data1$A2)\n",
    "\n",
    "input <- paste(\"/data/ALS_50k/DementiaSeq_LBD/GWAS_LBD.v2/scripts/rsid_perChr/rsid.chr\",args[2],\".txt\",sep=\"\")\n",
    "rsid <- fread(input, header = F)\n",
    "colnames(rsid) <- c(\"markerID\",\"rsID\",\"CHROM\",\"POS_hg38\",\"REF\",\"ALT\")\n",
    "\n",
    "# Merge files - make sure to flip alleles to make sure to account for any REF/ALT mismatch\n",
    "merge1 <- merge(data1, rsid[,2:6], by.x=c(\"CHROM\",\"POS_hg38\",\"A2\",\"A1\"),by.y=c(\"CHROM\",\"POS_hg38\",\"REF\",\"ALT\"))\n",
    "\n",
    "merge2 <- subset(data1, !(data1$MarkerName %in% merge1$MarkerName))\n",
    "merge3 <- merge(merge2, rsid[,2:6], by.x=c(\"CHROM\",\"POS_hg38\",\"A2\",\"A1\"),by.y=c(\"CHROM\",\"POS_hg38\",\"ALT\",\"REF\"))\n",
    "\n",
    "merge4 <- rbind(merge1,merge3)\n",
    "\n",
    "merge5 <- subset(data1, !(data1$MarkerName %in% merge4$MarkerName)) %>% mutate(rsID = \"NA\")\n",
    "\n",
    "mergeFINAL <- rbind(merge4,merge5) %>% arrange(CHROM,POS_hg38)\n",
    "\n",
    "write.table(mergeFINAL, paste(args[3],\".rsid.chr\",args[2],\".txt\",sep=\"\"), quote=F,row.names=F,col.names=T,sep=\"\\t\")\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "hearing-quarter",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[+] Loading gcc  7.3.0  ... \n",
      "[+] Loading GSL 2.4 for GCC 7.2.0 ... \n",
      "[-] Unloading gcc  7.3.0  ... \n",
      "[+] Loading gcc  7.3.0  ... \n",
      "[+] Loading openmpi 3.0.2  for GCC 7.3.0 \n",
      "[+] Loading ImageMagick  7.0.8  on cn1820 \n",
      "[+] Loading HDF5  1.10.4 \n",
      "[+] Loading pandoc  2.11.4  on cn1820 \n",
      "[+] Loading R 3.5.2 \n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "module load R/3.5.2\n",
    "\n",
    "# Run rsid script above\n",
    "for cohort in {AD.Jansen,PD.Nalls,ALS.Nicolas}\n",
    "do\n",
    "    for CHNUM in {1..22}\n",
    "    do\n",
    "    Rscript getRsid.summStats.R \\\n",
    "    ${cohort}.hg38.TIDY.txt \\\n",
    "    ${CHNUM} \\\n",
    "    ${cohort}.hg38\n",
    "    done\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ranging-forum",
   "metadata": {},
   "source": [
    "Merge per chr summ stats file to single large file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "executive-initial",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "head -n 1 AD.Jansen.hg38.rsid.chr1.txt > AD.Jansen.hg38.rsid.txt\n",
    "head -n 1 PD.Nalls.hg38.rsid.chr1.txt > PD.Nalls.hg38.rsid.txt\n",
    "head -n 1 ALS.Nicolas.hg38.rsid.chr1.txt > ALS.Nicolas.hg38.rsid.txt\n",
    "\n",
    "for cohort in {AD.Jansen,PD.Nalls,ALS.Nicolas}\n",
    "do\n",
    "    for CHNUM in {1..22}\n",
    "    do\n",
    "    tail -n +2 ${cohort}.hg38.rsid.chr${CHNUM}.txt >> ${cohort}.hg38.rsid.txt\n",
    "    rm ${cohort}.hg38.rsid.chr${CHNUM}.txt\n",
    "    done\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "racial-poverty",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   9732198 AD.Jansen.hg38.rsid.txt\n",
      "   8949247 ALS.Nicolas.hg38.rsid.txt\n",
      "   8330285 PD.Nalls.hg38.rsid.txt\n",
      "  27011730 total\n"
     ]
    }
   ],
   "source": [
    "!wc -l *.hg38.rsid.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "seventh-sending",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS_hg38</th>\n",
       "      <th>A2</th>\n",
       "      <th>A1</th>\n",
       "      <th>MarkerName</th>\n",
       "      <th>rsid</th>\n",
       "      <th>A1_Freq</th>\n",
       "      <th>N_total</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "      <th>rsID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>779885</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1:715265_T_C</td>\n",
       "      <td>rs12184267</td>\n",
       "      <td>0.040807</td>\n",
       "      <td>381761.0</td>\n",
       "      <td>0.012275</td>\n",
       "      <td>0.005785</td>\n",
       "      <td>0.03384</td>\n",
       "      <td>rs12184267</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>779987</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1:715367_G_A</td>\n",
       "      <td>rs12184277</td>\n",
       "      <td>0.041069</td>\n",
       "      <td>382151.0</td>\n",
       "      <td>0.011285</td>\n",
       "      <td>0.005764</td>\n",
       "      <td>0.05024</td>\n",
       "      <td>rs12184277</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>782105</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1:717485_A_C</td>\n",
       "      <td>rs12184279</td>\n",
       "      <td>0.040576</td>\n",
       "      <td>382180.0</td>\n",
       "      <td>0.011087</td>\n",
       "      <td>0.005797</td>\n",
       "      <td>0.05582</td>\n",
       "      <td>rs12184279</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>785001</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>1:720381_T_G</td>\n",
       "      <td>rs116801199</td>\n",
       "      <td>0.042162</td>\n",
       "      <td>382954.0</td>\n",
       "      <td>0.013052</td>\n",
       "      <td>0.005686</td>\n",
       "      <td>0.02171</td>\n",
       "      <td>rs116801199</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>785910</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1:721290_C_G</td>\n",
       "      <td>rs12565286</td>\n",
       "      <td>0.042378</td>\n",
       "      <td>382779.0</td>\n",
       "      <td>0.013137</td>\n",
       "      <td>0.005673</td>\n",
       "      <td>0.02058</td>\n",
       "      <td>rs12565286</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM  POS_hg38 A2 A1    MarkerName         rsid   A1_Freq   N_total  \\\n",
       "0      1    779885  C  T  1:715265_T_C   rs12184267  0.040807  381761.0   \n",
       "1      1    779987  A  G  1:715367_G_A   rs12184277  0.041069  382151.0   \n",
       "2      1    782105  C  A  1:717485_A_C   rs12184279  0.040576  382180.0   \n",
       "3      1    785001  G  T  1:720381_T_G  rs116801199  0.042162  382954.0   \n",
       "4      1    785910  G  C  1:721290_C_G   rs12565286  0.042378  382779.0   \n",
       "\n",
       "       BETA        SE        P         rsID  \n",
       "0  0.012275  0.005785  0.03384   rs12184267  \n",
       "1  0.011285  0.005764  0.05024   rs12184277  \n",
       "2  0.011087  0.005797  0.05582   rs12184279  \n",
       "3  0.013052  0.005686  0.02171  rs116801199  \n",
       "4  0.013137  0.005673  0.02058   rs12565286  "
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "pd.read_csv(\"AD.Jansen.hg38.rsid.txt\",sep=\"\\t\").head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "checked-action",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS_hg38</th>\n",
       "      <th>A2</th>\n",
       "      <th>A1</th>\n",
       "      <th>MarkerName</th>\n",
       "      <th>rsid</th>\n",
       "      <th>A1_Freq</th>\n",
       "      <th>N_total</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "      <th>rsID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1200</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr1:143118961</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0330</td>\n",
       "      <td>2137</td>\n",
       "      <td>-0.1062</td>\n",
       "      <td>0.1689</td>\n",
       "      <td>0.5293</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1636</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>chr1:143119397</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.9841</td>\n",
       "      <td>2137</td>\n",
       "      <td>0.1292</td>\n",
       "      <td>0.2423</td>\n",
       "      <td>0.5938</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1964</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>chr1:142537398</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.9857</td>\n",
       "      <td>2137</td>\n",
       "      <td>-0.0156</td>\n",
       "      <td>0.2525</td>\n",
       "      <td>0.9509</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>2976</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr1:143120737</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0112</td>\n",
       "      <td>2137</td>\n",
       "      <td>-0.1148</td>\n",
       "      <td>0.2871</td>\n",
       "      <td>0.6894</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>2979</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr1:13124908</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0165</td>\n",
       "      <td>464905</td>\n",
       "      <td>-0.0448</td>\n",
       "      <td>0.1241</td>\n",
       "      <td>0.7179</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM  POS_hg38 A2 A1      MarkerName  rsid  A1_Freq  N_total    BETA  \\\n",
       "0      1      1200  G  A  chr1:143118961   NaN   0.0330     2137 -0.1062   \n",
       "1      1      1636  G  T  chr1:143119397   NaN   0.9841     2137  0.1292   \n",
       "2      1      1964  T  A  chr1:142537398   NaN   0.9857     2137 -0.0156   \n",
       "3      1      2976  C  T  chr1:143120737   NaN   0.0112     2137 -0.1148   \n",
       "4      1      2979  G  A   chr1:13124908   NaN   0.0165   464905 -0.0448   \n",
       "\n",
       "       SE       P rsID  \n",
       "0  0.1689  0.5293  NaN  \n",
       "1  0.2423  0.5938  NaN  \n",
       "2  0.2525  0.9509  NaN  \n",
       "3  0.2871  0.6894  NaN  \n",
       "4  0.1241  0.7179  NaN  "
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "pd.read_csv(\"PD.Nalls.hg38.rsid.txt\",sep=\"\\t\").head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "comic-roots",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS_hg38</th>\n",
       "      <th>A2</th>\n",
       "      <th>A1</th>\n",
       "      <th>MarkerName</th>\n",
       "      <th>rsid</th>\n",
       "      <th>A1_Freq</th>\n",
       "      <th>N_total</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "      <th>rsID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>832873</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1:768253</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6548</td>\n",
       "      <td>80610</td>\n",
       "      <td>0.0248</td>\n",
       "      <td>0.0398</td>\n",
       "      <td>0.5336</td>\n",
       "      <td>rs2977608</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>839494</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1:774874</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1935</td>\n",
       "      <td>80610</td>\n",
       "      <td>0.0335</td>\n",
       "      <td>0.0478</td>\n",
       "      <td>0.4830</td>\n",
       "      <td>rs28810152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>845405</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>1:780785</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8150</td>\n",
       "      <td>80610</td>\n",
       "      <td>-0.0303</td>\n",
       "      <td>0.0464</td>\n",
       "      <td>0.5142</td>\n",
       "      <td>rs2977612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>847601</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1:782981</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1705</td>\n",
       "      <td>80610</td>\n",
       "      <td>0.0332</td>\n",
       "      <td>0.0491</td>\n",
       "      <td>0.4996</td>\n",
       "      <td>rs6594026</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>849670</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1:785050</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8030</td>\n",
       "      <td>80610</td>\n",
       "      <td>-0.0254</td>\n",
       "      <td>0.0422</td>\n",
       "      <td>0.5474</td>\n",
       "      <td>rs2905062</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM  POS_hg38 A2 A1 MarkerName  rsid  A1_Freq  N_total    BETA      SE  \\\n",
       "0      1    832873  C  A   1:768253   NaN   0.6548    80610  0.0248  0.0398   \n",
       "1      1    839494  C  A   1:774874   NaN   0.1935    80610  0.0335  0.0478   \n",
       "2      1    845405  T  A   1:780785   NaN   0.8150    80610 -0.0303  0.0464   \n",
       "3      1    847601  C  T   1:782981   NaN   0.1705    80610  0.0332  0.0491   \n",
       "4      1    849670  G  A   1:785050   NaN   0.8030    80610 -0.0254  0.0422   \n",
       "\n",
       "        P        rsID  \n",
       "0  0.5336   rs2977608  \n",
       "1  0.4830  rs28810152  \n",
       "2  0.5142   rs2977612  \n",
       "3  0.4996   rs6594026  \n",
       "4  0.5474   rs2905062  "
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "pd.read_csv(\"ALS.Nicolas.hg38.rsid.txt\",sep=\"\\t\").head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "systematic-contract",
   "metadata": {},
   "source": [
    "## Reformat summ stats\n",
    "\n",
    "Need to format files because the ldsc .sumstats format requires six pieces of information for each SNP:\n",
    "\n",
    "- A unique identifier (e.g., the rs number)\n",
    "- Allele 1 (effect allele)\n",
    "- Allele 2 (non-effect allele)\n",
    "- Sample size (which often varies from SNP to SNP)\n",
    "- A P-value\n",
    "- A signed summary statistic (beta, OR, log odds, Z-score, etc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "relevant-aggregate",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "R version 3.5.2 (2018-12-20) -- \"Eggshell Igloo\"\n",
      "Copyright (C) 2018 The R Foundation for Statistical Computing\n",
      "Platform: x86_64-pc-linux-gnu (64-bit)\n",
      "\n",
      "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
      "You are welcome to redistribute it under certain conditions.\n",
      "Type 'license()' or 'licence()' for distribution details.\n",
      "\n",
      "R is a collaborative project with many contributors.\n",
      "Type 'contributors()' for more information and\n",
      "'citation()' on how to cite R or R packages in publications.\n",
      "\n",
      "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
      "'help.start()' for an HTML browser interface to help.\n",
      "Type 'q()' to quit R.\n",
      "\n",
      "> \n",
      "> require(data.table)\n",
      "> require(tidyverse)\n",
      "> \n",
      "> # AD\n",
      "> AD <- fread(\"AD.Jansen.hg38.rsid.txt\",header=T) %>%\n",
      "+       mutate(marker = paste(CHROM,POS_hg38,A2,A1,sep=\":\"))\n",
      "> AD$SNP <- ifelse(is.na(AD$rsID), AD$marker, AD$rsID)      \n",
      "> AD.1 <- AD %>% select(SNP,A1,A2,N_total,P,BETA) %>% rename(N = N_total)\n",
      "> write.table(AD.1,\"AD.Jansen.hg38.rsid.forLDSC.txt\",quote=F,sep=\"\\t\",row.names=F,col.names=T)\n",
      "> \n",
      "> # PD\n",
      "> PD <- fread(\"PD.Nalls.hg38.rsid.txt\",header=T) %>%\n",
      "+       mutate(marker = paste(CHROM,POS_hg38,A2,A1,sep=\":\"))\n",
      "> PD$SNP <- ifelse(is.na(PD$rsID), PD$marker, PD$rsID)      \n",
      "> PD.1 <- PD %>% select(SNP,A1,A2,N_total,P,BETA) %>% rename(N = N_total)\n",
      "> write.table(PD.1,\"PD.Nalls.hg38.rsid.forLDSC.txt\",quote=F,sep=\"\\t\",row.names=F,col.names=T)\n",
      "> \n",
      "> # ALS\n",
      "> ALS <- fread(\"ALS.Nicolas.hg38.rsid.txt\",header=T) %>%\n",
      "+        mutate(marker = paste(CHROM,POS_hg38,A2,A1,sep=\":\"))\n",
      "> ALS$SNP <- ifelse(is.na(ALS$rsID), ALS$marker, ALS$rsID)      \n",
      "> ALS.1 <- ALS %>% select(SNP,A1,A2,N_total,P,BETA) %>% rename(N = N_total)\n",
      "> write.table(ALS.1,\"ALS.Nicolas.hg38.rsid.forLDSC.txt\",quote=F,sep=\"\\t\",row.names=F,col.names=T)\n",
      "> \n",
      "> ## LBD\n",
      "> LBD <- fread(\"LBDpath.LBDclin.ALLcontrols4K.unrelated.glm.noSpanningDels.snvCOVS.hwe1e-6.maf001cases.excludeCentromereFlank10kb.exclVDJ.GnomadTopmedOutliersRemoved.withValidation.FINAL.TIDY.rsid.txt\",header=T) %>%\n",
      "+        rename(POS_hg38 = POS, MarkerName = ID, rsid = rsID, A1_Freq = A1_FREQ, N_total = OBS_CT)\n",
      "> LBD <- LBD %>% mutate(marker = paste(CHROM,POS_hg38,A2,A1,sep=\":\"))\n",
      "> LBD$SNP <- ifelse(is.na(LBD$rsid), LBD$marker, LBD$rsid)  \n",
      "> LBD.1 <- LBD %>% select(SNP,A1,A2,N_total,P,BETA) %>% rename(N = N_total)\n",
      "> write.table(LBD.1,\"LBD.Chia.hg38.rsid.forLDSC.txt\",quote=F,sep=\"\\t\",row.names=F,col.names=T)\n",
      "> \n",
      "> # MG\n",
      "> MG <- fread(\"/data/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/USmerged.Itals.hg38/META_MG.USmerged.Itals.UNRELATED.hg38.Rsq03MAF00001.glm_filteredDirection.HetISq80.rsid.txt\",header=T)\n",
      "> MG$N_total <- 36370 + 1873\n",
      "> MG$SNP <- ifelse(is.na(MG$rsID), MG$MarkerName, MG$rsID)  \n",
      "> MG.1 <- MG %>% \n",
      "+         select(SNP,EffectAllele,OtherAllele,N_total,P,beta) %>%\n",
      "+         rename(A1 = EffectAllele, A2 = OtherAllele, BETA = beta) %>% rename(N = N_total)\n",
      "> write.table(MG.1,\"MG.Chia.hg38.rsid.forLDSC.txt\",quote=F,sep=\"\\t\",row.names=F,col.names=T)\n",
      "> \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[+] Loading gcc  7.3.0  ... \n",
      "[+] Loading GSL 2.4 for GCC 7.2.0 ... \n",
      "[-] Unloading gcc  7.3.0  ... \n",
      "[+] Loading gcc  7.3.0  ... \n",
      "[+] Loading openmpi 3.0.2  for GCC 7.3.0 \n",
      "[+] Loading ImageMagick  7.0.8  on cn1820 \n",
      "[+] Loading HDF5  1.10.4 \n",
      "[+] Loading pandoc  2.11.4  on cn1820 \n",
      "[+] Loading R 3.5.2 \n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "module load R/3.5.2\n",
    "R --vanilla --no-save\n",
    "\n",
    "require(data.table)\n",
    "require(tidyverse)\n",
    "\n",
    "# AD\n",
    "AD <- fread(\"AD.Jansen.hg38.rsid.txt\",header=T) %>%\n",
    "      mutate(marker = paste(CHROM,POS_hg38,A2,A1,sep=\":\"))\n",
    "AD$SNP <- ifelse(is.na(AD$rsID), AD$marker, AD$rsID)      \n",
    "AD.1 <- AD %>% select(SNP,A1,A2,N_total,P,BETA) %>% rename(N = N_total)\n",
    "write.table(AD.1,\"AD.Jansen.hg38.rsid.forLDSC.txt\",quote=F,sep=\"\\t\",row.names=F,col.names=T)\n",
    "\n",
    "# PD\n",
    "PD <- fread(\"PD.Nalls.hg38.rsid.txt\",header=T) %>%\n",
    "      mutate(marker = paste(CHROM,POS_hg38,A2,A1,sep=\":\"))\n",
    "PD$SNP <- ifelse(is.na(PD$rsID), PD$marker, PD$rsID)      \n",
    "PD.1 <- PD %>% select(SNP,A1,A2,N_total,P,BETA) %>% rename(N = N_total)\n",
    "write.table(PD.1,\"PD.Nalls.hg38.rsid.forLDSC.txt\",quote=F,sep=\"\\t\",row.names=F,col.names=T)\n",
    "\n",
    "# ALS\n",
    "ALS <- fread(\"ALS.Nicolas.hg38.rsid.txt\",header=T) %>%\n",
    "       mutate(marker = paste(CHROM,POS_hg38,A2,A1,sep=\":\"))\n",
    "ALS$SNP <- ifelse(is.na(ALS$rsID), ALS$marker, ALS$rsID)      \n",
    "ALS.1 <- ALS %>% select(SNP,A1,A2,N_total,P,BETA) %>% rename(N = N_total)\n",
    "write.table(ALS.1,\"ALS.Nicolas.hg38.rsid.forLDSC.txt\",quote=F,sep=\"\\t\",row.names=F,col.names=T)\n",
    "\n",
    "## LBD\n",
    "LBD <- fread(\"LBDpath.LBDclin.ALLcontrols4K.unrelated.glm.noSpanningDels.snvCOVS.hwe1e-6.maf001cases.excludeCentromereFlank10kb.exclVDJ.GnomadTopmedOutliersRemoved.withValidation.FINAL.TIDY.rsid.txt\",header=T) %>%\n",
    "       rename(POS_hg38 = POS, MarkerName = ID, rsid = rsID, A1_Freq = A1_FREQ, N_total = OBS_CT)\n",
    "LBD <- LBD %>% mutate(marker = paste(CHROM,POS_hg38,A2,A1,sep=\":\"))\n",
    "LBD$SNP <- ifelse(is.na(LBD$rsid), LBD$marker, LBD$rsid)  \n",
    "LBD.1 <- LBD %>% filter(A1_FREQ => 0.01 & A1_FREQ <=0.99) %>% select(SNP,A1,A2,N_total,P,BETA) %>% rename(N = N_total)\n",
    "write.table(LBD.1,\"LBD.Chia.hg38.rsid.forLDSC.txt\",quote=F,sep=\"\\t\",row.names=F,col.names=T)\n",
    "\n",
    "# MG\n",
    "MG <- fread(\"/data/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/USmerged.Itals.hg38/META_MG.USmerged.Itals.UNRELATED.hg38.Rsq03MAF00001.glm_filteredDirection.HetISq80.rsid.txt\",header=T)\n",
    "MG$N_total <- 36370 + 1873\n",
    "MG$SNP <- ifelse(is.na(MG$rsID), MG$MarkerName, MG$rsID)  \n",
    "MG.1 <- MG %>% \n",
    "        filter(maf_EA.CASE > 0.01) %>%\n",
    "        select(SNP,EffectAllele,OtherAllele,N_total,P,beta) %>%\n",
    "        rename(A1 = EffectAllele, A2 = OtherAllele, BETA = beta) %>% rename(N = N_total)\n",
    "write.table(MG.1,\"MG.Chia.hg38.rsid.forLDSC.txt\",quote=F,sep=\"\\t\",row.names=F,col.names=T)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "copyrighted-installation",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "R version 3.5.2 (2018-12-20) -- \"Eggshell Igloo\"\n",
      "Copyright (C) 2018 The R Foundation for Statistical Computing\n",
      "Platform: x86_64-pc-linux-gnu (64-bit)\n",
      "\n",
      "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
      "You are welcome to redistribute it under certain conditions.\n",
      "Type 'license()' or 'licence()' for distribution details.\n",
      "\n",
      "R is a collaborative project with many contributors.\n",
      "Type 'contributors()' for more information and\n",
      "'citation()' on how to cite R or R packages in publications.\n",
      "\n",
      "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
      "'help.start()' for an HTML browser interface to help.\n",
      "Type 'q()' to quit R.\n",
      "\n",
      "> \n",
      "> require(data.table)\n",
      "> require(tidyverse)\n",
      "> \n",
      "> # MG\n",
      "> MG <- fread(\"/data/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/USmerged.Itals.hg38/META_MG.USmerged.Itals.UNRELATED.hg38.Rsq03MAF00001.glm_filteredDirection.HetISq80.rsid.txt\",header=T)\n",
      "> MG$N_total <- 36370 + 1873\n",
      "> MG$SNP <- ifelse(is.na(MG$rsID), MG$MarkerName, MG$rsID)  \n",
      "> MG.1 <- MG %>% \n",
      "+         filter(maf_EA > 0.01) %>%\n",
      "+         select(SNP,EffectAllele,OtherAllele,N_total,P,beta) %>%\n",
      "+         rename(A1 = EffectAllele, A2 = OtherAllele, BETA = beta) %>% rename(N = N_total)\n",
      "> write.table(MG.1,\"MG.Chia.hg38.rsid.forLDSC.txt\",quote=F,sep=\"\\t\",row.names=F,col.names=T)\n",
      "> \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[+] Loading gcc  7.3.0  ... \n",
      "[+] Loading GSL 2.4 for GCC 7.2.0 ... \n",
      "[-] Unloading gcc  7.3.0  ... \n",
      "[+] Loading gcc  7.3.0  ... \n",
      "[+] Loading openmpi 3.0.2  for GCC 7.3.0 \n",
      "[+] Loading ImageMagick  7.0.8  on cn3147 \n",
      "[+] Loading HDF5  1.10.4 \n",
      "[+] Loading pandoc  2.13  on cn3147 \n",
      "[+] Loading R 3.5.2 \n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "module load R/3.5.2\n",
    "R --vanilla --no-save\n",
    "\n",
    "require(data.table)\n",
    "require(tidyverse)\n",
    "\n",
    "# MG\n",
    "MG <- fread(\"/data/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/USmerged.Itals.hg38/META_MG.USmerged.Itals.UNRELATED.hg38.Rsq03MAF00001.glm_filteredDirection.HetISq80.rsid.txt\",header=T)\n",
    "MG$N_total <- 36370 + 1873\n",
    "MG$SNP <- ifelse(is.na(MG$rsID), MG$MarkerName, MG$rsID)  \n",
    "MG.1 <- MG %>% \n",
    "        filter(maf_EA > 0.01) %>%\n",
    "        select(SNP,EffectAllele,OtherAllele,N_total,P,beta) %>%\n",
    "        rename(A1 = EffectAllele, A2 = OtherAllele, BETA = beta) %>% rename(N = N_total)\n",
    "write.table(MG.1,\"MG.Chia.hg38.rsid.forLDSC.txt\",quote=F,sep=\"\\t\",row.names=F,col.names=T)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "abroad-sussex",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "*********************************************************************\n",
      "* LD Score Regression (LDSC)\n",
      "* Version 1.0.1\n",
      "* (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane\n",
      "* Broad Institute of MIT and Harvard / MIT Department of Mathematics\n",
      "* GNU General Public License v3\n",
      "*********************************************************************\n",
      "Call: \n",
      "./munge_sumstats.py \\\n",
      "--out AD.Jansen \\\n",
      "--merge-alleles w_hm3.snplist \\\n",
      "--N 452010.0 \\\n",
      "--sumstats AD.Jansen.hg38.rsid.forLDSC.txt \n",
      "\n",
      "Interpreting column names as follows:\n",
      "N:\tSample size\n",
      "A1:\tAllele 1, interpreted as ref allele for signed sumstat.\n",
      "P:\tp-Value\n",
      "BETA:\t[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)\n",
      "A2:\tAllele 2, interpreted as non-ref allele for signed sumstat.\n",
      "SNP:\tVariant ID (e.g., rs number)\n",
      "\n",
      "Reading list of SNPs for allele merge from w_hm3.snplist\n",
      "Read 1217311 SNPs for allele merge.\n",
      "Reading sumstats from AD.Jansen.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.\n",
      "..WARNING: 3 SNPs had P outside of (0,1]. The P column may be mislabeled.\n",
      " done\n",
      "Read 9732197 SNPs from --sumstats file.\n",
      "Removed 8533711 SNPs not in --merge-alleles.\n",
      "Removed 0 SNPs with missing values.\n",
      "Removed 0 SNPs with INFO <= 0.9.\n",
      "Removed 0 SNPs with MAF <= 0.01.\n",
      "Removed 3 SNPs with out-of-bounds p-values.\n",
      "Removed 295 variants that were not SNPs or were strand-ambiguous.\n",
      "1198188 SNPs remain.\n",
      "Removed 359 SNPs with duplicated rs numbers (1197829 SNPs remain).\n",
      "Removed 46520 SNPs with N < 301340.126667 (1151309 SNPs remain).\n",
      "Median value of BETA was 5.16224534711e-05, which seems sensible.\n",
      "Removed 135 SNPs whose alleles did not match --merge-alleles (1151174 SNPs remain).\n",
      "Writing summary statistics for 1217311 SNPs (1151174 with nonmissing beta) to AD.Jansen.sumstats.gz.\n",
      "\n",
      "Metadata:\n",
      "Mean chi^2 = 1.125\n",
      "Lambda GC = 1.084\n",
      "Max chi^2 = 1001.735\n",
      "312 Genome-wide significant SNPs (some may have been removed by filtering).\n",
      "\n",
      "Conversion finished at Mon Mar 22 08:45:15 2021\n",
      "Total time elapsed: 1.0m:20.02s\n",
      "*********************************************************************\n",
      "* LD Score Regression (LDSC)\n",
      "* Version 1.0.1\n",
      "* (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane\n",
      "* Broad Institute of MIT and Harvard / MIT Department of Mathematics\n",
      "* GNU General Public License v3\n",
      "*********************************************************************\n",
      "Call: \n",
      "./munge_sumstats.py \\\n",
      "--out PD.Nalls \\\n",
      "--merge-alleles w_hm3.snplist \\\n",
      "--N 482730.0 \\\n",
      "--sumstats PD.Nalls.hg38.rsid.forLDSC.txt \n",
      "\n",
      "Interpreting column names as follows:\n",
      "N:\tSample size\n",
      "A1:\tAllele 1, interpreted as ref allele for signed sumstat.\n",
      "P:\tp-Value\n",
      "BETA:\t[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)\n",
      "A2:\tAllele 2, interpreted as non-ref allele for signed sumstat.\n",
      "SNP:\tVariant ID (e.g., rs number)\n",
      "\n",
      "Reading list of SNPs for allele merge from w_hm3.snplist\n",
      "Read 1217311 SNPs for allele merge.\n",
      "Reading sumstats from PD.Nalls.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.\n",
      ".. done\n",
      "Read 8330284 SNPs from --sumstats file.\n",
      "Removed 7137528 SNPs not in --merge-alleles.\n",
      "Removed 0 SNPs with missing values.\n",
      "Removed 0 SNPs with INFO <= 0.9.\n",
      "Removed 0 SNPs with MAF <= 0.01.\n",
      "Removed 0 SNPs with out-of-bounds p-values.\n",
      "Removed 659 variants that were not SNPs or were strand-ambiguous.\n",
      "1192097 SNPs remain.\n",
      "Removed 0 SNPs with duplicated rs numbers (1192097 SNPs remain).\n",
      "Removed 54873 SNPs with N < 321820.0 (1137224 SNPs remain).\n",
      "Median value of BETA was 0.0, which seems sensible.\n",
      "Removed 171 SNPs whose alleles did not match --merge-alleles (1137053 SNPs remain).\n",
      "Writing summary statistics for 1217311 SNPs (1137053 with nonmissing beta) to PD.Nalls.sumstats.gz.\n",
      "\n",
      "Metadata:\n",
      "Mean chi^2 = 1.137\n",
      "Lambda GC = 1.091\n",
      "Max chi^2 = 180.422\n",
      "276 Genome-wide significant SNPs (some may have been removed by filtering).\n",
      "\n",
      "Conversion finished at Mon Mar 22 08:46:29 2021\n",
      "Total time elapsed: 1.0m:11.31s\n",
      "*********************************************************************\n",
      "* LD Score Regression (LDSC)\n",
      "* Version 1.0.1\n",
      "* (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane\n",
      "* Broad Institute of MIT and Harvard / MIT Department of Mathematics\n",
      "* GNU General Public License v3\n",
      "*********************************************************************\n",
      "Call: \n",
      "./munge_sumstats.py \\\n",
      "--out ALS.Nicolas \\\n",
      "--merge-alleles w_hm3.snplist \\\n",
      "--N 80610.0 \\\n",
      "--sumstats ALS.Nicolas.hg38.rsid.forLDSC.txt \n",
      "\n",
      "Interpreting column names as follows:\n",
      "N:\tSample size\n",
      "A1:\tAllele 1, interpreted as ref allele for signed sumstat.\n",
      "P:\tp-Value\n",
      "BETA:\t[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)\n",
      "A2:\tAllele 2, interpreted as non-ref allele for signed sumstat.\n",
      "SNP:\tVariant ID (e.g., rs number)\n",
      "\n",
      "Reading list of SNPs for allele merge from w_hm3.snplist\n",
      "Read 1217311 SNPs for allele merge.\n",
      "Reading sumstats from ALS.Nicolas.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.\n",
      ".. done\n",
      "Read 8949246 SNPs from --sumstats file.\n",
      "Removed 7801463 SNPs not in --merge-alleles.\n",
      "Removed 0 SNPs with missing values.\n",
      "Removed 0 SNPs with INFO <= 0.9.\n",
      "Removed 0 SNPs with MAF <= 0.01.\n",
      "Removed 0 SNPs with out-of-bounds p-values.\n",
      "Removed 4 variants that were not SNPs or were strand-ambiguous.\n",
      "1147779 SNPs remain.\n",
      "Removed 0 SNPs with duplicated rs numbers (1147779 SNPs remain).\n",
      "Removed 0 SNPs with N < 53740.0 (1147779 SNPs remain).\n",
      "Median value of BETA was 0.0001, which seems sensible.\n",
      "Removed 2 SNPs whose alleles did not match --merge-alleles (1147777 SNPs remain).\n",
      "Writing summary statistics for 1217311 SNPs (1147777 with nonmissing beta) to ALS.Nicolas.sumstats.gz.\n",
      "\n",
      "Metadata:\n",
      "Mean chi^2 = 1.044\n",
      "Lambda GC = 1.034\n",
      "Max chi^2 = 50.171\n",
      "18 Genome-wide significant SNPs (some may have been removed by filtering).\n",
      "\n",
      "Conversion finished at Mon Mar 22 08:47:46 2021\n",
      "Total time elapsed: 1.0m:14.06s\n",
      "*********************************************************************\n",
      "* LD Score Regression (LDSC)\n",
      "* Version 1.0.1\n",
      "* (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane\n",
      "* Broad Institute of MIT and Harvard / MIT Department of Mathematics\n",
      "* GNU General Public License v3\n",
      "*********************************************************************\n",
      "Call: \n",
      "./munge_sumstats.py \\\n",
      "--out LBD.Chia \\\n",
      "--merge-alleles w_hm3.snplist \\\n",
      "--N 6502.0 \\\n",
      "--sumstats LBD.Chia.hg38.rsid.forLDSC.txt \n",
      "\n",
      "Interpreting column names as follows:\n",
      "N:\tSample size\n",
      "A1:\tAllele 1, interpreted as ref allele for signed sumstat.\n",
      "P:\tp-Value\n",
      "BETA:\t[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)\n",
      "A2:\tAllele 2, interpreted as non-ref allele for signed sumstat.\n",
      "SNP:\tVariant ID (e.g., rs number)\n",
      "\n",
      "Reading list of SNPs for allele merge from w_hm3.snplist\n",
      "Read 1217311 SNPs for allele merge.\n",
      "Reading sumstats from LBD.Chia.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.\n",
      ".. done\n",
      "Read 7843595 SNPs from --sumstats file.\n",
      "Removed 6762819 SNPs not in --merge-alleles.\n",
      "Removed 0 SNPs with missing values.\n",
      "Removed 0 SNPs with INFO <= 0.9.\n",
      "Removed 0 SNPs with MAF <= 0.01.\n",
      "Removed 0 SNPs with out-of-bounds p-values.\n",
      "Removed 3 variants that were not SNPs or were strand-ambiguous.\n",
      "1080773 SNPs remain.\n",
      "Removed 6 SNPs with duplicated rs numbers (1080767 SNPs remain).\n",
      "Removed 0 SNPs with N < 4412.0 (1080767 SNPs remain).\n",
      "Median value of BETA was 0.000859630411882, which seems sensible.\n",
      "Removed 19 SNPs whose alleles did not match --merge-alleles (1080748 SNPs remain).\n",
      "Writing summary statistics for 1217311 SNPs (1080748 with nonmissing beta) to LBD.Chia.sumstats.gz.\n",
      "\n",
      "Metadata:\n",
      "Mean chi^2 = 1.026\n",
      "Lambda GC = 1.023\n",
      "Max chi^2 = 215.691\n",
      "30 Genome-wide significant SNPs (some may have been removed by filtering).\n",
      "\n",
      "Conversion finished at Mon Mar 22 08:48:59 2021\n",
      "Total time elapsed: 1.0m:9.73s\n",
      "*********************************************************************\n",
      "* LD Score Regression (LDSC)\n",
      "* Version 1.0.1\n",
      "* (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane\n",
      "* Broad Institute of MIT and Harvard / MIT Department of Mathematics\n",
      "* GNU General Public License v3\n",
      "*********************************************************************\n",
      "Call: \n",
      "./munge_sumstats.py \\\n",
      "--out MG.Chia \\\n",
      "--merge-alleles w_hm3.snplist \\\n",
      "--N 38243.0 \\\n",
      "--sumstats MG.Chia.hg38.rsid.forLDSC.txt \n",
      "\n",
      "Interpreting column names as follows:\n",
      "N:\tSample size\n",
      "A1:\tAllele 1, interpreted as ref allele for signed sumstat.\n",
      "P:\tp-Value\n",
      "BETA:\t[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)\n",
      "A2:\tAllele 2, interpreted as non-ref allele for signed sumstat.\n",
      "SNP:\tVariant ID (e.g., rs number)\n",
      "\n",
      "Reading list of SNPs for allele merge from w_hm3.snplist\n",
      "Read 1217311 SNPs for allele merge.\n",
      "Reading sumstats from MG.Chia.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.\n",
      "..... done\n",
      "Read 24006245 SNPs from --sumstats file.\n",
      "Removed 22863916 SNPs not in --merge-alleles.\n",
      "Removed 0 SNPs with missing values.\n",
      "Removed 0 SNPs with INFO <= 0.9.\n",
      "Removed 0 SNPs with MAF <= 0.01.\n",
      "Removed 0 SNPs with out-of-bounds p-values.\n",
      "Removed 70 variants that were not SNPs or were strand-ambiguous.\n",
      "1142259 SNPs remain.\n",
      "Removed 189 SNPs with duplicated rs numbers (1142070 SNPs remain).\n",
      "Removed 0 SNPs with N < 25495.3333333 (1142070 SNPs remain).\n",
      "Median value of BETA was -0.0001, which seems sensible.\n",
      "Removed 93 SNPs whose alleles did not match --merge-alleles (1141977 SNPs remain).\n",
      "Writing summary statistics for 1217311 SNPs (1141977 with nonmissing beta) to MG.Chia.sumstats.gz.\n",
      "\n",
      "Metadata:\n",
      "Mean chi^2 = 1.073\n",
      "Lambda GC = 1.059\n",
      "Max chi^2 = 60.413\n",
      "7 Genome-wide significant SNPs (some may have been removed by filtering).\n",
      "\n",
      "Conversion finished at Mon Mar 22 08:50:51 2021\n",
      "Total time elapsed: 1.0m:48.5s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "source /data/chiarp/conda/etc/profile.d/conda.sh\n",
    "\n",
    "python ./ldsc/munge_sumstats.py \\\n",
    "--sumstats AD.Jansen.hg38.rsid.forLDSC.txt \\\n",
    "--N 452010 \\\n",
    "--out AD.Jansen \\\n",
    "--merge-alleles w_hm3.snplist\n",
    "\n",
    "python ./ldsc/munge_sumstats.py \\\n",
    "--sumstats PD.Nalls.hg38.rsid.forLDSC.txt \\\n",
    "--N 482730 \\\n",
    "--out PD.Nalls \\\n",
    "--merge-alleles w_hm3.snplist\n",
    "\n",
    "python ./ldsc/munge_sumstats.py \\\n",
    "--sumstats ALS.Nicolas.hg38.rsid.forLDSC.txt \\\n",
    "--N 80610 \\\n",
    "--out ALS.Nicolas \\\n",
    "--merge-alleles w_hm3.snplist\n",
    "\n",
    "python ./ldsc/munge_sumstats.py \\\n",
    "--sumstats LBD.Chia.hg38.rsid.forLDSC.txt \\\n",
    "--N 6502 \\\n",
    "--out LBD.Chia \\\n",
    "--merge-alleles w_hm3.snplist\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "alert-cleanup",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "*********************************************************************\n",
      "* LD Score Regression (LDSC)\n",
      "* Version 1.0.1\n",
      "* (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane\n",
      "* Broad Institute of MIT and Harvard / MIT Department of Mathematics\n",
      "* GNU General Public License v3\n",
      "*********************************************************************\n",
      "Call: \n",
      "./munge_sumstats.py \\\n",
      "--out MG.Chia \\\n",
      "--merge-alleles w_hm3.snplist \\\n",
      "--N 38243.0 \\\n",
      "--sumstats MG.Chia.hg38.rsid.forLDSC.txt \n",
      "\n",
      "Interpreting column names as follows:\n",
      "N:\tSample size\n",
      "A1:\tAllele 1, interpreted as ref allele for signed sumstat.\n",
      "P:\tp-Value\n",
      "BETA:\t[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)\n",
      "A2:\tAllele 2, interpreted as non-ref allele for signed sumstat.\n",
      "SNP:\tVariant ID (e.g., rs number)\n",
      "\n",
      "Reading list of SNPs for allele merge from w_hm3.snplist\n",
      "Read 1217311 SNPs for allele merge.\n",
      "Reading sumstats from MG.Chia.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.\n",
      ".. done\n",
      "Read 8387157 SNPs from --sumstats file.\n",
      "Removed 7272179 SNPs not in --merge-alleles.\n",
      "Removed 0 SNPs with missing values.\n",
      "Removed 0 SNPs with INFO <= 0.9.\n",
      "Removed 0 SNPs with MAF <= 0.01.\n",
      "Removed 0 SNPs with out-of-bounds p-values.\n",
      "Removed 8 variants that were not SNPs or were strand-ambiguous.\n",
      "1114970 SNPs remain.\n",
      "Removed 17 SNPs with duplicated rs numbers (1114953 SNPs remain).\n",
      "Removed 0 SNPs with N < 25495.3333333 (1114953 SNPs remain).\n",
      "Median value of BETA was -0.0002, which seems sensible.\n",
      "Removed 27 SNPs whose alleles did not match --merge-alleles (1114926 SNPs remain).\n",
      "Writing summary statistics for 1217311 SNPs (1114926 with nonmissing beta) to MG.Chia.sumstats.gz.\n",
      "\n",
      "Metadata:\n",
      "Mean chi^2 = 1.076\n",
      "Lambda GC = 1.062\n",
      "Max chi^2 = 60.413\n",
      "7 Genome-wide significant SNPs (some may have been removed by filtering).\n",
      "\n",
      "Conversion finished at Tue Apr  6 12:29:59 2021\n",
      "Total time elapsed: 1.0m:3.88s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "source /data/chiarp/conda/etc/profile.d/conda.sh\n",
    "python ./ldsc/munge_sumstats.py \\\n",
    "--sumstats MG.Chia.hg38.rsid.forLDSC.txt \\\n",
    "--N 38243 \\\n",
    "--out MG.Chia \\\n",
    "--merge-alleles w_hm3.snplist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "surgical-supply",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "*********************************************************************\n",
      "* LD Score Regression (LDSC)\n",
      "* Version 1.0.1\n",
      "* (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane\n",
      "* Broad Institute of MIT and Harvard / MIT Department of Mathematics\n",
      "* GNU General Public License v3\n",
      "*********************************************************************\n",
      "Call: \n",
      "./munge_sumstats.py \\\n",
      "--out ALS.VanRheenen \\\n",
      "--merge-alleles w_hm3.snplist \\\n",
      "--sumstats ALS.VanRheenen.hg38.rsid.forLDSC.txt \n",
      "\n",
      "Interpreting column names as follows:\n",
      "N:\tSample size\n",
      "A1:\tAllele 1, interpreted as ref allele for signed sumstat.\n",
      "P:\tp-Value\n",
      "BETA:\t[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)\n",
      "A2:\tAllele 2, interpreted as non-ref allele for signed sumstat.\n",
      "SNP:\tVariant ID (e.g., rs number)\n",
      "\n",
      "Reading list of SNPs for allele merge from w_hm3.snplist\n",
      "Read 1217311 SNPs for allele merge.\n",
      "Reading sumstats from ALS.VanRheenen.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.\n",
      ".. done\n",
      "Read 8709431 SNPs from --sumstats file.\n",
      "Removed 7535250 SNPs not in --merge-alleles.\n",
      "Removed 0 SNPs with missing values.\n",
      "Removed 0 SNPs with INFO <= 0.9.\n",
      "Removed 0 SNPs with MAF <= 0.01.\n",
      "Removed 0 SNPs with out-of-bounds p-values.\n",
      "Removed 0 variants that were not SNPs or were strand-ambiguous.\n",
      "1174181 SNPs remain.\n",
      "Removed 0 SNPs with duplicated rs numbers (1174181 SNPs remain).\n",
      "Removed 0 SNPs with N < 24034.6666667 (1174181 SNPs remain).\n",
      "Median value of BETA was 0.0, which seems sensible.\n",
      "Removed 0 SNPs whose alleles did not match --merge-alleles (1174181 SNPs remain).\n",
      "Writing summary statistics for 1217311 SNPs (1174181 with nonmissing beta) to ALS.VanRheenen.sumstats.gz.\n",
      "\n",
      "Metadata:\n",
      "Mean chi^2 = 1.03\n",
      "Lambda GC = 1.023\n",
      "Max chi^2 = 79.877\n",
      "28 Genome-wide significant SNPs (some may have been removed by filtering).\n",
      "\n",
      "Conversion finished at Mon Mar 22 12:10:51 2021\n",
      "Total time elapsed: 1.0m:13.13s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "source /data/chiarp/conda/etc/profile.d/conda.sh\n",
    "\n",
    "python ./ldsc/munge_sumstats.py \\\n",
    "--sumstats ALS.VanRheenen.hg38.rsid.forLDSC.txt \\\n",
    "--out ALS.VanRheenen \\\n",
    "--merge-alleles w_hm3.snplist"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "warming-intermediate",
   "metadata": {},
   "source": [
    "## Compute heritability and the LD Score regression intercept for MG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "related-lincoln",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "*********************************************************************\n",
      "* LD Score Regression (LDSC)\n",
      "* Version 1.0.1\n",
      "* (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane\n",
      "* Broad Institute of MIT and Harvard / MIT Department of Mathematics\n",
      "* GNU General Public License v3\n",
      "*********************************************************************\n",
      "Call: \n",
      "./ldsc.py \\\n",
      "--h2 MG.Chia.sumstats.gz \\\n",
      "--ref-ld-chr eur_w_ld_chr/ \\\n",
      "--out MG.Chia_h2 \\\n",
      "--w-ld-chr eur_w_ld_chr/ \n",
      "\n",
      "Beginning analysis at Tue Apr  6 12:48:50 2021\n",
      "Reading summary statistics from MG.Chia.sumstats.gz ...\n",
      "Read summary statistics for 1114926 SNPs.\n",
      "Reading reference panel LD Score from eur_w_ld_chr/[1-22] ... (ldscore_fromlist)\n",
      "Read reference panel LD Scores for 1290028 SNPs.\n",
      "Removing partitioned LD Scores with zero variance.\n",
      "Reading regression weight LD Score from eur_w_ld_chr/[1-22] ... (ldscore_fromlist)\n",
      "Read regression weight LD Scores for 1290028 SNPs.\n",
      "After merging with reference panel LD, 1107424 SNPs remain.\n",
      "After merging with regression SNP LD, 1107424 SNPs remain.\n",
      "Using two-step estimator with cutoff at 30.\n",
      "Total Observed scale h2: 0.0751 (0.0148)\n",
      "Lambda GC: 1.0618\n",
      "Mean Chi^2: 1.0742\n",
      "Intercept: 1.0174 (0.0066)\n",
      "Ratio: 0.235 (0.0888)\n",
      "Analysis finished at Tue Apr  6 12:49:05 2021\n",
      "Total time elapsed: 14.47s\n",
      "*********************************************************************\n",
      "* LD Score Regression (LDSC)\n",
      "* Version 1.0.1\n",
      "* (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane\n",
      "* Broad Institute of MIT and Harvard / MIT Department of Mathematics\n",
      "* GNU General Public License v3\n",
      "*********************************************************************\n",
      "Call: \n",
      "./ldsc.py \\\n",
      "--h2 MG.Chia.sumstats.gz \\\n",
      "--ref-ld-chr eur_w_ld_chr/ \\\n",
      "--out MG.Chia_h2 \\\n",
      "--w-ld-chr eur_w_ld_chr/ \n",
      "\n",
      "Beginning analysis at Tue Apr  6 12:48:50 2021\n",
      "Reading summary statistics from MG.Chia.sumstats.gz ...\n",
      "Read summary statistics for 1114926 SNPs.\n",
      "Reading reference panel LD Score from eur_w_ld_chr/[1-22] ... (ldscore_fromlist)\n",
      "Read reference panel LD Scores for 1290028 SNPs.\n",
      "Removing partitioned LD Scores with zero variance.\n",
      "Reading regression weight LD Score from eur_w_ld_chr/[1-22] ... (ldscore_fromlist)\n",
      "Read regression weight LD Scores for 1290028 SNPs.\n",
      "After merging with reference panel LD, 1107424 SNPs remain.\n",
      "After merging with regression SNP LD, 1107424 SNPs remain.\n",
      "Using two-step estimator with cutoff at 30.\n",
      "Total Observed scale h2: 0.0751 (0.0148)\n",
      "Lambda GC: 1.0618\n",
      "Mean Chi^2: 1.0742\n",
      "Intercept: 1.0174 (0.0066)\n",
      "Ratio: 0.235 (0.0888)\n",
      "Analysis finished at Tue Apr  6 12:49:05 2021\n",
      "Total time elapsed: 14.47s\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/gpfs/gsfs11/users/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/LDSC/ldsc/ldscore/irwls.py:161: FutureWarning: `rcond` parameter will change to the default of machine precision times ``max(M, N)`` where M and N are the input matrix dimensions.\n",
      "To use the future default and silence this warning we advise to pass `rcond=None`, to keep using the old, explicitly pass `rcond=-1`.\n",
      "  coef = np.linalg.lstsq(x, y)\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "source /data/chiarp/conda/etc/profile.d/conda.sh\n",
    "\n",
    "python ./ldsc/ldsc.py \\\n",
    "--h2 MG.Chia.sumstats.gz \\\n",
    "--ref-ld-chr eur_w_ld_chr/ \\\n",
    "--w-ld-chr eur_w_ld_chr/ \\\n",
    "--out MG.Chia_h2\n",
    "cat MG.Chia_h2.log"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "occasional-browse",
   "metadata": {},
   "source": [
    "## Run LDSC and genetic correlation on MG vs. neurological diseases\n",
    "### Estimating Heritability, Genetic Correlation and the LD Score Regression Intercept"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "imperial-campbell",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "*********************************************************************\n",
      "* LD Score Regression (LDSC)\n",
      "* Version 1.0.1\n",
      "* (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane\n",
      "* Broad Institute of MIT and Harvard / MIT Department of Mathematics\n",
      "* GNU General Public License v3\n",
      "*********************************************************************\n",
      "Call: \n",
      "./ldsc.py \\\n",
      "--ref-ld-chr eur_w_ld_chr/ \\\n",
      "--out MG_AD_PD_ALS_LBD \\\n",
      "--rg MG.Chia.sumstats.gz,AD.Jansen.sumstats.gz,PD.Nalls.sumstats.gz,ALS.VanRheenen.sumstats.gz,LBD.Chia.sumstats.gz \\\n",
      "--w-ld-chr eur_w_ld_chr/ \n",
      "\n",
      "Beginning analysis at Tue Apr  6 12:50:51 2021\n",
      "Reading summary statistics from MG.Chia.sumstats.gz ...\n",
      "Read summary statistics for 1114926 SNPs.\n",
      "Reading reference panel LD Score from eur_w_ld_chr/[1-22] ... (ldscore_fromlist)\n",
      "Read reference panel LD Scores for 1290028 SNPs.\n",
      "Removing partitioned LD Scores with zero variance.\n",
      "Reading regression weight LD Score from eur_w_ld_chr/[1-22] ... (ldscore_fromlist)\n",
      "Read regression weight LD Scores for 1290028 SNPs.\n",
      "After merging with reference panel LD, 1107424 SNPs remain.\n",
      "After merging with regression SNP LD, 1107424 SNPs remain.\n",
      "Computing rg for phenotype 2/5\n",
      "Reading summary statistics from AD.Jansen.sumstats.gz ...\n",
      "Read summary statistics for 1217311 SNPs.\n",
      "After merging with summary statistics, 1107424 SNPs remain.\n",
      "1065135 SNPs with valid alleles.\n",
      "\n",
      "Heritability of phenotype 1\n",
      "---------------------------\n",
      "Total Observed scale h2: 0.0754 (0.0165)\n",
      "Lambda GC: 1.0618\n",
      "Mean Chi^2: 1.0751\n",
      "Intercept: 1.0171 (0.0078)\n",
      "Ratio: 0.2272 (0.1044)\n",
      "\n",
      "Heritability of phenotype 2/5\n",
      "-----------------------------\n",
      "Total Observed scale h2: 0.0075 (0.0049)\n",
      "Lambda GC: 1.0833\n",
      "Mean Chi^2: 1.1236\n",
      "Intercept: 1.057 (0.0592)\n",
      "Ratio: 0.4613 (0.4791)\n",
      "\n",
      "Genetic Covariance\n",
      "------------------\n",
      "Total Observed scale gencov: 0.0002 (0.0029)\n",
      "Mean z1*z2: 0.0062\n",
      "Intercept: 0.005 (0.0047)\n",
      "\n",
      "Genetic Correlation\n",
      "-------------------\n",
      "Genetic Correlation: 0.0091 (0.123)\n",
      "Z-score: 0.0741\n",
      "P: 0.9409\n",
      "\n",
      "Computing rg for phenotype 3/5\n",
      "Reading summary statistics from PD.Nalls.sumstats.gz ...\n",
      "Read summary statistics for 1217311 SNPs.\n",
      "After merging with summary statistics, 1107424 SNPs remain.\n",
      "1064424 SNPs with valid alleles.\n",
      "\n",
      "Heritability of phenotype 3/5\n",
      "-----------------------------\n",
      "Total Observed scale h2: 0.0166 (0.0017)\n",
      "Lambda GC: 1.0864\n",
      "Mean Chi^2: 1.1347\n",
      "Intercept: 0.981 (0.0076)\n",
      "Ratio < 0 (usually indicates GC correction).\n",
      "\n",
      "Genetic Covariance\n",
      "------------------\n",
      "Total Observed scale gencov: 0.0001 (0.0031)\n",
      "Mean z1*z2: -0.0018\n",
      "Intercept: -0.0029 (0.0053)\n",
      "\n",
      "Genetic Correlation\n",
      "-------------------\n",
      "Genetic Correlation: 0.0032 (0.0883)\n",
      "Z-score: 0.0359\n",
      "P: 0.9714\n",
      "\n",
      "Computing rg for phenotype 4/5\n",
      "Reading summary statistics from ALS.VanRheenen.sumstats.gz ...\n",
      "Read summary statistics for 1217311 SNPs.\n",
      "After merging with summary statistics, 1107424 SNPs remain.\n",
      "1098401 SNPs with valid alleles.\n",
      "\n",
      "Heritability of phenotype 4/5\n",
      "-----------------------------\n",
      "Total Observed scale h2: 0.0485 (0.0131)\n",
      "Lambda GC: 1.0225\n",
      "Mean Chi^2: 1.0281\n",
      "Intercept: 0.9941 (0.0062)\n",
      "Ratio < 0 (usually indicates GC correction).\n",
      "\n",
      "Genetic Covariance\n",
      "------------------\n",
      "Total Observed scale gencov: 0.0171 (0.0102)\n",
      "Mean z1*z2: 0.0246\n",
      "Intercept: 0.0122 (0.0047)\n",
      "\n",
      "Genetic Correlation\n",
      "-------------------\n",
      "Genetic Correlation: 0.2858 (0.18)\n",
      "Z-score: 1.5881\n",
      "P: 0.1123\n",
      "\n",
      "Computing rg for phenotype 5/5\n",
      "Reading summary statistics from LBD.Chia.sumstats.gz ...\n",
      "Read summary statistics for 1217311 SNPs.\n",
      "After merging with summary statistics, 1107424 SNPs remain.\n",
      "1009819 SNPs with valid alleles.\n",
      "\n",
      "Heritability of phenotype 5/5\n",
      "-----------------------------\n",
      "Total Observed scale h2: 0.1293 (0.0835)\n",
      "Lambda GC: 1.0225\n",
      "Mean Chi^2: 1.0233\n",
      "Intercept: 1.0064 (0.0087)\n",
      "Ratio: 0.2728 (0.3714)\n",
      "\n",
      "Genetic Covariance\n",
      "------------------\n",
      "Total Observed scale gencov: 0.0349 (0.0229)\n",
      "Mean z1*z2: -0.0009\n",
      "Intercept: -0.0117 (0.0047)\n",
      "\n",
      "Genetic Correlation\n",
      "-------------------\n",
      "Genetic Correlation: 0.3657 (0.2797)\n",
      "Z-score: 1.3078\n",
      "P: 0.191\n",
      "\n",
      "\n",
      "Summary of Genetic Correlation Results\n",
      "                  p1                          p2      rg      se       z       p  h2_obs  h2_obs_se  h2_int  h2_int_se  gcov_int  gcov_int_se\n",
      " MG.Chia.sumstats.gz       AD.Jansen.sumstats.gz  0.0091  0.1230  0.0741  0.9409  0.0075     0.0049  1.0570     0.0592    0.0050       0.0047\n",
      " MG.Chia.sumstats.gz        PD.Nalls.sumstats.gz  0.0032  0.0883  0.0359  0.9714  0.0166     0.0017  0.9810     0.0076   -0.0029       0.0053\n",
      " MG.Chia.sumstats.gz  ALS.VanRheenen.sumstats.gz  0.2858  0.1800  1.5881  0.1123  0.0485     0.0131  0.9941     0.0062    0.0122       0.0047\n",
      " MG.Chia.sumstats.gz        LBD.Chia.sumstats.gz  0.3657  0.2797  1.3078  0.1910  0.1293     0.0835  1.0064     0.0087   -0.0117       0.0047\n",
      "\n",
      "Analysis finished at Tue Apr  6 12:51:30 2021\n",
      "Total time elapsed: 38.68s\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/gpfs/gsfs11/users/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/LDSC/ldsc/ldscore/sumstats.py:532: FutureWarning: Method .as_matrix will be removed in a future version. Use .values instead.\n",
      "  ref_ld = sumstats.as_matrix(columns=ref_ld_cnames)\n",
      "/gpfs/gsfs11/users/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/LDSC/ldsc/ldscore/irwls.py:161: FutureWarning: `rcond` parameter will change to the default of machine precision times ``max(M, N)`` where M and N are the input matrix dimensions.\n",
      "To use the future default and silence this warning we advise to pass `rcond=None`, to keep using the old, explicitly pass `rcond=-1`.\n",
      "  coef = np.linalg.lstsq(x, y)\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "source /data/chiarp/conda/etc/profile.d/conda.sh\n",
    "\n",
    "python ./ldsc/ldsc.py \\\n",
    "--rg MG.Chia.sumstats.gz,AD.Jansen.sumstats.gz,PD.Nalls.sumstats.gz,ALS.VanRheenen.sumstats.gz,LBD.Chia.sumstats.gz \\\n",
    "--ref-ld-chr eur_w_ld_chr/ \\\n",
    "--w-ld-chr eur_w_ld_chr/ \\\n",
    "--out MG_AD_PD_ALS_LBD"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "persistent-bidding",
   "metadata": {},
   "source": [
    "## Run LDSC and genetic correlation on autoimmune diseases\n",
    "\n",
    "Ran a term search i.e. `autoimmune disease` on GWAS catalaog to find summary stats to download. Filtered down list of studies which were dont on European samples with the largest cohort (where possible) and to those where summary stats were available for FTP download directly from GWAS catalog.\n",
    "\n",
    "- Crohns : `http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST004001-GCST005000/GCST004132/harmonised/28067908-GCST004132-EFO_0000384.h.tsv.gz`\n",
    "- Graves.Hashimoto.Thyroiditis : `http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST005001-GCST006000/GCST005524/harmonised/22922229-GCST005524-EFO_0006812.h.tsv.gz`\n",
    "- IBS : `http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST004001-GCST005000/GCST004131/harmonised/28067908-GCST004131-EFO_0003767.h.tsv.gz`\n",
    "- jArthritis : `http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST90010001-GCST90011000/GCST90010715/harmonised/33106285-GCST90010715-EFO_0002609.h.tsv.gz`\n",
    "- LatentAutoImmuneDiabetes : `http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST007001-GCST008000/GCST007245/CousminerDL_30254083.gz`\n",
    "- MicroscopicColitis : `http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST009001-GCST010000/GCST009081/wholegwas`\n",
    "- MS : `http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST005001-GCST006000/GCST005531/harmonised/24076602-GCST005531-EFO_0003885.h.tsv.gz`\n",
    "- NeuromyelitisOptica_AQP4_IgGpos : `http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST006001-GCST007000/GCST006937/harmonised/29769526-GCST006937-EFO_0009584.h.tsv.gz`\n",
    "- PsoriaticArthritis : `http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST007001-GCST008000/GCST007043/Table-S3.xlsx`\n",
    "- RA : `http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST002001-GCST003000/GCST002318/harmonised/24390342-GCST002318-EFO_0000685.h.tsv.gz`\n",
    "- SLE : `http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST005001-GCST006000/GCST005831/harmonised/29848360-GCST005831-EFO_0002690.h.tsv.gz`\n",
    "- T1D : `http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST005001-GCST006000/GCST005536/harmonised/25751624-GCST005536-EFO_0001359.h.tsv.gz`\n",
    "- UC : `http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST004001-GCST005000/GCST004133/harmonised/28067908-GCST004133-EFO_0000729.h.tsv.gz`\n",
    "- Vitiligo : `http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST004001-GCST005000/GCST004785/GWAS123chr1.cmh.gz` ## chr 1-22,X\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "simple-parish",
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir GWAScatalog"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "capital-disposal",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "IOPub data rate exceeded.\n",
      "The notebook server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--NotebookApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "NotebookApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "NotebookApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd GWAScatalog/\n",
    "wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST004001-GCST005000/GCST004132/harmonised/28067908-GCST004132-EFO_0000384.h.tsv.gz\n",
    "wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST005001-GCST006000/GCST005524/harmonised/22922229-GCST005524-EFO_0006812.h.tsv.gz\n",
    "wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST004001-GCST005000/GCST004131/harmonised/28067908-GCST004131-EFO_0003767.h.tsv.gz\n",
    "wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST90010001-GCST90011000/GCST90010715/harmonised/33106285-GCST90010715-EFO_0002609.h.tsv.gz\n",
    "wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST007001-GCST008000/GCST007245/CousminerDL_30254083.gz\n",
    "wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST009001-GCST010000/GCST009081/wholegwas\n",
    "wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST005001-GCST006000/GCST005531/harmonised/24076602-GCST005531-EFO_0003885.h.tsv.gz\n",
    "wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST006001-GCST007000/GCST006937/harmonised/29769526-GCST006937-EFO_0009584.h.tsv.gz\n",
    "wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST007001-GCST008000/GCST007043/Table-S3.xlsx\n",
    "wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST002001-GCST003000/GCST002318/harmonised/24390342-GCST002318-EFO_0000685.h.tsv.gz\n",
    "wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST005001-GCST006000/GCST005831/harmonised/29848360-GCST005831-EFO_0002690.h.tsv.gz\n",
    "wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST005001-GCST006000/GCST005536/harmonised/25751624-GCST005536-EFO_0001359.h.tsv.gz\n",
    "wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST004001-GCST005000/GCST004133/harmonised/28067908-GCST004133-EFO_0000729.h.tsv.gz\n",
    "wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST004001-GCST005000/GCST004785/GWAS123chr1.cmh.gz\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "vocal-smile",
   "metadata": {},
   "source": [
    "- SNP = A unique identifier (e.g., the rs number)\n",
    "- A1 =  effect allele\n",
    "- A2 =  non-effect allele\n",
    "- N = Sample size (which often varies from SNP to SNP)\n",
    "- P = P-value\n",
    "- BETA = A signed summary statistic (beta, OR, log odds, Z-score, etc)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "hearing-spank",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "module load R/3.5.2\n",
    "R --vanilla --no-save\n",
    "\n",
    "require(data.table)\n",
    "require(tidyverse)\n",
    "\n",
    "# Crohn's\n",
    "data <- fread(\"28067908-GCST004132-EFO_0000384.h.tsv.gz\",header=T)\n",
    "data$SNP = ifelse(is.na(data$hm_rsid), data$hm_variant_id, data$hm_rsid)\n",
    "data1 <- data %>%\n",
    "         mutate(N = 40266) %>%\n",
    "         select(SNP,hm_effect_allele,hm_other_allele,N,p_value,hm_beta) %>%\n",
    "         rename(A1 = hm_effect_allele, A2 = hm_other_allele, P = p_value, BETA = hm_beta)\n",
    "write.table(data1, \"Crohns.deLange.hg38.rsid.forLDSC.txt\",sep=\"\\t\",quote=F,row.names=F,col.names=T)\n",
    "\n",
    "# Graves.Hashimoto.Thyroiditis - does not have effect allele information\n",
    "\n",
    "# IBS\n",
    "data <- fread(\"28067908-GCST004131-EFO_0003767.h.tsv.gz\",header=T)\n",
    "data$SNP = ifelse(is.na(data$hm_rsid), data$hm_variant_id, data$hm_rsid)\n",
    "data1 <- data %>%\n",
    "         mutate(N = 59957) %>%\n",
    "         select(SNP,hm_effect_allele,hm_other_allele,N,p_value,hm_beta) %>%\n",
    "         rename(A1 = hm_effect_allele, A2 = hm_other_allele, P = p_value, BETA = hm_beta)\n",
    "write.table(data1, \"IBS.deLange.hg38.rsid.forLDSC.txt\",sep=\"\\t\",quote=F,row.names=F,col.names=T)\n",
    "\n",
    "\n",
    "# juvenile idiopathic arthritis\n",
    "data <- fread(\"33106285-GCST90010715-EFO_0002609.h.tsv.gz\",header=T)\n",
    "data$SNP = ifelse(is.na(data$hm_rsid), data$hm_variant_id, data$hm_rsid)\n",
    "data1 <- data %>%\n",
    "         mutate(N = 12501) %>%\n",
    "         select(SNP,hm_effect_allele,hm_other_allele,N,p_value,hm_beta) %>%\n",
    "         rename(A1 = hm_effect_allele, A2 = hm_other_allele, P = p_value, BETA = hm_beta)\n",
    "write.table(data1, \"jArthritis.LopezIsac.hg38.rsid.forLDSC.txt\",sep=\"\\t\",quote=F,row.names=F,col.names=T)\n",
    "\n",
    "\n",
    "# MicroscopicColitis\n",
    "data <- fread(\"wholegwas\",header=T)\n",
    "data1 <- data %>%\n",
    "         mutate(N = 445655) %>%\n",
    "         select(SNP,ALLELE1,ALLELE0,N,P_BOLT_LMM,BETA) %>%\n",
    "         rename(A1 = ALLELE1, A2 = ALLELE0, P = P_BOLT_LMM)\n",
    "write.table(data1, \"MicroscopicColitis.Green.hg38.rsid.forLDSC.txt\",sep=\"\\t\",quote=F,row.names=F,col.names=T)\n",
    "\n",
    "\n",
    "# MS\n",
    "data <- fread(\"24076602-GCST005531-EFO_0003885.h.tsv.gz\",header=T)\n",
    "data$SNP = ifelse(is.na(data$hm_rsid), data$hm_variant_id, data$hm_rsid)\n",
    "data1 <- data %>%\n",
    "         mutate(N = 38582) %>%\n",
    "         select(SNP,hm_effect_allele,hm_other_allele,N,p_value,hm_beta) %>%\n",
    "         rename(A1 = hm_effect_allele, A2 = hm_other_allele, P = p_value, BETA = hm_beta)\n",
    "write.table(data1, \"MS.Beecham.hg38.rsid.forLDSC.txt\",sep=\"\\t\",quote=F,row.names=F,col.names=T)\n",
    "\n",
    "\n",
    "# NeuromyelitisOptica_AQP4_IgGpos\n",
    "data <- fread(\"29769526-GCST006937-EFO_0009584.h.tsv.gz\",header=T)\n",
    "data$SNP = ifelse(is.na(data$hm_rsid), data$hm_variant_id, data$hm_rsid)\n",
    "data1 <- data %>%\n",
    "         mutate(N = n) %>%\n",
    "         select(SNP,hm_effect_allele,hm_other_allele,N,p_value,hm_beta) %>%\n",
    "         rename(A1 = hm_effect_allele, A2 = hm_other_allele, P = p_value, BETA = hm_beta)\n",
    "write.table(data1, \"NeuromyelitisOptica_AQP4_IgGpos.Estrada.hg38.rsid.forLDSC.txt\",sep=\"\\t\",quote=F,row.names=F,col.names=T)\n",
    "\n",
    "\n",
    "# RA\n",
    "data <- fread(\"24390342-GCST002318-EFO_0000685.h.tsv.gz\",header=T)\n",
    "data$SNP = ifelse(is.na(data$hm_rsid), data$hm_variant_id, data$hm_rsid)\n",
    "data1 <- data %>%\n",
    "         mutate(N = 58284,\n",
    "                BETA = log(hm_odds_ratio)) %>%\n",
    "         select(SNP,hm_effect_allele,hm_other_allele,N,p_value,BETA) %>%\n",
    "         rename(A1 = hm_effect_allele, A2 = hm_other_allele, P = p_value) %>%\n",
    "         filter(SNP != is.na(SNP))\n",
    "write.table(data1, \"RA.Okada.hg38.rsid.forLDSC.txt\",sep=\"\\t\",quote=F,row.names=F,col.names=T)\n",
    "\n",
    "\n",
    "# SLE\n",
    "data <- fread(\"29848360-GCST005831-EFO_0002690.h.tsv.gz\",header=T)\n",
    "data$SNP = ifelse(is.na(data$hm_rsid), data$hm_variant_id, data$hm_rsid)\n",
    "data1 <- data %>%\n",
    "         mutate(N = 16966) %>%\n",
    "         select(SNP,effect_allele,other_allele,N,p_value,z) %>%\n",
    "         rename(A1 = effect_allele, A2 = other_allele, P = p_value, Z = z)\n",
    "write.table(data1, \"SLE.Julia.hg38.rsid.forLDSC.txt\",sep=\"\\t\",quote=F,row.names=F,col.names=T)\n",
    "\n",
    "\n",
    "# T1D\n",
    "data <- fread(\"25751624-GCST005536-EFO_0001359.h.tsv.gz\",header=T)\n",
    "data$SNP = ifelse(is.na(data$hm_rsid), data$hm_variant_id, data$hm_rsid)\n",
    "data1 <- data %>%\n",
    "         mutate(N = 29652) %>%\n",
    "         select(SNP,hm_effect_allele,hm_other_allele,N,p_value,hm_beta) %>%\n",
    "         rename(A1 = hm_effect_allele, A2 = hm_other_allele, P = p_value, BETA = hm_beta)\n",
    "write.table(data1, \"T1D.OnengutGumuscu.hg38.rsid.forLDSC.txt\",sep=\"\\t\",quote=F,row.names=F,col.names=T)\n",
    "\n",
    "\n",
    "# UC\n",
    "data <- fread(\"28067908-GCST004133-EFO_0000729.h.tsv.gz\",header=T)\n",
    "data$SNP = ifelse(is.na(data$hm_rsid), data$hm_variant_id, data$hm_rsid)\n",
    "data1 <- data %>%\n",
    "         mutate(N = 45975) %>%\n",
    "         select(SNP,hm_effect_allele,hm_other_allele,N,p_value,hm_beta) %>%\n",
    "         rename(A1 = hm_effect_allele, A2 = hm_other_allele, P = p_value, BETA = hm_beta)\n",
    "write.table(data1, \"UC.deLange.hg38.rsid.forLDSC.txt\",sep=\"\\t\",quote=F,row.names=F,col.names=T)\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "interesting-might",
   "metadata": {},
   "source": [
    "### Reformat autoimmune summ stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "practical-motor",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "*********************************************************************\n",
      "* LD Score Regression (LDSC)\n",
      "* Version 1.0.1\n",
      "* (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane\n",
      "* Broad Institute of MIT and Harvard / MIT Department of Mathematics\n",
      "* GNU General Public License v3\n",
      "*********************************************************************\n",
      "Call: \n",
      "./munge_sumstats.py \\\n",
      "--out Crohns.deLange \\\n",
      "--merge-alleles w_hm3.snplist \\\n",
      "--sumstats GWAScatalog/Crohns.deLange.hg38.rsid.forLDSC.txt \n",
      "\n",
      "Interpreting column names as follows:\n",
      "N:\tSample size\n",
      "A1:\tAllele 1, interpreted as ref allele for signed sumstat.\n",
      "P:\tp-Value\n",
      "BETA:\t[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)\n",
      "A2:\tAllele 2, interpreted as non-ref allele for signed sumstat.\n",
      "SNP:\tVariant ID (e.g., rs number)\n",
      "\n",
      "Reading list of SNPs for allele merge from w_hm3.snplist\n",
      "Read 1217311 SNPs for allele merge.\n",
      "Reading sumstats from GWAScatalog/Crohns.deLange.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.\n",
      ".. done\n",
      "Read 9469592 SNPs from --sumstats file.\n",
      "Removed 8297513 SNPs not in --merge-alleles.\n",
      "Removed 11594 SNPs with missing values.\n",
      "Removed 0 SNPs with INFO <= 0.9.\n",
      "Removed 0 SNPs with MAF <= 0.01.\n",
      "Removed 0 SNPs with out-of-bounds p-values.\n",
      "Removed 0 variants that were not SNPs or were strand-ambiguous.\n",
      "1160485 SNPs remain.\n",
      "Removed 0 SNPs with duplicated rs numbers (1160485 SNPs remain).\n",
      "Removed 0 SNPs with N < 26844.0 (1160485 SNPs remain).\n",
      "Median value of BETA was 0.0002, which seems sensible.\n",
      "Removed 0 SNPs whose alleles did not match --merge-alleles (1160485 SNPs remain).\n",
      "Writing summary statistics for 1217311 SNPs (1160485 with nonmissing beta) to Crohns.deLange.sumstats.gz.\n",
      "\n",
      "Metadata:\n",
      "Mean chi^2 = 1.456\n",
      "Lambda GC = 1.26\n",
      "Max chi^2 = 489.534\n",
      "2076 Genome-wide significant SNPs (some may have been removed by filtering).\n",
      "\n",
      "Conversion finished at Mon Mar 29 03:50:10 2021\n",
      "Total time elapsed: 1.0m:8.63s\n",
      "*********************************************************************\n",
      "* LD Score Regression (LDSC)\n",
      "* Version 1.0.1\n",
      "* (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane\n",
      "* Broad Institute of MIT and Harvard / MIT Department of Mathematics\n",
      "* GNU General Public License v3\n",
      "*********************************************************************\n",
      "Call: \n",
      "./munge_sumstats.py \\\n",
      "--out jArthritis.LopezIsac \\\n",
      "--merge-alleles w_hm3.snplist \\\n",
      "--sumstats GWAScatalog/jArthritis.LopezIsac.hg38.rsid.forLDSC.txt \n",
      "\n",
      "Interpreting column names as follows:\n",
      "N:\tSample size\n",
      "A1:\tAllele 1, interpreted as ref allele for signed sumstat.\n",
      "P:\tp-Value\n",
      "BETA:\t[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)\n",
      "A2:\tAllele 2, interpreted as non-ref allele for signed sumstat.\n",
      "SNP:\tVariant ID (e.g., rs number)\n",
      "\n",
      "Reading list of SNPs for allele merge from w_hm3.snplist\n",
      "Read 1217311 SNPs for allele merge.\n",
      "Reading sumstats from GWAScatalog/jArthritis.LopezIsac.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.\n",
      ".. done\n",
      "\n",
      "ERROR converting summary statistics:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"./ldsc/munge_sumstats.py\", line 686, in munge_sumstats\n",
      "    dat = parse_dat(dat_gen, cname_translation, merge_alleles, log, args)\n",
      "  File \"./ldsc/munge_sumstats.py\", line 301, in parse_dat\n",
      "    dat = pd.concat(dat_list, axis=0).reset_index(drop=True)\n",
      "  File \"/usr/local/Anaconda/envs/py2.7/lib/python2.7/site-packages/pandas/core/reshape/concat.py\", line 228, in concat\n",
      "    copy=copy, sort=sort)\n",
      "  File \"/usr/local/Anaconda/envs/py2.7/lib/python2.7/site-packages/pandas/core/reshape/concat.py\", line 262, in __init__\n",
      "    raise ValueError('No objects to concatenate')\n",
      "ValueError: No objects to concatenate\n",
      "\n",
      "\n",
      "Conversion finished at Mon Mar 29 03:50:27 2021\n",
      "Total time elapsed: 14.79s\n",
      "*********************************************************************\n",
      "* LD Score Regression (LDSC)\n",
      "* Version 1.0.1\n",
      "* (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane\n",
      "* Broad Institute of MIT and Harvard / MIT Department of Mathematics\n",
      "* GNU General Public License v3\n",
      "*********************************************************************\n",
      "Call: \n",
      "./munge_sumstats.py \\\n",
      "--out IBS.deLange \\\n",
      "--merge-alleles w_hm3.snplist \\\n",
      "--sumstats GWAScatalog/IBS.deLange.hg38.rsid.forLDSC.txt \n",
      "\n",
      "Interpreting column names as follows:\n",
      "N:\tSample size\n",
      "A1:\tAllele 1, interpreted as ref allele for signed sumstat.\n",
      "P:\tp-Value\n",
      "BETA:\t[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)\n",
      "A2:\tAllele 2, interpreted as non-ref allele for signed sumstat.\n",
      "SNP:\tVariant ID (e.g., rs number)\n",
      "\n",
      "Reading list of SNPs for allele merge from w_hm3.snplist\n",
      "Read 1217311 SNPs for allele merge.\n",
      "Reading sumstats from GWAScatalog/IBS.deLange.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.\n",
      ".. done\n",
      "Read 9633630 SNPs from --sumstats file.\n",
      "Removed 8458334 SNPs not in --merge-alleles.\n",
      "Removed 14614 SNPs with missing values.\n",
      "Removed 0 SNPs with INFO <= 0.9.\n",
      "Removed 0 SNPs with MAF <= 0.01.\n",
      "Removed 0 SNPs with out-of-bounds p-values.\n",
      "Removed 0 variants that were not SNPs or were strand-ambiguous.\n",
      "1160682 SNPs remain.\n",
      "Removed 0 SNPs with duplicated rs numbers (1160682 SNPs remain).\n",
      "Removed 0 SNPs with N < 39971.3333333 (1160682 SNPs remain).\n",
      "Median value of BETA was -0.0001, which seems sensible.\n",
      "Removed 0 SNPs whose alleles did not match --merge-alleles (1160682 SNPs remain).\n",
      "Writing summary statistics for 1217311 SNPs (1160682 with nonmissing beta) to IBS.deLange.sumstats.gz.\n",
      "\n",
      "Metadata:\n",
      "Mean chi^2 = 1.503\n",
      "Lambda GC = 1.291\n",
      "Max chi^2 = 501.36\n",
      "2383 Genome-wide significant SNPs (some may have been removed by filtering).\n",
      "\n",
      "Conversion finished at Mon Mar 29 03:51:36 2021\n",
      "Total time elapsed: 1.0m:8.43s\n",
      "*********************************************************************\n",
      "* LD Score Regression (LDSC)\n",
      "* Version 1.0.1\n",
      "* (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane\n",
      "* Broad Institute of MIT and Harvard / MIT Department of Mathematics\n",
      "* GNU General Public License v3\n",
      "*********************************************************************\n",
      "Call: \n",
      "./munge_sumstats.py \\\n",
      "--out MicroscopicColitis.Green \\\n",
      "--merge-alleles w_hm3.snplist \\\n",
      "--sumstats GWAScatalog/MicroscopicColitis.Green.hg38.rsid.forLDSC.txt \n",
      "\n",
      "Interpreting column names as follows:\n",
      "N:\tSample size\n",
      "A1:\tAllele 1, interpreted as ref allele for signed sumstat.\n",
      "P:\tp-Value\n",
      "BETA:\t[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)\n",
      "A2:\tAllele 2, interpreted as non-ref allele for signed sumstat.\n",
      "SNP:\tVariant ID (e.g., rs number)\n",
      "\n",
      "Reading list of SNPs for allele merge from w_hm3.snplist\n",
      "Read 1217311 SNPs for allele merge.\n",
      "Reading sumstats from GWAScatalog/MicroscopicColitis.Green.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.\n",
      ".. done\n",
      "Read 7935431 SNPs from --sumstats file.\n",
      "Removed 6800745 SNPs not in --merge-alleles.\n",
      "Removed 0 SNPs with missing values.\n",
      "Removed 0 SNPs with INFO <= 0.9.\n",
      "Removed 0 SNPs with MAF <= 0.01.\n",
      "Removed 0 SNPs with out-of-bounds p-values.\n",
      "Removed 115 variants that were not SNPs or were strand-ambiguous.\n",
      "1134571 SNPs remain.\n",
      "Removed 159 SNPs with duplicated rs numbers (1134412 SNPs remain).\n",
      "Removed 0 SNPs with N < 297103.333333 (1134412 SNPs remain).\n",
      "Median value of BETA was 6.58051e-07, which seems sensible.\n",
      "Removed 90 SNPs whose alleles did not match --merge-alleles (1134322 SNPs remain).\n",
      "Writing summary statistics for 1217311 SNPs (1134322 with nonmissing beta) to MicroscopicColitis.Green.sumstats.gz.\n",
      "\n",
      "Metadata:\n",
      "Mean chi^2 = 1.007\n",
      "WARNING: mean chi^2 may be too small.\n",
      "Lambda GC = 1.0\n",
      "Max chi^2 = 28.6\n",
      "0 Genome-wide significant SNPs (some may have been removed by filtering).\n",
      "\n",
      "Conversion finished at Mon Mar 29 03:53:34 2021\n",
      "Total time elapsed: 1.0m:55.62s\n",
      "*********************************************************************\n",
      "* LD Score Regression (LDSC)\n",
      "* Version 1.0.1\n",
      "* (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane\n",
      "* Broad Institute of MIT and Harvard / MIT Department of Mathematics\n",
      "* GNU General Public License v3\n",
      "*********************************************************************\n",
      "Call: \n",
      "./munge_sumstats.py \\\n",
      "--out MS.Beecham \\\n",
      "--merge-alleles w_hm3.snplist \\\n",
      "--sumstats GWAScatalog/MS.Beecham.hg38.rsid.forLDSC.txt \n",
      "\n",
      "Interpreting column names as follows:\n",
      "N:\tSample size\n",
      "A1:\tAllele 1, interpreted as ref allele for signed sumstat.\n",
      "P:\tp-Value\n",
      "BETA:\t[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)\n",
      "A2:\tAllele 2, interpreted as non-ref allele for signed sumstat.\n",
      "SNP:\tVariant ID (e.g., rs number)\n",
      "\n",
      "Reading list of SNPs for allele merge from w_hm3.snplist\n",
      "Read 1217311 SNPs for allele merge.\n",
      "Reading sumstats from GWAScatalog/MS.Beecham.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.\n",
      ".WARNING: 1 SNPs had P outside of (0,1]. The P column may be mislabeled.\n",
      " done\n",
      "Read 155249 SNPs from --sumstats file.\n",
      "Removed 80737 SNPs not in --merge-alleles.\n",
      "Removed 23160 SNPs with missing values.\n",
      "Removed 0 SNPs with INFO <= 0.9.\n",
      "Removed 0 SNPs with MAF <= 0.01.\n",
      "Removed 1 SNPs with out-of-bounds p-values.\n",
      "Removed 0 variants that were not SNPs or were strand-ambiguous.\n",
      "51351 SNPs remain.\n",
      "Removed 0 SNPs with duplicated rs numbers (51351 SNPs remain).\n",
      "Removed 0 SNPs with N < 25721.3333333 (51351 SNPs remain).\n",
      "Median value of BETA was 0.000999500333083, which seems sensible.\n",
      "Removed 1 SNPs whose alleles did not match --merge-alleles (51350 SNPs remain).\n",
      "Writing summary statistics for 1217311 SNPs (51350 with nonmissing beta) to MS.Beecham.sumstats.gz.\n",
      "\n",
      "Metadata:\n",
      "Mean chi^2 = 3.325\n",
      "Lambda GC = 1.499\n",
      "Max chi^2 = 1342.232\n",
      "655 Genome-wide significant SNPs (some may have been removed by filtering).\n",
      "\n",
      "Conversion finished at Mon Mar 29 03:55:11 2021\n",
      "Total time elapsed: 1.0m:33.14s\n",
      "*********************************************************************\n",
      "* LD Score Regression (LDSC)\n",
      "* Version 1.0.1\n",
      "* (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane\n",
      "* Broad Institute of MIT and Harvard / MIT Department of Mathematics\n",
      "* GNU General Public License v3\n",
      "*********************************************************************\n",
      "Call: \n",
      "./munge_sumstats.py \\\n",
      "--out NeuromyelitisOptica_AQP4_IgGpos.Estrada \\\n",
      "--merge-alleles w_hm3.snplist \\\n",
      "--sumstats GWAScatalog/NeuromyelitisOptica_AQP4_IgGpos.Estrada.hg38.rsid.forLDSC.txt \n",
      "\n",
      "Interpreting column names as follows:\n",
      "N:\tSample size\n",
      "A1:\tAllele 1, interpreted as ref allele for signed sumstat.\n",
      "P:\tp-Value\n",
      "BETA:\t[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)\n",
      "A2:\tAllele 2, interpreted as non-ref allele for signed sumstat.\n",
      "SNP:\tVariant ID (e.g., rs number)\n",
      "\n",
      "Reading list of SNPs for allele merge from w_hm3.snplist\n",
      "Read 1217311 SNPs for allele merge.\n",
      "Reading sumstats from GWAScatalog/NeuromyelitisOptica_AQP4_IgGpos.Estrada.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.\n",
      ".. done\n",
      "Read 6804545 SNPs from --sumstats file.\n",
      "Removed 5646909 SNPs not in --merge-alleles.\n",
      "Removed 13446 SNPs with missing values.\n",
      "Removed 0 SNPs with INFO <= 0.9.\n",
      "Removed 0 SNPs with MAF <= 0.01.\n",
      "Removed 0 SNPs with out-of-bounds p-values.\n",
      "Removed 0 variants that were not SNPs or were strand-ambiguous.\n",
      "1144190 SNPs remain.\n",
      "Removed 0 SNPs with duplicated rs numbers (1144190 SNPs remain).\n",
      "Removed 0 SNPs with N < 917.333333333 (1144190 SNPs remain).\n",
      "Median value of BETA was 0.0051, which seems sensible.\n",
      "Removed 0 SNPs whose alleles did not match --merge-alleles (1144190 SNPs remain).\n",
      "Writing summary statistics for 1217311 SNPs (1144190 with nonmissing beta) to NeuromyelitisOptica_AQP4_IgGpos.Estrada.sumstats.gz.\n",
      "\n",
      "Metadata:\n",
      "Mean chi^2 = 0.967\n",
      "WARNING: mean chi^2 may be too small.\n",
      "Lambda GC = 0.967\n",
      "Max chi^2 = 65.633\n",
      "33 Genome-wide significant SNPs (some may have been removed by filtering).\n",
      "\n",
      "Conversion finished at Mon Mar 29 03:58:34 2021\n",
      "Total time elapsed: 3.0m:20.09s\n",
      "*********************************************************************\n",
      "* LD Score Regression (LDSC)\n",
      "* Version 1.0.1\n",
      "* (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane\n",
      "* Broad Institute of MIT and Harvard / MIT Department of Mathematics\n",
      "* GNU General Public License v3\n",
      "*********************************************************************\n",
      "Call: \n",
      "./munge_sumstats.py \\\n",
      "--out RA.Okada \\\n",
      "--merge-alleles w_hm3.snplist \\\n",
      "--sumstats GWAScatalog/RA.Okada.hg38.rsid.forLDSC.txt \n",
      "\n",
      "Interpreting column names as follows:\n",
      "N:\tSample size\n",
      "A1:\tAllele 1, interpreted as ref allele for signed sumstat.\n",
      "P:\tp-Value\n",
      "BETA:\t[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)\n",
      "A2:\tAllele 2, interpreted as non-ref allele for signed sumstat.\n",
      "SNP:\tVariant ID (e.g., rs number)\n",
      "\n",
      "Reading list of SNPs for allele merge from w_hm3.snplist\n",
      "Read 1217311 SNPs for allele merge.\n",
      "Reading sumstats from GWAScatalog/RA.Okada.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.\n",
      ".. done\n",
      "Read 9700598 SNPs from --sumstats file.\n",
      "Removed 8506767 SNPs not in --merge-alleles.\n",
      "Removed 0 SNPs with missing values.\n",
      "Removed 0 SNPs with INFO <= 0.9.\n",
      "Removed 0 SNPs with MAF <= 0.01.\n",
      "Removed 0 SNPs with out-of-bounds p-values.\n",
      "Removed 1 variants that were not SNPs or were strand-ambiguous.\n",
      "1193830 SNPs remain.\n",
      "Removed 11 SNPs with duplicated rs numbers (1193819 SNPs remain).\n",
      "Removed 0 SNPs with N < 38856.0 (1193819 SNPs remain).\n",
      "Median value of BETA was 0.0, which seems sensible.\n",
      "Removed 2 SNPs whose alleles did not match --merge-alleles (1193817 SNPs remain).\n",
      "Writing summary statistics for 1217311 SNPs (1193817 with nonmissing beta) to RA.Okada.sumstats.gz.\n",
      "\n",
      "Metadata:\n",
      "Mean chi^2 = 1.212\n",
      "Lambda GC = 1.048\n",
      "Max chi^2 = 1143.797\n",
      "1334 Genome-wide significant SNPs (some may have been removed by filtering).\n",
      "\n",
      "Conversion finished at Mon Mar 29 04:01:03 2021\n",
      "Total time elapsed: 2.0m:25.61s\n",
      "*********************************************************************\n",
      "* LD Score Regression (LDSC)\n",
      "* Version 1.0.1\n",
      "* (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane\n",
      "* Broad Institute of MIT and Harvard / MIT Department of Mathematics\n",
      "* GNU General Public License v3\n",
      "*********************************************************************\n",
      "Call: \n",
      "./munge_sumstats.py \\\n",
      "--out SLE.Julia \\\n",
      "--merge-alleles w_hm3.snplist \\\n",
      "--sumstats GWAScatalog/SLE.Julia.hg38.rsid.forLDSC.txt \n",
      "\n",
      "Interpreting column names as follows:\n",
      "N:\tSample size\n",
      "A1:\tAllele 1, interpreted as ref allele for signed sumstat.\n",
      "P:\tp-Value\n",
      "A2:\tAllele 2, interpreted as non-ref allele for signed sumstat.\n",
      "SNP:\tVariant ID (e.g., rs number)\n",
      "Z:\tZ-score (0 --> no effect; above 0 --> A1 is trait/risk increasing)\n",
      "\n",
      "Reading list of SNPs for allele merge from w_hm3.snplist\n",
      "Read 1217311 SNPs for allele merge.\n",
      "Reading sumstats from GWAScatalog/SLE.Julia.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.\n",
      ".. done\n",
      "Read 8294672 SNPs from --sumstats file.\n",
      "Removed 7128945 SNPs not in --merge-alleles.\n",
      "Removed 19002 SNPs with missing values.\n",
      "Removed 0 SNPs with INFO <= 0.9.\n",
      "Removed 0 SNPs with MAF <= 0.01.\n",
      "Removed 0 SNPs with out-of-bounds p-values.\n",
      "Removed 0 variants that were not SNPs or were strand-ambiguous.\n",
      "1146725 SNPs remain.\n",
      "Removed 1 SNPs with duplicated rs numbers (1146724 SNPs remain).\n",
      "Removed 0 SNPs with N < 11310.6666667 (1146724 SNPs remain).\n",
      "Median value of Z was 0.001, which seems sensible.\n",
      "Removed 0 SNPs whose alleles did not match --merge-alleles (1146724 SNPs remain).\n",
      "Writing summary statistics for 1217311 SNPs (1146724 with nonmissing beta) to SLE.Julia.sumstats.gz.\n",
      "\n",
      "Metadata:\n",
      "Mean chi^2 = 1.299\n",
      "Lambda GC = 1.188\n",
      "Max chi^2 = 515.624\n",
      "1081 Genome-wide significant SNPs (some may have been removed by filtering).\n",
      "\n",
      "Conversion finished at Mon Mar 29 04:06:16 2021\n",
      "Total time elapsed: 5.0m:9.76s\n",
      "*********************************************************************\n",
      "* LD Score Regression (LDSC)\n",
      "* Version 1.0.1\n",
      "* (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane\n",
      "* Broad Institute of MIT and Harvard / MIT Department of Mathematics\n",
      "* GNU General Public License v3\n",
      "*********************************************************************\n",
      "Call: \n",
      "./munge_sumstats.py \\\n",
      "--out T1D.OnengutGumuscu \\\n",
      "--merge-alleles w_hm3.snplist \\\n",
      "--sumstats GWAScatalog/T1D.OnengutGumuscu.hg38.rsid.forLDSC.txt \n",
      "\n",
      "Interpreting column names as follows:\n",
      "N:\tSample size\n",
      "A1:\tAllele 1, interpreted as ref allele for signed sumstat.\n",
      "P:\tp-Value\n",
      "BETA:\t[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)\n",
      "A2:\tAllele 2, interpreted as non-ref allele for signed sumstat.\n",
      "SNP:\tVariant ID (e.g., rs number)\n",
      "\n",
      "Reading list of SNPs for allele merge from w_hm3.snplist\n",
      "Read 1217311 SNPs for allele merge.\n",
      "Reading sumstats from GWAScatalog/T1D.OnengutGumuscu.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.\n",
      ". done\n",
      "Read 121619 SNPs from --sumstats file.\n",
      "Removed 53156 SNPs not in --merge-alleles.\n",
      "Removed 20518 SNPs with missing values.\n",
      "Removed 0 SNPs with INFO <= 0.9.\n",
      "Removed 0 SNPs with MAF <= 0.01.\n",
      "Removed 0 SNPs with out-of-bounds p-values.\n",
      "Removed 0 variants that were not SNPs or were strand-ambiguous.\n",
      "47945 SNPs remain.\n",
      "Removed 59 SNPs with duplicated rs numbers (47886 SNPs remain).\n",
      "Removed 0 SNPs with N < 19768.0 (47886 SNPs remain).\n",
      "Median value of BETA was -0.00100050033358, which seems sensible.\n",
      "Removed 0 SNPs whose alleles did not match --merge-alleles (47886 SNPs remain).\n",
      "Writing summary statistics for 1217311 SNPs (47886 with nonmissing beta) to T1D.OnengutGumuscu.sumstats.gz.\n",
      "\n",
      "Metadata:\n",
      "Mean chi^2 = 2.932\n",
      "Lambda GC = 1.645\n",
      "Max chi^2 = 892.191\n",
      "652 Genome-wide significant SNPs (some may have been removed by filtering).\n",
      "\n",
      "Conversion finished at Mon Mar 29 04:11:19 2021\n",
      "Total time elapsed: 4.0m:51.62s\n",
      "*********************************************************************\n",
      "* LD Score Regression (LDSC)\n",
      "* Version 1.0.1\n",
      "* (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane\n",
      "* Broad Institute of MIT and Harvard / MIT Department of Mathematics\n",
      "* GNU General Public License v3\n",
      "*********************************************************************\n",
      "Call: \n",
      "./munge_sumstats.py \\\n",
      "--out UC.deLange \\\n",
      "--merge-alleles w_hm3.snplist \\\n",
      "--sumstats GWAScatalog/UC.deLange.hg38.rsid.forLDSC.txt \n",
      "\n",
      "Interpreting column names as follows:\n",
      "N:\tSample size\n",
      "A1:\tAllele 1, interpreted as ref allele for signed sumstat.\n",
      "P:\tp-Value\n",
      "BETA:\t[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)\n",
      "A2:\tAllele 2, interpreted as non-ref allele for signed sumstat.\n",
      "SNP:\tVariant ID (e.g., rs number)\n",
      "\n",
      "Reading list of SNPs for allele merge from w_hm3.snplist\n",
      "Read 1217311 SNPs for allele merge.\n",
      "Reading sumstats from GWAScatalog/UC.deLange.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.\n",
      ".. done\n",
      "Read 9486539 SNPs from --sumstats file.\n",
      "Removed 8314044 SNPs not in --merge-alleles.\n",
      "Removed 11980 SNPs with missing values.\n",
      "Removed 0 SNPs with INFO <= 0.9.\n",
      "Removed 0 SNPs with MAF <= 0.01.\n",
      "Removed 0 SNPs with out-of-bounds p-values.\n",
      "Removed 0 variants that were not SNPs or were strand-ambiguous.\n",
      "1160515 SNPs remain.\n",
      "Removed 0 SNPs with duplicated rs numbers (1160515 SNPs remain).\n",
      "Removed 0 SNPs with N < 30650.0 (1160515 SNPs remain).\n",
      "Median value of BETA was -0.0001, which seems sensible.\n",
      "Removed 0 SNPs whose alleles did not match --merge-alleles (1160515 SNPs remain).\n",
      "Writing summary statistics for 1217311 SNPs (1160515 with nonmissing beta) to UC.deLange.sumstats.gz.\n",
      "\n",
      "Metadata:\n",
      "Mean chi^2 = 1.321\n",
      "Lambda GC = 1.208\n",
      "Max chi^2 = 279.731\n",
      "1042 Genome-wide significant SNPs (some may have been removed by filtering).\n",
      "\n",
      "Conversion finished at Mon Mar 29 04:14:59 2021\n",
      "Total time elapsed: 3.0m:30.76s\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"./ldsc/munge_sumstats.py\", line 745, in <module>\n",
      "    munge_sumstats(parser.parse_args(), p=True)\n",
      "  File \"./ldsc/munge_sumstats.py\", line 686, in munge_sumstats\n",
      "    dat = parse_dat(dat_gen, cname_translation, merge_alleles, log, args)\n",
      "  File \"./ldsc/munge_sumstats.py\", line 301, in parse_dat\n",
      "    dat = pd.concat(dat_list, axis=0).reset_index(drop=True)\n",
      "  File \"/usr/local/Anaconda/envs/py2.7/lib/python2.7/site-packages/pandas/core/reshape/concat.py\", line 228, in concat\n",
      "    copy=copy, sort=sort)\n",
      "  File \"/usr/local/Anaconda/envs/py2.7/lib/python2.7/site-packages/pandas/core/reshape/concat.py\", line 262, in __init__\n",
      "    raise ValueError('No objects to concatenate')\n",
      "ValueError: No objects to concatenate\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "source /data/chiarp/conda/etc/profile.d/conda.sh\n",
    "\n",
    "python ./ldsc/munge_sumstats.py \\\n",
    "--sumstats GWAScatalog/Crohns.deLange.hg38.rsid.forLDSC.txt \\\n",
    "--out Crohns.deLange \\\n",
    "--merge-alleles w_hm3.snplist\n",
    "\n",
    "python ./ldsc/munge_sumstats.py \\\n",
    "--sumstats GWAScatalog/jArthritis.LopezIsac.hg38.rsid.forLDSC.txt \\\n",
    "--out jArthritis.LopezIsac \\\n",
    "--merge-alleles w_hm3.snplist\n",
    "\n",
    "python ./ldsc/munge_sumstats.py \\\n",
    "--sumstats GWAScatalog/IBS.deLange.hg38.rsid.forLDSC.txt \\\n",
    "--out IBS.deLange \\\n",
    "--merge-alleles w_hm3.snplist\n",
    "\n",
    "python ./ldsc/munge_sumstats.py \\\n",
    "--sumstats GWAScatalog/MicroscopicColitis.Green.hg38.rsid.forLDSC.txt \\\n",
    "--out MicroscopicColitis.Green \\\n",
    "--merge-alleles w_hm3.snplist\n",
    "\n",
    "python ./ldsc/munge_sumstats.py \\\n",
    "--sumstats GWAScatalog/MS.Beecham.hg38.rsid.forLDSC.txt \\\n",
    "--out MS.Beecham \\\n",
    "--merge-alleles w_hm3.snplist\n",
    "\n",
    "python ./ldsc/munge_sumstats.py \\\n",
    "--sumstats GWAScatalog/NeuromyelitisOptica_AQP4_IgGpos.Estrada.hg38.rsid.forLDSC.txt \\\n",
    "--out NeuromyelitisOptica_AQP4_IgGpos.Estrada \\\n",
    "--merge-alleles w_hm3.snplist\n",
    "\n",
    "python ./ldsc/munge_sumstats.py \\\n",
    "--sumstats GWAScatalog/RA.Okada.hg38.rsid.forLDSC.txt \\\n",
    "--out RA.Okada \\\n",
    "--merge-alleles w_hm3.snplist\n",
    "\n",
    "python ./ldsc/munge_sumstats.py \\\n",
    "--sumstats GWAScatalog/SLE.Julia.hg38.rsid.forLDSC.txt \\\n",
    "--out SLE.Julia \\\n",
    "--merge-alleles w_hm3.snplist\n",
    "\n",
    "python ./ldsc/munge_sumstats.py \\\n",
    "--sumstats GWAScatalog/T1D.OnengutGumuscu.hg38.rsid.forLDSC.txt \\\n",
    "--out T1D.OnengutGumuscu \\\n",
    "--merge-alleles w_hm3.snplist\n",
    "\n",
    "python ./ldsc/munge_sumstats.py \\\n",
    "--sumstats GWAScatalog/UC.deLange.hg38.rsid.forLDSC.txt \\\n",
    "--out UC.deLange \\\n",
    "--merge-alleles w_hm3.snplist"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "chief-shooting",
   "metadata": {},
   "source": [
    "### Estimating Heritability, Genetic Correlation with autoimmune diseases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "surrounded-dimension",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "*********************************************************************\n",
      "* LD Score Regression (LDSC)\n",
      "* Version 1.0.1\n",
      "* (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane\n",
      "* Broad Institute of MIT and Harvard / MIT Department of Mathematics\n",
      "* GNU General Public License v3\n",
      "*********************************************************************\n",
      "Call: \n",
      "./ldsc.py \\\n",
      "--ref-ld-chr eur_w_ld_chr/ \\\n",
      "--out MG_AutoImmuneDiseases \\\n",
      "--rg MG.Chia.sumstats.gz,Crohns.deLange.sumstats.gz,NeuromyelitisOptica_AQP4_IgGpos.Estrada.sumstats.gz,RA.Okada.sumstats.gz,SLE.Julia.sumstats.gz,T1D.OnengutGumuscu.sumstats.gz,UC.deLange.sumstats.gz \\\n",
      "--w-ld-chr eur_w_ld_chr/ \n",
      "\n",
      "Beginning analysis at Tue Apr  6 12:58:47 2021\n",
      "Reading summary statistics from MG.Chia.sumstats.gz ...\n",
      "Read summary statistics for 1114926 SNPs.\n",
      "Reading reference panel LD Score from eur_w_ld_chr/[1-22] ... (ldscore_fromlist)\n",
      "Read reference panel LD Scores for 1290028 SNPs.\n",
      "Removing partitioned LD Scores with zero variance.\n",
      "Reading regression weight LD Score from eur_w_ld_chr/[1-22] ... (ldscore_fromlist)\n",
      "Read regression weight LD Scores for 1290028 SNPs.\n",
      "After merging with reference panel LD, 1107424 SNPs remain.\n",
      "After merging with regression SNP LD, 1107424 SNPs remain.\n",
      "Computing rg for phenotype 2/7\n",
      "Reading summary statistics from Crohns.deLange.sumstats.gz ...\n",
      "Read summary statistics for 1217311 SNPs.\n",
      "After merging with summary statistics, 1107424 SNPs remain.\n",
      "1067853 SNPs with valid alleles.\n",
      "\n",
      "Heritability of phenotype 1\n",
      "---------------------------\n",
      "Total Observed scale h2: 0.075 (0.015)\n",
      "Lambda GC: 1.0618\n",
      "Mean Chi^2: 1.0745\n",
      "Intercept: 1.0178 (0.0071)\n",
      "Ratio: 0.2393 (0.0947)\n",
      "\n",
      "Heritability of phenotype 2/7\n",
      "-----------------------------\n",
      "Total Observed scale h2: 0.4422 (0.0524)\n",
      "Lambda GC: 1.2564\n",
      "Mean Chi^2: 1.4482\n",
      "Intercept: 1.0975 (0.0153)\n",
      "Ratio: 0.2175 (0.0341)\n",
      "\n",
      "Genetic Covariance\n",
      "------------------\n",
      "Total Observed scale gencov: 0.0126 (0.0158)\n",
      "Mean z1*z2: 0.0107\n",
      "Intercept: 0.0014 (0.0063)\n",
      "\n",
      "Genetic Correlation\n",
      "-------------------\n",
      "Genetic Correlation: 0.0693 (0.0883)\n",
      "Z-score: 0.7844\n",
      "P: 0.4328\n",
      "\n",
      "Computing rg for phenotype 3/7\n",
      "Reading summary statistics from NeuromyelitisOptica_AQP4_IgGpos.Estrada.sumstats.gz ...\n",
      "Read summary statistics for 1217311 SNPs.\n",
      "After merging with summary statistics, 1107424 SNPs remain.\n",
      "1070202 SNPs with valid alleles.\n",
      "\n",
      "Heritability of phenotype 3/7\n",
      "-----------------------------\n",
      "Total Observed scale h2: -0.1747 (0.3013)\n",
      "Lambda GC: 0.9605\n",
      "Mean Chi^2: 0.9567\n",
      "Intercept: 0.9614 (0.0061)\n",
      "Ratio: NA (mean chi^2 < 1)\n",
      "\n",
      "Genetic Covariance\n",
      "------------------\n",
      "Total Observed scale gencov: 0.0468 (0.0474)\n",
      "Mean z1*z2: 0.0029\n",
      "Intercept: -0.0038 (0.0046)\n",
      "\n",
      "Genetic Correlation\n",
      "-------------------\n",
      "Genetic Correlation: nan (nan) (h2  out of bounds) \n",
      "Z-score: nan (nan) (h2  out of bounds)\n",
      "P: nan (nan) (h2  out of bounds)\n",
      "WARNING: One of the h2's was out of bounds.\n",
      "This usually indicates a data-munging error or that h2 or N is low.\n",
      "\n",
      "Computing rg for phenotype 4/7\n",
      "Reading summary statistics from RA.Okada.sumstats.gz ...\n",
      "Read summary statistics for 1217311 SNPs.\n",
      "After merging with summary statistics, 1107424 SNPs remain.\n",
      "1099228 SNPs with valid alleles.\n",
      "\n",
      "Heritability of phenotype 4/7\n",
      "-----------------------------\n",
      "Total Observed scale h2: 0.1696 (0.0229)\n",
      "Lambda GC: 1.0466\n",
      "Mean Chi^2: 1.1445\n",
      "Intercept: 0.9484 (0.0117)\n",
      "Ratio < 0 (usually indicates GC correction).\n",
      "\n",
      "Genetic Covariance\n",
      "------------------\n",
      "Total Observed scale gencov: 0.0466 (0.0125)\n",
      "Mean z1*z2: 0.0444\n",
      "Intercept: 0.0003 (0.0058)\n",
      "\n",
      "Genetic Correlation\n",
      "-------------------\n",
      "Genetic Correlation: 0.4135 (0.0929)\n",
      "Z-score: 4.4531\n",
      "P: 8.4652e-06\n",
      "\n",
      "Computing rg for phenotype 5/7\n",
      "Reading summary statistics from SLE.Julia.sumstats.gz ...\n",
      "Read summary statistics for 1217311 SNPs.\n",
      "After merging with summary statistics, 1107424 SNPs remain.\n",
      "1056030 SNPs with valid alleles.\n",
      "\n",
      "Heritability of phenotype 5/7\n",
      "-----------------------------\n",
      "Total Observed scale h2: 0.4178 (0.0645)\n",
      "Lambda GC: 1.1811\n",
      "Mean Chi^2: 1.2485\n",
      "Intercept: 1.1006 (0.0121)\n",
      "Ratio: 0.4047 (0.0485)\n",
      "\n",
      "Genetic Covariance\n",
      "------------------\n",
      "Total Observed scale gencov: -0.0373 (0.0202)\n",
      "Mean z1*z2: -0.033\n",
      "Intercept: -0.0138 (0.0063)\n",
      "\n",
      "Genetic Correlation\n",
      "-------------------\n",
      "Genetic Correlation: -0.2126 (0.1185)\n",
      "Z-score: -1.7942\n",
      "P: 0.0728\n",
      "\n",
      "Computing rg for phenotype 6/7\n",
      "Reading summary statistics from T1D.OnengutGumuscu.sumstats.gz ...\n",
      "Read summary statistics for 1217311 SNPs.\n",
      "After merging with summary statistics, 1107424 SNPs remain.\n",
      "44608 SNPs with valid alleles.\n",
      "WARNING: number of SNPs less than 200k; this is almost always bad.\n",
      "\n",
      "Heritability of phenotype 6/7\n",
      "-----------------------------\n",
      "Total Observed scale h2: 2.1786 (0.5807)\n",
      "Lambda GC: 1.641\n",
      "Mean Chi^2: 2.8979\n",
      "Intercept: 1.3416 (0.2462)\n",
      "Ratio: 0.18 (0.1297)\n",
      "\n",
      "Genetic Covariance\n",
      "------------------\n",
      "Total Observed scale gencov: 0.3856 (0.1396)\n",
      "Mean z1*z2: 0.36\n",
      "Intercept: -0.0212 (0.043)\n",
      "\n",
      "Genetic Correlation\n",
      "-------------------\n",
      "Genetic Correlation: 0.4356 (0.1066)\n",
      "Z-score: 4.087\n",
      "P: 4.3707e-05\n",
      "\n",
      "Computing rg for phenotype 7/7\n",
      "Reading summary statistics from UC.deLange.sumstats.gz ...\n",
      "Read summary statistics for 1217311 SNPs.\n",
      "After merging with summary statistics, 1107424 SNPs remain.\n",
      "1067835 SNPs with valid alleles.\n",
      "\n",
      "Heritability of phenotype 7/7\n",
      "-----------------------------\n",
      "Total Observed scale h2: 0.2335 (0.0256)\n",
      "Lambda GC: 1.2038\n",
      "Mean Chi^2: 1.3086\n",
      "Intercept: 1.101 (0.0138)\n",
      "Ratio: 0.3274 (0.0449)\n",
      "\n",
      "Genetic Covariance\n",
      "------------------\n",
      "Total Observed scale gencov: 0.0267 (0.011)\n",
      "Mean z1*z2: 0.0119\n",
      "Intercept: -0.0097 (0.0052)\n",
      "\n",
      "Genetic Correlation\n",
      "-------------------\n",
      "Genetic Correlation: 0.2015 (0.0851)\n",
      "Z-score: 2.3671\n",
      "P: 0.0179\n",
      "\n",
      "\n",
      "Summary of Genetic Correlation Results\n",
      "                  p1                                                   p2       rg       se       z          p  h2_obs  h2_obs_se  h2_int  h2_int_se  gcov_int  gcov_int_se\n",
      " MG.Chia.sumstats.gz                           Crohns.deLange.sumstats.gz  0.06927   0.0883  0.7844     0.4328  0.4422     0.0524  1.0975     0.0153    0.0014       0.0063\n",
      " MG.Chia.sumstats.gz  NeuromyelitisOptica_AQP4_IgGpos.Estrada.sumstats.gz       NA       NA      NA         NA -0.1747     0.3013  0.9614     0.0061   -0.0038       0.0046\n",
      " MG.Chia.sumstats.gz                                 RA.Okada.sumstats.gz   0.4135  0.09286   4.453  8.465e-06  0.1696     0.0229  0.9484     0.0117    0.0003       0.0058\n",
      " MG.Chia.sumstats.gz                                SLE.Julia.sumstats.gz  -0.2126   0.1185  -1.794    0.07278  0.4178     0.0645  1.1006     0.0121   -0.0138       0.0063\n",
      " MG.Chia.sumstats.gz                       T1D.OnengutGumuscu.sumstats.gz   0.4356   0.1066   4.087  4.371e-05  2.1786     0.5807  1.3416     0.2462   -0.0212       0.0430\n",
      " MG.Chia.sumstats.gz                               UC.deLange.sumstats.gz   0.2015  0.08513   2.367    0.01793  0.2335     0.0256  1.1010     0.0138   -0.0097       0.0052\n",
      "\n",
      "Analysis finished at Tue Apr  6 12:59:36 2021\n",
      "Total time elapsed: 48.76s\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/gpfs/gsfs11/users/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/LDSC/ldsc/ldscore/sumstats.py:532: FutureWarning: Method .as_matrix will be removed in a future version. Use .values instead.\n",
      "  ref_ld = sumstats.as_matrix(columns=ref_ld_cnames)\n",
      "/gpfs/gsfs11/users/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/LDSC/ldsc/ldscore/irwls.py:161: FutureWarning: `rcond` parameter will change to the default of machine precision times ``max(M, N)`` where M and N are the input matrix dimensions.\n",
      "To use the future default and silence this warning we advise to pass `rcond=None`, to keep using the old, explicitly pass `rcond=-1`.\n",
      "  coef = np.linalg.lstsq(x, y)\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "source /data/chiarp/conda/etc/profile.d/conda.sh\n",
    "\n",
    "python ./ldsc/ldsc.py \\\n",
    "--rg MG.Chia.sumstats.gz,Crohns.deLange.sumstats.gz,NeuromyelitisOptica_AQP4_IgGpos.Estrada.sumstats.gz,RA.Okada.sumstats.gz,SLE.Julia.sumstats.gz,T1D.OnengutGumuscu.sumstats.gz,UC.deLange.sumstats.gz \\\n",
    "--ref-ld-chr eur_w_ld_chr/ \\\n",
    "--w-ld-chr eur_w_ld_chr/ \\\n",
    "--out MG_AutoImmuneDiseases"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "unnecessary-commissioner",
   "metadata": {},
   "source": [
    "### Estimating Heritability, Genetic Correlation with neurodisease and autoimmune diseases\n",
    "\n",
    "\n",
    "Discussed with Bryan, and he suggested that I take out the following diseases from the autoimmune group: \n",
    "\n",
    "microscopic colitis and IBS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "convertible-haven",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "*********************************************************************\n",
      "* LD Score Regression (LDSC)\n",
      "* Version 1.0.1\n",
      "* (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane\n",
      "* Broad Institute of MIT and Harvard / MIT Department of Mathematics\n",
      "* GNU General Public License v3\n",
      "*********************************************************************\n",
      "Call: \n",
      "./ldsc.py \\\n",
      "--ref-ld-chr eur_w_ld_chr/ \\\n",
      "--out MG_Neuro.AutoImmuneDiseases \\\n",
      "--rg MG.Chia.sumstats.gz,AD.Jansen.sumstats.gz,PD.Nalls.sumstats.gz,ALS.VanRheenen.sumstats.gz,LBD.Chia.sumstats.gz,Crohns.deLange.sumstats.gz,MS.Beecham.sumstats.gz,RA.Okada.sumstats.gz,SLE.Julia.sumstats.gz,T1D.OnengutGumuscu.sumstats.gz,UC.deLange.sumstats.gz \\\n",
      "--w-ld-chr eur_w_ld_chr/ \n",
      "\n",
      "Beginning analysis at Tue Apr  6 12:59:38 2021\n",
      "Reading summary statistics from MG.Chia.sumstats.gz ...\n",
      "Read summary statistics for 1114926 SNPs.\n",
      "Reading reference panel LD Score from eur_w_ld_chr/[1-22] ... (ldscore_fromlist)\n",
      "Read reference panel LD Scores for 1290028 SNPs.\n",
      "Removing partitioned LD Scores with zero variance.\n",
      "Reading regression weight LD Score from eur_w_ld_chr/[1-22] ... (ldscore_fromlist)\n",
      "Read regression weight LD Scores for 1290028 SNPs.\n",
      "After merging with reference panel LD, 1107424 SNPs remain.\n",
      "After merging with regression SNP LD, 1107424 SNPs remain.\n",
      "Computing rg for phenotype 2/11\n",
      "Reading summary statistics from AD.Jansen.sumstats.gz ...\n",
      "Read summary statistics for 1217311 SNPs.\n",
      "After merging with summary statistics, 1107424 SNPs remain.\n",
      "1065135 SNPs with valid alleles.\n",
      "\n",
      "Heritability of phenotype 1\n",
      "---------------------------\n",
      "Total Observed scale h2: 0.0754 (0.0165)\n",
      "Lambda GC: 1.0618\n",
      "Mean Chi^2: 1.0751\n",
      "Intercept: 1.0171 (0.0078)\n",
      "Ratio: 0.2272 (0.1044)\n",
      "\n",
      "Heritability of phenotype 2/11\n",
      "------------------------------\n",
      "Total Observed scale h2: 0.0075 (0.0049)\n",
      "Lambda GC: 1.0833\n",
      "Mean Chi^2: 1.1236\n",
      "Intercept: 1.057 (0.0592)\n",
      "Ratio: 0.4613 (0.4791)\n",
      "\n",
      "Genetic Covariance\n",
      "------------------\n",
      "Total Observed scale gencov: 0.0002 (0.0029)\n",
      "Mean z1*z2: 0.0062\n",
      "Intercept: 0.005 (0.0047)\n",
      "\n",
      "Genetic Correlation\n",
      "-------------------\n",
      "Genetic Correlation: 0.0091 (0.123)\n",
      "Z-score: 0.0741\n",
      "P: 0.9409\n",
      "\n",
      "Computing rg for phenotype 3/11\n",
      "Reading summary statistics from PD.Nalls.sumstats.gz ...\n",
      "Read summary statistics for 1217311 SNPs.\n",
      "After merging with summary statistics, 1107424 SNPs remain.\n",
      "1064424 SNPs with valid alleles.\n",
      "\n",
      "Heritability of phenotype 3/11\n",
      "------------------------------\n",
      "Total Observed scale h2: 0.0166 (0.0017)\n",
      "Lambda GC: 1.0864\n",
      "Mean Chi^2: 1.1347\n",
      "Intercept: 0.981 (0.0076)\n",
      "Ratio < 0 (usually indicates GC correction).\n",
      "\n",
      "Genetic Covariance\n",
      "------------------\n",
      "Total Observed scale gencov: 0.0001 (0.0031)\n",
      "Mean z1*z2: -0.0018\n",
      "Intercept: -0.0029 (0.0053)\n",
      "\n",
      "Genetic Correlation\n",
      "-------------------\n",
      "Genetic Correlation: 0.0032 (0.0883)\n",
      "Z-score: 0.0359\n",
      "P: 0.9714\n",
      "\n",
      "Computing rg for phenotype 4/11\n",
      "Reading summary statistics from ALS.VanRheenen.sumstats.gz ...\n",
      "Read summary statistics for 1217311 SNPs.\n",
      "After merging with summary statistics, 1107424 SNPs remain.\n",
      "1098401 SNPs with valid alleles.\n",
      "\n",
      "Heritability of phenotype 4/11\n",
      "------------------------------\n",
      "Total Observed scale h2: 0.0485 (0.0131)\n",
      "Lambda GC: 1.0225\n",
      "Mean Chi^2: 1.0281\n",
      "Intercept: 0.9941 (0.0062)\n",
      "Ratio < 0 (usually indicates GC correction).\n",
      "\n",
      "Genetic Covariance\n",
      "------------------\n",
      "Total Observed scale gencov: 0.0171 (0.0102)\n",
      "Mean z1*z2: 0.0246\n",
      "Intercept: 0.0122 (0.0047)\n",
      "\n",
      "Genetic Correlation\n",
      "-------------------\n",
      "Genetic Correlation: 0.2858 (0.18)\n",
      "Z-score: 1.5881\n",
      "P: 0.1123\n",
      "\n",
      "Computing rg for phenotype 5/11\n",
      "Reading summary statistics from LBD.Chia.sumstats.gz ...\n",
      "Read summary statistics for 1217311 SNPs.\n",
      "After merging with summary statistics, 1107424 SNPs remain.\n",
      "1009819 SNPs with valid alleles.\n",
      "\n",
      "Heritability of phenotype 5/11\n",
      "------------------------------\n",
      "Total Observed scale h2: 0.1293 (0.0835)\n",
      "Lambda GC: 1.0225\n",
      "Mean Chi^2: 1.0233\n",
      "Intercept: 1.0064 (0.0087)\n",
      "Ratio: 0.2728 (0.3714)\n",
      "\n",
      "Genetic Covariance\n",
      "------------------\n",
      "Total Observed scale gencov: 0.0349 (0.0229)\n",
      "Mean z1*z2: -0.0009\n",
      "Intercept: -0.0117 (0.0047)\n",
      "\n",
      "Genetic Correlation\n",
      "-------------------\n",
      "Genetic Correlation: 0.3657 (0.2797)\n",
      "Z-score: 1.3078\n",
      "P: 0.191\n",
      "\n",
      "Computing rg for phenotype 6/11\n",
      "Reading summary statistics from Crohns.deLange.sumstats.gz ...\n",
      "Read summary statistics for 1217311 SNPs.\n",
      "After merging with summary statistics, 1107424 SNPs remain.\n",
      "1067853 SNPs with valid alleles.\n",
      "\n",
      "Heritability of phenotype 6/11\n",
      "------------------------------\n",
      "Total Observed scale h2: 0.4422 (0.0524)\n",
      "Lambda GC: 1.2564\n",
      "Mean Chi^2: 1.4482\n",
      "Intercept: 1.0975 (0.0153)\n",
      "Ratio: 0.2175 (0.0341)\n",
      "\n",
      "Genetic Covariance\n",
      "------------------\n",
      "Total Observed scale gencov: 0.0126 (0.0158)\n",
      "Mean z1*z2: 0.0107\n",
      "Intercept: 0.0014 (0.0063)\n",
      "\n",
      "Genetic Correlation\n",
      "-------------------\n",
      "Genetic Correlation: 0.0693 (0.0883)\n",
      "Z-score: 0.7844\n",
      "P: 0.4328\n",
      "\n",
      "Computing rg for phenotype 7/11\n",
      "Reading summary statistics from MS.Beecham.sumstats.gz ...\n",
      "Read summary statistics for 1217311 SNPs.\n",
      "After merging with summary statistics, 1107424 SNPs remain.\n",
      "46758 SNPs with valid alleles.\n",
      "WARNING: number of SNPs less than 200k; this is almost always bad.\n",
      "\n",
      "Heritability of phenotype 7/11\n",
      "------------------------------\n",
      "Total Observed scale h2: 1.5156 (0.2219)\n",
      "Lambda GC: 1.4494\n",
      "Mean Chi^2: 2.1573\n",
      "Intercept: 1.0313 (0.0998)\n",
      "Ratio: 0.0271 (0.0862)\n",
      "\n",
      "Genetic Covariance\n",
      "------------------\n",
      "Total Observed scale gencov: 0.0849 (0.0654)\n",
      "Mean z1*z2: 0.0818\n",
      "Intercept: -0.0232 (0.0267)\n",
      "\n",
      "Genetic Correlation\n",
      "-------------------\n",
      "Genetic Correlation: 0.1121 (0.0825)\n",
      "Z-score: 1.3585\n",
      "P: 0.1743\n",
      "\n",
      "Computing rg for phenotype 8/11\n",
      "Reading summary statistics from RA.Okada.sumstats.gz ...\n",
      "Read summary statistics for 1217311 SNPs.\n",
      "After merging with summary statistics, 1107424 SNPs remain.\n",
      "1099228 SNPs with valid alleles.\n",
      "\n",
      "Heritability of phenotype 8/11\n",
      "------------------------------\n",
      "Total Observed scale h2: 0.1696 (0.0229)\n",
      "Lambda GC: 1.0466\n",
      "Mean Chi^2: 1.1445\n",
      "Intercept: 0.9484 (0.0117)\n",
      "Ratio < 0 (usually indicates GC correction).\n",
      "\n",
      "Genetic Covariance\n",
      "------------------\n",
      "Total Observed scale gencov: 0.0466 (0.0125)\n",
      "Mean z1*z2: 0.0444\n",
      "Intercept: 0.0003 (0.0058)\n",
      "\n",
      "Genetic Correlation\n",
      "-------------------\n",
      "Genetic Correlation: 0.4135 (0.0929)\n",
      "Z-score: 4.4531\n",
      "P: 8.4652e-06\n",
      "\n",
      "Computing rg for phenotype 9/11\n",
      "Reading summary statistics from SLE.Julia.sumstats.gz ...\n",
      "Read summary statistics for 1217311 SNPs.\n",
      "After merging with summary statistics, 1107424 SNPs remain.\n",
      "1056030 SNPs with valid alleles.\n",
      "\n",
      "Heritability of phenotype 9/11\n",
      "------------------------------\n",
      "Total Observed scale h2: 0.4178 (0.0645)\n",
      "Lambda GC: 1.1811\n",
      "Mean Chi^2: 1.2485\n",
      "Intercept: 1.1006 (0.0121)\n",
      "Ratio: 0.4047 (0.0485)\n",
      "\n",
      "Genetic Covariance\n",
      "------------------\n",
      "Total Observed scale gencov: -0.0373 (0.0202)\n",
      "Mean z1*z2: -0.033\n",
      "Intercept: -0.0138 (0.0063)\n",
      "\n",
      "Genetic Correlation\n",
      "-------------------\n",
      "Genetic Correlation: -0.2126 (0.1185)\n",
      "Z-score: -1.7942\n",
      "P: 0.0728\n",
      "\n",
      "Computing rg for phenotype 10/11\n",
      "Reading summary statistics from T1D.OnengutGumuscu.sumstats.gz ...\n",
      "Read summary statistics for 1217311 SNPs.\n",
      "After merging with summary statistics, 1107424 SNPs remain.\n",
      "44608 SNPs with valid alleles.\n",
      "WARNING: number of SNPs less than 200k; this is almost always bad.\n",
      "\n",
      "Heritability of phenotype 10/11\n",
      "-------------------------------\n",
      "Total Observed scale h2: 2.1786 (0.5807)\n",
      "Lambda GC: 1.641\n",
      "Mean Chi^2: 2.8979\n",
      "Intercept: 1.3416 (0.2462)\n",
      "Ratio: 0.18 (0.1297)\n",
      "\n",
      "Genetic Covariance\n",
      "------------------\n",
      "Total Observed scale gencov: 0.3856 (0.1396)\n",
      "Mean z1*z2: 0.36\n",
      "Intercept: -0.0212 (0.043)\n",
      "\n",
      "Genetic Correlation\n",
      "-------------------\n",
      "Genetic Correlation: 0.4356 (0.1066)\n",
      "Z-score: 4.087\n",
      "P: 4.3707e-05\n",
      "\n",
      "Computing rg for phenotype 11/11\n",
      "Reading summary statistics from UC.deLange.sumstats.gz ...\n",
      "Read summary statistics for 1217311 SNPs.\n",
      "After merging with summary statistics, 1107424 SNPs remain.\n",
      "1067835 SNPs with valid alleles.\n",
      "\n",
      "Heritability of phenotype 11/11\n",
      "-------------------------------\n",
      "Total Observed scale h2: 0.2335 (0.0256)\n",
      "Lambda GC: 1.2038\n",
      "Mean Chi^2: 1.3086\n",
      "Intercept: 1.101 (0.0138)\n",
      "Ratio: 0.3274 (0.0449)\n",
      "\n",
      "Genetic Covariance\n",
      "------------------\n",
      "Total Observed scale gencov: 0.0267 (0.011)\n",
      "Mean z1*z2: 0.0119\n",
      "Intercept: -0.0097 (0.0052)\n",
      "\n",
      "Genetic Correlation\n",
      "-------------------\n",
      "Genetic Correlation: 0.2015 (0.0851)\n",
      "Z-score: 2.3671\n",
      "P: 0.0179\n",
      "\n",
      "\n",
      "Summary of Genetic Correlation Results\n",
      "                  p1                              p2      rg      se       z           p  h2_obs  h2_obs_se  h2_int  h2_int_se  gcov_int  gcov_int_se\n",
      " MG.Chia.sumstats.gz           AD.Jansen.sumstats.gz  0.0091  0.1230  0.0741  9.4092e-01  0.0075     0.0049  1.0570     0.0592    0.0050       0.0047\n",
      " MG.Chia.sumstats.gz            PD.Nalls.sumstats.gz  0.0032  0.0883  0.0359  9.7138e-01  0.0166     0.0017  0.9810     0.0076   -0.0029       0.0053\n",
      " MG.Chia.sumstats.gz      ALS.VanRheenen.sumstats.gz  0.2858  0.1800  1.5881  1.1227e-01  0.0485     0.0131  0.9941     0.0062    0.0122       0.0047\n",
      " MG.Chia.sumstats.gz            LBD.Chia.sumstats.gz  0.3657  0.2797  1.3078  1.9095e-01  0.1293     0.0835  1.0064     0.0087   -0.0117       0.0047\n",
      " MG.Chia.sumstats.gz      Crohns.deLange.sumstats.gz  0.0693  0.0883  0.7844  4.3279e-01  0.4422     0.0524  1.0975     0.0153    0.0014       0.0063\n",
      " MG.Chia.sumstats.gz          MS.Beecham.sumstats.gz  0.1121  0.0825  1.3585  1.7430e-01  1.5156     0.2219  1.0313     0.0998   -0.0232       0.0267\n",
      " MG.Chia.sumstats.gz            RA.Okada.sumstats.gz  0.4135  0.0929  4.4531  8.4652e-06  0.1696     0.0229  0.9484     0.0117    0.0003       0.0058\n",
      " MG.Chia.sumstats.gz           SLE.Julia.sumstats.gz -0.2126  0.1185 -1.7942  7.2775e-02  0.4178     0.0645  1.1006     0.0121   -0.0138       0.0063\n",
      " MG.Chia.sumstats.gz  T1D.OnengutGumuscu.sumstats.gz  0.4356  0.1066  4.0870  4.3707e-05  2.1786     0.5807  1.3416     0.2462   -0.0212       0.0430\n",
      " MG.Chia.sumstats.gz          UC.deLange.sumstats.gz  0.2015  0.0851  2.3671  1.7926e-02  0.2335     0.0256  1.1010     0.0138   -0.0097       0.0052\n",
      "\n",
      "Analysis finished at Tue Apr  6 13:00:48 2021\n",
      "Total time elapsed: 1.0m:10.63s\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/gpfs/gsfs11/users/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/LDSC/ldsc/ldscore/sumstats.py:532: FutureWarning: Method .as_matrix will be removed in a future version. Use .values instead.\n",
      "  ref_ld = sumstats.as_matrix(columns=ref_ld_cnames)\n",
      "/gpfs/gsfs11/users/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/LDSC/ldsc/ldscore/irwls.py:161: FutureWarning: `rcond` parameter will change to the default of machine precision times ``max(M, N)`` where M and N are the input matrix dimensions.\n",
      "To use the future default and silence this warning we advise to pass `rcond=None`, to keep using the old, explicitly pass `rcond=-1`.\n",
      "  coef = np.linalg.lstsq(x, y)\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "source /data/chiarp/conda/etc/profile.d/conda.sh\n",
    "\n",
    "python ./ldsc/ldsc.py \\\n",
    "--rg MG.Chia.sumstats.gz,AD.Jansen.sumstats.gz,PD.Nalls.sumstats.gz,ALS.VanRheenen.sumstats.gz,LBD.Chia.sumstats.gz,Crohns.deLange.sumstats.gz,MS.Beecham.sumstats.gz,RA.Okada.sumstats.gz,SLE.Julia.sumstats.gz,T1D.OnengutGumuscu.sumstats.gz,UC.deLange.sumstats.gz \\\n",
    "--ref-ld-chr eur_w_ld_chr/ \\\n",
    "--w-ld-chr eur_w_ld_chr/ \\\n",
    "--out MG_Neuro.AutoImmuneDiseases"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "integrated-salem",
   "metadata": {},
   "source": [
    "### Create pretty genetic correlation table\n",
    "\n",
    "Note:\n",
    "if need to plot correlation plot for larger dataset:\n",
    "`https://github.com/mkanai/ldsc-corrplot-rg.git`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "french-saturn",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "R version 3.5.2 (2018-12-20) -- \"Eggshell Igloo\"\n",
      "Copyright (C) 2018 The R Foundation for Statistical Computing\n",
      "Platform: x86_64-pc-linux-gnu (64-bit)\n",
      "\n",
      "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
      "You are welcome to redistribute it under certain conditions.\n",
      "Type 'license()' or 'licence()' for distribution details.\n",
      "\n",
      "R is a collaborative project with many contributors.\n",
      "Type 'contributors()' for more information and\n",
      "'citation()' on how to cite R or R packages in publications.\n",
      "\n",
      "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
      "'help.start()' for an HTML browser interface to help.\n",
      "Type 'q()' to quit R.\n",
      "\n",
      "> \n",
      "> require(data.table)\n",
      "> require(tidyverse)\n",
      "> library(stringr)\n",
      "> library(\"ggrepel\")\n",
      "> require(\"data.table\")\n",
      "> library(viridis)\n",
      "> library(gridExtra)\n",
      "> library(grid)\n",
      "> library(gt)\n",
      "> \n",
      "> data <- fread(\"GeneticCorrelation.MG.vs.Neuro.AutoImmune.txt\",header=T)\n",
      "> data$q <- p.adjust(data$p, method = \"fdr\", n = length(data$p))\n",
      "> data$rg <- format(round(data$rg,4),nsmall=4)\n",
      "> data$se <- format(round(data$se,4),nsmall=4)\n",
      "> data$z <- format(round(data$z,4),nsmall=4)\n",
      "> data1 <- data %>% select(p1_category,p1,p2_category,p2,rg,se,z,p,q) %>%\n",
      "+          rename(`Trait 1` = p1,\n",
      "+                 `Trait 2` = p2,\n",
      "+                 `Trait 1 category` = p1_category,\n",
      "+                 `Trait 2 category` = p2_category,\n",
      "+                 `Genetic correlation (rg)` = rg,\n",
      "+                  SE = se,\n",
      "+                  Z = z,\n",
      "+                  P = p,\n",
      "+                  `P(FDR adjusted)` = q)\n",
      "> write.table(data1,\"GeneticCorrelation.MG.vs.Neuro.AutoImmune.forPlotting.txt\",quote=F,sep=\"\\t\",row.names=F,col.names=T)\n",
      "> \n",
      "> plot <-   data1 %>%\n",
      "+           select(`Trait 1`,`Trait 2`,`Trait 2 category`,`Genetic correlation (rg)`,SE,Z,P,`P(FDR adjusted)`) %>%\n",
      "+           gt() %>%\n",
      "+           tab_header(title = md(\"**Genetic correlation: MG vs neurological and autoimmune diseases**\"),\n",
      "+                      subtitle = NULL) %>%\n",
      "+           fmt_scientific(columns = vars(P,`P(FDR adjusted)`),decimals = 2) %>%\n",
      "+           cols_align(align = \"center\", columns = vars(`Trait 2 category`,`Genetic correlation (rg)`,SE,Z,P,`P(FDR adjusted)`)) %>%\n",
      "+           cols_width(vars(`Trait 1`) ~ px(140),\n",
      "+                      vars(`Trait 2 category`) ~ px(160),\n",
      "+                      vars(`Genetic correlation (rg)`) ~ px(80),\n",
      "+                      vars(SE) ~ px(80),\n",
      "+                      vars(Z) ~ px(80),\n",
      "+                      vars(P) ~ px(110),\n",
      "+                      vars(`P(FDR adjusted)`) ~ px(110)) %>%\n",
      "+           tab_options(column_labels.background.color = \"#EFFBFC\") %>%\n",
      "+           tab_style(style = list(cell_text(weight = \"bold\")),\n",
      "+                     locations = cells_column_labels(columns = everything())) %>%\n",
      "+           gtsave(\"TableXX.GeneticCorrelation.rtf\")\n",
      "> plot\n",
      "NULL\n",
      "> \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[+] Loading gcc  7.3.0  ... \n",
      "[+] Loading GSL 2.4 for GCC 7.2.0 ... \n",
      "[-] Unloading gcc  7.3.0  ... \n",
      "[+] Loading gcc  7.3.0  ... \n",
      "[+] Loading openmpi 3.0.2  for GCC 7.3.0 \n",
      "[+] Loading ImageMagick  7.0.8  on cn1809 \n",
      "[+] Loading HDF5  1.10.4 \n",
      "[+] Loading pandoc  2.13  on cn1809 \n",
      "[+] Loading R 3.5.2 \n",
      "Loading required package: data.table\n",
      "Loading required package: tidyverse\n",
      "-- Attaching packages --------------------------------------- tidyverse 1.2.1 --\n",
      "v ggplot2 3.3.2     v purrr   0.3.4\n",
      "v tibble  3.0.3     v dplyr   0.8.5\n",
      "v tidyr   0.8.3     v stringr 1.4.0\n",
      "v readr   1.3.1     v forcats 0.5.0\n",
      "-- Conflicts ------------------------------------------ tidyverse_conflicts() --\n",
      "x dplyr::between()   masks data.table::between()\n",
      "x dplyr::filter()    masks stats::filter()\n",
      "x dplyr::first()     masks data.table::first()\n",
      "x dplyr::lag()       masks stats::lag()\n",
      "x dplyr::last()      masks data.table::last()\n",
      "x purrr::transpose() masks data.table::transpose()\n",
      "Loading required package: viridisLite\n",
      "\n",
      "Attaching package: 'gridExtra'\n",
      "\n",
      "The following object is masked from 'package:dplyr':\n",
      "\n",
      "    combine\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "## prep file for plotting\n",
    "\n",
    "module load R/3.5.2\n",
    "R --vanilla --no-save\n",
    "\n",
    "require(data.table)\n",
    "require(tidyverse)\n",
    "library(stringr)\n",
    "library(\"ggrepel\")\n",
    "require(\"data.table\")\n",
    "library(viridis)\n",
    "library(gridExtra)\n",
    "library(grid)\n",
    "library(gt)\n",
    "\n",
    "data <- fread(\"GeneticCorrelation.MG.vs.Neuro.AutoImmune.txt\",header=T)\n",
    "data$q <- p.adjust(data$p, method = \"fdr\", n = length(data$p))\n",
    "data$rg <- format(round(data$rg,4),nsmall=4)\n",
    "data$se <- format(round(data$se,4),nsmall=4)\n",
    "data$z <- format(round(data$z,4),nsmall=4)\n",
    "data1 <- data %>% select(p1_category,p1,p2_category,p2,rg,se,z,p,q) %>%\n",
    "         rename(`Trait 1` = p1,\n",
    "                `Trait 2` = p2,\n",
    "                `Trait 1 category` = p1_category,\n",
    "                `Trait 2 category` = p2_category,\n",
    "                `Genetic correlation (rg)` = rg,\n",
    "                 SE = se,\n",
    "                 Z = z,\n",
    "                 P = p,\n",
    "                 `P(FDR adjusted)` = q)\n",
    "write.table(data1,\"GeneticCorrelation.MG.vs.Neuro.AutoImmune.forPlotting.txt\",quote=F,sep=\"\\t\",row.names=F,col.names=T)\n",
    "\n",
    "plot <-   data1 %>%\n",
    "          select(`Trait 1`,`Trait 2`,`Trait 2 category`,`Genetic correlation (rg)`,SE,Z,P,`P(FDR adjusted)`) %>%\n",
    "          gt() %>%\n",
    "          tab_header(title = md(\"**Genetic correlation: MG vs neurological and autoimmune diseases**\"),\n",
    "                     subtitle = NULL) %>%\n",
    "          fmt_scientific(columns = vars(P,`P(FDR adjusted)`),decimals = 2) %>%\n",
    "          cols_align(align = \"center\", columns = vars(`Trait 2 category`,`Genetic correlation (rg)`,SE,Z,P,`P(FDR adjusted)`)) %>%\n",
    "          cols_width(vars(`Trait 1`) ~ px(140),\n",
    "                     vars(`Trait 2 category`) ~ px(160),\n",
    "                     vars(`Genetic correlation (rg)`) ~ px(80),\n",
    "                     vars(SE) ~ px(80),\n",
    "                     vars(Z) ~ px(80),\n",
    "                     vars(P) ~ px(110),\n",
    "                     vars(`P(FDR adjusted)`) ~ px(110)) %>%\n",
    "          tab_options(column_labels.background.color = \"#EFFBFC\") %>%\n",
    "          tab_style(style = list(cell_text(weight = \"bold\")),\n",
    "                    locations = cells_column_labels(columns = everything())) %>%\n",
    "          gtsave(\"TableXX.GeneticCorrelation.rtf\")\n",
    "plot\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "python/3.5",
   "language": "python",
   "name": "py3.5"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.6"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": false,
   "sideBar": true,
   "skip_h1_title": true,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": true,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "303.513px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
