# Genetic correlation using LDSC between MG and other dementia-related disorders

**Start date:** 03-19-2021

**Updated date:** 03-19-2021

**Analysed by:** Ruth Chia

**Working directory:** `/data/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/LDSC`
___


<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Rationale/background" data-toc-modified-id="Rationale/background-1">Rationale/background</a></span></li><li><span><a href="#What-to-do-here" data-toc-modified-id="What-to-do-here-2">What to do here</a></span></li><li><span><a href="#Files-needed-here" data-toc-modified-id="Files-needed-here-3">Files needed here</a></span></li><li><span><a href="#Clone-LDSC-github-repository-to-here" data-toc-modified-id="Clone-LDSC-github-repository-to-here-4">Clone LDSC github repository to here</a></span></li><li><span><a href="#Liftover-hg19-summ-stats-to-hg38" data-toc-modified-id="Liftover-hg19-summ-stats-to-hg38-5">Liftover hg19 summ stats to hg38</a></span><ul class="toc-item"><li><span><a href="#Run-liftover" data-toc-modified-id="Run-liftover-5.1">Run liftover</a></span></li><li><span><a href="#Update-summ-stats-to-hg38" data-toc-modified-id="Update-summ-stats-to-hg38-5.2">Update summ stats to hg38</a></span></li></ul></li><li><span><a href="#Merge-summary-stats-with-hg38-rsid-file-to-get-rsids" data-toc-modified-id="Merge-summary-stats-with-hg38-rsid-file-to-get-rsids-6">Merge summary stats with hg38 rsid file to get rsids</a></span></li><li><span><a href="#Reformat-summ-stats" data-toc-modified-id="Reformat-summ-stats-7">Reformat summ stats</a></span></li><li><span><a href="#Compute-heritability-and-the-LD-Score-regression-intercept-for-MG" data-toc-modified-id="Compute-heritability-and-the-LD-Score-regression-intercept-for-MG-8">Compute heritability and the LD Score regression intercept for MG</a></span></li><li><span><a href="#Run-LDSC-and-genetic-correlation-on-MG-vs.-neurological-diseases" data-toc-modified-id="Run-LDSC-and-genetic-correlation-on-MG-vs.-neurological-diseases-9">Run LDSC and genetic correlation on MG vs. neurological diseases</a></span><ul class="toc-item"><li><span><a href="#Estimating-Heritability,-Genetic-Correlation-and-the-LD-Score-Regression-Intercept" data-toc-modified-id="Estimating-Heritability,-Genetic-Correlation-and-the-LD-Score-Regression-Intercept-9.1">Estimating Heritability, Genetic Correlation and the LD Score Regression Intercept</a></span></li></ul></li><li><span><a href="#Run-LDSC-and-genetic-correlation-on-autoimmune-diseases" data-toc-modified-id="Run-LDSC-and-genetic-correlation-on-autoimmune-diseases-10">Run LDSC and genetic correlation on autoimmune diseases</a></span><ul class="toc-item"><li><span><a href="#Reformat-autoimmune-summ-stats" data-toc-modified-id="Reformat-autoimmune-summ-stats-10.1">Reformat autoimmune summ stats</a></span></li><li><span><a href="#Estimating-Heritability,-Genetic-Correlation-with-autoimmune-diseases" data-toc-modified-id="Estimating-Heritability,-Genetic-Correlation-with-autoimmune-diseases-10.2">Estimating Heritability, Genetic Correlation with autoimmune diseases</a></span></li><li><span><a href="#Estimating-Heritability,-Genetic-Correlation-with-neurodisease-and-autoimmune-diseases" data-toc-modified-id="Estimating-Heritability,-Genetic-Correlation-with-neurodisease-and-autoimmune-diseases-10.3">Estimating Heritability, Genetic Correlation with neurodisease and autoimmune diseases</a></span></li><li><span><a href="#Create-pretty-genetic-correlation-table" data-toc-modified-id="Create-pretty-genetic-correlation-table-10.4">Create pretty genetic correlation table</a></span></li></ul></li></ul></div>

## Rationale/background
Run LDSC to determine the genetic correlation between MG and:
1. Neurological diseases - AD,PD,LBD and ALS/FTD (disorders with a component of dementia as a disease phenotype)
2. Autoimmune diseases
Question to answer: ***Is there a correlation?***

The reason for doing this is because there is some evidence that there is an association between MG and cognitive function:
1. [Association between myasthenia gravis and cognitive function: A systematic review and meta-analysis](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4445185/)

2. [Assessment of cognitive function in patients with myasthenia gravis](https://nnjournal.net/article/view/90)

3. [Cognitive performance in patients with Myasthenia Gravis: an association with glucocorticosteroid use and depression](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7500821/)

4. [Cognition in myasthenia](https://n.neurology.org/content/neurology/39/7/1002.3.full.pdf)


## What to do here
1. Format summ stats file 
2. Run LDSC (Wiki how-to to run [Heritability and Genetic Correlation](https://github.com/bulik/ldsc/wiki/Heritability-and-Genetic-Correlation))
3. Plot


## Files needed here
Will need summary stats file for AD (Jansen et al), PD (Nalls et al), LBD (Chia et al) and ALS/FTD (Nicolas et al)

1. AD: `/data/ALS_50k/DementiaSeq_LBD/GWAS_LBD.v2/FIGURES_PUBLICATION/AdditionalInfo/LBDoverlap.AD.PD.GWAS/AD_sumstats_Jansenetal.txt`. Note that this based on hg19 and has rsid. Will need to update to hg38..


2. PD: `/data/ALS_50k/DementiaSeq_LBD/GWAS_LBD.v2/FIGURES_PUBLICATION/AdditionalInfo/LBDoverlap.AD.PD.GWAS/nallsEtAl2019_excluding23andMe_allVariants.tab`. Note that this based on hg19 and not does not have rsid. Will need to update to hg38 and include rsid.


3. LBD: `/data/ALS_50k/DementiaSeq_LBD/GWAS_LBD.v2/LBDpath.LBDclin.ALLcontrols4K.unrelated.glm.noSpanningDels.snvCOVS.hwe1e-6.maf001cases.excludeCentromereFlank10kb.exclVDJ.GnomadTopmedOutliersRemoved.withValidation.FINAL.TIDY.rsid.txt`. This is based on hg38 and has rsid.


4. ALS/FTD : as originally located `/Volumes/labchip/ALS_Project_3rd_June_2007/CURRENT_PROJECTS/mega_GWAS/summaryStats/fromSaraB_4sept2019/MODIFIED.CLEAN.MetaAmericans4PCs10BIT3PCsBFr3PCsUK2PCswheaderRsq0.3B5NoMAFwoVRSE2.TBL.tab`. Made a copy to this working directory on biowulf `/data/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/LDSC/MODIFIED.CLEAN.MetaAmericans4PCs10BIT3PCsBFr3PCsUK2PCswheaderRsq0.3B5NoMAFwoVRSE2.TBL.tab`. Note that this summ stats is in hg19 and has no rsid. Will need to update to hg38 and include rsid.


5. Summary stats from GWAS catalog for autoimmune diseases.


## Clone LDSC github repository to here


```python
!git clone https://github.com/bulik/ldsc.git
```


```python
!wget https://data.broadinstitute.org/alkesgroup/LDSCORE/eur_w_ld_chr.tar.bz2
!tar -jxvf eur_w_ld_chr.tar.bz2
```


```python
!wget https://data.broadinstitute.org/alkesgroup/LDSCORE/w_hm3.snplist.bz2
!bunzip2 w_hm3.snplist.bz2
```

    --2021-03-19 14:48:57--  https://data.broadinstitute.org/alkesgroup/LDSCORE/w_hm3.snplist.bz2
    Resolving dtn08-e0 (dtn08-e0)... 10.1.200.244
    Connecting to dtn08-e0 (dtn08-e0)|10.1.200.244|:3128... connected.
    Proxy request sent, awaiting response... 301 Moved Permanently
    Location: https://alkesgroup.broadinstitute.org/LDSCORE/w_hm3.snplist.bz2 [following]
    --2021-03-19 14:48:57--  https://alkesgroup.broadinstitute.org/LDSCORE/w_hm3.snplist.bz2
    Connecting to dtn08-e0 (dtn08-e0)|10.1.200.244|:3128... connected.
    Proxy request sent, awaiting response... 200 OK
    Length: 4320662 (4.1M) [application/octet-stream]
    Saving to: 'w_hm3.snplist.bz2'
    
    100%[======================================>] 4,320,662   --.-K/s   in 0.05s   
    
    2021-03-19 14:48:57 (88.4 MB/s) - 'w_hm3.snplist.bz2' saved [4320662/4320662]
    


## Liftover hg19 summ stats to hg38


```python
!cp /data/ALS_50k/DementiaSeq_LBD/GWAS_LBD.v2/FIGURES_PUBLICATION/AdditionalInfo/LBDoverlap.AD.PD.GWAS/AD_sumstats_Jansenetal.txt .
!cp /data/ALS_50k/DementiaSeq_LBD/GWAS_LBD.v2/FIGURES_PUBLICATION/AdditionalInfo/LBDoverlap.AD.PD.GWAS/nallsEtAl2019_excluding23andMe_allVariants.tab .
!cp /data/ALS_50k/DementiaSeq_LBD/GWAS_LBD.v2/LBDpath.LBDclin.ALLcontrols4K.unrelated.glm.noSpanningDels.snvCOVS.hwe1e-6.maf001cases.excludeCentromereFlank10kb.exclVDJ.GnomadTopmedOutliersRemoved.withValidation.FINAL.TIDY.rsid.txt .
```


```bash
%%bash
module load R/3.5.2
R --vanilla --no-save

require(data.table)
require(tidyverse)

# create function to run prep file for liftover hg19 to hg38

ForLiftover.hg19.to.hg38 <- function(data,outName) {
    data <- data %>% rename(CHROM_hg19 = CHROM, POS_hg19 = POS) %>%
            mutate(chrom = paste("chr",CHROM_hg19,sep=""))
    bed <- data[,c("chrom","POS_hg19","MarkerName")]
    bed$`POS_hg19_1` <- as.numeric(bed$POS_hg19) + 1
    bed <- bed[,c(1,2,4,3)]
    options(scipen=999, justified="none")
    write.table(bed, paste(outName,".hg19.ForLiftOver.bed",sep=""), col.names=F,row.names=F,quote=F,sep="\t")
}

# Read in files and rename columne names so that there is a CHROM and POS column

## AD
AD <- fread("AD_sumstats_Jansenetal.txt",header=F)
colnames(AD) <- c("MarkerName","CHROM","POS","A1","A2","SNP","Z","P","Nsum","Neff","dir","MAF","BETA","SE")

## PD
PD <- fread("/data/ALS_50k/DementiaSeq_LBD/GWAS_LBD.v2/FIGURES_PUBLICATION/AdditionalInfo/LBDoverlap.AD.PD.GWAS/nallsEtAl2019_excluding23andMe_allVariants.tab",header=T)
PD <- PD %>%
      mutate(snp = SNP) %>%
      separate(snp, c("CHROM","POS"),sep=":") %>%
      rename(MarkerName = SNP)
PD$CHROM <- gsub("chr","",PD$CHROM)

## ALS/FTD
ALS <- fread("MODIFIED.CLEAN.MetaAmericans4PCs10BIT3PCsBFr3PCsUK2PCswheaderRsq0.3B5NoMAFwoVRSE2.TBL.tab",header=T)
ALS <- ALS %>%
       mutate(snp = MarkerName) %>%
       separate(snp, c("CHROM","POS"),sep=":")

AD.1 <- ForLiftover.hg19.to.hg38(AD,"AD.Jansen")
PD.1 <- ForLiftover.hg19.to.hg38(PD,"PD.Nalls")
ALS.1 <- ForLiftover.hg19.to.hg38(ALS,"ALS.Nicolas")
```

    
    R version 3.5.2 (2018-12-20) -- "Eggshell Igloo"
    Copyright (C) 2018 The R Foundation for Statistical Computing
    Platform: x86_64-pc-linux-gnu (64-bit)
    
    R is free software and comes with ABSOLUTELY NO WARRANTY.
    You are welcome to redistribute it under certain conditions.
    Type 'license()' or 'licence()' for distribution details.
    
    R is a collaborative project with many contributors.
    Type 'contributors()' for more information and
    'citation()' on how to cite R or R packages in publications.
    
    Type 'demo()' for some demos, 'help()' for on-line help, or
    'help.start()' for an HTML browser interface to help.
    Type 'q()' to quit R.
    
    > 
    > require(data.table)
    > require(tidyverse)
    > 
    > # create function to run prep file for liftover hg19 to hg38
    > 
    > ForLiftover.hg19.to.hg38 <- function(data,outName) {
    +     data <- data %>% rename(CHROM_hg19 = CHROM, POS_hg19 = POS) %>%
    +             mutate(chrom = paste("chr",CHROM_hg19,sep=""))
    +     bed <- data[,c("chrom","POS_hg19","MarkerName")]
    +     bed$`POS_hg19_1` <- as.numeric(bed$POS_hg19) + 1
    +     bed <- bed[,c(1,2,4,3)]
    +     options(scipen=999, justified="none")
    +     write.table(bed, paste(outName,".hg19.ForLiftOver.bed",sep=""), col.names=F,row.names=F,quote=F,sep="\t")
    + }
    > 
    > # Read in files and rename columne names so that there is a CHROM and POS column
    > 
    > ## AD
    > AD <- fread("AD_sumstats_Jansenetal.txt",header=F)
    > colnames(AD) <- c("MarkerName","CHROM","POS","A1","A2","SNP","Z","P","Nsum","Neff","dir","MAF","BETA","SE")
    > 
    > ## PD
    > PD <- fread("/data/ALS_50k/DementiaSeq_LBD/GWAS_LBD.v2/FIGURES_PUBLICATION/AdditionalInfo/LBDoverlap.AD.PD.GWAS/nallsEtAl2019_excluding23andMe_allVariants.tab",header=T)
    > PD <- PD %>%
    +       mutate(snp = SNP) %>%
    +       separate(snp, c("CHROM","POS"),sep=":") %>%
    +       rename(MarkerName = SNP)
    > PD$CHROM <- gsub("chr","",PD$CHROM)
    > 
    > ## ALS/FTD
    > ALS <- fread("MODIFIED.CLEAN.MetaAmericans4PCs10BIT3PCsBFr3PCsUK2PCswheaderRsq0.3B5NoMAFwoVRSE2.TBL.tab",header=T)
    > ALS <- ALS %>%
    +        mutate(snp = MarkerName) %>%
    +        separate(snp, c("CHROM","POS"),sep=":")
    > 
    > AD.1 <- ForLiftover.hg19.to.hg38(AD,"AD.Jansen")
    > PD.1 <- ForLiftover.hg19.to.hg38(PD,"PD.Nalls")
    > ALS.1 <- ForLiftover.hg19.to.hg38(ALS,"ALS.Nicolas")
    > 


    [+] Loading gcc  7.3.0  ... 
    [+] Loading GSL 2.4 for GCC 7.2.0 ... 
    [-] Unloading gcc  7.3.0  ... 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading openmpi 3.0.2  for GCC 7.3.0 
    [+] Loading ImageMagick  7.0.8  on cn2074 
    [+] Loading HDF5  1.10.4 
    [+] Loading pandoc  2.11.4  on cn2074 
    [+] Loading R 3.5.2 
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()


### Run liftover


```bash
%%bash
mkdir LiftOver
wget http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/liftOver -O LiftOver/liftOver
wget --timestamping 'ftp://hgdownload.soe.ucsc.edu/goldenPath/hg19/liftOver/hg19ToHg38.over.chain.gz' -O LiftOver/hg19ToHg38.over.chain.gz
chmod u+x LiftOver/liftOver

for cohort in {AD.Jansen,PD.Nalls,ALS.Nicolas}
do
LiftOver/liftOver ${cohort}.hg19.ForLiftOver.bed LiftOver/hg19ToHg38.over.chain.gz ${cohort}.hg38.ForLiftOver.bed unlifted.${cohort}.bed
grep -v "#Deleted" unlifted.${cohort}.bed | awk '{print $4}' > unlifted.${cohort}_not_in_b38.txt
awk '{print $4,$2}' ${cohort}.hg38.ForLiftOver.bed > ${cohort}.hg19.ForLiftOver.bed_update-map_to_b38.txt
done
```

    Reading liftover chains
    Mapping coordinates
    Reading liftover chains
    Mapping coordinates
    Reading liftover chains
    Mapping coordinates



```bash
%%bash
for cohort in {AD.Jansen,PD.Nalls,ALS.Nicolas}
do
wc -l unlifted.${cohort}.bed
wc -l ${cohort}.hg19.ForLiftOver.bed_update-map_to_b38.txt
done
```

    7036 unlifted.AD.Jansen.bed
    13363781 AD.Jansen.hg19.ForLiftOver.bed_update-map_to_b38.txt
    2910 unlifted.PD.Nalls.bed
    17509162 PD.Nalls.hg19.ForLiftOver.bed_update-map_to_b38.txt
    5744 unlifted.ALS.Nicolas.bed
    10479018 ALS.Nicolas.hg19.ForLiftOver.bed_update-map_to_b38.txt


### Update summ stats to hg38


```bash
%%bash
module load R/3.5.2
R --vanilla --no-save

require(data.table)
require(tidyverse)

# AD
data <- fread("AD_sumstats_Jansenetal.txt",header=F)
colnames(data) <- c("MarkerName","CHROM","POS","A1","A2","SNP","Z","P","Nsum","Neff","dir","MAF","BETA","SE")
map  <- fread("AD.Jansen.hg19.ForLiftOver.bed_update-map_to_b38.txt",header=F) %>%
        rename(MarkerName = V1, POS_hg38 = V2)

## Merge to update positions
data1 <- merge(data,map,by="MarkerName")
data2 <- data1 %>% 
         filter(MAF >= 0.01 & MAF <= 0.99) %>%
         select(CHROM,POS_hg38,MarkerName,SNP,A1,A2,MAF,Neff,BETA,SE,P) %>%
         rename(A1_Freq = MAF, N_total = Neff, rsid = SNP)
write.table(data2,"AD.Jansen.hg38.TIDY.txt",quote=F,sep="\t",row.names=F,col.names=T)



# PD
data <- fread("/data/ALS_50k/DementiaSeq_LBD/GWAS_LBD.v2/FIGURES_PUBLICATION/AdditionalInfo/LBDoverlap.AD.PD.GWAS/nallsEtAl2019_excluding23andMe_allVariants.tab",header=T)
data <- data %>%
        mutate(snp = SNP) %>%
        separate(snp, c("CHROM","POS"),sep=":") %>%
        mutate(CHROM = str_replace(CHROM, "chr",""))
map  <- fread("PD.Nalls.hg19.ForLiftOver.bed_update-map_to_b38.txt",header=F) %>%
        rename(MarkerName = V1, POS_hg38 = V2)

## Merge to update positions
data1 <- merge(data,map,by.x="SNP", by.y="MarkerName")
data2 <- data1 %>%
         filter(freq >= 0.01 & freq <= 0.99) %>%
         mutate(N_total = N_cases + N_controls) %>%
         rename(MarkerName = SNP, BETA = b, SE = se, P = p, A1_Freq = freq) %>%
         mutate(rsid = "NA") %>%
         select(CHROM,POS_hg38,MarkerName,rsid,A1,A2,A1_Freq,N_total,BETA,SE,P)
write.table(data2,"PD.Nalls.hg38.TIDY.txt",quote=F,sep="\t",row.names=F,col.names=T)


## ALS/FTD
data <- fread("MODIFIED.CLEAN.MetaAmericans4PCs10BIT3PCsBFr3PCsUK2PCswheaderRsq0.3B5NoMAFwoVRSE2.TBL.tab",header=T)
data <- data %>%
       mutate(snp = MarkerName) %>%
       separate(snp, c("CHROM","POS"),sep=":")
map  <- fread("ALS.Nicolas.hg19.ForLiftOver.bed_update-map_to_b38.txt",header=F) %>%
        rename(MarkerName = V1, POS_hg38 = V2)

## Merge to update positions
data1 <- merge(data,map,by.x="MarkerName", by.y="MarkerName")
data2 <- data1[!grepl("\\?",data1$Direction),]
data2$N_cases <- 20806
data2$N_controls <- 59804
data2$N_total <- data2$N_cases + data2$N_controls
data3 <- data2 %>%
         filter(effectAlleleFreq >= 0.01 & effectAlleleFreq <= 0.99) %>%    
         rename(A1 = Allele1, A2 = Allele2, BETA = Effect, SE = StdErr, P = `P-value`, A1_Freq = effectAlleleFreq) %>%
         mutate(rsid = "NA") %>%
         select(CHROM,POS_hg38,MarkerName,rsid,A1,A2,A1_Freq,N_total,BETA,SE,P)
write.table(data3,"ALS.Nicolas.hg38.TIDY.txt",quote=F,sep="\t",row.names=F,col.names=T)
         
```

    
    R version 3.5.2 (2018-12-20) -- "Eggshell Igloo"
    Copyright (C) 2018 The R Foundation for Statistical Computing
    Platform: x86_64-pc-linux-gnu (64-bit)
    
    R is free software and comes with ABSOLUTELY NO WARRANTY.
    You are welcome to redistribute it under certain conditions.
    Type 'license()' or 'licence()' for distribution details.
    
    R is a collaborative project with many contributors.
    Type 'contributors()' for more information and
    'citation()' on how to cite R or R packages in publications.
    
    Type 'demo()' for some demos, 'help()' for on-line help, or
    'help.start()' for an HTML browser interface to help.
    Type 'q()' to quit R.
    
    > 
    > require(data.table)
    > require(tidyverse)
    > 
    > # AD
    > data <- fread("AD_sumstats_Jansenetal.txt",header=F)
    > colnames(data) <- c("MarkerName","CHROM","POS","A1","A2","SNP","Z","P","Nsum","Neff","dir","MAF","BETA","SE")
    > map  <- fread("AD.Jansen.hg19.ForLiftOver.bed_update-map_to_b38.txt",header=F) %>%
    +         rename(MarkerName = V1, POS_hg38 = V2)
    > 
    > ## Merge to update positions
    > data1 <- merge(data,map,by="MarkerName")
    > data2 <- data1 %>% 
    +          filter(MAF >= 0.01 & MAF <= 0.99) %>%
    +          select(CHROM,POS_hg38,MarkerName,SNP,A1,A2,MAF,Neff,BETA,SE,P) %>%
    +          rename(A1_Freq = MAF, N_total = Neff, rsid = SNP)
    > write.table(data2,"AD.Jansen.hg38.TIDY.txt",quote=F,sep="\t",row.names=F,col.names=T)
    > 
    > 
    > 
    > # PD
    > data <- fread("/data/ALS_50k/DementiaSeq_LBD/GWAS_LBD.v2/FIGURES_PUBLICATION/AdditionalInfo/LBDoverlap.AD.PD.GWAS/nallsEtAl2019_excluding23andMe_allVariants.tab",header=T)
    > data <- data %>%
    +         mutate(snp = SNP) %>%
    +         separate(snp, c("CHROM","POS"),sep=":") %>%
    +         mutate(CHROM = str_replace(CHROM, "chr",""))
    > map  <- fread("PD.Nalls.hg19.ForLiftOver.bed_update-map_to_b38.txt",header=F) %>%
    +         rename(MarkerName = V1, POS_hg38 = V2)
    > 
    > ## Merge to update positions
    > data1 <- merge(data,map,by.x="SNP", by.y="MarkerName")
    > data2 <- data1 %>%
    +          filter(freq >= 0.01 & freq <= 0.99) %>%
    +          mutate(N_total = N_cases + N_controls) %>%
    +          rename(MarkerName = SNP, BETA = b, SE = se, P = p, A1_Freq = freq) %>%
    +          mutate(rsid = "NA") %>%
    +          select(CHROM,POS_hg38,MarkerName,rsid,A1,A2,A1_Freq,N_total,BETA,SE,P)
    > write.table(data2,"PD.Nalls.hg38.TIDY.txt",quote=F,sep="\t",row.names=F,col.names=T)
    > 
    > 
    > ## ALS/FTD
    > data <- fread("MODIFIED.CLEAN.MetaAmericans4PCs10BIT3PCsBFr3PCsUK2PCswheaderRsq0.3B5NoMAFwoVRSE2.TBL.tab",header=T)
    > data <- data %>%
    +        mutate(snp = MarkerName) %>%
    +        separate(snp, c("CHROM","POS"),sep=":")
    > map  <- fread("ALS.Nicolas.hg19.ForLiftOver.bed_update-map_to_b38.txt",header=F) %>%
    +         rename(MarkerName = V1, POS_hg38 = V2)
    > 
    > ## Merge to update positions
    > data1 <- merge(data,map,by.x="MarkerName", by.y="MarkerName")
    > data2 <- data1[!grepl("\\?",data1$Direction),]
    > data2$N_cases <- 20806
    > data2$N_controls <- 59804
    > data2$N_total <- data2$N_cases + data2$N_controls
    > data3 <- data2 %>%
    +          filter(effectAlleleFreq >= 0.01 & effectAlleleFreq <= 0.99) %>%    
    +          rename(A1 = Allele1, A2 = Allele2, BETA = Effect, SE = StdErr, P = `P-value`, A1_Freq = effectAlleleFreq) %>%
    +          mutate(rsid = "NA") %>%
    +          select(CHROM,POS_hg38,MarkerName,rsid,A1,A2,A1_Freq,N_total,BETA,SE,P)
    > write.table(data3,"ALS.Nicolas.hg38.TIDY.txt",quote=F,sep="\t",row.names=F,col.names=T)
    >          
    > 


    [+] Loading gcc  7.3.0  ... 
    [+] Loading GSL 2.4 for GCC 7.2.0 ... 
    [-] Unloading gcc  7.3.0  ... 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading openmpi 3.0.2  for GCC 7.3.0 
    [+] Loading ImageMagick  7.0.8  on cn1820 
    [+] Loading HDF5  1.10.4 
    [+] Loading pandoc  2.11.4  on cn1820 
    [+] Loading R 3.5.2 
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()



```bash
%%bash
# Prep ALS GWAS from Van Rheenen - already has rsid. No need to liftover to hg38
# file: `/data/NDRS_LNG/ALS_pathway_plink_files_UPDATED.March2020/Analysis.GLM.hg19/MetaAnalysis.VanRheenen/VanRheenen.meta_freqsfixed.tab`

module load R/3.5.2
R --vanilla --no-save

require(data.table)
require(tidyverse)

data <- fread("/data/NDRS_LNG/ALS_pathway_plink_files_UPDATED.March2020/Analysis.GLM.hg19/MetaAnalysis.VanRheenen/VanRheenen.meta_freqsfixed.tab",header=T)
data1 <- data %>%
        rename(SNP = snp, A1 = a1, A2 = a2, BETA = b, P = p, MAF = effectAlleleFreq) %>%
        mutate(N = cases + controls) %>%
        filter(MAF >= 0.01 & MAF <= 0.99) %>%
        select(SNP,A1,A2,N,P,BETA)
write.table(data1, "ALS.VanRheenen.hg38.rsid.forLDSC.txt",quote=F,sep="\t",row.names=F,col.names=T)

```

    
    R version 3.5.2 (2018-12-20) -- "Eggshell Igloo"
    Copyright (C) 2018 The R Foundation for Statistical Computing
    Platform: x86_64-pc-linux-gnu (64-bit)
    
    R is free software and comes with ABSOLUTELY NO WARRANTY.
    You are welcome to redistribute it under certain conditions.
    Type 'license()' or 'licence()' for distribution details.
    
    R is a collaborative project with many contributors.
    Type 'contributors()' for more information and
    'citation()' on how to cite R or R packages in publications.
    
    Type 'demo()' for some demos, 'help()' for on-line help, or
    'help.start()' for an HTML browser interface to help.
    Type 'q()' to quit R.
    
    > 
    > require(data.table)
    > require(tidyverse)
    > 
    > data <- fread("/data/NDRS_LNG/ALS_pathway_plink_files_UPDATED.March2020/Analysis.GLM.hg19/MetaAnalysis.VanRheenen/VanRheenen.meta_freqsfixed.tab",header=T)
    > data1 <- data %>%
    +         rename(SNP = snp, A1 = a1, A2 = a2, BETA = b, P = p, MAF = effectAlleleFreq) %>%
    +         mutate(N = cases + controls) %>%
    +         filter(MAF >= 0.01 & MAF <= 0.99) %>%
    +         select(SNP,A1,A2,N,P,BETA)
    > write.table(data1, "ALS.VanRheenen.hg38.rsid.forLDSC.txt",quote=F,sep="\t",row.names=F,col.names=T)
    > 


    [+] Loading gcc  7.3.0  ... 
    [+] Loading GSL 2.4 for GCC 7.2.0 ... 
    [-] Unloading gcc  7.3.0  ... 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading openmpi 3.0.2  for GCC 7.3.0 
    [+] Loading ImageMagick  7.0.8  on cn1820 
    [+] Loading HDF5  1.10.4 
    [+] Loading pandoc  2.11.4  on cn1820 
    [+] Loading R 3.5.2 
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()


## Merge summary stats with hg38 rsid file to get rsids

**Notes:**

Location of rsid hg38 file: 
- master file: `/data/ALS_50k/DementiaSeq_LBD/GWAS_LBD.v2/scripts/hg38_rsID_AUG2019_goodFormat.ALLFINAL.txt`
- chunked per chr: `/data/ALS_50k/DementiaSeq_LBD/GWAS_LBD.v2/scripts/rsid_perChr/rsid.chr*.txt`

Script to merge by chromosome: `getRsid.summStats.R`

```bash
#!/usr/bin/env Rscript

# read command line
args <- commandArgs(trailingOnly=TRUE)
if (length(args) != 3) {
stop("USAGE: Rscript getRsid.summStats.R inputSummStatsFile CHNUM OutputFileName")
}

# Load required libraries
require(data.table)
require(tidyverse)

# Read in files
data <- fread(args[1], header=T)
data1 <- subset(data, data$CHROM == args[2])
data1$A1 <- toupper(data1$A1)
data1$A2 <- toupper(data1$A2)

input <- paste("/data/ALS_50k/DementiaSeq_LBD/GWAS_LBD.v2/scripts/rsid_perChr/rsid.chr",args[2],".txt",sep="")
rsid <- fread(input, header = F)
colnames(rsid) <- c("markerID","rsID","CHROM","POS_hg38","REF","ALT")

# Merge files - make sure to flip alleles to make sure to account for any REF/ALT mismatch
merge1 <- merge(data1, rsid[,2:6], by.x=c("CHROM","POS_hg38","A2","A1"),by.y=c("CHROM","POS_hg38","REF","ALT"))

merge2 <- subset(data1, !(data1$MarkerName %in% merge1$MarkerName))
merge3 <- merge(merge2, rsid[,2:6], by.x=c("CHROM","POS_hg38","A2","A1"),by.y=c("CHROM","POS_hg38","ALT","REF"))

merge4 <- rbind(merge1,merge3)

merge5 <- subset(data1, !(data1$MarkerName %in% merge4$MarkerName)) %>% mutate(rsID = "NA")

mergeFINAL <- rbind(merge4,merge5) %>% arrange(CHROM,POS_hg38)

write.table(mergeFINAL, paste(args[3],".rsid.chr",args[2],".txt",sep=""), quote=F,row.names=F,col.names=T,sep="\t")
```


```bash
%%bash
module load R/3.5.2

# Run rsid script above
for cohort in {AD.Jansen,PD.Nalls,ALS.Nicolas}
do
    for CHNUM in {1..22}
    do
    Rscript getRsid.summStats.R \
    ${cohort}.hg38.TIDY.txt \
    ${CHNUM} \
    ${cohort}.hg38
    done
done
```

    [+] Loading gcc  7.3.0  ... 
    [+] Loading GSL 2.4 for GCC 7.2.0 ... 
    [-] Unloading gcc  7.3.0  ... 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading openmpi 3.0.2  for GCC 7.3.0 
    [+] Loading ImageMagick  7.0.8  on cn1820 
    [+] Loading HDF5  1.10.4 
    [+] Loading pandoc  2.11.4  on cn1820 
    [+] Loading R 3.5.2 
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()


Merge per chr summ stats file to single large file


```bash
%%bash
head -n 1 AD.Jansen.hg38.rsid.chr1.txt > AD.Jansen.hg38.rsid.txt
head -n 1 PD.Nalls.hg38.rsid.chr1.txt > PD.Nalls.hg38.rsid.txt
head -n 1 ALS.Nicolas.hg38.rsid.chr1.txt > ALS.Nicolas.hg38.rsid.txt

for cohort in {AD.Jansen,PD.Nalls,ALS.Nicolas}
do
    for CHNUM in {1..22}
    do
    tail -n +2 ${cohort}.hg38.rsid.chr${CHNUM}.txt >> ${cohort}.hg38.rsid.txt
    rm ${cohort}.hg38.rsid.chr${CHNUM}.txt
    done
done
```


```python
!wc -l *.hg38.rsid.txt
```

       9732198 AD.Jansen.hg38.rsid.txt
       8949247 ALS.Nicolas.hg38.rsid.txt
       8330285 PD.Nalls.hg38.rsid.txt
      27011730 total



```python
import pandas as pd
pd.read_csv("AD.Jansen.hg38.rsid.txt",sep="\t").head()
```




<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>CHROM</th>
      <th>POS_hg38</th>
      <th>A2</th>
      <th>A1</th>
      <th>MarkerName</th>
      <th>rsid</th>
      <th>A1_Freq</th>
      <th>N_total</th>
      <th>BETA</th>
      <th>SE</th>
      <th>P</th>
      <th>rsID</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>1</td>
      <td>779885</td>
      <td>C</td>
      <td>T</td>
      <td>1:715265_T_C</td>
      <td>rs12184267</td>
      <td>0.040807</td>
      <td>381761.0</td>
      <td>0.012275</td>
      <td>0.005785</td>
      <td>0.03384</td>
      <td>rs12184267</td>
    </tr>
    <tr>
      <th>1</th>
      <td>1</td>
      <td>779987</td>
      <td>A</td>
      <td>G</td>
      <td>1:715367_G_A</td>
      <td>rs12184277</td>
      <td>0.041069</td>
      <td>382151.0</td>
      <td>0.011285</td>
      <td>0.005764</td>
      <td>0.05024</td>
      <td>rs12184277</td>
    </tr>
    <tr>
      <th>2</th>
      <td>1</td>
      <td>782105</td>
      <td>C</td>
      <td>A</td>
      <td>1:717485_A_C</td>
      <td>rs12184279</td>
      <td>0.040576</td>
      <td>382180.0</td>
      <td>0.011087</td>
      <td>0.005797</td>
      <td>0.05582</td>
      <td>rs12184279</td>
    </tr>
    <tr>
      <th>3</th>
      <td>1</td>
      <td>785001</td>
      <td>G</td>
      <td>T</td>
      <td>1:720381_T_G</td>
      <td>rs116801199</td>
      <td>0.042162</td>
      <td>382954.0</td>
      <td>0.013052</td>
      <td>0.005686</td>
      <td>0.02171</td>
      <td>rs116801199</td>
    </tr>
    <tr>
      <th>4</th>
      <td>1</td>
      <td>785910</td>
      <td>G</td>
      <td>C</td>
      <td>1:721290_C_G</td>
      <td>rs12565286</td>
      <td>0.042378</td>
      <td>382779.0</td>
      <td>0.013137</td>
      <td>0.005673</td>
      <td>0.02058</td>
      <td>rs12565286</td>
    </tr>
  </tbody>
</table>
</div>




```python
import pandas as pd
pd.read_csv("PD.Nalls.hg38.rsid.txt",sep="\t").head()
```




<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>CHROM</th>
      <th>POS_hg38</th>
      <th>A2</th>
      <th>A1</th>
      <th>MarkerName</th>
      <th>rsid</th>
      <th>A1_Freq</th>
      <th>N_total</th>
      <th>BETA</th>
      <th>SE</th>
      <th>P</th>
      <th>rsID</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>1</td>
      <td>1200</td>
      <td>G</td>
      <td>A</td>
      <td>chr1:143118961</td>
      <td>NaN</td>
      <td>0.0330</td>
      <td>2137</td>
      <td>-0.1062</td>
      <td>0.1689</td>
      <td>0.5293</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1</th>
      <td>1</td>
      <td>1636</td>
      <td>G</td>
      <td>T</td>
      <td>chr1:143119397</td>
      <td>NaN</td>
      <td>0.9841</td>
      <td>2137</td>
      <td>0.1292</td>
      <td>0.2423</td>
      <td>0.5938</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>2</th>
      <td>1</td>
      <td>1964</td>
      <td>T</td>
      <td>A</td>
      <td>chr1:142537398</td>
      <td>NaN</td>
      <td>0.9857</td>
      <td>2137</td>
      <td>-0.0156</td>
      <td>0.2525</td>
      <td>0.9509</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>3</th>
      <td>1</td>
      <td>2976</td>
      <td>C</td>
      <td>T</td>
      <td>chr1:143120737</td>
      <td>NaN</td>
      <td>0.0112</td>
      <td>2137</td>
      <td>-0.1148</td>
      <td>0.2871</td>
      <td>0.6894</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>4</th>
      <td>1</td>
      <td>2979</td>
      <td>G</td>
      <td>A</td>
      <td>chr1:13124908</td>
      <td>NaN</td>
      <td>0.0165</td>
      <td>464905</td>
      <td>-0.0448</td>
      <td>0.1241</td>
      <td>0.7179</td>
      <td>NaN</td>
    </tr>
  </tbody>
</table>
</div>




```python
import pandas as pd
pd.read_csv("ALS.Nicolas.hg38.rsid.txt",sep="\t").head()
```




<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>CHROM</th>
      <th>POS_hg38</th>
      <th>A2</th>
      <th>A1</th>
      <th>MarkerName</th>
      <th>rsid</th>
      <th>A1_Freq</th>
      <th>N_total</th>
      <th>BETA</th>
      <th>SE</th>
      <th>P</th>
      <th>rsID</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>1</td>
      <td>832873</td>
      <td>C</td>
      <td>A</td>
      <td>1:768253</td>
      <td>NaN</td>
      <td>0.6548</td>
      <td>80610</td>
      <td>0.0248</td>
      <td>0.0398</td>
      <td>0.5336</td>
      <td>rs2977608</td>
    </tr>
    <tr>
      <th>1</th>
      <td>1</td>
      <td>839494</td>
      <td>C</td>
      <td>A</td>
      <td>1:774874</td>
      <td>NaN</td>
      <td>0.1935</td>
      <td>80610</td>
      <td>0.0335</td>
      <td>0.0478</td>
      <td>0.4830</td>
      <td>rs28810152</td>
    </tr>
    <tr>
      <th>2</th>
      <td>1</td>
      <td>845405</td>
      <td>T</td>
      <td>A</td>
      <td>1:780785</td>
      <td>NaN</td>
      <td>0.8150</td>
      <td>80610</td>
      <td>-0.0303</td>
      <td>0.0464</td>
      <td>0.5142</td>
      <td>rs2977612</td>
    </tr>
    <tr>
      <th>3</th>
      <td>1</td>
      <td>847601</td>
      <td>C</td>
      <td>T</td>
      <td>1:782981</td>
      <td>NaN</td>
      <td>0.1705</td>
      <td>80610</td>
      <td>0.0332</td>
      <td>0.0491</td>
      <td>0.4996</td>
      <td>rs6594026</td>
    </tr>
    <tr>
      <th>4</th>
      <td>1</td>
      <td>849670</td>
      <td>G</td>
      <td>A</td>
      <td>1:785050</td>
      <td>NaN</td>
      <td>0.8030</td>
      <td>80610</td>
      <td>-0.0254</td>
      <td>0.0422</td>
      <td>0.5474</td>
      <td>rs2905062</td>
    </tr>
  </tbody>
</table>
</div>



## Reformat summ stats

Need to format files because the ldsc .sumstats format requires six pieces of information for each SNP:

- A unique identifier (e.g., the rs number)
- Allele 1 (effect allele)
- Allele 2 (non-effect allele)
- Sample size (which often varies from SNP to SNP)
- A P-value
- A signed summary statistic (beta, OR, log odds, Z-score, etc)


```bash
%%bash
module load R/3.5.2
R --vanilla --no-save

require(data.table)
require(tidyverse)

# AD
AD <- fread("AD.Jansen.hg38.rsid.txt",header=T) %>%
      mutate(marker = paste(CHROM,POS_hg38,A2,A1,sep=":"))
AD$SNP <- ifelse(is.na(AD$rsID), AD$marker, AD$rsID)      
AD.1 <- AD %>% select(SNP,A1,A2,N_total,P,BETA) %>% rename(N = N_total)
write.table(AD.1,"AD.Jansen.hg38.rsid.forLDSC.txt",quote=F,sep="\t",row.names=F,col.names=T)

# PD
PD <- fread("PD.Nalls.hg38.rsid.txt",header=T) %>%
      mutate(marker = paste(CHROM,POS_hg38,A2,A1,sep=":"))
PD$SNP <- ifelse(is.na(PD$rsID), PD$marker, PD$rsID)      
PD.1 <- PD %>% select(SNP,A1,A2,N_total,P,BETA) %>% rename(N = N_total)
write.table(PD.1,"PD.Nalls.hg38.rsid.forLDSC.txt",quote=F,sep="\t",row.names=F,col.names=T)

# ALS
ALS <- fread("ALS.Nicolas.hg38.rsid.txt",header=T) %>%
       mutate(marker = paste(CHROM,POS_hg38,A2,A1,sep=":"))
ALS$SNP <- ifelse(is.na(ALS$rsID), ALS$marker, ALS$rsID)      
ALS.1 <- ALS %>% select(SNP,A1,A2,N_total,P,BETA) %>% rename(N = N_total)
write.table(ALS.1,"ALS.Nicolas.hg38.rsid.forLDSC.txt",quote=F,sep="\t",row.names=F,col.names=T)

## LBD
LBD <- fread("LBDpath.LBDclin.ALLcontrols4K.unrelated.glm.noSpanningDels.snvCOVS.hwe1e-6.maf001cases.excludeCentromereFlank10kb.exclVDJ.GnomadTopmedOutliersRemoved.withValidation.FINAL.TIDY.rsid.txt",header=T) %>%
       rename(POS_hg38 = POS, MarkerName = ID, rsid = rsID, A1_Freq = A1_FREQ, N_total = OBS_CT)
LBD <- LBD %>% mutate(marker = paste(CHROM,POS_hg38,A2,A1,sep=":"))
LBD$SNP <- ifelse(is.na(LBD$rsid), LBD$marker, LBD$rsid)  
LBD.1 <- LBD %>% filter(A1_FREQ => 0.01 & A1_FREQ <=0.99) %>% select(SNP,A1,A2,N_total,P,BETA) %>% rename(N = N_total)
write.table(LBD.1,"LBD.Chia.hg38.rsid.forLDSC.txt",quote=F,sep="\t",row.names=F,col.names=T)

# MG
MG <- fread("/data/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/USmerged.Itals.hg38/META_MG.USmerged.Itals.UNRELATED.hg38.Rsq03MAF00001.glm_filteredDirection.HetISq80.rsid.txt",header=T)
MG$N_total <- 36370 + 1873
MG$SNP <- ifelse(is.na(MG$rsID), MG$MarkerName, MG$rsID)  
MG.1 <- MG %>% 
        filter(maf_EA.CASE > 0.01) %>%
        select(SNP,EffectAllele,OtherAllele,N_total,P,beta) %>%
        rename(A1 = EffectAllele, A2 = OtherAllele, BETA = beta) %>% rename(N = N_total)
write.table(MG.1,"MG.Chia.hg38.rsid.forLDSC.txt",quote=F,sep="\t",row.names=F,col.names=T)
```

    
    R version 3.5.2 (2018-12-20) -- "Eggshell Igloo"
    Copyright (C) 2018 The R Foundation for Statistical Computing
    Platform: x86_64-pc-linux-gnu (64-bit)
    
    R is free software and comes with ABSOLUTELY NO WARRANTY.
    You are welcome to redistribute it under certain conditions.
    Type 'license()' or 'licence()' for distribution details.
    
    R is a collaborative project with many contributors.
    Type 'contributors()' for more information and
    'citation()' on how to cite R or R packages in publications.
    
    Type 'demo()' for some demos, 'help()' for on-line help, or
    'help.start()' for an HTML browser interface to help.
    Type 'q()' to quit R.
    
    > 
    > require(data.table)
    > require(tidyverse)
    > 
    > # AD
    > AD <- fread("AD.Jansen.hg38.rsid.txt",header=T) %>%
    +       mutate(marker = paste(CHROM,POS_hg38,A2,A1,sep=":"))
    > AD$SNP <- ifelse(is.na(AD$rsID), AD$marker, AD$rsID)      
    > AD.1 <- AD %>% select(SNP,A1,A2,N_total,P,BETA) %>% rename(N = N_total)
    > write.table(AD.1,"AD.Jansen.hg38.rsid.forLDSC.txt",quote=F,sep="\t",row.names=F,col.names=T)
    > 
    > # PD
    > PD <- fread("PD.Nalls.hg38.rsid.txt",header=T) %>%
    +       mutate(marker = paste(CHROM,POS_hg38,A2,A1,sep=":"))
    > PD$SNP <- ifelse(is.na(PD$rsID), PD$marker, PD$rsID)      
    > PD.1 <- PD %>% select(SNP,A1,A2,N_total,P,BETA) %>% rename(N = N_total)
    > write.table(PD.1,"PD.Nalls.hg38.rsid.forLDSC.txt",quote=F,sep="\t",row.names=F,col.names=T)
    > 
    > # ALS
    > ALS <- fread("ALS.Nicolas.hg38.rsid.txt",header=T) %>%
    +        mutate(marker = paste(CHROM,POS_hg38,A2,A1,sep=":"))
    > ALS$SNP <- ifelse(is.na(ALS$rsID), ALS$marker, ALS$rsID)      
    > ALS.1 <- ALS %>% select(SNP,A1,A2,N_total,P,BETA) %>% rename(N = N_total)
    > write.table(ALS.1,"ALS.Nicolas.hg38.rsid.forLDSC.txt",quote=F,sep="\t",row.names=F,col.names=T)
    > 
    > ## LBD
    > LBD <- fread("LBDpath.LBDclin.ALLcontrols4K.unrelated.glm.noSpanningDels.snvCOVS.hwe1e-6.maf001cases.excludeCentromereFlank10kb.exclVDJ.GnomadTopmedOutliersRemoved.withValidation.FINAL.TIDY.rsid.txt",header=T) %>%
    +        rename(POS_hg38 = POS, MarkerName = ID, rsid = rsID, A1_Freq = A1_FREQ, N_total = OBS_CT)
    > LBD <- LBD %>% mutate(marker = paste(CHROM,POS_hg38,A2,A1,sep=":"))
    > LBD$SNP <- ifelse(is.na(LBD$rsid), LBD$marker, LBD$rsid)  
    > LBD.1 <- LBD %>% select(SNP,A1,A2,N_total,P,BETA) %>% rename(N = N_total)
    > write.table(LBD.1,"LBD.Chia.hg38.rsid.forLDSC.txt",quote=F,sep="\t",row.names=F,col.names=T)
    > 
    > # MG
    > MG <- fread("/data/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/USmerged.Itals.hg38/META_MG.USmerged.Itals.UNRELATED.hg38.Rsq03MAF00001.glm_filteredDirection.HetISq80.rsid.txt",header=T)
    > MG$N_total <- 36370 + 1873
    > MG$SNP <- ifelse(is.na(MG$rsID), MG$MarkerName, MG$rsID)  
    > MG.1 <- MG %>% 
    +         select(SNP,EffectAllele,OtherAllele,N_total,P,beta) %>%
    +         rename(A1 = EffectAllele, A2 = OtherAllele, BETA = beta) %>% rename(N = N_total)
    > write.table(MG.1,"MG.Chia.hg38.rsid.forLDSC.txt",quote=F,sep="\t",row.names=F,col.names=T)
    > 


    [+] Loading gcc  7.3.0  ... 
    [+] Loading GSL 2.4 for GCC 7.2.0 ... 
    [-] Unloading gcc  7.3.0  ... 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading openmpi 3.0.2  for GCC 7.3.0 
    [+] Loading ImageMagick  7.0.8  on cn1820 
    [+] Loading HDF5  1.10.4 
    [+] Loading pandoc  2.11.4  on cn1820 
    [+] Loading R 3.5.2 
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()



```bash
%%bash
module load R/3.5.2
R --vanilla --no-save

require(data.table)
require(tidyverse)

# MG
MG <- fread("/data/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/USmerged.Itals.hg38/META_MG.USmerged.Itals.UNRELATED.hg38.Rsq03MAF00001.glm_filteredDirection.HetISq80.rsid.txt",header=T)
MG$N_total <- 36370 + 1873
MG$SNP <- ifelse(is.na(MG$rsID), MG$MarkerName, MG$rsID)  
MG.1 <- MG %>% 
        filter(maf_EA > 0.01) %>%
        select(SNP,EffectAllele,OtherAllele,N_total,P,beta) %>%
        rename(A1 = EffectAllele, A2 = OtherAllele, BETA = beta) %>% rename(N = N_total)
write.table(MG.1,"MG.Chia.hg38.rsid.forLDSC.txt",quote=F,sep="\t",row.names=F,col.names=T)
```

    
    R version 3.5.2 (2018-12-20) -- "Eggshell Igloo"
    Copyright (C) 2018 The R Foundation for Statistical Computing
    Platform: x86_64-pc-linux-gnu (64-bit)
    
    R is free software and comes with ABSOLUTELY NO WARRANTY.
    You are welcome to redistribute it under certain conditions.
    Type 'license()' or 'licence()' for distribution details.
    
    R is a collaborative project with many contributors.
    Type 'contributors()' for more information and
    'citation()' on how to cite R or R packages in publications.
    
    Type 'demo()' for some demos, 'help()' for on-line help, or
    'help.start()' for an HTML browser interface to help.
    Type 'q()' to quit R.
    
    > 
    > require(data.table)
    > require(tidyverse)
    > 
    > # MG
    > MG <- fread("/data/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/USmerged.Itals.hg38/META_MG.USmerged.Itals.UNRELATED.hg38.Rsq03MAF00001.glm_filteredDirection.HetISq80.rsid.txt",header=T)
    > MG$N_total <- 36370 + 1873
    > MG$SNP <- ifelse(is.na(MG$rsID), MG$MarkerName, MG$rsID)  
    > MG.1 <- MG %>% 
    +         filter(maf_EA > 0.01) %>%
    +         select(SNP,EffectAllele,OtherAllele,N_total,P,beta) %>%
    +         rename(A1 = EffectAllele, A2 = OtherAllele, BETA = beta) %>% rename(N = N_total)
    > write.table(MG.1,"MG.Chia.hg38.rsid.forLDSC.txt",quote=F,sep="\t",row.names=F,col.names=T)
    > 


    [+] Loading gcc  7.3.0  ... 
    [+] Loading GSL 2.4 for GCC 7.2.0 ... 
    [-] Unloading gcc  7.3.0  ... 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading openmpi 3.0.2  for GCC 7.3.0 
    [+] Loading ImageMagick  7.0.8  on cn3147 
    [+] Loading HDF5  1.10.4 
    [+] Loading pandoc  2.13  on cn3147 
    [+] Loading R 3.5.2 
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()



```bash
%%bash

source /data/chiarp/conda/etc/profile.d/conda.sh

python ./ldsc/munge_sumstats.py \
--sumstats AD.Jansen.hg38.rsid.forLDSC.txt \
--N 452010 \
--out AD.Jansen \
--merge-alleles w_hm3.snplist

python ./ldsc/munge_sumstats.py \
--sumstats PD.Nalls.hg38.rsid.forLDSC.txt \
--N 482730 \
--out PD.Nalls \
--merge-alleles w_hm3.snplist

python ./ldsc/munge_sumstats.py \
--sumstats ALS.Nicolas.hg38.rsid.forLDSC.txt \
--N 80610 \
--out ALS.Nicolas \
--merge-alleles w_hm3.snplist

python ./ldsc/munge_sumstats.py \
--sumstats LBD.Chia.hg38.rsid.forLDSC.txt \
--N 6502 \
--out LBD.Chia \
--merge-alleles w_hm3.snplist

```

    *********************************************************************
    * LD Score Regression (LDSC)
    * Version 1.0.1
    * (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane
    * Broad Institute of MIT and Harvard / MIT Department of Mathematics
    * GNU General Public License v3
    *********************************************************************
    Call: 
    ./munge_sumstats.py \
    --out AD.Jansen \
    --merge-alleles w_hm3.snplist \
    --N 452010.0 \
    --sumstats AD.Jansen.hg38.rsid.forLDSC.txt 
    
    Interpreting column names as follows:
    N:	Sample size
    A1:	Allele 1, interpreted as ref allele for signed sumstat.
    P:	p-Value
    BETA:	[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)
    A2:	Allele 2, interpreted as non-ref allele for signed sumstat.
    SNP:	Variant ID (e.g., rs number)
    
    Reading list of SNPs for allele merge from w_hm3.snplist
    Read 1217311 SNPs for allele merge.
    Reading sumstats from AD.Jansen.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.
    ..WARNING: 3 SNPs had P outside of (0,1]. The P column may be mislabeled.
     done
    Read 9732197 SNPs from --sumstats file.
    Removed 8533711 SNPs not in --merge-alleles.
    Removed 0 SNPs with missing values.
    Removed 0 SNPs with INFO <= 0.9.
    Removed 0 SNPs with MAF <= 0.01.
    Removed 3 SNPs with out-of-bounds p-values.
    Removed 295 variants that were not SNPs or were strand-ambiguous.
    1198188 SNPs remain.
    Removed 359 SNPs with duplicated rs numbers (1197829 SNPs remain).
    Removed 46520 SNPs with N < 301340.126667 (1151309 SNPs remain).
    Median value of BETA was 5.16224534711e-05, which seems sensible.
    Removed 135 SNPs whose alleles did not match --merge-alleles (1151174 SNPs remain).
    Writing summary statistics for 1217311 SNPs (1151174 with nonmissing beta) to AD.Jansen.sumstats.gz.
    
    Metadata:
    Mean chi^2 = 1.125
    Lambda GC = 1.084
    Max chi^2 = 1001.735
    312 Genome-wide significant SNPs (some may have been removed by filtering).
    
    Conversion finished at Mon Mar 22 08:45:15 2021
    Total time elapsed: 1.0m:20.02s
    *********************************************************************
    * LD Score Regression (LDSC)
    * Version 1.0.1
    * (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane
    * Broad Institute of MIT and Harvard / MIT Department of Mathematics
    * GNU General Public License v3
    *********************************************************************
    Call: 
    ./munge_sumstats.py \
    --out PD.Nalls \
    --merge-alleles w_hm3.snplist \
    --N 482730.0 \
    --sumstats PD.Nalls.hg38.rsid.forLDSC.txt 
    
    Interpreting column names as follows:
    N:	Sample size
    A1:	Allele 1, interpreted as ref allele for signed sumstat.
    P:	p-Value
    BETA:	[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)
    A2:	Allele 2, interpreted as non-ref allele for signed sumstat.
    SNP:	Variant ID (e.g., rs number)
    
    Reading list of SNPs for allele merge from w_hm3.snplist
    Read 1217311 SNPs for allele merge.
    Reading sumstats from PD.Nalls.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.
    .. done
    Read 8330284 SNPs from --sumstats file.
    Removed 7137528 SNPs not in --merge-alleles.
    Removed 0 SNPs with missing values.
    Removed 0 SNPs with INFO <= 0.9.
    Removed 0 SNPs with MAF <= 0.01.
    Removed 0 SNPs with out-of-bounds p-values.
    Removed 659 variants that were not SNPs or were strand-ambiguous.
    1192097 SNPs remain.
    Removed 0 SNPs with duplicated rs numbers (1192097 SNPs remain).
    Removed 54873 SNPs with N < 321820.0 (1137224 SNPs remain).
    Median value of BETA was 0.0, which seems sensible.
    Removed 171 SNPs whose alleles did not match --merge-alleles (1137053 SNPs remain).
    Writing summary statistics for 1217311 SNPs (1137053 with nonmissing beta) to PD.Nalls.sumstats.gz.
    
    Metadata:
    Mean chi^2 = 1.137
    Lambda GC = 1.091
    Max chi^2 = 180.422
    276 Genome-wide significant SNPs (some may have been removed by filtering).
    
    Conversion finished at Mon Mar 22 08:46:29 2021
    Total time elapsed: 1.0m:11.31s
    *********************************************************************
    * LD Score Regression (LDSC)
    * Version 1.0.1
    * (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane
    * Broad Institute of MIT and Harvard / MIT Department of Mathematics
    * GNU General Public License v3
    *********************************************************************
    Call: 
    ./munge_sumstats.py \
    --out ALS.Nicolas \
    --merge-alleles w_hm3.snplist \
    --N 80610.0 \
    --sumstats ALS.Nicolas.hg38.rsid.forLDSC.txt 
    
    Interpreting column names as follows:
    N:	Sample size
    A1:	Allele 1, interpreted as ref allele for signed sumstat.
    P:	p-Value
    BETA:	[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)
    A2:	Allele 2, interpreted as non-ref allele for signed sumstat.
    SNP:	Variant ID (e.g., rs number)
    
    Reading list of SNPs for allele merge from w_hm3.snplist
    Read 1217311 SNPs for allele merge.
    Reading sumstats from ALS.Nicolas.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.
    .. done
    Read 8949246 SNPs from --sumstats file.
    Removed 7801463 SNPs not in --merge-alleles.
    Removed 0 SNPs with missing values.
    Removed 0 SNPs with INFO <= 0.9.
    Removed 0 SNPs with MAF <= 0.01.
    Removed 0 SNPs with out-of-bounds p-values.
    Removed 4 variants that were not SNPs or were strand-ambiguous.
    1147779 SNPs remain.
    Removed 0 SNPs with duplicated rs numbers (1147779 SNPs remain).
    Removed 0 SNPs with N < 53740.0 (1147779 SNPs remain).
    Median value of BETA was 0.0001, which seems sensible.
    Removed 2 SNPs whose alleles did not match --merge-alleles (1147777 SNPs remain).
    Writing summary statistics for 1217311 SNPs (1147777 with nonmissing beta) to ALS.Nicolas.sumstats.gz.
    
    Metadata:
    Mean chi^2 = 1.044
    Lambda GC = 1.034
    Max chi^2 = 50.171
    18 Genome-wide significant SNPs (some may have been removed by filtering).
    
    Conversion finished at Mon Mar 22 08:47:46 2021
    Total time elapsed: 1.0m:14.06s
    *********************************************************************
    * LD Score Regression (LDSC)
    * Version 1.0.1
    * (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane
    * Broad Institute of MIT and Harvard / MIT Department of Mathematics
    * GNU General Public License v3
    *********************************************************************
    Call: 
    ./munge_sumstats.py \
    --out LBD.Chia \
    --merge-alleles w_hm3.snplist \
    --N 6502.0 \
    --sumstats LBD.Chia.hg38.rsid.forLDSC.txt 
    
    Interpreting column names as follows:
    N:	Sample size
    A1:	Allele 1, interpreted as ref allele for signed sumstat.
    P:	p-Value
    BETA:	[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)
    A2:	Allele 2, interpreted as non-ref allele for signed sumstat.
    SNP:	Variant ID (e.g., rs number)
    
    Reading list of SNPs for allele merge from w_hm3.snplist
    Read 1217311 SNPs for allele merge.
    Reading sumstats from LBD.Chia.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.
    .. done
    Read 7843595 SNPs from --sumstats file.
    Removed 6762819 SNPs not in --merge-alleles.
    Removed 0 SNPs with missing values.
    Removed 0 SNPs with INFO <= 0.9.
    Removed 0 SNPs with MAF <= 0.01.
    Removed 0 SNPs with out-of-bounds p-values.
    Removed 3 variants that were not SNPs or were strand-ambiguous.
    1080773 SNPs remain.
    Removed 6 SNPs with duplicated rs numbers (1080767 SNPs remain).
    Removed 0 SNPs with N < 4412.0 (1080767 SNPs remain).
    Median value of BETA was 0.000859630411882, which seems sensible.
    Removed 19 SNPs whose alleles did not match --merge-alleles (1080748 SNPs remain).
    Writing summary statistics for 1217311 SNPs (1080748 with nonmissing beta) to LBD.Chia.sumstats.gz.
    
    Metadata:
    Mean chi^2 = 1.026
    Lambda GC = 1.023
    Max chi^2 = 215.691
    30 Genome-wide significant SNPs (some may have been removed by filtering).
    
    Conversion finished at Mon Mar 22 08:48:59 2021
    Total time elapsed: 1.0m:9.73s
    *********************************************************************
    * LD Score Regression (LDSC)
    * Version 1.0.1
    * (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane
    * Broad Institute of MIT and Harvard / MIT Department of Mathematics
    * GNU General Public License v3
    *********************************************************************
    Call: 
    ./munge_sumstats.py \
    --out MG.Chia \
    --merge-alleles w_hm3.snplist \
    --N 38243.0 \
    --sumstats MG.Chia.hg38.rsid.forLDSC.txt 
    
    Interpreting column names as follows:
    N:	Sample size
    A1:	Allele 1, interpreted as ref allele for signed sumstat.
    P:	p-Value
    BETA:	[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)
    A2:	Allele 2, interpreted as non-ref allele for signed sumstat.
    SNP:	Variant ID (e.g., rs number)
    
    Reading list of SNPs for allele merge from w_hm3.snplist
    Read 1217311 SNPs for allele merge.
    Reading sumstats from MG.Chia.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.
    ..... done
    Read 24006245 SNPs from --sumstats file.
    Removed 22863916 SNPs not in --merge-alleles.
    Removed 0 SNPs with missing values.
    Removed 0 SNPs with INFO <= 0.9.
    Removed 0 SNPs with MAF <= 0.01.
    Removed 0 SNPs with out-of-bounds p-values.
    Removed 70 variants that were not SNPs or were strand-ambiguous.
    1142259 SNPs remain.
    Removed 189 SNPs with duplicated rs numbers (1142070 SNPs remain).
    Removed 0 SNPs with N < 25495.3333333 (1142070 SNPs remain).
    Median value of BETA was -0.0001, which seems sensible.
    Removed 93 SNPs whose alleles did not match --merge-alleles (1141977 SNPs remain).
    Writing summary statistics for 1217311 SNPs (1141977 with nonmissing beta) to MG.Chia.sumstats.gz.
    
    Metadata:
    Mean chi^2 = 1.073
    Lambda GC = 1.059
    Max chi^2 = 60.413
    7 Genome-wide significant SNPs (some may have been removed by filtering).
    
    Conversion finished at Mon Mar 22 08:50:51 2021
    Total time elapsed: 1.0m:48.5s



```bash
%%bash

source /data/chiarp/conda/etc/profile.d/conda.sh
python ./ldsc/munge_sumstats.py \
--sumstats MG.Chia.hg38.rsid.forLDSC.txt \
--N 38243 \
--out MG.Chia \
--merge-alleles w_hm3.snplist
```

    *********************************************************************
    * LD Score Regression (LDSC)
    * Version 1.0.1
    * (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane
    * Broad Institute of MIT and Harvard / MIT Department of Mathematics
    * GNU General Public License v3
    *********************************************************************
    Call: 
    ./munge_sumstats.py \
    --out MG.Chia \
    --merge-alleles w_hm3.snplist \
    --N 38243.0 \
    --sumstats MG.Chia.hg38.rsid.forLDSC.txt 
    
    Interpreting column names as follows:
    N:	Sample size
    A1:	Allele 1, interpreted as ref allele for signed sumstat.
    P:	p-Value
    BETA:	[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)
    A2:	Allele 2, interpreted as non-ref allele for signed sumstat.
    SNP:	Variant ID (e.g., rs number)
    
    Reading list of SNPs for allele merge from w_hm3.snplist
    Read 1217311 SNPs for allele merge.
    Reading sumstats from MG.Chia.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.
    .. done
    Read 8387157 SNPs from --sumstats file.
    Removed 7272179 SNPs not in --merge-alleles.
    Removed 0 SNPs with missing values.
    Removed 0 SNPs with INFO <= 0.9.
    Removed 0 SNPs with MAF <= 0.01.
    Removed 0 SNPs with out-of-bounds p-values.
    Removed 8 variants that were not SNPs or were strand-ambiguous.
    1114970 SNPs remain.
    Removed 17 SNPs with duplicated rs numbers (1114953 SNPs remain).
    Removed 0 SNPs with N < 25495.3333333 (1114953 SNPs remain).
    Median value of BETA was -0.0002, which seems sensible.
    Removed 27 SNPs whose alleles did not match --merge-alleles (1114926 SNPs remain).
    Writing summary statistics for 1217311 SNPs (1114926 with nonmissing beta) to MG.Chia.sumstats.gz.
    
    Metadata:
    Mean chi^2 = 1.076
    Lambda GC = 1.062
    Max chi^2 = 60.413
    7 Genome-wide significant SNPs (some may have been removed by filtering).
    
    Conversion finished at Tue Apr  6 12:29:59 2021
    Total time elapsed: 1.0m:3.88s



```bash
%%bash

source /data/chiarp/conda/etc/profile.d/conda.sh

python ./ldsc/munge_sumstats.py \
--sumstats ALS.VanRheenen.hg38.rsid.forLDSC.txt \
--out ALS.VanRheenen \
--merge-alleles w_hm3.snplist
```

    *********************************************************************
    * LD Score Regression (LDSC)
    * Version 1.0.1
    * (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane
    * Broad Institute of MIT and Harvard / MIT Department of Mathematics
    * GNU General Public License v3
    *********************************************************************
    Call: 
    ./munge_sumstats.py \
    --out ALS.VanRheenen \
    --merge-alleles w_hm3.snplist \
    --sumstats ALS.VanRheenen.hg38.rsid.forLDSC.txt 
    
    Interpreting column names as follows:
    N:	Sample size
    A1:	Allele 1, interpreted as ref allele for signed sumstat.
    P:	p-Value
    BETA:	[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)
    A2:	Allele 2, interpreted as non-ref allele for signed sumstat.
    SNP:	Variant ID (e.g., rs number)
    
    Reading list of SNPs for allele merge from w_hm3.snplist
    Read 1217311 SNPs for allele merge.
    Reading sumstats from ALS.VanRheenen.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.
    .. done
    Read 8709431 SNPs from --sumstats file.
    Removed 7535250 SNPs not in --merge-alleles.
    Removed 0 SNPs with missing values.
    Removed 0 SNPs with INFO <= 0.9.
    Removed 0 SNPs with MAF <= 0.01.
    Removed 0 SNPs with out-of-bounds p-values.
    Removed 0 variants that were not SNPs or were strand-ambiguous.
    1174181 SNPs remain.
    Removed 0 SNPs with duplicated rs numbers (1174181 SNPs remain).
    Removed 0 SNPs with N < 24034.6666667 (1174181 SNPs remain).
    Median value of BETA was 0.0, which seems sensible.
    Removed 0 SNPs whose alleles did not match --merge-alleles (1174181 SNPs remain).
    Writing summary statistics for 1217311 SNPs (1174181 with nonmissing beta) to ALS.VanRheenen.sumstats.gz.
    
    Metadata:
    Mean chi^2 = 1.03
    Lambda GC = 1.023
    Max chi^2 = 79.877
    28 Genome-wide significant SNPs (some may have been removed by filtering).
    
    Conversion finished at Mon Mar 22 12:10:51 2021
    Total time elapsed: 1.0m:13.13s


## Compute heritability and the LD Score regression intercept for MG


```bash
%%bash

source /data/chiarp/conda/etc/profile.d/conda.sh

python ./ldsc/ldsc.py \
--h2 MG.Chia.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out MG.Chia_h2
cat MG.Chia_h2.log
```

    *********************************************************************
    * LD Score Regression (LDSC)
    * Version 1.0.1
    * (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane
    * Broad Institute of MIT and Harvard / MIT Department of Mathematics
    * GNU General Public License v3
    *********************************************************************
    Call: 
    ./ldsc.py \
    --h2 MG.Chia.sumstats.gz \
    --ref-ld-chr eur_w_ld_chr/ \
    --out MG.Chia_h2 \
    --w-ld-chr eur_w_ld_chr/ 
    
    Beginning analysis at Tue Apr  6 12:48:50 2021
    Reading summary statistics from MG.Chia.sumstats.gz ...
    Read summary statistics for 1114926 SNPs.
    Reading reference panel LD Score from eur_w_ld_chr/[1-22] ... (ldscore_fromlist)
    Read reference panel LD Scores for 1290028 SNPs.
    Removing partitioned LD Scores with zero variance.
    Reading regression weight LD Score from eur_w_ld_chr/[1-22] ... (ldscore_fromlist)
    Read regression weight LD Scores for 1290028 SNPs.
    After merging with reference panel LD, 1107424 SNPs remain.
    After merging with regression SNP LD, 1107424 SNPs remain.
    Using two-step estimator with cutoff at 30.
    Total Observed scale h2: 0.0751 (0.0148)
    Lambda GC: 1.0618
    Mean Chi^2: 1.0742
    Intercept: 1.0174 (0.0066)
    Ratio: 0.235 (0.0888)
    Analysis finished at Tue Apr  6 12:49:05 2021
    Total time elapsed: 14.47s
    *********************************************************************
    * LD Score Regression (LDSC)
    * Version 1.0.1
    * (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane
    * Broad Institute of MIT and Harvard / MIT Department of Mathematics
    * GNU General Public License v3
    *********************************************************************
    Call: 
    ./ldsc.py \
    --h2 MG.Chia.sumstats.gz \
    --ref-ld-chr eur_w_ld_chr/ \
    --out MG.Chia_h2 \
    --w-ld-chr eur_w_ld_chr/ 
    
    Beginning analysis at Tue Apr  6 12:48:50 2021
    Reading summary statistics from MG.Chia.sumstats.gz ...
    Read summary statistics for 1114926 SNPs.
    Reading reference panel LD Score from eur_w_ld_chr/[1-22] ... (ldscore_fromlist)
    Read reference panel LD Scores for 1290028 SNPs.
    Removing partitioned LD Scores with zero variance.
    Reading regression weight LD Score from eur_w_ld_chr/[1-22] ... (ldscore_fromlist)
    Read regression weight LD Scores for 1290028 SNPs.
    After merging with reference panel LD, 1107424 SNPs remain.
    After merging with regression SNP LD, 1107424 SNPs remain.
    Using two-step estimator with cutoff at 30.
    Total Observed scale h2: 0.0751 (0.0148)
    Lambda GC: 1.0618
    Mean Chi^2: 1.0742
    Intercept: 1.0174 (0.0066)
    Ratio: 0.235 (0.0888)
    Analysis finished at Tue Apr  6 12:49:05 2021
    Total time elapsed: 14.47s


    /gpfs/gsfs11/users/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/LDSC/ldsc/ldscore/irwls.py:161: FutureWarning: `rcond` parameter will change to the default of machine precision times ``max(M, N)`` where M and N are the input matrix dimensions.
    To use the future default and silence this warning we advise to pass `rcond=None`, to keep using the old, explicitly pass `rcond=-1`.
      coef = np.linalg.lstsq(x, y)


## Run LDSC and genetic correlation on MG vs. neurological diseases
### Estimating Heritability, Genetic Correlation and the LD Score Regression Intercept


```bash
%%bash

source /data/chiarp/conda/etc/profile.d/conda.sh

python ./ldsc/ldsc.py \
--rg MG.Chia.sumstats.gz,AD.Jansen.sumstats.gz,PD.Nalls.sumstats.gz,ALS.VanRheenen.sumstats.gz,LBD.Chia.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out MG_AD_PD_ALS_LBD
```

    *********************************************************************
    * LD Score Regression (LDSC)
    * Version 1.0.1
    * (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane
    * Broad Institute of MIT and Harvard / MIT Department of Mathematics
    * GNU General Public License v3
    *********************************************************************
    Call: 
    ./ldsc.py \
    --ref-ld-chr eur_w_ld_chr/ \
    --out MG_AD_PD_ALS_LBD \
    --rg MG.Chia.sumstats.gz,AD.Jansen.sumstats.gz,PD.Nalls.sumstats.gz,ALS.VanRheenen.sumstats.gz,LBD.Chia.sumstats.gz \
    --w-ld-chr eur_w_ld_chr/ 
    
    Beginning analysis at Tue Apr  6 12:50:51 2021
    Reading summary statistics from MG.Chia.sumstats.gz ...
    Read summary statistics for 1114926 SNPs.
    Reading reference panel LD Score from eur_w_ld_chr/[1-22] ... (ldscore_fromlist)
    Read reference panel LD Scores for 1290028 SNPs.
    Removing partitioned LD Scores with zero variance.
    Reading regression weight LD Score from eur_w_ld_chr/[1-22] ... (ldscore_fromlist)
    Read regression weight LD Scores for 1290028 SNPs.
    After merging with reference panel LD, 1107424 SNPs remain.
    After merging with regression SNP LD, 1107424 SNPs remain.
    Computing rg for phenotype 2/5
    Reading summary statistics from AD.Jansen.sumstats.gz ...
    Read summary statistics for 1217311 SNPs.
    After merging with summary statistics, 1107424 SNPs remain.
    1065135 SNPs with valid alleles.
    
    Heritability of phenotype 1
    ---------------------------
    Total Observed scale h2: 0.0754 (0.0165)
    Lambda GC: 1.0618
    Mean Chi^2: 1.0751
    Intercept: 1.0171 (0.0078)
    Ratio: 0.2272 (0.1044)
    
    Heritability of phenotype 2/5
    -----------------------------
    Total Observed scale h2: 0.0075 (0.0049)
    Lambda GC: 1.0833
    Mean Chi^2: 1.1236
    Intercept: 1.057 (0.0592)
    Ratio: 0.4613 (0.4791)
    
    Genetic Covariance
    ------------------
    Total Observed scale gencov: 0.0002 (0.0029)
    Mean z1*z2: 0.0062
    Intercept: 0.005 (0.0047)
    
    Genetic Correlation
    -------------------
    Genetic Correlation: 0.0091 (0.123)
    Z-score: 0.0741
    P: 0.9409
    
    Computing rg for phenotype 3/5
    Reading summary statistics from PD.Nalls.sumstats.gz ...
    Read summary statistics for 1217311 SNPs.
    After merging with summary statistics, 1107424 SNPs remain.
    1064424 SNPs with valid alleles.
    
    Heritability of phenotype 3/5
    -----------------------------
    Total Observed scale h2: 0.0166 (0.0017)
    Lambda GC: 1.0864
    Mean Chi^2: 1.1347
    Intercept: 0.981 (0.0076)
    Ratio < 0 (usually indicates GC correction).
    
    Genetic Covariance
    ------------------
    Total Observed scale gencov: 0.0001 (0.0031)
    Mean z1*z2: -0.0018
    Intercept: -0.0029 (0.0053)
    
    Genetic Correlation
    -------------------
    Genetic Correlation: 0.0032 (0.0883)
    Z-score: 0.0359
    P: 0.9714
    
    Computing rg for phenotype 4/5
    Reading summary statistics from ALS.VanRheenen.sumstats.gz ...
    Read summary statistics for 1217311 SNPs.
    After merging with summary statistics, 1107424 SNPs remain.
    1098401 SNPs with valid alleles.
    
    Heritability of phenotype 4/5
    -----------------------------
    Total Observed scale h2: 0.0485 (0.0131)
    Lambda GC: 1.0225
    Mean Chi^2: 1.0281
    Intercept: 0.9941 (0.0062)
    Ratio < 0 (usually indicates GC correction).
    
    Genetic Covariance
    ------------------
    Total Observed scale gencov: 0.0171 (0.0102)
    Mean z1*z2: 0.0246
    Intercept: 0.0122 (0.0047)
    
    Genetic Correlation
    -------------------
    Genetic Correlation: 0.2858 (0.18)
    Z-score: 1.5881
    P: 0.1123
    
    Computing rg for phenotype 5/5
    Reading summary statistics from LBD.Chia.sumstats.gz ...
    Read summary statistics for 1217311 SNPs.
    After merging with summary statistics, 1107424 SNPs remain.
    1009819 SNPs with valid alleles.
    
    Heritability of phenotype 5/5
    -----------------------------
    Total Observed scale h2: 0.1293 (0.0835)
    Lambda GC: 1.0225
    Mean Chi^2: 1.0233
    Intercept: 1.0064 (0.0087)
    Ratio: 0.2728 (0.3714)
    
    Genetic Covariance
    ------------------
    Total Observed scale gencov: 0.0349 (0.0229)
    Mean z1*z2: -0.0009
    Intercept: -0.0117 (0.0047)
    
    Genetic Correlation
    -------------------
    Genetic Correlation: 0.3657 (0.2797)
    Z-score: 1.3078
    P: 0.191
    
    
    Summary of Genetic Correlation Results
                      p1                          p2      rg      se       z       p  h2_obs  h2_obs_se  h2_int  h2_int_se  gcov_int  gcov_int_se
     MG.Chia.sumstats.gz       AD.Jansen.sumstats.gz  0.0091  0.1230  0.0741  0.9409  0.0075     0.0049  1.0570     0.0592    0.0050       0.0047
     MG.Chia.sumstats.gz        PD.Nalls.sumstats.gz  0.0032  0.0883  0.0359  0.9714  0.0166     0.0017  0.9810     0.0076   -0.0029       0.0053
     MG.Chia.sumstats.gz  ALS.VanRheenen.sumstats.gz  0.2858  0.1800  1.5881  0.1123  0.0485     0.0131  0.9941     0.0062    0.0122       0.0047
     MG.Chia.sumstats.gz        LBD.Chia.sumstats.gz  0.3657  0.2797  1.3078  0.1910  0.1293     0.0835  1.0064     0.0087   -0.0117       0.0047
    
    Analysis finished at Tue Apr  6 12:51:30 2021
    Total time elapsed: 38.68s


    /gpfs/gsfs11/users/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/LDSC/ldsc/ldscore/sumstats.py:532: FutureWarning: Method .as_matrix will be removed in a future version. Use .values instead.
      ref_ld = sumstats.as_matrix(columns=ref_ld_cnames)
    /gpfs/gsfs11/users/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/LDSC/ldsc/ldscore/irwls.py:161: FutureWarning: `rcond` parameter will change to the default of machine precision times ``max(M, N)`` where M and N are the input matrix dimensions.
    To use the future default and silence this warning we advise to pass `rcond=None`, to keep using the old, explicitly pass `rcond=-1`.
      coef = np.linalg.lstsq(x, y)


## Run LDSC and genetic correlation on autoimmune diseases

Ran a term search i.e. `autoimmune disease` on GWAS catalaog to find summary stats to download. Filtered down list of studies which were dont on European samples with the largest cohort (where possible) and to those where summary stats were available for FTP download directly from GWAS catalog.

- Crohns : `http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST004001-GCST005000/GCST004132/harmonised/28067908-GCST004132-EFO_0000384.h.tsv.gz`
- Graves.Hashimoto.Thyroiditis : `http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST005001-GCST006000/GCST005524/harmonised/22922229-GCST005524-EFO_0006812.h.tsv.gz`
- IBS : `http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST004001-GCST005000/GCST004131/harmonised/28067908-GCST004131-EFO_0003767.h.tsv.gz`
- jArthritis : `http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST90010001-GCST90011000/GCST90010715/harmonised/33106285-GCST90010715-EFO_0002609.h.tsv.gz`
- LatentAutoImmuneDiabetes : `http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST007001-GCST008000/GCST007245/CousminerDL_30254083.gz`
- MicroscopicColitis : `http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST009001-GCST010000/GCST009081/wholegwas`
- MS : `http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST005001-GCST006000/GCST005531/harmonised/24076602-GCST005531-EFO_0003885.h.tsv.gz`
- NeuromyelitisOptica_AQP4_IgGpos : `http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST006001-GCST007000/GCST006937/harmonised/29769526-GCST006937-EFO_0009584.h.tsv.gz`
- PsoriaticArthritis : `http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST007001-GCST008000/GCST007043/Table-S3.xlsx`
- RA : `http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST002001-GCST003000/GCST002318/harmonised/24390342-GCST002318-EFO_0000685.h.tsv.gz`
- SLE : `http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST005001-GCST006000/GCST005831/harmonised/29848360-GCST005831-EFO_0002690.h.tsv.gz`
- T1D : `http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST005001-GCST006000/GCST005536/harmonised/25751624-GCST005536-EFO_0001359.h.tsv.gz`
- UC : `http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST004001-GCST005000/GCST004133/harmonised/28067908-GCST004133-EFO_0000729.h.tsv.gz`
- Vitiligo : `http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST004001-GCST005000/GCST004785/GWAS123chr1.cmh.gz` ## chr 1-22,X



```python
!mkdir GWAScatalog
```


```bash
%%bash
cd GWAScatalog/
wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST004001-GCST005000/GCST004132/harmonised/28067908-GCST004132-EFO_0000384.h.tsv.gz
wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST005001-GCST006000/GCST005524/harmonised/22922229-GCST005524-EFO_0006812.h.tsv.gz
wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST004001-GCST005000/GCST004131/harmonised/28067908-GCST004131-EFO_0003767.h.tsv.gz
wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST90010001-GCST90011000/GCST90010715/harmonised/33106285-GCST90010715-EFO_0002609.h.tsv.gz
wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST007001-GCST008000/GCST007245/CousminerDL_30254083.gz
wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST009001-GCST010000/GCST009081/wholegwas
wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST005001-GCST006000/GCST005531/harmonised/24076602-GCST005531-EFO_0003885.h.tsv.gz
wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST006001-GCST007000/GCST006937/harmonised/29769526-GCST006937-EFO_0009584.h.tsv.gz
wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST007001-GCST008000/GCST007043/Table-S3.xlsx
wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST002001-GCST003000/GCST002318/harmonised/24390342-GCST002318-EFO_0000685.h.tsv.gz
wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST005001-GCST006000/GCST005831/harmonised/29848360-GCST005831-EFO_0002690.h.tsv.gz
wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST005001-GCST006000/GCST005536/harmonised/25751624-GCST005536-EFO_0001359.h.tsv.gz
wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST004001-GCST005000/GCST004133/harmonised/28067908-GCST004133-EFO_0000729.h.tsv.gz
wget http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST004001-GCST005000/GCST004785/GWAS123chr1.cmh.gz

```

    IOPub data rate exceeded.
    The notebook server will temporarily stop sending output
    to the client in order to avoid crashing it.
    To change this limit, set the config variable
    `--NotebookApp.iopub_data_rate_limit`.
    
    Current values:
    NotebookApp.iopub_data_rate_limit=1000000.0 (bytes/sec)
    NotebookApp.rate_limit_window=3.0 (secs)
    


- SNP = A unique identifier (e.g., the rs number)
- A1 =  effect allele
- A2 =  non-effect allele
- N = Sample size (which often varies from SNP to SNP)
- P = P-value
- BETA = A signed summary statistic (beta, OR, log odds, Z-score, etc)



```bash
%%bash
module load R/3.5.2
R --vanilla --no-save

require(data.table)
require(tidyverse)

# Crohn's
data <- fread("28067908-GCST004132-EFO_0000384.h.tsv.gz",header=T)
data$SNP = ifelse(is.na(data$hm_rsid), data$hm_variant_id, data$hm_rsid)
data1 <- data %>%
         mutate(N = 40266) %>%
         select(SNP,hm_effect_allele,hm_other_allele,N,p_value,hm_beta) %>%
         rename(A1 = hm_effect_allele, A2 = hm_other_allele, P = p_value, BETA = hm_beta)
write.table(data1, "Crohns.deLange.hg38.rsid.forLDSC.txt",sep="\t",quote=F,row.names=F,col.names=T)

# Graves.Hashimoto.Thyroiditis - does not have effect allele information

# IBS
data <- fread("28067908-GCST004131-EFO_0003767.h.tsv.gz",header=T)
data$SNP = ifelse(is.na(data$hm_rsid), data$hm_variant_id, data$hm_rsid)
data1 <- data %>%
         mutate(N = 59957) %>%
         select(SNP,hm_effect_allele,hm_other_allele,N,p_value,hm_beta) %>%
         rename(A1 = hm_effect_allele, A2 = hm_other_allele, P = p_value, BETA = hm_beta)
write.table(data1, "IBS.deLange.hg38.rsid.forLDSC.txt",sep="\t",quote=F,row.names=F,col.names=T)


# juvenile idiopathic arthritis
data <- fread("33106285-GCST90010715-EFO_0002609.h.tsv.gz",header=T)
data$SNP = ifelse(is.na(data$hm_rsid), data$hm_variant_id, data$hm_rsid)
data1 <- data %>%
         mutate(N = 12501) %>%
         select(SNP,hm_effect_allele,hm_other_allele,N,p_value,hm_beta) %>%
         rename(A1 = hm_effect_allele, A2 = hm_other_allele, P = p_value, BETA = hm_beta)
write.table(data1, "jArthritis.LopezIsac.hg38.rsid.forLDSC.txt",sep="\t",quote=F,row.names=F,col.names=T)


# MicroscopicColitis
data <- fread("wholegwas",header=T)
data1 <- data %>%
         mutate(N = 445655) %>%
         select(SNP,ALLELE1,ALLELE0,N,P_BOLT_LMM,BETA) %>%
         rename(A1 = ALLELE1, A2 = ALLELE0, P = P_BOLT_LMM)
write.table(data1, "MicroscopicColitis.Green.hg38.rsid.forLDSC.txt",sep="\t",quote=F,row.names=F,col.names=T)


# MS
data <- fread("24076602-GCST005531-EFO_0003885.h.tsv.gz",header=T)
data$SNP = ifelse(is.na(data$hm_rsid), data$hm_variant_id, data$hm_rsid)
data1 <- data %>%
         mutate(N = 38582) %>%
         select(SNP,hm_effect_allele,hm_other_allele,N,p_value,hm_beta) %>%
         rename(A1 = hm_effect_allele, A2 = hm_other_allele, P = p_value, BETA = hm_beta)
write.table(data1, "MS.Beecham.hg38.rsid.forLDSC.txt",sep="\t",quote=F,row.names=F,col.names=T)


# NeuromyelitisOptica_AQP4_IgGpos
data <- fread("29769526-GCST006937-EFO_0009584.h.tsv.gz",header=T)
data$SNP = ifelse(is.na(data$hm_rsid), data$hm_variant_id, data$hm_rsid)
data1 <- data %>%
         mutate(N = n) %>%
         select(SNP,hm_effect_allele,hm_other_allele,N,p_value,hm_beta) %>%
         rename(A1 = hm_effect_allele, A2 = hm_other_allele, P = p_value, BETA = hm_beta)
write.table(data1, "NeuromyelitisOptica_AQP4_IgGpos.Estrada.hg38.rsid.forLDSC.txt",sep="\t",quote=F,row.names=F,col.names=T)


# RA
data <- fread("24390342-GCST002318-EFO_0000685.h.tsv.gz",header=T)
data$SNP = ifelse(is.na(data$hm_rsid), data$hm_variant_id, data$hm_rsid)
data1 <- data %>%
         mutate(N = 58284,
                BETA = log(hm_odds_ratio)) %>%
         select(SNP,hm_effect_allele,hm_other_allele,N,p_value,BETA) %>%
         rename(A1 = hm_effect_allele, A2 = hm_other_allele, P = p_value) %>%
         filter(SNP != is.na(SNP))
write.table(data1, "RA.Okada.hg38.rsid.forLDSC.txt",sep="\t",quote=F,row.names=F,col.names=T)


# SLE
data <- fread("29848360-GCST005831-EFO_0002690.h.tsv.gz",header=T)
data$SNP = ifelse(is.na(data$hm_rsid), data$hm_variant_id, data$hm_rsid)
data1 <- data %>%
         mutate(N = 16966) %>%
         select(SNP,effect_allele,other_allele,N,p_value,z) %>%
         rename(A1 = effect_allele, A2 = other_allele, P = p_value, Z = z)
write.table(data1, "SLE.Julia.hg38.rsid.forLDSC.txt",sep="\t",quote=F,row.names=F,col.names=T)


# T1D
data <- fread("25751624-GCST005536-EFO_0001359.h.tsv.gz",header=T)
data$SNP = ifelse(is.na(data$hm_rsid), data$hm_variant_id, data$hm_rsid)
data1 <- data %>%
         mutate(N = 29652) %>%
         select(SNP,hm_effect_allele,hm_other_allele,N,p_value,hm_beta) %>%
         rename(A1 = hm_effect_allele, A2 = hm_other_allele, P = p_value, BETA = hm_beta)
write.table(data1, "T1D.OnengutGumuscu.hg38.rsid.forLDSC.txt",sep="\t",quote=F,row.names=F,col.names=T)


# UC
data <- fread("28067908-GCST004133-EFO_0000729.h.tsv.gz",header=T)
data$SNP = ifelse(is.na(data$hm_rsid), data$hm_variant_id, data$hm_rsid)
data1 <- data %>%
         mutate(N = 45975) %>%
         select(SNP,hm_effect_allele,hm_other_allele,N,p_value,hm_beta) %>%
         rename(A1 = hm_effect_allele, A2 = hm_other_allele, P = p_value, BETA = hm_beta)
write.table(data1, "UC.deLange.hg38.rsid.forLDSC.txt",sep="\t",quote=F,row.names=F,col.names=T)


```

### Reformat autoimmune summ stats


```bash
%%bash

source /data/chiarp/conda/etc/profile.d/conda.sh

python ./ldsc/munge_sumstats.py \
--sumstats GWAScatalog/Crohns.deLange.hg38.rsid.forLDSC.txt \
--out Crohns.deLange \
--merge-alleles w_hm3.snplist

python ./ldsc/munge_sumstats.py \
--sumstats GWAScatalog/jArthritis.LopezIsac.hg38.rsid.forLDSC.txt \
--out jArthritis.LopezIsac \
--merge-alleles w_hm3.snplist

python ./ldsc/munge_sumstats.py \
--sumstats GWAScatalog/IBS.deLange.hg38.rsid.forLDSC.txt \
--out IBS.deLange \
--merge-alleles w_hm3.snplist

python ./ldsc/munge_sumstats.py \
--sumstats GWAScatalog/MicroscopicColitis.Green.hg38.rsid.forLDSC.txt \
--out MicroscopicColitis.Green \
--merge-alleles w_hm3.snplist

python ./ldsc/munge_sumstats.py \
--sumstats GWAScatalog/MS.Beecham.hg38.rsid.forLDSC.txt \
--out MS.Beecham \
--merge-alleles w_hm3.snplist

python ./ldsc/munge_sumstats.py \
--sumstats GWAScatalog/NeuromyelitisOptica_AQP4_IgGpos.Estrada.hg38.rsid.forLDSC.txt \
--out NeuromyelitisOptica_AQP4_IgGpos.Estrada \
--merge-alleles w_hm3.snplist

python ./ldsc/munge_sumstats.py \
--sumstats GWAScatalog/RA.Okada.hg38.rsid.forLDSC.txt \
--out RA.Okada \
--merge-alleles w_hm3.snplist

python ./ldsc/munge_sumstats.py \
--sumstats GWAScatalog/SLE.Julia.hg38.rsid.forLDSC.txt \
--out SLE.Julia \
--merge-alleles w_hm3.snplist

python ./ldsc/munge_sumstats.py \
--sumstats GWAScatalog/T1D.OnengutGumuscu.hg38.rsid.forLDSC.txt \
--out T1D.OnengutGumuscu \
--merge-alleles w_hm3.snplist

python ./ldsc/munge_sumstats.py \
--sumstats GWAScatalog/UC.deLange.hg38.rsid.forLDSC.txt \
--out UC.deLange \
--merge-alleles w_hm3.snplist
```

    *********************************************************************
    * LD Score Regression (LDSC)
    * Version 1.0.1
    * (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane
    * Broad Institute of MIT and Harvard / MIT Department of Mathematics
    * GNU General Public License v3
    *********************************************************************
    Call: 
    ./munge_sumstats.py \
    --out Crohns.deLange \
    --merge-alleles w_hm3.snplist \
    --sumstats GWAScatalog/Crohns.deLange.hg38.rsid.forLDSC.txt 
    
    Interpreting column names as follows:
    N:	Sample size
    A1:	Allele 1, interpreted as ref allele for signed sumstat.
    P:	p-Value
    BETA:	[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)
    A2:	Allele 2, interpreted as non-ref allele for signed sumstat.
    SNP:	Variant ID (e.g., rs number)
    
    Reading list of SNPs for allele merge from w_hm3.snplist
    Read 1217311 SNPs for allele merge.
    Reading sumstats from GWAScatalog/Crohns.deLange.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.
    .. done
    Read 9469592 SNPs from --sumstats file.
    Removed 8297513 SNPs not in --merge-alleles.
    Removed 11594 SNPs with missing values.
    Removed 0 SNPs with INFO <= 0.9.
    Removed 0 SNPs with MAF <= 0.01.
    Removed 0 SNPs with out-of-bounds p-values.
    Removed 0 variants that were not SNPs or were strand-ambiguous.
    1160485 SNPs remain.
    Removed 0 SNPs with duplicated rs numbers (1160485 SNPs remain).
    Removed 0 SNPs with N < 26844.0 (1160485 SNPs remain).
    Median value of BETA was 0.0002, which seems sensible.
    Removed 0 SNPs whose alleles did not match --merge-alleles (1160485 SNPs remain).
    Writing summary statistics for 1217311 SNPs (1160485 with nonmissing beta) to Crohns.deLange.sumstats.gz.
    
    Metadata:
    Mean chi^2 = 1.456
    Lambda GC = 1.26
    Max chi^2 = 489.534
    2076 Genome-wide significant SNPs (some may have been removed by filtering).
    
    Conversion finished at Mon Mar 29 03:50:10 2021
    Total time elapsed: 1.0m:8.63s
    *********************************************************************
    * LD Score Regression (LDSC)
    * Version 1.0.1
    * (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane
    * Broad Institute of MIT and Harvard / MIT Department of Mathematics
    * GNU General Public License v3
    *********************************************************************
    Call: 
    ./munge_sumstats.py \
    --out jArthritis.LopezIsac \
    --merge-alleles w_hm3.snplist \
    --sumstats GWAScatalog/jArthritis.LopezIsac.hg38.rsid.forLDSC.txt 
    
    Interpreting column names as follows:
    N:	Sample size
    A1:	Allele 1, interpreted as ref allele for signed sumstat.
    P:	p-Value
    BETA:	[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)
    A2:	Allele 2, interpreted as non-ref allele for signed sumstat.
    SNP:	Variant ID (e.g., rs number)
    
    Reading list of SNPs for allele merge from w_hm3.snplist
    Read 1217311 SNPs for allele merge.
    Reading sumstats from GWAScatalog/jArthritis.LopezIsac.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.
    .. done
    
    ERROR converting summary statistics:
    
    Traceback (most recent call last):
      File "./ldsc/munge_sumstats.py", line 686, in munge_sumstats
        dat = parse_dat(dat_gen, cname_translation, merge_alleles, log, args)
      File "./ldsc/munge_sumstats.py", line 301, in parse_dat
        dat = pd.concat(dat_list, axis=0).reset_index(drop=True)
      File "/usr/local/Anaconda/envs/py2.7/lib/python2.7/site-packages/pandas/core/reshape/concat.py", line 228, in concat
        copy=copy, sort=sort)
      File "/usr/local/Anaconda/envs/py2.7/lib/python2.7/site-packages/pandas/core/reshape/concat.py", line 262, in __init__
        raise ValueError('No objects to concatenate')
    ValueError: No objects to concatenate
    
    
    Conversion finished at Mon Mar 29 03:50:27 2021
    Total time elapsed: 14.79s
    *********************************************************************
    * LD Score Regression (LDSC)
    * Version 1.0.1
    * (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane
    * Broad Institute of MIT and Harvard / MIT Department of Mathematics
    * GNU General Public License v3
    *********************************************************************
    Call: 
    ./munge_sumstats.py \
    --out IBS.deLange \
    --merge-alleles w_hm3.snplist \
    --sumstats GWAScatalog/IBS.deLange.hg38.rsid.forLDSC.txt 
    
    Interpreting column names as follows:
    N:	Sample size
    A1:	Allele 1, interpreted as ref allele for signed sumstat.
    P:	p-Value
    BETA:	[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)
    A2:	Allele 2, interpreted as non-ref allele for signed sumstat.
    SNP:	Variant ID (e.g., rs number)
    
    Reading list of SNPs for allele merge from w_hm3.snplist
    Read 1217311 SNPs for allele merge.
    Reading sumstats from GWAScatalog/IBS.deLange.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.
    .. done
    Read 9633630 SNPs from --sumstats file.
    Removed 8458334 SNPs not in --merge-alleles.
    Removed 14614 SNPs with missing values.
    Removed 0 SNPs with INFO <= 0.9.
    Removed 0 SNPs with MAF <= 0.01.
    Removed 0 SNPs with out-of-bounds p-values.
    Removed 0 variants that were not SNPs or were strand-ambiguous.
    1160682 SNPs remain.
    Removed 0 SNPs with duplicated rs numbers (1160682 SNPs remain).
    Removed 0 SNPs with N < 39971.3333333 (1160682 SNPs remain).
    Median value of BETA was -0.0001, which seems sensible.
    Removed 0 SNPs whose alleles did not match --merge-alleles (1160682 SNPs remain).
    Writing summary statistics for 1217311 SNPs (1160682 with nonmissing beta) to IBS.deLange.sumstats.gz.
    
    Metadata:
    Mean chi^2 = 1.503
    Lambda GC = 1.291
    Max chi^2 = 501.36
    2383 Genome-wide significant SNPs (some may have been removed by filtering).
    
    Conversion finished at Mon Mar 29 03:51:36 2021
    Total time elapsed: 1.0m:8.43s
    *********************************************************************
    * LD Score Regression (LDSC)
    * Version 1.0.1
    * (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane
    * Broad Institute of MIT and Harvard / MIT Department of Mathematics
    * GNU General Public License v3
    *********************************************************************
    Call: 
    ./munge_sumstats.py \
    --out MicroscopicColitis.Green \
    --merge-alleles w_hm3.snplist \
    --sumstats GWAScatalog/MicroscopicColitis.Green.hg38.rsid.forLDSC.txt 
    
    Interpreting column names as follows:
    N:	Sample size
    A1:	Allele 1, interpreted as ref allele for signed sumstat.
    P:	p-Value
    BETA:	[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)
    A2:	Allele 2, interpreted as non-ref allele for signed sumstat.
    SNP:	Variant ID (e.g., rs number)
    
    Reading list of SNPs for allele merge from w_hm3.snplist
    Read 1217311 SNPs for allele merge.
    Reading sumstats from GWAScatalog/MicroscopicColitis.Green.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.
    .. done
    Read 7935431 SNPs from --sumstats file.
    Removed 6800745 SNPs not in --merge-alleles.
    Removed 0 SNPs with missing values.
    Removed 0 SNPs with INFO <= 0.9.
    Removed 0 SNPs with MAF <= 0.01.
    Removed 0 SNPs with out-of-bounds p-values.
    Removed 115 variants that were not SNPs or were strand-ambiguous.
    1134571 SNPs remain.
    Removed 159 SNPs with duplicated rs numbers (1134412 SNPs remain).
    Removed 0 SNPs with N < 297103.333333 (1134412 SNPs remain).
    Median value of BETA was 6.58051e-07, which seems sensible.
    Removed 90 SNPs whose alleles did not match --merge-alleles (1134322 SNPs remain).
    Writing summary statistics for 1217311 SNPs (1134322 with nonmissing beta) to MicroscopicColitis.Green.sumstats.gz.
    
    Metadata:
    Mean chi^2 = 1.007
    WARNING: mean chi^2 may be too small.
    Lambda GC = 1.0
    Max chi^2 = 28.6
    0 Genome-wide significant SNPs (some may have been removed by filtering).
    
    Conversion finished at Mon Mar 29 03:53:34 2021
    Total time elapsed: 1.0m:55.62s
    *********************************************************************
    * LD Score Regression (LDSC)
    * Version 1.0.1
    * (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane
    * Broad Institute of MIT and Harvard / MIT Department of Mathematics
    * GNU General Public License v3
    *********************************************************************
    Call: 
    ./munge_sumstats.py \
    --out MS.Beecham \
    --merge-alleles w_hm3.snplist \
    --sumstats GWAScatalog/MS.Beecham.hg38.rsid.forLDSC.txt 
    
    Interpreting column names as follows:
    N:	Sample size
    A1:	Allele 1, interpreted as ref allele for signed sumstat.
    P:	p-Value
    BETA:	[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)
    A2:	Allele 2, interpreted as non-ref allele for signed sumstat.
    SNP:	Variant ID (e.g., rs number)
    
    Reading list of SNPs for allele merge from w_hm3.snplist
    Read 1217311 SNPs for allele merge.
    Reading sumstats from GWAScatalog/MS.Beecham.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.
    .WARNING: 1 SNPs had P outside of (0,1]. The P column may be mislabeled.
     done
    Read 155249 SNPs from --sumstats file.
    Removed 80737 SNPs not in --merge-alleles.
    Removed 23160 SNPs with missing values.
    Removed 0 SNPs with INFO <= 0.9.
    Removed 0 SNPs with MAF <= 0.01.
    Removed 1 SNPs with out-of-bounds p-values.
    Removed 0 variants that were not SNPs or were strand-ambiguous.
    51351 SNPs remain.
    Removed 0 SNPs with duplicated rs numbers (51351 SNPs remain).
    Removed 0 SNPs with N < 25721.3333333 (51351 SNPs remain).
    Median value of BETA was 0.000999500333083, which seems sensible.
    Removed 1 SNPs whose alleles did not match --merge-alleles (51350 SNPs remain).
    Writing summary statistics for 1217311 SNPs (51350 with nonmissing beta) to MS.Beecham.sumstats.gz.
    
    Metadata:
    Mean chi^2 = 3.325
    Lambda GC = 1.499
    Max chi^2 = 1342.232
    655 Genome-wide significant SNPs (some may have been removed by filtering).
    
    Conversion finished at Mon Mar 29 03:55:11 2021
    Total time elapsed: 1.0m:33.14s
    *********************************************************************
    * LD Score Regression (LDSC)
    * Version 1.0.1
    * (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane
    * Broad Institute of MIT and Harvard / MIT Department of Mathematics
    * GNU General Public License v3
    *********************************************************************
    Call: 
    ./munge_sumstats.py \
    --out NeuromyelitisOptica_AQP4_IgGpos.Estrada \
    --merge-alleles w_hm3.snplist \
    --sumstats GWAScatalog/NeuromyelitisOptica_AQP4_IgGpos.Estrada.hg38.rsid.forLDSC.txt 
    
    Interpreting column names as follows:
    N:	Sample size
    A1:	Allele 1, interpreted as ref allele for signed sumstat.
    P:	p-Value
    BETA:	[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)
    A2:	Allele 2, interpreted as non-ref allele for signed sumstat.
    SNP:	Variant ID (e.g., rs number)
    
    Reading list of SNPs for allele merge from w_hm3.snplist
    Read 1217311 SNPs for allele merge.
    Reading sumstats from GWAScatalog/NeuromyelitisOptica_AQP4_IgGpos.Estrada.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.
    .. done
    Read 6804545 SNPs from --sumstats file.
    Removed 5646909 SNPs not in --merge-alleles.
    Removed 13446 SNPs with missing values.
    Removed 0 SNPs with INFO <= 0.9.
    Removed 0 SNPs with MAF <= 0.01.
    Removed 0 SNPs with out-of-bounds p-values.
    Removed 0 variants that were not SNPs or were strand-ambiguous.
    1144190 SNPs remain.
    Removed 0 SNPs with duplicated rs numbers (1144190 SNPs remain).
    Removed 0 SNPs with N < 917.333333333 (1144190 SNPs remain).
    Median value of BETA was 0.0051, which seems sensible.
    Removed 0 SNPs whose alleles did not match --merge-alleles (1144190 SNPs remain).
    Writing summary statistics for 1217311 SNPs (1144190 with nonmissing beta) to NeuromyelitisOptica_AQP4_IgGpos.Estrada.sumstats.gz.
    
    Metadata:
    Mean chi^2 = 0.967
    WARNING: mean chi^2 may be too small.
    Lambda GC = 0.967
    Max chi^2 = 65.633
    33 Genome-wide significant SNPs (some may have been removed by filtering).
    
    Conversion finished at Mon Mar 29 03:58:34 2021
    Total time elapsed: 3.0m:20.09s
    *********************************************************************
    * LD Score Regression (LDSC)
    * Version 1.0.1
    * (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane
    * Broad Institute of MIT and Harvard / MIT Department of Mathematics
    * GNU General Public License v3
    *********************************************************************
    Call: 
    ./munge_sumstats.py \
    --out RA.Okada \
    --merge-alleles w_hm3.snplist \
    --sumstats GWAScatalog/RA.Okada.hg38.rsid.forLDSC.txt 
    
    Interpreting column names as follows:
    N:	Sample size
    A1:	Allele 1, interpreted as ref allele for signed sumstat.
    P:	p-Value
    BETA:	[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)
    A2:	Allele 2, interpreted as non-ref allele for signed sumstat.
    SNP:	Variant ID (e.g., rs number)
    
    Reading list of SNPs for allele merge from w_hm3.snplist
    Read 1217311 SNPs for allele merge.
    Reading sumstats from GWAScatalog/RA.Okada.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.
    .. done
    Read 9700598 SNPs from --sumstats file.
    Removed 8506767 SNPs not in --merge-alleles.
    Removed 0 SNPs with missing values.
    Removed 0 SNPs with INFO <= 0.9.
    Removed 0 SNPs with MAF <= 0.01.
    Removed 0 SNPs with out-of-bounds p-values.
    Removed 1 variants that were not SNPs or were strand-ambiguous.
    1193830 SNPs remain.
    Removed 11 SNPs with duplicated rs numbers (1193819 SNPs remain).
    Removed 0 SNPs with N < 38856.0 (1193819 SNPs remain).
    Median value of BETA was 0.0, which seems sensible.
    Removed 2 SNPs whose alleles did not match --merge-alleles (1193817 SNPs remain).
    Writing summary statistics for 1217311 SNPs (1193817 with nonmissing beta) to RA.Okada.sumstats.gz.
    
    Metadata:
    Mean chi^2 = 1.212
    Lambda GC = 1.048
    Max chi^2 = 1143.797
    1334 Genome-wide significant SNPs (some may have been removed by filtering).
    
    Conversion finished at Mon Mar 29 04:01:03 2021
    Total time elapsed: 2.0m:25.61s
    *********************************************************************
    * LD Score Regression (LDSC)
    * Version 1.0.1
    * (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane
    * Broad Institute of MIT and Harvard / MIT Department of Mathematics
    * GNU General Public License v3
    *********************************************************************
    Call: 
    ./munge_sumstats.py \
    --out SLE.Julia \
    --merge-alleles w_hm3.snplist \
    --sumstats GWAScatalog/SLE.Julia.hg38.rsid.forLDSC.txt 
    
    Interpreting column names as follows:
    N:	Sample size
    A1:	Allele 1, interpreted as ref allele for signed sumstat.
    P:	p-Value
    A2:	Allele 2, interpreted as non-ref allele for signed sumstat.
    SNP:	Variant ID (e.g., rs number)
    Z:	Z-score (0 --> no effect; above 0 --> A1 is trait/risk increasing)
    
    Reading list of SNPs for allele merge from w_hm3.snplist
    Read 1217311 SNPs for allele merge.
    Reading sumstats from GWAScatalog/SLE.Julia.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.
    .. done
    Read 8294672 SNPs from --sumstats file.
    Removed 7128945 SNPs not in --merge-alleles.
    Removed 19002 SNPs with missing values.
    Removed 0 SNPs with INFO <= 0.9.
    Removed 0 SNPs with MAF <= 0.01.
    Removed 0 SNPs with out-of-bounds p-values.
    Removed 0 variants that were not SNPs or were strand-ambiguous.
    1146725 SNPs remain.
    Removed 1 SNPs with duplicated rs numbers (1146724 SNPs remain).
    Removed 0 SNPs with N < 11310.6666667 (1146724 SNPs remain).
    Median value of Z was 0.001, which seems sensible.
    Removed 0 SNPs whose alleles did not match --merge-alleles (1146724 SNPs remain).
    Writing summary statistics for 1217311 SNPs (1146724 with nonmissing beta) to SLE.Julia.sumstats.gz.
    
    Metadata:
    Mean chi^2 = 1.299
    Lambda GC = 1.188
    Max chi^2 = 515.624
    1081 Genome-wide significant SNPs (some may have been removed by filtering).
    
    Conversion finished at Mon Mar 29 04:06:16 2021
    Total time elapsed: 5.0m:9.76s
    *********************************************************************
    * LD Score Regression (LDSC)
    * Version 1.0.1
    * (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane
    * Broad Institute of MIT and Harvard / MIT Department of Mathematics
    * GNU General Public License v3
    *********************************************************************
    Call: 
    ./munge_sumstats.py \
    --out T1D.OnengutGumuscu \
    --merge-alleles w_hm3.snplist \
    --sumstats GWAScatalog/T1D.OnengutGumuscu.hg38.rsid.forLDSC.txt 
    
    Interpreting column names as follows:
    N:	Sample size
    A1:	Allele 1, interpreted as ref allele for signed sumstat.
    P:	p-Value
    BETA:	[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)
    A2:	Allele 2, interpreted as non-ref allele for signed sumstat.
    SNP:	Variant ID (e.g., rs number)
    
    Reading list of SNPs for allele merge from w_hm3.snplist
    Read 1217311 SNPs for allele merge.
    Reading sumstats from GWAScatalog/T1D.OnengutGumuscu.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.
    . done
    Read 121619 SNPs from --sumstats file.
    Removed 53156 SNPs not in --merge-alleles.
    Removed 20518 SNPs with missing values.
    Removed 0 SNPs with INFO <= 0.9.
    Removed 0 SNPs with MAF <= 0.01.
    Removed 0 SNPs with out-of-bounds p-values.
    Removed 0 variants that were not SNPs or were strand-ambiguous.
    47945 SNPs remain.
    Removed 59 SNPs with duplicated rs numbers (47886 SNPs remain).
    Removed 0 SNPs with N < 19768.0 (47886 SNPs remain).
    Median value of BETA was -0.00100050033358, which seems sensible.
    Removed 0 SNPs whose alleles did not match --merge-alleles (47886 SNPs remain).
    Writing summary statistics for 1217311 SNPs (47886 with nonmissing beta) to T1D.OnengutGumuscu.sumstats.gz.
    
    Metadata:
    Mean chi^2 = 2.932
    Lambda GC = 1.645
    Max chi^2 = 892.191
    652 Genome-wide significant SNPs (some may have been removed by filtering).
    
    Conversion finished at Mon Mar 29 04:11:19 2021
    Total time elapsed: 4.0m:51.62s
    *********************************************************************
    * LD Score Regression (LDSC)
    * Version 1.0.1
    * (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane
    * Broad Institute of MIT and Harvard / MIT Department of Mathematics
    * GNU General Public License v3
    *********************************************************************
    Call: 
    ./munge_sumstats.py \
    --out UC.deLange \
    --merge-alleles w_hm3.snplist \
    --sumstats GWAScatalog/UC.deLange.hg38.rsid.forLDSC.txt 
    
    Interpreting column names as follows:
    N:	Sample size
    A1:	Allele 1, interpreted as ref allele for signed sumstat.
    P:	p-Value
    BETA:	[linear/logistic] regression coefficient (0 --> no effect; above 0 --> A1 is trait/risk increasing)
    A2:	Allele 2, interpreted as non-ref allele for signed sumstat.
    SNP:	Variant ID (e.g., rs number)
    
    Reading list of SNPs for allele merge from w_hm3.snplist
    Read 1217311 SNPs for allele merge.
    Reading sumstats from GWAScatalog/UC.deLange.hg38.rsid.forLDSC.txt into memory 5000000 SNPs at a time.
    .. done
    Read 9486539 SNPs from --sumstats file.
    Removed 8314044 SNPs not in --merge-alleles.
    Removed 11980 SNPs with missing values.
    Removed 0 SNPs with INFO <= 0.9.
    Removed 0 SNPs with MAF <= 0.01.
    Removed 0 SNPs with out-of-bounds p-values.
    Removed 0 variants that were not SNPs or were strand-ambiguous.
    1160515 SNPs remain.
    Removed 0 SNPs with duplicated rs numbers (1160515 SNPs remain).
    Removed 0 SNPs with N < 30650.0 (1160515 SNPs remain).
    Median value of BETA was -0.0001, which seems sensible.
    Removed 0 SNPs whose alleles did not match --merge-alleles (1160515 SNPs remain).
    Writing summary statistics for 1217311 SNPs (1160515 with nonmissing beta) to UC.deLange.sumstats.gz.
    
    Metadata:
    Mean chi^2 = 1.321
    Lambda GC = 1.208
    Max chi^2 = 279.731
    1042 Genome-wide significant SNPs (some may have been removed by filtering).
    
    Conversion finished at Mon Mar 29 04:14:59 2021
    Total time elapsed: 3.0m:30.76s


    Traceback (most recent call last):
      File "./ldsc/munge_sumstats.py", line 745, in <module>
        munge_sumstats(parser.parse_args(), p=True)
      File "./ldsc/munge_sumstats.py", line 686, in munge_sumstats
        dat = parse_dat(dat_gen, cname_translation, merge_alleles, log, args)
      File "./ldsc/munge_sumstats.py", line 301, in parse_dat
        dat = pd.concat(dat_list, axis=0).reset_index(drop=True)
      File "/usr/local/Anaconda/envs/py2.7/lib/python2.7/site-packages/pandas/core/reshape/concat.py", line 228, in concat
        copy=copy, sort=sort)
      File "/usr/local/Anaconda/envs/py2.7/lib/python2.7/site-packages/pandas/core/reshape/concat.py", line 262, in __init__
        raise ValueError('No objects to concatenate')
    ValueError: No objects to concatenate


### Estimating Heritability, Genetic Correlation with autoimmune diseases


```bash
%%bash

source /data/chiarp/conda/etc/profile.d/conda.sh

python ./ldsc/ldsc.py \
--rg MG.Chia.sumstats.gz,Crohns.deLange.sumstats.gz,NeuromyelitisOptica_AQP4_IgGpos.Estrada.sumstats.gz,RA.Okada.sumstats.gz,SLE.Julia.sumstats.gz,T1D.OnengutGumuscu.sumstats.gz,UC.deLange.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out MG_AutoImmuneDiseases
```

    *********************************************************************
    * LD Score Regression (LDSC)
    * Version 1.0.1
    * (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane
    * Broad Institute of MIT and Harvard / MIT Department of Mathematics
    * GNU General Public License v3
    *********************************************************************
    Call: 
    ./ldsc.py \
    --ref-ld-chr eur_w_ld_chr/ \
    --out MG_AutoImmuneDiseases \
    --rg MG.Chia.sumstats.gz,Crohns.deLange.sumstats.gz,NeuromyelitisOptica_AQP4_IgGpos.Estrada.sumstats.gz,RA.Okada.sumstats.gz,SLE.Julia.sumstats.gz,T1D.OnengutGumuscu.sumstats.gz,UC.deLange.sumstats.gz \
    --w-ld-chr eur_w_ld_chr/ 
    
    Beginning analysis at Tue Apr  6 12:58:47 2021
    Reading summary statistics from MG.Chia.sumstats.gz ...
    Read summary statistics for 1114926 SNPs.
    Reading reference panel LD Score from eur_w_ld_chr/[1-22] ... (ldscore_fromlist)
    Read reference panel LD Scores for 1290028 SNPs.
    Removing partitioned LD Scores with zero variance.
    Reading regression weight LD Score from eur_w_ld_chr/[1-22] ... (ldscore_fromlist)
    Read regression weight LD Scores for 1290028 SNPs.
    After merging with reference panel LD, 1107424 SNPs remain.
    After merging with regression SNP LD, 1107424 SNPs remain.
    Computing rg for phenotype 2/7
    Reading summary statistics from Crohns.deLange.sumstats.gz ...
    Read summary statistics for 1217311 SNPs.
    After merging with summary statistics, 1107424 SNPs remain.
    1067853 SNPs with valid alleles.
    
    Heritability of phenotype 1
    ---------------------------
    Total Observed scale h2: 0.075 (0.015)
    Lambda GC: 1.0618
    Mean Chi^2: 1.0745
    Intercept: 1.0178 (0.0071)
    Ratio: 0.2393 (0.0947)
    
    Heritability of phenotype 2/7
    -----------------------------
    Total Observed scale h2: 0.4422 (0.0524)
    Lambda GC: 1.2564
    Mean Chi^2: 1.4482
    Intercept: 1.0975 (0.0153)
    Ratio: 0.2175 (0.0341)
    
    Genetic Covariance
    ------------------
    Total Observed scale gencov: 0.0126 (0.0158)
    Mean z1*z2: 0.0107
    Intercept: 0.0014 (0.0063)
    
    Genetic Correlation
    -------------------
    Genetic Correlation: 0.0693 (0.0883)
    Z-score: 0.7844
    P: 0.4328
    
    Computing rg for phenotype 3/7
    Reading summary statistics from NeuromyelitisOptica_AQP4_IgGpos.Estrada.sumstats.gz ...
    Read summary statistics for 1217311 SNPs.
    After merging with summary statistics, 1107424 SNPs remain.
    1070202 SNPs with valid alleles.
    
    Heritability of phenotype 3/7
    -----------------------------
    Total Observed scale h2: -0.1747 (0.3013)
    Lambda GC: 0.9605
    Mean Chi^2: 0.9567
    Intercept: 0.9614 (0.0061)
    Ratio: NA (mean chi^2 < 1)
    
    Genetic Covariance
    ------------------
    Total Observed scale gencov: 0.0468 (0.0474)
    Mean z1*z2: 0.0029
    Intercept: -0.0038 (0.0046)
    
    Genetic Correlation
    -------------------
    Genetic Correlation: nan (nan) (h2  out of bounds) 
    Z-score: nan (nan) (h2  out of bounds)
    P: nan (nan) (h2  out of bounds)
    WARNING: One of the h2's was out of bounds.
    This usually indicates a data-munging error or that h2 or N is low.
    
    Computing rg for phenotype 4/7
    Reading summary statistics from RA.Okada.sumstats.gz ...
    Read summary statistics for 1217311 SNPs.
    After merging with summary statistics, 1107424 SNPs remain.
    1099228 SNPs with valid alleles.
    
    Heritability of phenotype 4/7
    -----------------------------
    Total Observed scale h2: 0.1696 (0.0229)
    Lambda GC: 1.0466
    Mean Chi^2: 1.1445
    Intercept: 0.9484 (0.0117)
    Ratio < 0 (usually indicates GC correction).
    
    Genetic Covariance
    ------------------
    Total Observed scale gencov: 0.0466 (0.0125)
    Mean z1*z2: 0.0444
    Intercept: 0.0003 (0.0058)
    
    Genetic Correlation
    -------------------
    Genetic Correlation: 0.4135 (0.0929)
    Z-score: 4.4531
    P: 8.4652e-06
    
    Computing rg for phenotype 5/7
    Reading summary statistics from SLE.Julia.sumstats.gz ...
    Read summary statistics for 1217311 SNPs.
    After merging with summary statistics, 1107424 SNPs remain.
    1056030 SNPs with valid alleles.
    
    Heritability of phenotype 5/7
    -----------------------------
    Total Observed scale h2: 0.4178 (0.0645)
    Lambda GC: 1.1811
    Mean Chi^2: 1.2485
    Intercept: 1.1006 (0.0121)
    Ratio: 0.4047 (0.0485)
    
    Genetic Covariance
    ------------------
    Total Observed scale gencov: -0.0373 (0.0202)
    Mean z1*z2: -0.033
    Intercept: -0.0138 (0.0063)
    
    Genetic Correlation
    -------------------
    Genetic Correlation: -0.2126 (0.1185)
    Z-score: -1.7942
    P: 0.0728
    
    Computing rg for phenotype 6/7
    Reading summary statistics from T1D.OnengutGumuscu.sumstats.gz ...
    Read summary statistics for 1217311 SNPs.
    After merging with summary statistics, 1107424 SNPs remain.
    44608 SNPs with valid alleles.
    WARNING: number of SNPs less than 200k; this is almost always bad.
    
    Heritability of phenotype 6/7
    -----------------------------
    Total Observed scale h2: 2.1786 (0.5807)
    Lambda GC: 1.641
    Mean Chi^2: 2.8979
    Intercept: 1.3416 (0.2462)
    Ratio: 0.18 (0.1297)
    
    Genetic Covariance
    ------------------
    Total Observed scale gencov: 0.3856 (0.1396)
    Mean z1*z2: 0.36
    Intercept: -0.0212 (0.043)
    
    Genetic Correlation
    -------------------
    Genetic Correlation: 0.4356 (0.1066)
    Z-score: 4.087
    P: 4.3707e-05
    
    Computing rg for phenotype 7/7
    Reading summary statistics from UC.deLange.sumstats.gz ...
    Read summary statistics for 1217311 SNPs.
    After merging with summary statistics, 1107424 SNPs remain.
    1067835 SNPs with valid alleles.
    
    Heritability of phenotype 7/7
    -----------------------------
    Total Observed scale h2: 0.2335 (0.0256)
    Lambda GC: 1.2038
    Mean Chi^2: 1.3086
    Intercept: 1.101 (0.0138)
    Ratio: 0.3274 (0.0449)
    
    Genetic Covariance
    ------------------
    Total Observed scale gencov: 0.0267 (0.011)
    Mean z1*z2: 0.0119
    Intercept: -0.0097 (0.0052)
    
    Genetic Correlation
    -------------------
    Genetic Correlation: 0.2015 (0.0851)
    Z-score: 2.3671
    P: 0.0179
    
    
    Summary of Genetic Correlation Results
                      p1                                                   p2       rg       se       z          p  h2_obs  h2_obs_se  h2_int  h2_int_se  gcov_int  gcov_int_se
     MG.Chia.sumstats.gz                           Crohns.deLange.sumstats.gz  0.06927   0.0883  0.7844     0.4328  0.4422     0.0524  1.0975     0.0153    0.0014       0.0063
     MG.Chia.sumstats.gz  NeuromyelitisOptica_AQP4_IgGpos.Estrada.sumstats.gz       NA       NA      NA         NA -0.1747     0.3013  0.9614     0.0061   -0.0038       0.0046
     MG.Chia.sumstats.gz                                 RA.Okada.sumstats.gz   0.4135  0.09286   4.453  8.465e-06  0.1696     0.0229  0.9484     0.0117    0.0003       0.0058
     MG.Chia.sumstats.gz                                SLE.Julia.sumstats.gz  -0.2126   0.1185  -1.794    0.07278  0.4178     0.0645  1.1006     0.0121   -0.0138       0.0063
     MG.Chia.sumstats.gz                       T1D.OnengutGumuscu.sumstats.gz   0.4356   0.1066   4.087  4.371e-05  2.1786     0.5807  1.3416     0.2462   -0.0212       0.0430
     MG.Chia.sumstats.gz                               UC.deLange.sumstats.gz   0.2015  0.08513   2.367    0.01793  0.2335     0.0256  1.1010     0.0138   -0.0097       0.0052
    
    Analysis finished at Tue Apr  6 12:59:36 2021
    Total time elapsed: 48.76s


    /gpfs/gsfs11/users/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/LDSC/ldsc/ldscore/sumstats.py:532: FutureWarning: Method .as_matrix will be removed in a future version. Use .values instead.
      ref_ld = sumstats.as_matrix(columns=ref_ld_cnames)
    /gpfs/gsfs11/users/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/LDSC/ldsc/ldscore/irwls.py:161: FutureWarning: `rcond` parameter will change to the default of machine precision times ``max(M, N)`` where M and N are the input matrix dimensions.
    To use the future default and silence this warning we advise to pass `rcond=None`, to keep using the old, explicitly pass `rcond=-1`.
      coef = np.linalg.lstsq(x, y)


### Estimating Heritability, Genetic Correlation with neurodisease and autoimmune diseases


Discussed with Bryan, and he suggested that I take out the following diseases from the autoimmune group: 

microscopic colitis and IBS


```bash
%%bash

source /data/chiarp/conda/etc/profile.d/conda.sh

python ./ldsc/ldsc.py \
--rg MG.Chia.sumstats.gz,AD.Jansen.sumstats.gz,PD.Nalls.sumstats.gz,ALS.VanRheenen.sumstats.gz,LBD.Chia.sumstats.gz,Crohns.deLange.sumstats.gz,MS.Beecham.sumstats.gz,RA.Okada.sumstats.gz,SLE.Julia.sumstats.gz,T1D.OnengutGumuscu.sumstats.gz,UC.deLange.sumstats.gz \
--ref-ld-chr eur_w_ld_chr/ \
--w-ld-chr eur_w_ld_chr/ \
--out MG_Neuro.AutoImmuneDiseases
```

    *********************************************************************
    * LD Score Regression (LDSC)
    * Version 1.0.1
    * (C) 2014-2019 Brendan Bulik-Sullivan and Hilary Finucane
    * Broad Institute of MIT and Harvard / MIT Department of Mathematics
    * GNU General Public License v3
    *********************************************************************
    Call: 
    ./ldsc.py \
    --ref-ld-chr eur_w_ld_chr/ \
    --out MG_Neuro.AutoImmuneDiseases \
    --rg MG.Chia.sumstats.gz,AD.Jansen.sumstats.gz,PD.Nalls.sumstats.gz,ALS.VanRheenen.sumstats.gz,LBD.Chia.sumstats.gz,Crohns.deLange.sumstats.gz,MS.Beecham.sumstats.gz,RA.Okada.sumstats.gz,SLE.Julia.sumstats.gz,T1D.OnengutGumuscu.sumstats.gz,UC.deLange.sumstats.gz \
    --w-ld-chr eur_w_ld_chr/ 
    
    Beginning analysis at Tue Apr  6 12:59:38 2021
    Reading summary statistics from MG.Chia.sumstats.gz ...
    Read summary statistics for 1114926 SNPs.
    Reading reference panel LD Score from eur_w_ld_chr/[1-22] ... (ldscore_fromlist)
    Read reference panel LD Scores for 1290028 SNPs.
    Removing partitioned LD Scores with zero variance.
    Reading regression weight LD Score from eur_w_ld_chr/[1-22] ... (ldscore_fromlist)
    Read regression weight LD Scores for 1290028 SNPs.
    After merging with reference panel LD, 1107424 SNPs remain.
    After merging with regression SNP LD, 1107424 SNPs remain.
    Computing rg for phenotype 2/11
    Reading summary statistics from AD.Jansen.sumstats.gz ...
    Read summary statistics for 1217311 SNPs.
    After merging with summary statistics, 1107424 SNPs remain.
    1065135 SNPs with valid alleles.
    
    Heritability of phenotype 1
    ---------------------------
    Total Observed scale h2: 0.0754 (0.0165)
    Lambda GC: 1.0618
    Mean Chi^2: 1.0751
    Intercept: 1.0171 (0.0078)
    Ratio: 0.2272 (0.1044)
    
    Heritability of phenotype 2/11
    ------------------------------
    Total Observed scale h2: 0.0075 (0.0049)
    Lambda GC: 1.0833
    Mean Chi^2: 1.1236
    Intercept: 1.057 (0.0592)
    Ratio: 0.4613 (0.4791)
    
    Genetic Covariance
    ------------------
    Total Observed scale gencov: 0.0002 (0.0029)
    Mean z1*z2: 0.0062
    Intercept: 0.005 (0.0047)
    
    Genetic Correlation
    -------------------
    Genetic Correlation: 0.0091 (0.123)
    Z-score: 0.0741
    P: 0.9409
    
    Computing rg for phenotype 3/11
    Reading summary statistics from PD.Nalls.sumstats.gz ...
    Read summary statistics for 1217311 SNPs.
    After merging with summary statistics, 1107424 SNPs remain.
    1064424 SNPs with valid alleles.
    
    Heritability of phenotype 3/11
    ------------------------------
    Total Observed scale h2: 0.0166 (0.0017)
    Lambda GC: 1.0864
    Mean Chi^2: 1.1347
    Intercept: 0.981 (0.0076)
    Ratio < 0 (usually indicates GC correction).
    
    Genetic Covariance
    ------------------
    Total Observed scale gencov: 0.0001 (0.0031)
    Mean z1*z2: -0.0018
    Intercept: -0.0029 (0.0053)
    
    Genetic Correlation
    -------------------
    Genetic Correlation: 0.0032 (0.0883)
    Z-score: 0.0359
    P: 0.9714
    
    Computing rg for phenotype 4/11
    Reading summary statistics from ALS.VanRheenen.sumstats.gz ...
    Read summary statistics for 1217311 SNPs.
    After merging with summary statistics, 1107424 SNPs remain.
    1098401 SNPs with valid alleles.
    
    Heritability of phenotype 4/11
    ------------------------------
    Total Observed scale h2: 0.0485 (0.0131)
    Lambda GC: 1.0225
    Mean Chi^2: 1.0281
    Intercept: 0.9941 (0.0062)
    Ratio < 0 (usually indicates GC correction).
    
    Genetic Covariance
    ------------------
    Total Observed scale gencov: 0.0171 (0.0102)
    Mean z1*z2: 0.0246
    Intercept: 0.0122 (0.0047)
    
    Genetic Correlation
    -------------------
    Genetic Correlation: 0.2858 (0.18)
    Z-score: 1.5881
    P: 0.1123
    
    Computing rg for phenotype 5/11
    Reading summary statistics from LBD.Chia.sumstats.gz ...
    Read summary statistics for 1217311 SNPs.
    After merging with summary statistics, 1107424 SNPs remain.
    1009819 SNPs with valid alleles.
    
    Heritability of phenotype 5/11
    ------------------------------
    Total Observed scale h2: 0.1293 (0.0835)
    Lambda GC: 1.0225
    Mean Chi^2: 1.0233
    Intercept: 1.0064 (0.0087)
    Ratio: 0.2728 (0.3714)
    
    Genetic Covariance
    ------------------
    Total Observed scale gencov: 0.0349 (0.0229)
    Mean z1*z2: -0.0009
    Intercept: -0.0117 (0.0047)
    
    Genetic Correlation
    -------------------
    Genetic Correlation: 0.3657 (0.2797)
    Z-score: 1.3078
    P: 0.191
    
    Computing rg for phenotype 6/11
    Reading summary statistics from Crohns.deLange.sumstats.gz ...
    Read summary statistics for 1217311 SNPs.
    After merging with summary statistics, 1107424 SNPs remain.
    1067853 SNPs with valid alleles.
    
    Heritability of phenotype 6/11
    ------------------------------
    Total Observed scale h2: 0.4422 (0.0524)
    Lambda GC: 1.2564
    Mean Chi^2: 1.4482
    Intercept: 1.0975 (0.0153)
    Ratio: 0.2175 (0.0341)
    
    Genetic Covariance
    ------------------
    Total Observed scale gencov: 0.0126 (0.0158)
    Mean z1*z2: 0.0107
    Intercept: 0.0014 (0.0063)
    
    Genetic Correlation
    -------------------
    Genetic Correlation: 0.0693 (0.0883)
    Z-score: 0.7844
    P: 0.4328
    
    Computing rg for phenotype 7/11
    Reading summary statistics from MS.Beecham.sumstats.gz ...
    Read summary statistics for 1217311 SNPs.
    After merging with summary statistics, 1107424 SNPs remain.
    46758 SNPs with valid alleles.
    WARNING: number of SNPs less than 200k; this is almost always bad.
    
    Heritability of phenotype 7/11
    ------------------------------
    Total Observed scale h2: 1.5156 (0.2219)
    Lambda GC: 1.4494
    Mean Chi^2: 2.1573
    Intercept: 1.0313 (0.0998)
    Ratio: 0.0271 (0.0862)
    
    Genetic Covariance
    ------------------
    Total Observed scale gencov: 0.0849 (0.0654)
    Mean z1*z2: 0.0818
    Intercept: -0.0232 (0.0267)
    
    Genetic Correlation
    -------------------
    Genetic Correlation: 0.1121 (0.0825)
    Z-score: 1.3585
    P: 0.1743
    
    Computing rg for phenotype 8/11
    Reading summary statistics from RA.Okada.sumstats.gz ...
    Read summary statistics for 1217311 SNPs.
    After merging with summary statistics, 1107424 SNPs remain.
    1099228 SNPs with valid alleles.
    
    Heritability of phenotype 8/11
    ------------------------------
    Total Observed scale h2: 0.1696 (0.0229)
    Lambda GC: 1.0466
    Mean Chi^2: 1.1445
    Intercept: 0.9484 (0.0117)
    Ratio < 0 (usually indicates GC correction).
    
    Genetic Covariance
    ------------------
    Total Observed scale gencov: 0.0466 (0.0125)
    Mean z1*z2: 0.0444
    Intercept: 0.0003 (0.0058)
    
    Genetic Correlation
    -------------------
    Genetic Correlation: 0.4135 (0.0929)
    Z-score: 4.4531
    P: 8.4652e-06
    
    Computing rg for phenotype 9/11
    Reading summary statistics from SLE.Julia.sumstats.gz ...
    Read summary statistics for 1217311 SNPs.
    After merging with summary statistics, 1107424 SNPs remain.
    1056030 SNPs with valid alleles.
    
    Heritability of phenotype 9/11
    ------------------------------
    Total Observed scale h2: 0.4178 (0.0645)
    Lambda GC: 1.1811
    Mean Chi^2: 1.2485
    Intercept: 1.1006 (0.0121)
    Ratio: 0.4047 (0.0485)
    
    Genetic Covariance
    ------------------
    Total Observed scale gencov: -0.0373 (0.0202)
    Mean z1*z2: -0.033
    Intercept: -0.0138 (0.0063)
    
    Genetic Correlation
    -------------------
    Genetic Correlation: -0.2126 (0.1185)
    Z-score: -1.7942
    P: 0.0728
    
    Computing rg for phenotype 10/11
    Reading summary statistics from T1D.OnengutGumuscu.sumstats.gz ...
    Read summary statistics for 1217311 SNPs.
    After merging with summary statistics, 1107424 SNPs remain.
    44608 SNPs with valid alleles.
    WARNING: number of SNPs less than 200k; this is almost always bad.
    
    Heritability of phenotype 10/11
    -------------------------------
    Total Observed scale h2: 2.1786 (0.5807)
    Lambda GC: 1.641
    Mean Chi^2: 2.8979
    Intercept: 1.3416 (0.2462)
    Ratio: 0.18 (0.1297)
    
    Genetic Covariance
    ------------------
    Total Observed scale gencov: 0.3856 (0.1396)
    Mean z1*z2: 0.36
    Intercept: -0.0212 (0.043)
    
    Genetic Correlation
    -------------------
    Genetic Correlation: 0.4356 (0.1066)
    Z-score: 4.087
    P: 4.3707e-05
    
    Computing rg for phenotype 11/11
    Reading summary statistics from UC.deLange.sumstats.gz ...
    Read summary statistics for 1217311 SNPs.
    After merging with summary statistics, 1107424 SNPs remain.
    1067835 SNPs with valid alleles.
    
    Heritability of phenotype 11/11
    -------------------------------
    Total Observed scale h2: 0.2335 (0.0256)
    Lambda GC: 1.2038
    Mean Chi^2: 1.3086
    Intercept: 1.101 (0.0138)
    Ratio: 0.3274 (0.0449)
    
    Genetic Covariance
    ------------------
    Total Observed scale gencov: 0.0267 (0.011)
    Mean z1*z2: 0.0119
    Intercept: -0.0097 (0.0052)
    
    Genetic Correlation
    -------------------
    Genetic Correlation: 0.2015 (0.0851)
    Z-score: 2.3671
    P: 0.0179
    
    
    Summary of Genetic Correlation Results
                      p1                              p2      rg      se       z           p  h2_obs  h2_obs_se  h2_int  h2_int_se  gcov_int  gcov_int_se
     MG.Chia.sumstats.gz           AD.Jansen.sumstats.gz  0.0091  0.1230  0.0741  9.4092e-01  0.0075     0.0049  1.0570     0.0592    0.0050       0.0047
     MG.Chia.sumstats.gz            PD.Nalls.sumstats.gz  0.0032  0.0883  0.0359  9.7138e-01  0.0166     0.0017  0.9810     0.0076   -0.0029       0.0053
     MG.Chia.sumstats.gz      ALS.VanRheenen.sumstats.gz  0.2858  0.1800  1.5881  1.1227e-01  0.0485     0.0131  0.9941     0.0062    0.0122       0.0047
     MG.Chia.sumstats.gz            LBD.Chia.sumstats.gz  0.3657  0.2797  1.3078  1.9095e-01  0.1293     0.0835  1.0064     0.0087   -0.0117       0.0047
     MG.Chia.sumstats.gz      Crohns.deLange.sumstats.gz  0.0693  0.0883  0.7844  4.3279e-01  0.4422     0.0524  1.0975     0.0153    0.0014       0.0063
     MG.Chia.sumstats.gz          MS.Beecham.sumstats.gz  0.1121  0.0825  1.3585  1.7430e-01  1.5156     0.2219  1.0313     0.0998   -0.0232       0.0267
     MG.Chia.sumstats.gz            RA.Okada.sumstats.gz  0.4135  0.0929  4.4531  8.4652e-06  0.1696     0.0229  0.9484     0.0117    0.0003       0.0058
     MG.Chia.sumstats.gz           SLE.Julia.sumstats.gz -0.2126  0.1185 -1.7942  7.2775e-02  0.4178     0.0645  1.1006     0.0121   -0.0138       0.0063
     MG.Chia.sumstats.gz  T1D.OnengutGumuscu.sumstats.gz  0.4356  0.1066  4.0870  4.3707e-05  2.1786     0.5807  1.3416     0.2462   -0.0212       0.0430
     MG.Chia.sumstats.gz          UC.deLange.sumstats.gz  0.2015  0.0851  2.3671  1.7926e-02  0.2335     0.0256  1.1010     0.0138   -0.0097       0.0052
    
    Analysis finished at Tue Apr  6 13:00:48 2021
    Total time elapsed: 1.0m:10.63s


    /gpfs/gsfs11/users/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/LDSC/ldsc/ldscore/sumstats.py:532: FutureWarning: Method .as_matrix will be removed in a future version. Use .values instead.
      ref_ld = sumstats.as_matrix(columns=ref_ld_cnames)
    /gpfs/gsfs11/users/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/LDSC/ldsc/ldscore/irwls.py:161: FutureWarning: `rcond` parameter will change to the default of machine precision times ``max(M, N)`` where M and N are the input matrix dimensions.
    To use the future default and silence this warning we advise to pass `rcond=None`, to keep using the old, explicitly pass `rcond=-1`.
      coef = np.linalg.lstsq(x, y)


### Create pretty genetic correlation table

Note:
if need to plot correlation plot for larger dataset:
`https://github.com/mkanai/ldsc-corrplot-rg.git`


```bash
%%bash
## prep file for plotting

module load R/3.5.2
R --vanilla --no-save

require(data.table)
require(tidyverse)
library(stringr)
library("ggrepel")
require("data.table")
library(viridis)
library(gridExtra)
library(grid)
library(gt)

data <- fread("GeneticCorrelation.MG.vs.Neuro.AutoImmune.txt",header=T)
data$q <- p.adjust(data$p, method = "fdr", n = length(data$p))
data$rg <- format(round(data$rg,4),nsmall=4)
data$se <- format(round(data$se,4),nsmall=4)
data$z <- format(round(data$z,4),nsmall=4)
data1 <- data %>% select(p1_category,p1,p2_category,p2,rg,se,z,p,q) %>%
         rename(`Trait 1` = p1,
                `Trait 2` = p2,
                `Trait 1 category` = p1_category,
                `Trait 2 category` = p2_category,
                `Genetic correlation (rg)` = rg,
                 SE = se,
                 Z = z,
                 P = p,
                 `P(FDR adjusted)` = q)
write.table(data1,"GeneticCorrelation.MG.vs.Neuro.AutoImmune.forPlotting.txt",quote=F,sep="\t",row.names=F,col.names=T)

plot <-   data1 %>%
          select(`Trait 1`,`Trait 2`,`Trait 2 category`,`Genetic correlation (rg)`,SE,Z,P,`P(FDR adjusted)`) %>%
          gt() %>%
          tab_header(title = md("**Genetic correlation: MG vs neurological and autoimmune diseases**"),
                     subtitle = NULL) %>%
          fmt_scientific(columns = vars(P,`P(FDR adjusted)`),decimals = 2) %>%
          cols_align(align = "center", columns = vars(`Trait 2 category`,`Genetic correlation (rg)`,SE,Z,P,`P(FDR adjusted)`)) %>%
          cols_width(vars(`Trait 1`) ~ px(140),
                     vars(`Trait 2 category`) ~ px(160),
                     vars(`Genetic correlation (rg)`) ~ px(80),
                     vars(SE) ~ px(80),
                     vars(Z) ~ px(80),
                     vars(P) ~ px(110),
                     vars(`P(FDR adjusted)`) ~ px(110)) %>%
          tab_options(column_labels.background.color = "#EFFBFC") %>%
          tab_style(style = list(cell_text(weight = "bold")),
                    locations = cells_column_labels(columns = everything())) %>%
          gtsave("TableXX.GeneticCorrelation.rtf")
plot


```

    
    R version 3.5.2 (2018-12-20) -- "Eggshell Igloo"
    Copyright (C) 2018 The R Foundation for Statistical Computing
    Platform: x86_64-pc-linux-gnu (64-bit)
    
    R is free software and comes with ABSOLUTELY NO WARRANTY.
    You are welcome to redistribute it under certain conditions.
    Type 'license()' or 'licence()' for distribution details.
    
    R is a collaborative project with many contributors.
    Type 'contributors()' for more information and
    'citation()' on how to cite R or R packages in publications.
    
    Type 'demo()' for some demos, 'help()' for on-line help, or
    'help.start()' for an HTML browser interface to help.
    Type 'q()' to quit R.
    
    > 
    > require(data.table)
    > require(tidyverse)
    > library(stringr)
    > library("ggrepel")
    > require("data.table")
    > library(viridis)
    > library(gridExtra)
    > library(grid)
    > library(gt)
    > 
    > data <- fread("GeneticCorrelation.MG.vs.Neuro.AutoImmune.txt",header=T)
    > data$q <- p.adjust(data$p, method = "fdr", n = length(data$p))
    > data$rg <- format(round(data$rg,4),nsmall=4)
    > data$se <- format(round(data$se,4),nsmall=4)
    > data$z <- format(round(data$z,4),nsmall=4)
    > data1 <- data %>% select(p1_category,p1,p2_category,p2,rg,se,z,p,q) %>%
    +          rename(`Trait 1` = p1,
    +                 `Trait 2` = p2,
    +                 `Trait 1 category` = p1_category,
    +                 `Trait 2 category` = p2_category,
    +                 `Genetic correlation (rg)` = rg,
    +                  SE = se,
    +                  Z = z,
    +                  P = p,
    +                  `P(FDR adjusted)` = q)
    > write.table(data1,"GeneticCorrelation.MG.vs.Neuro.AutoImmune.forPlotting.txt",quote=F,sep="\t",row.names=F,col.names=T)
    > 
    > plot <-   data1 %>%
    +           select(`Trait 1`,`Trait 2`,`Trait 2 category`,`Genetic correlation (rg)`,SE,Z,P,`P(FDR adjusted)`) %>%
    +           gt() %>%
    +           tab_header(title = md("**Genetic correlation: MG vs neurological and autoimmune diseases**"),
    +                      subtitle = NULL) %>%
    +           fmt_scientific(columns = vars(P,`P(FDR adjusted)`),decimals = 2) %>%
    +           cols_align(align = "center", columns = vars(`Trait 2 category`,`Genetic correlation (rg)`,SE,Z,P,`P(FDR adjusted)`)) %>%
    +           cols_width(vars(`Trait 1`) ~ px(140),
    +                      vars(`Trait 2 category`) ~ px(160),
    +                      vars(`Genetic correlation (rg)`) ~ px(80),
    +                      vars(SE) ~ px(80),
    +                      vars(Z) ~ px(80),
    +                      vars(P) ~ px(110),
    +                      vars(`P(FDR adjusted)`) ~ px(110)) %>%
    +           tab_options(column_labels.background.color = "#EFFBFC") %>%
    +           tab_style(style = list(cell_text(weight = "bold")),
    +                     locations = cells_column_labels(columns = everything())) %>%
    +           gtsave("TableXX.GeneticCorrelation.rtf")
    > plot
    NULL
    > 


    [+] Loading gcc  7.3.0  ... 
    [+] Loading GSL 2.4 for GCC 7.2.0 ... 
    [-] Unloading gcc  7.3.0  ... 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading openmpi 3.0.2  for GCC 7.3.0 
    [+] Loading ImageMagick  7.0.8  on cn1809 
    [+] Loading HDF5  1.10.4 
    [+] Loading pandoc  2.13  on cn1809 
    [+] Loading R 3.5.2 
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()
    Loading required package: viridisLite
    
    Attaching package: 'gridExtra'
    
    The following object is masked from 'package:dplyr':
    
        combine
    

