# Meta Analysis MG: USmerged,Itals,UKBB (imputed hg38)
# Overall MG (female cases vs all controls)

**Start date:** 04-05-2021

**End date:** 04-05-2021

**Analysed by:** Ruth Chia

**Working directory:** `/data/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/USmerged.Itals.UKBB.hg38_Females.SexStrat`

___


<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Summary-stats-file-needed" data-toc-modified-id="Summary-stats-file-needed-1">Summary stats file needed</a></span></li><li><span><a href="#What-I-need-to-do" data-toc-modified-id="What-I-need-to-do-2">What I need to do</a></span></li><li><span><a href="#Subset-US-and-Itals-to-contain-only-index-vars-of-interest" data-toc-modified-id="Subset-US-and-Itals-to-contain-only-index-vars-of-interest-3">Subset US and Itals to contain only index vars of interest</a></span></li><li><span><a href="#Run-metal-(index-vars-from-meta-MG)" data-toc-modified-id="Run-metal-(index-vars-from-meta-MG)-4">Run metal (index vars from meta MG)</a></span></li><li><span><a href="#Include-maf-cases-and-controls-in-meta-results" data-toc-modified-id="Include-maf-cases-and-controls-in-meta-results-5">Include maf cases and controls in meta results</a></span></li></ul></div>

## Summary stats file needed
1. to subset from MG.USmerged: `/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Analysis.GLM.SexStratified.hg38/US.JointPostImputation/FemalesCases.vs.AllControls/toMeta.USmerged.FemaleMG.AllControls.hg38.glm.MetaMGindexVars.tab`
2. to subset from MG.Itals: `/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Analysis.GLM.SexStratified.hg38/FemalesCases.vs.AllControls.rerun/toMeta.Itals.FemaleMG.AllControls.hg38.glm.MetaMGindexVars.tab`
3. UKBB: `/data/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/USmerged.Itals.hg38/UKBB.replication/toMeta.MG.Females.UKBB.NEW.matchedRatio20.noDups.UNRELATED.hg38.DiscoverySignifHits.tab`
    
## What I need to do
1. Prepare summary stats for meta analysis
2. Run metal

## Subset US and Itals to contain only index vars of interest


```python
!cut -f4 /data/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/USmerged.Itals.hg38/UKBB.replication/toMeta.MG.Females.UKBB.NEW.matchedRatio20.noDups.UNRELATED.hg38.DiscoverySignifHits.tab > IndexVars.txt
!grep -Ewf IndexVars.txt /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Analysis.GLM.SexStratified.hg38/US.JointPostImputation/FemalesCases.vs.AllControls/toMeta.USmerged.FemaleMG.AllControls.hg38.glm.MetaMGindexVars.tab > toMeta.USmerged.MG.Females.hg38.DiscoverySignifHits.tab
!grep -Ewf IndexVars.txt /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Analysis.GLM.SexStratified.hg38/FemalesCases.vs.AllControls.rerun/toMeta.Itals.FemaleMG.AllControls.hg38.glm.MetaMGindexVars.tab > toMeta.Itals.MG.Females.hg38.DiscoverySignifHits.tab
!wc -l toMeta.*.MG*.hg38.DiscoverySignifHits.tab
```

       9 toMeta.Itals.MG.Females.hg38.DiscoverySignifHits.tab
       9 toMeta.USmerged.MG.Females.hg38.DiscoverySignifHits.tab
      18 total


## Run metal (index vars from meta MG)

metal script: `metal_MG.US.Itals.newUKBB.hg38.v1.txt`

```bash
# THIS SCRIPT EXECUTES AN ANALYSIS OF THREE STUDIES
# (i.e. MG.US and MG.Itals filtered for --maf 0.0001, Rsq>03)

# THE RESULTS FOR EACH STUDY ARE STORED IN FILES:
# toMeta.USmerged.MG.Females.hg38.DiscoverySignifHits.tab
# toMeta.Itals.MG.Females.hg38.DiscoverySignifHits.tab
# /data/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/USmerged.Itals.hg38/UKBB.replication/toMeta.MG.Females.UKBB.NEW.matchedRatio20.noDups.UNRELATED.hg38.DiscoverySignifHits.tab

SCHEME  STDERR
AVERAGEFREQ ON
MINMAXFREQ ON

# LOAD THE FIRST INPUT FILE
# UNCOMMENT THE NEXT LINE TO ENABLE GenomicControl CORRECTION
# GENOMICCONTROL ON

# === DESCRIBE AND PROCESS THE FIRST INPUT FILE ===
MARKER marker
ALLELE EffectAllele OtherAllele
FREQ   Freq_EffectAllele.CASE
EFFECT BETA
STDERR SE
PVALUE P
#WEIGHT OBS_CT
PROCESS toMeta.USmerged.MG.Females.hg38.DiscoverySignifHits.tab

# === DESCRIBE AND PROCESS ADDITIONAL INPUT FILE ===
PROCESS toMeta.Itals.MG.Females.hg38.DiscoverySignifHits.tab
PROCESS /data/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/USmerged.Itals.hg38/UKBB.replication/toMeta.MG.Females.UKBB.NEW.matchedRatio20.noDups.UNRELATED.hg38.DiscoverySignifHits.tab

OUTFILE META_USmerged.Itals.newUKBB.MG.Females.UNRELATED.hg38.MetaMGindexVars.FreqCases.glm .tbl
ANALYZE HETEROGENEITY
```




```bash
%%bash
module load metal
metal metal_MG.US.Itals.newUKBB.hg38.v1.txt
```

    MetaAnalysis Helper - (c) 2007 - 2009 Goncalo Abecasis
    This version released on 2018-08-28
    
    # This program faciliates meta-analysis of genome-wide association studies.
    # Commonly used commands are listed below:
    #
    # Options for describing input files ...
    #   SEPARATOR        [WHITESPACE|COMMA|BOTH|TAB] (default = WHITESPACE)
    #   COLUMNCOUNTING   [STRICT|LENIENT]            (default = 'STRICT')
    #   MARKERLABEL      [LABEL]                     (default = 'MARKER')
    #   ALLELELABELS     [LABEL1 LABEL2]             (default = 'ALLELE1','ALLELE2')
    #   EFFECTLABEL      [LABEL|log(LABEL)]          (default = 'EFFECT')
    #   FLIP
    #
    # Options for filtering input files ...
    #   ADDFILTER        [LABEL CONDITION VALUE]     (example = ADDFILTER N > 10)
    #                    (available conditions are <, >, <=, >=, =, !=, IN)
    #   REMOVEFILTERS
    #
    # Options for sample size weighted meta-analysis ...
    #   WEIGHTLABEL      [LABEL]                     (default = 'N')
    #   PVALUELABEL      [LABEL]                     (default = 'PVALUE')
    #   DEFAULTWEIGHT    [NUMBER]                    (default = 1.0)
    #   MINWEIGHT        [NUMBER]                    (default = 1.0)
    #
    # Options for inverse variance weighted meta-analysis ...
    #   STDERRLABEL      [LABEL]                     (default = 'STDERR')
    #   SCHEME           [SAMPLESIZE|STDERR]         (default = SAMPLESIZE)
    #
    # Options to enable tracking of allele frequencies ...
    #   AVERAGEFREQ      [ON|OFF]                    (default = OFF)
    #   MINMAXFREQ       [ON|OFF]                    (default = OFF)
    #   FREQLABEL        [LABEL]                     (default = 'FREQ')
    #
    # Options to enable tracking of user defined variables ...
    #   CUSTOMVARIABLE   [VARNAME]
    #   LABEL            [VARNAME] AS [HEADER]
    #
    # Options to enable tracking of chromosomes and positions ...
    #   TRACKPOSITIONS   [ON|OFF]                    (default = OFF
    #   CHROMOSOMELABEL  [LABEL]                     (default = 'CHROMOSOME')
    #   POSITIONLABEL    [LABEL]                     (default = 'POSITION')
    #
    # Options to enable explicit strand information ...
    #   USESTRAND        [ON|OFF]                    (default = OFF)
    #   STRANDLABEL      [LABEL]                     (default = 'STRAND')
    #
    # Automatic genomic control correction of input statistics ...
    #   GENOMICCONTROL   [ON|OFF|VALUE|LIST snps.txt](default = OFF)
    #
    # Options to account for samples overlap ...
    #   OVERLAP          [ON|OFF]                    (default = OFF)
    #   ZCUTOFF          [NUMBER]                    (default = 1.0)
    #
    # Options for general analysis control ...
    #   PROCESSFILE      [FILENAME]
    #   OUTFILE          [PREFIX SUFFIX]             (default = 'METAANALYSIS','.TBL')
    #   MAXWARNINGS      [NUMBER]                    (default = 20)
    #   VERBOSE          [ON|OFF]                    (default = 'OFF')
    #   LOGPVALUE        [ON|OFF]                    (default = 'OFF')
    #   ANALYZE          [HETEROGENEITY]
    #   CLEAR
    
    # Options for general run control ...
    #   SOURCE           [SCRIPTFILE]
    #   RETURN
    #   QUIT
    
    # Processing commands in metal_MG.US.Itals.newUKBB.hg38.v1.txt ...
    ## Meta-analysis will be based on effect sizes and their standard errors ...
    ## Averaging of allele frequencies enabled
    ## Tracking of extreme allele frequencies enabled
    ## Set marker header to marker ...
    ## Set allele headers to EffectAllele and OtherAllele ...
    ## Set frequency header to Freq_EffectAllele.CASE ...
    ## Set effect header to BETA ...
    ## Set standard error header to SE ...
    ## Set p-value header to P ...
    ###########################################################################
    ## Processing file 'toMeta.USmerged.MG.Females.hg38.DiscoverySignifHits.tab'
    ## Processed 8 markers ...
    
    ###########################################################################
    ## Processing file 'toMeta.Itals.MG.Females.hg38.DiscoverySignifHits.tab'
    ## Processed 8 markers ...
    
    ###########################################################################
    ## Processing file '/data/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/USmerged.Itals.hg38/UKBB.replication/toMeta.MG.Females.UKBB.NEW.matchedRatio20.noDups.UNRELATED.hg38.DiscoverySignifHits.tab'
    ## Processed 8 markers ...
    
    ## Set output file prefix and suffix to META_USmerged.Itals.newUKBB.MG.Females.UNRELATED.hg38.MetaMGindexVars.FreqCases.glm and .tbl ...
    ###########################################################################
    ## Running second pass analysis to evaluate heterogeneity...
    ## Processing file '/data/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/USmerged.Itals.hg38/UKBB.replication/toMeta.MG.Females.UKBB.NEW.matchedRatio20.noDups.UNRELATED.hg38.DiscoverySignifHits.tab'
    ## Processing file 'toMeta.Itals.MG.Females.hg38.DiscoverySignifHits.tab'
    ## Processing file 'toMeta.USmerged.MG.Females.hg38.DiscoverySignifHits.tab'
    
    ###########################################################################
    ## Executing meta-analysis ...
    ## Complete results will be stored in file 'META_USmerged.Itals.newUKBB.MG.Females.UNRELATED.hg38.MetaMGindexVars.FreqCases.glm1.tbl'
    ## Column descriptions will be stored in file 'META_USmerged.Itals.newUKBB.MG.Females.UNRELATED.hg38.MetaMGindexVars.FreqCases.glm1.tbl.info'
    ## Completed meta-analysis for 8 markers!
    ## Smallest p-value is 1.193e-13 at marker '6:31358836:G:A'
    
    ## Clearing all stored statistics ...
    # Clearing user defined filters ...


    [+] Loading metal  2018-08-28 



```python
import pandas as pd
import numpy as np

pd.read_csv("META_USmerged.Itals.newUKBB.MG.Females.UNRELATED.hg38.MetaMGindexVars.FreqCases.glm1.tbl",sep="\t").sort_values(by=['MarkerName'])
```




<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>MarkerName</th>
      <th>Allele1</th>
      <th>Allele2</th>
      <th>Freq1</th>
      <th>FreqSE</th>
      <th>MinFreq</th>
      <th>MaxFreq</th>
      <th>Effect</th>
      <th>StdErr</th>
      <th>P-value</th>
      <th>Direction</th>
      <th>HetISq</th>
      <th>HetChiSq</th>
      <th>HetDf</th>
      <th>HetPVal</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>2</th>
      <td>10:7410781:A:G</td>
      <td>a</td>
      <td>g</td>
      <td>0.7382</td>
      <td>0.0343</td>
      <td>0.7026</td>
      <td>0.8006</td>
      <td>-0.1931</td>
      <td>0.0540</td>
      <td>3.469000e-04</td>
      <td>--+</td>
      <td>42.9</td>
      <td>3.505</td>
      <td>2</td>
      <td>0.173300</td>
    </tr>
    <tr>
      <th>0</th>
      <td>11:95578258:T:C</td>
      <td>t</td>
      <td>c</td>
      <td>0.7956</td>
      <td>0.0215</td>
      <td>0.7857</td>
      <td>0.8468</td>
      <td>-0.1775</td>
      <td>0.0589</td>
      <td>2.594000e-03</td>
      <td>--+</td>
      <td>60.1</td>
      <td>5.009</td>
      <td>2</td>
      <td>0.081700</td>
    </tr>
    <tr>
      <th>5</th>
      <td>18:62342581:T:C</td>
      <td>t</td>
      <td>c</td>
      <td>0.4997</td>
      <td>0.0231</td>
      <td>0.4786</td>
      <td>0.5427</td>
      <td>-0.1358</td>
      <td>0.0473</td>
      <td>4.063000e-03</td>
      <td>--+</td>
      <td>71.7</td>
      <td>7.061</td>
      <td>2</td>
      <td>0.029290</td>
    </tr>
    <tr>
      <th>3</th>
      <td>1:113834946:G:A</td>
      <td>a</td>
      <td>g</td>
      <td>0.1155</td>
      <td>0.0312</td>
      <td>0.0593</td>
      <td>0.1474</td>
      <td>0.3006</td>
      <td>0.0780</td>
      <td>1.149000e-04</td>
      <td>+++</td>
      <td>0.0</td>
      <td>0.114</td>
      <td>2</td>
      <td>0.944500</td>
    </tr>
    <tr>
      <th>1</th>
      <td>2:174764492:G:A</td>
      <td>a</td>
      <td>g</td>
      <td>0.0491</td>
      <td>0.0139</td>
      <td>0.0264</td>
      <td>0.0625</td>
      <td>0.2321</td>
      <td>0.1153</td>
      <td>4.422000e-02</td>
      <td>++-</td>
      <td>0.0</td>
      <td>1.905</td>
      <td>2</td>
      <td>0.385800</td>
    </tr>
    <tr>
      <th>6</th>
      <td>6:31358836:G:A</td>
      <td>a</td>
      <td>g</td>
      <td>0.4089</td>
      <td>0.0566</td>
      <td>0.3448</td>
      <td>0.4679</td>
      <td>0.3649</td>
      <td>0.0492</td>
      <td>1.193000e-13</td>
      <td>++-</td>
      <td>86.3</td>
      <td>14.555</td>
      <td>2</td>
      <td>0.000691</td>
    </tr>
    <tr>
      <th>4</th>
      <td>6:32620936:T:C</td>
      <td>t</td>
      <td>c</td>
      <td>0.9687</td>
      <td>0.0099</td>
      <td>0.9607</td>
      <td>0.9838</td>
      <td>0.6449</td>
      <td>0.1386</td>
      <td>3.248000e-06</td>
      <td>+++</td>
      <td>50.0</td>
      <td>4.001</td>
      <td>2</td>
      <td>0.135300</td>
    </tr>
    <tr>
      <th>7</th>
      <td>8:108247378:T:C</td>
      <td>t</td>
      <td>c</td>
      <td>0.9915</td>
      <td>0.0029</td>
      <td>0.9892</td>
      <td>1.0000</td>
      <td>-0.5632</td>
      <td>0.3197</td>
      <td>7.816000e-02</td>
      <td>--+</td>
      <td>0.0</td>
      <td>0.134</td>
      <td>2</td>
      <td>0.935300</td>
    </tr>
  </tbody>
</table>
</div>



## Include maf cases and controls in meta results

script to reformat, add maf for cases and controls before plotting: `ReformatMetaResults.addMAF.v4.R`

```Rscript

#!/usr/bin/env Rscript

args <- commandArgs(trailingOnly=TRUE)
if (length(args) != 5) {
  stop("USAGE: Rscript ReformatMetaResults.addMAF.v3.R args[1] args[2] args[3] args[4] args[5]
  where args[1] = metaResults
        args[2] = toMetaInput.US
        args[3] = toMetaInput.Itals
        args[4] = toMetaInput.UKBB
        args[5] = outputFileName")
}

# Load libraries
require(data.table)
require(tidyverse)


# Set variables
metaResults <- args[1]
toMetaInput.US <- args[2]
toMetaInput.Itals <- args[3]
toMetaInput.UKBB <- args[4]
outputFileName <- args[5]

# Read in files and reformat meta
meta <- fread(metaResults, header=T) %>%
        mutate(snp = MarkerName) %>%
        separate(snp,c("CHROM","POS","OA","EA"),sep=":") %>%
        rename(P = `P-value`)

## format meta results such that the effect allele matches original allele designation 
## given in MarkerName as submitted in the original files used for meta analysis 
## i.e. chr:pos:OtherAllele:EffectAllele
meta$OA <- tolower(meta$OA)
meta$EA <- tolower(meta$EA)
meta$EffectAllele <- ifelse(meta$EA == meta$Allele1, meta$Allele1, meta$Allele2)
meta$OtherAllele <- ifelse(meta$EA == meta$Allele1, meta$Allele2, meta$Allele1)
meta$maf_EA.cases <- ifelse(meta$EA == meta$Allele1, meta$Freq1, 1 - meta$Freq1)
meta$beta <- ifelse(meta$EffectAllele == meta$Allele1, meta$Effect, meta$Effect*-1)

data <- meta %>%
        mutate(CHROM = as.numeric(gsub("chr","",CHROM), POS = as.numeric(POS))) %>%
        filter(POS != is.na(POS)) %>%
        select(CHROM,POS,MarkerName,EffectAllele,OtherAllele,maf_EA.cases,beta,StdErr,P,Direction,HetISq) %>%
        arrange(CHROM,POS)

dim(meta)
dim(data)

# Get frequency information
us <- fread(toMetaInput.US,header=T) %>%
      select(marker,Freq_EffectAllele,Freq_EffectAllele.CASE,Freq_EffectAllele.CTRL,
             ALLELE_CT,CASE_ALLELE_CT,CTRL_ALLELE_CT,
             EffectAllele_CT,EffectAllele_CT.CASE,EffectAllele_CT.CTRL)
colnames(us)[2:10] <- paste(colnames(us)[2:10],"_US",sep="")

itals <- fread(toMetaInput.Itals,header=T) %>%
         select(marker,Freq_EffectAllele,Freq_EffectAllele.CASE,Freq_EffectAllele.CTRL,
                ALLELE_CT,CASE_ALLELE_CT,CTRL_ALLELE_CT,
                EffectAllele_CT,EffectAllele_CT.CASE,EffectAllele_CT.CTRL)
colnames(itals)[2:10] <- paste(colnames(itals)[2:10],"_Italy",sep="")

ukbb <- fread(toMetaInput.UKBB,header=T) %>%
         select(marker,Freq_EffectAllele,Freq_EffectAllele.CASE,Freq_EffectAllele.CTRL,
                ALLELE_CT,CASE_ALLELE_CT,CTRL_ALLELE_CT,
                EffectAllele_CT,EffectAllele_CT.CASE,EffectAllele_CT.CTRL)
colnames(ukbb)[2:10] <- paste(colnames(ukbb)[2:10],"_UKBB",sep="")


both0 <- merge(us,itals,by="marker")
both <- merge(both0,ukbb,by="marker")

dim(us)
dim(itals)
dim(ukbb)
dim(both)

both$Freq_EffectAllele <- (both$Freq_EffectAllele_US + both$Freq_EffectAllele_Italy + both$Freq_EffectAllele_UKBB)/3
both$Freq_EffectAllele.CASE <- (both$Freq_EffectAllele.CASE_US + both$Freq_EffectAllele.CASE_Italy + both$Freq_EffectAllele.CASE_UKBB)/3
both$Freq_EffectAllele.CTRL <- (both$Freq_EffectAllele.CTRL_US + both$Freq_EffectAllele.CTRL_Italy + both$Freq_EffectAllele.CTRL_UKBB)/3

both$ALLELE_CT <- both$ALLELE_CT_US + both$ALLELE_CT_Italy + both$ALLELE_CT_UKBB
both$ALLELE_CT.CASE <- both$CASE_ALLELE_CT_US + both$CASE_ALLELE_CT_Italy + both$CASE_ALLELE_CT_UKBB
both$ALLELE_CT.CTRL <- both$CTRL_ALLELE_CT_US + both$CTRL_ALLELE_CT_Italy + both$CTRL_ALLELE_CT_UKBB

both$EffectAllele_CT <- both$EffectAllele_CT_US + both$EffectAllele_CT_Italy + both$EffectAllele_CT_UKBB
both$EffectAllele_CT.CASE <- both$EffectAllele_CT.CASE_US + both$EffectAllele_CT.CASE_Italy + both$EffectAllele_CT.CASE_UKBB
both$EffectAllele_CT.CTRL <- both$EffectAllele_CT.CTRL_US + both$EffectAllele_CT.CTRL_Italy + both$EffectAllele_CT.CTRL_UKBB

both$maf_EA <- both$EffectAllele_CT/both$ALLELE_CT
both$maf_EA.CASE <- both$EffectAllele_CT.CASE/both$ALLELE_CT.CASE
both$maf_EA.CTRL <- both$EffectAllele_CT.CTRL/both$ALLELE_CT.CTRL
write.table(both,"MAF.USmerged.Itals.txt",sep="\t",quote=F,row.names=F,col.names=T)
            
freq <- both %>%
        select(marker,Freq_EffectAllele,Freq_EffectAllele.CASE,Freq_EffectAllele.CTRL,
              ALLELE_CT,ALLELE_CT.CASE,ALLELE_CT.CTRL,
              EffectAllele_CT,EffectAllele_CT.CASE,EffectAllele_CT.CTRL,
              maf_EA,maf_EA.CASE,maf_EA.CTRL)


# merge formatted meta results with frequency file
tmp <- merge(data,freq,by.x="MarkerName",by.y="marker")
data2 <- tmp %>% 
         select(CHROM,POS,MarkerName,EffectAllele,OtherAllele,
                maf_EA,maf_EA.CASE,maf_EA.CTRL,beta,StdErr,P,Direction,HetISq)

write.table(data2,paste(args[5],".TIDY.txt",sep=""),quote=F,row.names=F,col.names=T,sep="\t")
```


```bash
%%bash
module load R/3.5.2
DIR="/data/NDRS_LNG/MyastheniaGravis/updated.April2020"

Rscript ReformatMetaResults.addMAF.v4.R \
META_USmerged.Itals.newUKBB.MG.Females.UNRELATED.hg38.MetaMGindexVars.FreqCases.glm1.tbl \
toMeta.USmerged.MG.Females.hg38.DiscoverySignifHits.tab \
toMeta.Itals.MG.Females.hg38.DiscoverySignifHits.tab \
$DIR/MetaAnalysis.hg38.ByCohort.rerun/USmerged.Itals.hg38/UKBB.replication/toMeta.MG.Females.UKBB.NEW.matchedRatio20.noDups.UNRELATED.hg38.DiscoverySignifHits.tab \
META_MG.Females.USmerged.Itals.newUKBB.UNRELATED.hg38.Rsq03

```

    [1]  8 23
    [1]  8 11
    [1]  8 10
    [1]  8 10
    [1]  8 10
    [1]  8 28


    [+] Loading gcc  7.3.0  ... 
    [+] Loading GSL 2.4 for GCC 7.2.0 ... 
    [-] Unloading gcc  7.3.0  ... 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading openmpi 3.0.2  for GCC 7.3.0 
    [+] Loading ImageMagick  7.0.8  on cn1940 
    [+] Loading HDF5  1.10.4 
    [+] Loading pandoc  2.13  on cn1940 
    [+] Loading R 3.5.2 
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()



```python
import pandas as pd
import numpy as np

pd.read_csv("META_MG.Females.USmerged.Itals.newUKBB.UNRELATED.hg38.Rsq03.TIDY.txt",sep="\t").sort_values(by=['CHROM','POS'])
```




<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>CHROM</th>
      <th>POS</th>
      <th>MarkerName</th>
      <th>EffectAllele</th>
      <th>OtherAllele</th>
      <th>maf_EA</th>
      <th>maf_EA.CASE</th>
      <th>maf_EA.CTRL</th>
      <th>beta</th>
      <th>StdErr</th>
      <th>P</th>
      <th>Direction</th>
      <th>HetISq</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>3</th>
      <td>1</td>
      <td>113834946</td>
      <td>1:113834946:G:A</td>
      <td>a</td>
      <td>g</td>
      <td>0.095504</td>
      <td>0.098865</td>
      <td>0.095422</td>
      <td>0.3006</td>
      <td>0.0780</td>
      <td>1.149000e-04</td>
      <td>+++</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>4</th>
      <td>2</td>
      <td>174764492</td>
      <td>2:174764492:G:A</td>
      <td>a</td>
      <td>g</td>
      <td>0.035552</td>
      <td>0.046897</td>
      <td>0.035275</td>
      <td>0.2321</td>
      <td>0.1153</td>
      <td>4.422000e-02</td>
      <td>++-</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>5</th>
      <td>6</td>
      <td>31358836</td>
      <td>6:31358836:G:A</td>
      <td>a</td>
      <td>g</td>
      <td>0.354311</td>
      <td>0.399263</td>
      <td>0.353217</td>
      <td>0.3649</td>
      <td>0.0492</td>
      <td>1.193000e-13</td>
      <td>++-</td>
      <td>86.3</td>
    </tr>
    <tr>
      <th>6</th>
      <td>6</td>
      <td>32620936</td>
      <td>6:32620936:T:C</td>
      <td>c</td>
      <td>t</td>
      <td>0.058975</td>
      <td>0.026963</td>
      <td>0.059754</td>
      <td>-0.6449</td>
      <td>0.1386</td>
      <td>3.248000e-06</td>
      <td>+++</td>
      <td>50.0</td>
    </tr>
    <tr>
      <th>7</th>
      <td>8</td>
      <td>108247378</td>
      <td>8:108247378:T:C</td>
      <td>c</td>
      <td>t</td>
      <td>0.003570</td>
      <td>0.006626</td>
      <td>0.003496</td>
      <td>0.5632</td>
      <td>0.3197</td>
      <td>7.816000e-02</td>
      <td>--+</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>0</th>
      <td>10</td>
      <td>7410781</td>
      <td>10:7410781:A:G</td>
      <td>g</td>
      <td>a</td>
      <td>0.213330</td>
      <td>0.263009</td>
      <td>0.212121</td>
      <td>0.1931</td>
      <td>0.0540</td>
      <td>3.469000e-04</td>
      <td>--+</td>
      <td>42.9</td>
    </tr>
    <tr>
      <th>1</th>
      <td>11</td>
      <td>95578258</td>
      <td>11:95578258:T:C</td>
      <td>c</td>
      <td>t</td>
      <td>0.174194</td>
      <td>0.203406</td>
      <td>0.173483</td>
      <td>0.1775</td>
      <td>0.0589</td>
      <td>2.594000e-03</td>
      <td>--+</td>
      <td>60.1</td>
    </tr>
    <tr>
      <th>2</th>
      <td>18</td>
      <td>62342581</td>
      <td>18:62342581:T:C</td>
      <td>c</td>
      <td>t</td>
      <td>0.463851</td>
      <td>0.499633</td>
      <td>0.462980</td>
      <td>0.1358</td>
      <td>0.0473</td>
      <td>4.063000e-03</td>
      <td>--+</td>
      <td>71.7</td>
    </tr>
  </tbody>
</table>
</div>


