# Meta Analysis MG_Females: US.merged and Itals (imputed hg38)

**Start date:** 03-15-2021

**Updated date:** 03-15-2021

**Analysed by:** Ruth Chia

**Working directory:** `/data/NDRS_LNG/MyastheniaGravis/updated.April2020/MetaAnalysis.hg38.ByCohort.rerun/SexStratified.GLM.hg38/FemaleCases.vs.AllControls`
___


<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Summary-stats-file-needed" data-toc-modified-id="Summary-stats-file-needed-1">Summary stats file needed</a></span></li><li><span><a href="#What-I-need-to-do" data-toc-modified-id="What-I-need-to-do-2">What I need to do</a></span></li><li><span><a href="#Run-metal" data-toc-modified-id="Run-metal-3">Run metal</a></span></li><li><span><a href="#Format-metal-results-to-match-other-summ-stats" data-toc-modified-id="Format-metal-results-to-match-other-summ-stats-4">Format metal results to match other summ stats</a></span></li></ul></div>

## Summary stats file needed
1. MG.USmerged: `/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Analysis.GLM.SexStratified.hg38/US.JointPostImputation/FemalesCases.vs.AllControls/toMeta.USmerged.FemaleMG.AllControls.hg38.glm.MetaMGindexVars.tab`
2. MG.Itals: `/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Analysis.GLM.SexStratified.hg38/FemalesCases.vs.AllControls.rerun/toMeta.Itals.FemaleMG.AllControls.hg38.glm.MetaMGindexVars.tab`
    
## What I need to do
1. Prepare summary stats for meta analysis
2. Run metal


## Run metal

metal script: `metal_MG.US.Itals.hg38.v1.txt`

```bash
# THIS SCRIPT EXECUTES AN ANALYSIS OF TWO STUDIES
# (i.e. MG.US and MG.Itals filtered for --maf 0.0001, Rsq>03)

# THE RESULTS FOR EACH STUDY ARE STORED IN FILES:
# /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Analysis.GLM.SexStratified.hg38/US.JointPostImputation/FemalesCases.vs.AllControls/toMeta.USmerged.FemaleMG.AllControls.hg38.glm.MetaMGindexVars.tab
# /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Analysis.GLM.SexStratified.hg38/FemalesCases.vs.AllControls.rerun/toMeta.Itals.FemaleMG.AllControls.hg38.glm.MetaMGindexVars.tab

SCHEME  STDERR
AVERAGEFREQ ON
MINMAXFREQ ON

# LOAD THE FIRST INPUT FILE
# UNCOMMENT THE NEXT LINE TO ENABLE GenomicControl CORRECTION
# GENOMICCONTROL ON

# === DESCRIBE AND PROCESS THE FIRST INPUT FILE ===
MARKER marker
ALLELE EffectAllele OtherAllele
FREQ   Freq_EffectAllele.CASE
EFFECT BETA
STDERR SE
PVALUE P
#WEIGHT OBS_CT
PROCESS /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Analysis.GLM.SexStratified.hg38/US.JointPostImputation/FemalesCases.vs.AllControls/toMeta.USmerged.FemaleMG.AllControls.hg38.glm.MetaMGindexVars.tab

# === DESCRIBE AND PROCESS ADDITIONAL INPUT FILE ===
PROCESS /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Analysis.GLM.SexStratified.hg38/FemalesCases.vs.AllControls.rerun/toMeta.Itals.FemaleMG.AllControls.hg38.glm.MetaMGindexVars.tab

OUTFILE META_USmerged.Itals.FemaleMG.AllControls.UNRELATED.hg38.Rsq03MAF00001.FreqCases.glm .tbl
ANALYZE HETEROGENEITY
```

metal script: `metal_MG.US.Itals.hg38.v2.txt`

```bash
# THIS SCRIPT EXECUTES AN ANALYSIS OF TWO STUDIES
# (i.e. MG.US and MG.Itals filtered for --maf 0.0001, Rsq>03)

# THE RESULTS FOR EACH STUDY ARE STORED IN FILES:
# /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Analysis.GLM.SexStratified.hg38/US.JointPostImputation/FemalesCases.vs.AllControls/toMeta.USmerged.FemaleMG.AllControls.hg38.glm.MetaMGindexVars.tab
# /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Analysis.GLM.SexStratified.hg38/FemalesCases.vs.AllControls.rerun/toMeta.Itals.FemaleMG.AllControls.hg38.glm.MetaMGindexVars.tab

SCHEME  STDERR
AVERAGEFREQ ON
MINMAXFREQ ON

# LOAD THE FIRST INPUT FILE
# UNCOMMENT THE NEXT LINE TO ENABLE GenomicControl CORRECTION
# GENOMICCONTROL ON

# === DESCRIBE AND PROCESS THE FIRST INPUT FILE ===
MARKER marker
ALLELE EffectAllele OtherAllele
FREQ   Freq_EffectAllele.CTRL
EFFECT BETA
STDERR SE
PVALUE P
#WEIGHT OBS_CT
PROCESS /data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Analysis.GLM.SexStratified.hg38/US.JointPostImputation/FemalesCases.vs.AllControls/toMeta.USmerged.FemaleMG.AllControls.hg38.glm.MetaMGindexVars.tab

# === DESCRIBE AND PROCESS ADDITIONAL INPUT FILE ===
PROCESS /data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Analysis.GLM.SexStratified.hg38/FemalesCases.vs.AllControls.rerun/toMeta.Itals.FemaleMG.AllControls.hg38.glm.MetaMGindexVars.tab

OUTFILE META_USmerged.Itals.FemaleMG.AllControls.UNRELATED.hg38.Rsq03MAF00001.FreqCtrl.glm .tbl
ANALYZE HETEROGENEITY
```



```bash
%%bash
module load metal
metal metal_MG.US.Itals.hg38.v1.txt
metal metal_MG.US.Itals.hg38.v2.txt

```

    MetaAnalysis Helper - (c) 2007 - 2009 Goncalo Abecasis
    This version released on 2018-08-28
    
    # This program faciliates meta-analysis of genome-wide association studies.
    # Commonly used commands are listed below:
    #
    # Options for describing input files ...
    #   SEPARATOR        [WHITESPACE|COMMA|BOTH|TAB] (default = WHITESPACE)
    #   COLUMNCOUNTING   [STRICT|LENIENT]            (default = 'STRICT')
    #   MARKERLABEL      [LABEL]                     (default = 'MARKER')
    #   ALLELELABELS     [LABEL1 LABEL2]             (default = 'ALLELE1','ALLELE2')
    #   EFFECTLABEL      [LABEL|log(LABEL)]          (default = 'EFFECT')
    #   FLIP
    #
    # Options for filtering input files ...
    #   ADDFILTER        [LABEL CONDITION VALUE]     (example = ADDFILTER N > 10)
    #                    (available conditions are <, >, <=, >=, =, !=, IN)
    #   REMOVEFILTERS
    #
    # Options for sample size weighted meta-analysis ...
    #   WEIGHTLABEL      [LABEL]                     (default = 'N')
    #   PVALUELABEL      [LABEL]                     (default = 'PVALUE')
    #   DEFAULTWEIGHT    [NUMBER]                    (default = 1.0)
    #   MINWEIGHT        [NUMBER]                    (default = 1.0)
    #
    # Options for inverse variance weighted meta-analysis ...
    #   STDERRLABEL      [LABEL]                     (default = 'STDERR')
    #   SCHEME           [SAMPLESIZE|STDERR]         (default = SAMPLESIZE)
    #
    # Options to enable tracking of allele frequencies ...
    #   AVERAGEFREQ      [ON|OFF]                    (default = OFF)
    #   MINMAXFREQ       [ON|OFF]                    (default = OFF)
    #   FREQLABEL        [LABEL]                     (default = 'FREQ')
    #
    # Options to enable tracking of user defined variables ...
    #   CUSTOMVARIABLE   [VARNAME]
    #   LABEL            [VARNAME] AS [HEADER]
    #
    # Options to enable tracking of chromosomes and positions ...
    #   TRACKPOSITIONS   [ON|OFF]                    (default = OFF
    #   CHROMOSOMELABEL  [LABEL]                     (default = 'CHROMOSOME')
    #   POSITIONLABEL    [LABEL]                     (default = 'POSITION')
    #
    # Options to enable explicit strand information ...
    #   USESTRAND        [ON|OFF]                    (default = OFF)
    #   STRANDLABEL      [LABEL]                     (default = 'STRAND')
    #
    # Automatic genomic control correction of input statistics ...
    #   GENOMICCONTROL   [ON|OFF|VALUE|LIST snps.txt](default = OFF)
    #
    # Options to account for samples overlap ...
    #   OVERLAP          [ON|OFF]                    (default = OFF)
    #   ZCUTOFF          [NUMBER]                    (default = 1.0)
    #
    # Options for general analysis control ...
    #   PROCESSFILE      [FILENAME]
    #   OUTFILE          [PREFIX SUFFIX]             (default = 'METAANALYSIS','.TBL')
    #   MAXWARNINGS      [NUMBER]                    (default = 20)
    #   VERBOSE          [ON|OFF]                    (default = 'OFF')
    #   LOGPVALUE        [ON|OFF]                    (default = 'OFF')
    #   ANALYZE          [HETEROGENEITY]
    #   CLEAR
    
    # Options for general run control ...
    #   SOURCE           [SCRIPTFILE]
    #   RETURN
    #   QUIT
    
    # Processing commands in metal_MG.US.Itals.hg38.v1.txt ...
    ## Meta-analysis will be based on effect sizes and their standard errors ...
    ## Averaging of allele frequencies enabled
    ## Tracking of extreme allele frequencies enabled
    ## Set marker header to marker ...
    ## Set allele headers to EffectAllele and OtherAllele ...
    ## Set frequency header to Freq_EffectAllele.CASE ...
    ## Set effect header to BETA ...
    ## Set standard error header to SE ...
    ## Set p-value header to P ...
    ###########################################################################
    ## Processing file '/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Analysis.GLM.SexStratified.hg38/US.JointPostImputation/FemalesCases.vs.AllControls/toMeta.USmerged.FemaleMG.AllControls.hg38.glm.MetaMGindexVars.tab'
    ## Processed 8 markers ...
    
    ###########################################################################
    ## Processing file '/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Analysis.GLM.SexStratified.hg38/FemalesCases.vs.AllControls.rerun/toMeta.Itals.FemaleMG.AllControls.hg38.glm.MetaMGindexVars.tab'
    ## Processed 8 markers ...
    
    ## Set output file prefix and suffix to META_USmerged.Itals.FemaleMG.AllControls.UNRELATED.hg38.Rsq03MAF00001.FreqCases.glm and .tbl ...
    ###########################################################################
    ## Running second pass analysis to evaluate heterogeneity...
    ## Processing file '/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Analysis.GLM.SexStratified.hg38/FemalesCases.vs.AllControls.rerun/toMeta.Itals.FemaleMG.AllControls.hg38.glm.MetaMGindexVars.tab'
    ## Processing file '/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Analysis.GLM.SexStratified.hg38/US.JointPostImputation/FemalesCases.vs.AllControls/toMeta.USmerged.FemaleMG.AllControls.hg38.glm.MetaMGindexVars.tab'
    
    ###########################################################################
    ## Executing meta-analysis ...
    ## Complete results will be stored in file 'META_USmerged.Itals.FemaleMG.AllControls.UNRELATED.hg38.Rsq03MAF00001.FreqCases.glm1.tbl'
    ## Column descriptions will be stored in file 'META_USmerged.Itals.FemaleMG.AllControls.UNRELATED.hg38.Rsq03MAF00001.FreqCases.glm1.tbl.info'
    ## Completed meta-analysis for 8 markers!
    ## Smallest p-value is 7.146e-18 at marker '6:31358836:G:A'
    
    ## Clearing all stored statistics ...
    # Clearing user defined filters ...
    MetaAnalysis Helper - (c) 2007 - 2009 Goncalo Abecasis
    This version released on 2018-08-28
    
    # This program faciliates meta-analysis of genome-wide association studies.
    # Commonly used commands are listed below:
    #
    # Options for describing input files ...
    #   SEPARATOR        [WHITESPACE|COMMA|BOTH|TAB] (default = WHITESPACE)
    #   COLUMNCOUNTING   [STRICT|LENIENT]            (default = 'STRICT')
    #   MARKERLABEL      [LABEL]                     (default = 'MARKER')
    #   ALLELELABELS     [LABEL1 LABEL2]             (default = 'ALLELE1','ALLELE2')
    #   EFFECTLABEL      [LABEL|log(LABEL)]          (default = 'EFFECT')
    #   FLIP
    #
    # Options for filtering input files ...
    #   ADDFILTER        [LABEL CONDITION VALUE]     (example = ADDFILTER N > 10)
    #                    (available conditions are <, >, <=, >=, =, !=, IN)
    #   REMOVEFILTERS
    #
    # Options for sample size weighted meta-analysis ...
    #   WEIGHTLABEL      [LABEL]                     (default = 'N')
    #   PVALUELABEL      [LABEL]                     (default = 'PVALUE')
    #   DEFAULTWEIGHT    [NUMBER]                    (default = 1.0)
    #   MINWEIGHT        [NUMBER]                    (default = 1.0)
    #
    # Options for inverse variance weighted meta-analysis ...
    #   STDERRLABEL      [LABEL]                     (default = 'STDERR')
    #   SCHEME           [SAMPLESIZE|STDERR]         (default = SAMPLESIZE)
    #
    # Options to enable tracking of allele frequencies ...
    #   AVERAGEFREQ      [ON|OFF]                    (default = OFF)
    #   MINMAXFREQ       [ON|OFF]                    (default = OFF)
    #   FREQLABEL        [LABEL]                     (default = 'FREQ')
    #
    # Options to enable tracking of user defined variables ...
    #   CUSTOMVARIABLE   [VARNAME]
    #   LABEL            [VARNAME] AS [HEADER]
    #
    # Options to enable tracking of chromosomes and positions ...
    #   TRACKPOSITIONS   [ON|OFF]                    (default = OFF
    #   CHROMOSOMELABEL  [LABEL]                     (default = 'CHROMOSOME')
    #   POSITIONLABEL    [LABEL]                     (default = 'POSITION')
    #
    # Options to enable explicit strand information ...
    #   USESTRAND        [ON|OFF]                    (default = OFF)
    #   STRANDLABEL      [LABEL]                     (default = 'STRAND')
    #
    # Automatic genomic control correction of input statistics ...
    #   GENOMICCONTROL   [ON|OFF|VALUE|LIST snps.txt](default = OFF)
    #
    # Options to account for samples overlap ...
    #   OVERLAP          [ON|OFF]                    (default = OFF)
    #   ZCUTOFF          [NUMBER]                    (default = 1.0)
    #
    # Options for general analysis control ...
    #   PROCESSFILE      [FILENAME]
    #   OUTFILE          [PREFIX SUFFIX]             (default = 'METAANALYSIS','.TBL')
    #   MAXWARNINGS      [NUMBER]                    (default = 20)
    #   VERBOSE          [ON|OFF]                    (default = 'OFF')
    #   LOGPVALUE        [ON|OFF]                    (default = 'OFF')
    #   ANALYZE          [HETEROGENEITY]
    #   CLEAR
    
    # Options for general run control ...
    #   SOURCE           [SCRIPTFILE]
    #   RETURN
    #   QUIT
    
    # Processing commands in metal_MG.US.Itals.hg38.v2.txt ...
    ## Meta-analysis will be based on effect sizes and their standard errors ...
    ## Averaging of allele frequencies enabled
    ## Tracking of extreme allele frequencies enabled
    ## Set marker header to marker ...
    ## Set allele headers to EffectAllele and OtherAllele ...
    ## Set frequency header to Freq_EffectAllele.CTRL ...
    ## Set effect header to BETA ...
    ## Set standard error header to SE ...
    ## Set p-value header to P ...
    ###########################################################################
    ## Processing file '/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Analysis.GLM.SexStratified.hg38/US.JointPostImputation/FemalesCases.vs.AllControls/toMeta.USmerged.FemaleMG.AllControls.hg38.glm.MetaMGindexVars.tab'
    ## Processed 8 markers ...
    
    ###########################################################################
    ## Processing file '/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Analysis.GLM.SexStratified.hg38/FemalesCases.vs.AllControls.rerun/toMeta.Itals.FemaleMG.AllControls.hg38.glm.MetaMGindexVars.tab'
    ## Processed 8 markers ...
    
    ## Set output file prefix and suffix to META_USmerged.Itals.FemaleMG.AllControls.UNRELATED.hg38.Rsq03MAF00001.FreqCtrl.glm and .tbl ...
    ###########################################################################
    ## Running second pass analysis to evaluate heterogeneity...
    ## Processing file '/data/NDRS_LNG/MyastheniaGravis/updated.April2020/Itals/Analysis.GLM.SexStratified.hg38/FemalesCases.vs.AllControls.rerun/toMeta.Itals.FemaleMG.AllControls.hg38.glm.MetaMGindexVars.tab'
    ## Processing file '/data/NDRS_LNG/MyastheniaGravis/updated.April2020/US/Analysis.GLM.SexStratified.hg38/US.JointPostImputation/FemalesCases.vs.AllControls/toMeta.USmerged.FemaleMG.AllControls.hg38.glm.MetaMGindexVars.tab'
    
    ###########################################################################
    ## Executing meta-analysis ...
    ## Complete results will be stored in file 'META_USmerged.Itals.FemaleMG.AllControls.UNRELATED.hg38.Rsq03MAF00001.FreqCtrl.glm1.tbl'
    ## Column descriptions will be stored in file 'META_USmerged.Itals.FemaleMG.AllControls.UNRELATED.hg38.Rsq03MAF00001.FreqCtrl.glm1.tbl.info'
    ## Completed meta-analysis for 8 markers!
    ## Smallest p-value is 7.146e-18 at marker '6:31358836:G:A'
    
    ## Clearing all stored statistics ...
    # Clearing user defined filters ...


    [+] Loading metal  2018-08-28 



```python
import pandas as pd
import numpy as np

pd.read_csv("META_USmerged.Itals.FemaleMG.AllControls.UNRELATED.hg38.Rsq03MAF00001.FreqCases.glm1.tbl",sep="\t").sort_values(by=['MarkerName'])
```




<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>MarkerName</th>
      <th>Allele1</th>
      <th>Allele2</th>
      <th>Freq1</th>
      <th>FreqSE</th>
      <th>MinFreq</th>
      <th>MaxFreq</th>
      <th>Effect</th>
      <th>StdErr</th>
      <th>P-value</th>
      <th>Direction</th>
      <th>HetISq</th>
      <th>HetChiSq</th>
      <th>HetDf</th>
      <th>HetPVal</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>2</th>
      <td>10:7410781:A:G</td>
      <td>a</td>
      <td>g</td>
      <td>0.7263</td>
      <td>0.0231</td>
      <td>0.7026</td>
      <td>0.7488</td>
      <td>-0.2313</td>
      <td>0.0585</td>
      <td>7.729000e-05</td>
      <td>--</td>
      <td>0.0</td>
      <td>0.203</td>
      <td>1</td>
      <td>0.65200</td>
    </tr>
    <tr>
      <th>0</th>
      <td>11:95578258:T:C</td>
      <td>t</td>
      <td>c</td>
      <td>0.7866</td>
      <td>0.0010</td>
      <td>0.7857</td>
      <td>0.7877</td>
      <td>-0.2359</td>
      <td>0.0637</td>
      <td>2.114000e-04</td>
      <td>--</td>
      <td>0.0</td>
      <td>0.243</td>
      <td>1</td>
      <td>0.62230</td>
    </tr>
    <tr>
      <th>5</th>
      <td>18:62342581:T:C</td>
      <td>t</td>
      <td>c</td>
      <td>0.4905</td>
      <td>0.0128</td>
      <td>0.4786</td>
      <td>0.5043</td>
      <td>-0.1835</td>
      <td>0.0519</td>
      <td>4.076000e-04</td>
      <td>--</td>
      <td>67.4</td>
      <td>3.064</td>
      <td>1</td>
      <td>0.08007</td>
    </tr>
    <tr>
      <th>3</th>
      <td>1:113834946:G:A</td>
      <td>a</td>
      <td>g</td>
      <td>0.1040</td>
      <td>0.0289</td>
      <td>0.0593</td>
      <td>0.1226</td>
      <td>0.2893</td>
      <td>0.0903</td>
      <td>1.349000e-03</td>
      <td>++</td>
      <td>0.0</td>
      <td>0.009</td>
      <td>1</td>
      <td>0.92310</td>
    </tr>
    <tr>
      <th>1</th>
      <td>2:174764492:G:A</td>
      <td>a</td>
      <td>g</td>
      <td>0.0521</td>
      <td>0.0121</td>
      <td>0.0381</td>
      <td>0.0625</td>
      <td>0.2961</td>
      <td>0.1215</td>
      <td>1.483000e-02</td>
      <td>++</td>
      <td>0.0</td>
      <td>0.669</td>
      <td>1</td>
      <td>0.41340</td>
    </tr>
    <tr>
      <th>6</th>
      <td>6:31358836:G:A</td>
      <td>a</td>
      <td>g</td>
      <td>0.4154</td>
      <td>0.0609</td>
      <td>0.3448</td>
      <td>0.4679</td>
      <td>0.4687</td>
      <td>0.0544</td>
      <td>7.146000e-18</td>
      <td>++</td>
      <td>0.0</td>
      <td>0.377</td>
      <td>1</td>
      <td>0.53940</td>
    </tr>
    <tr>
      <th>4</th>
      <td>6:32620936:T:C</td>
      <td>t</td>
      <td>c</td>
      <td>0.9677</td>
      <td>0.0106</td>
      <td>0.9607</td>
      <td>0.9838</td>
      <td>0.6361</td>
      <td>0.1516</td>
      <td>2.728000e-05</td>
      <td>++</td>
      <td>75.4</td>
      <td>4.057</td>
      <td>1</td>
      <td>0.04399</td>
    </tr>
    <tr>
      <th>7</th>
      <td>8:108247378:T:C</td>
      <td>t</td>
      <td>c</td>
      <td>0.9915</td>
      <td>0.0029</td>
      <td>0.9892</td>
      <td>0.9952</td>
      <td>-0.5194</td>
      <td>0.3192</td>
      <td>1.037000e-01</td>
      <td>--</td>
      <td>0.0</td>
      <td>0.017</td>
      <td>1</td>
      <td>0.89540</td>
    </tr>
  </tbody>
</table>
</div>




```python
import pandas as pd
import numpy as np

pd.read_csv("META_USmerged.Itals.FemaleMG.AllControls.UNRELATED.hg38.Rsq03MAF00001.FreqCtrl.glm1.tbl",sep="\t").sort_values(by=['MarkerName'])
```




<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>MarkerName</th>
      <th>Allele1</th>
      <th>Allele2</th>
      <th>Freq1</th>
      <th>FreqSE</th>
      <th>MinFreq</th>
      <th>MaxFreq</th>
      <th>Effect</th>
      <th>StdErr</th>
      <th>P-value</th>
      <th>Direction</th>
      <th>HetISq</th>
      <th>HetChiSq</th>
      <th>HetDf</th>
      <th>HetPVal</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>2</th>
      <td>10:7410781:A:G</td>
      <td>a</td>
      <td>g</td>
      <td>0.7730</td>
      <td>0.0161</td>
      <td>0.7565</td>
      <td>0.7887</td>
      <td>-0.2313</td>
      <td>0.0585</td>
      <td>7.729000e-05</td>
      <td>--</td>
      <td>0.0</td>
      <td>0.203</td>
      <td>1</td>
      <td>0.65200</td>
    </tr>
    <tr>
      <th>0</th>
      <td>11:95578258:T:C</td>
      <td>t</td>
      <td>c</td>
      <td>0.8232</td>
      <td>0.0039</td>
      <td>0.8189</td>
      <td>0.8267</td>
      <td>-0.2359</td>
      <td>0.0637</td>
      <td>2.114000e-04</td>
      <td>--</td>
      <td>0.0</td>
      <td>0.243</td>
      <td>1</td>
      <td>0.62230</td>
    </tr>
    <tr>
      <th>5</th>
      <td>18:62342581:T:C</td>
      <td>t</td>
      <td>c</td>
      <td>0.5310</td>
      <td>0.0102</td>
      <td>0.5201</td>
      <td>0.5405</td>
      <td>-0.1835</td>
      <td>0.0519</td>
      <td>4.076000e-04</td>
      <td>--</td>
      <td>67.4</td>
      <td>3.064</td>
      <td>1</td>
      <td>0.08007</td>
    </tr>
    <tr>
      <th>3</th>
      <td>1:113834946:G:A</td>
      <td>a</td>
      <td>g</td>
      <td>0.0804</td>
      <td>0.0253</td>
      <td>0.0412</td>
      <td>0.0968</td>
      <td>0.2893</td>
      <td>0.0903</td>
      <td>1.349000e-03</td>
      <td>++</td>
      <td>0.0</td>
      <td>0.009</td>
      <td>1</td>
      <td>0.92310</td>
    </tr>
    <tr>
      <th>1</th>
      <td>2:174764492:G:A</td>
      <td>a</td>
      <td>g</td>
      <td>0.0418</td>
      <td>0.0054</td>
      <td>0.0355</td>
      <td>0.0464</td>
      <td>0.2961</td>
      <td>0.1215</td>
      <td>1.483000e-02</td>
      <td>++</td>
      <td>0.0</td>
      <td>0.669</td>
      <td>1</td>
      <td>0.41340</td>
    </tr>
    <tr>
      <th>6</th>
      <td>6:31358836:G:A</td>
      <td>a</td>
      <td>g</td>
      <td>0.2992</td>
      <td>0.0658</td>
      <td>0.2230</td>
      <td>0.3559</td>
      <td>0.4687</td>
      <td>0.0544</td>
      <td>7.146000e-18</td>
      <td>++</td>
      <td>0.0</td>
      <td>0.377</td>
      <td>1</td>
      <td>0.53940</td>
    </tr>
    <tr>
      <th>4</th>
      <td>6:32620936:T:C</td>
      <td>t</td>
      <td>c</td>
      <td>0.9437</td>
      <td>0.0097</td>
      <td>0.9373</td>
      <td>0.9584</td>
      <td>0.6361</td>
      <td>0.1516</td>
      <td>2.728000e-05</td>
      <td>++</td>
      <td>75.4</td>
      <td>4.057</td>
      <td>1</td>
      <td>0.04399</td>
    </tr>
    <tr>
      <th>7</th>
      <td>8:108247378:T:C</td>
      <td>t</td>
      <td>c</td>
      <td>0.9949</td>
      <td>0.0011</td>
      <td>0.9940</td>
      <td>0.9963</td>
      <td>-0.5194</td>
      <td>0.3192</td>
      <td>1.037000e-01</td>
      <td>--</td>
      <td>0.0</td>
      <td>0.017</td>
      <td>1</td>
      <td>0.89540</td>
    </tr>
  </tbody>
</table>
</div>



## Format metal results to match other summ stats


```python
!cat ReformatMetaResults.addMAF.v2.R
```

    #!/usr/bin/env Rscript
    
    args <- commandArgs(trailingOnly=TRUE)
    if (length(args) != 4) {
      stop("USAGE: Rscript ReformatMetaResults.addMAF.R args[1] args[2] args[3] args[4]
      where args[1] = metaResults
            args[2] = toMetaInput.US
            args[3] = toMetaInput.Itals
            args[4] = outputFileName")
    }
    
    # Load libraries
    require(data.table)
    require(tidyverse)
    
    
    # Set variables
    metaResults <- args[1]
    toMetaInput.US <- args[2]
    toMetaInput.Itals <- args[3]
    outputFileName <- args[4]
    
    # Read in files and reformat meta
    meta <- fread(metaResults, header=T) %>%
            mutate(snp = MarkerName) %>%
            separate(snp,c("CHROM","POS","OA","EA"),sep=":") %>%
            rename(P = `P-value`)
    
    ## format meta results such that the effect allele matches original allele designation 
    ## given in MarkerName as submitted in the original files used for meta analysis 
    ## i.e. chr:pos:OtherAllele:EffectAllele
    meta$OA <- tolower(meta$OA)
    meta$EA <- tolower(meta$EA)
    meta$EffectAllele <- ifelse(meta$EA == meta$Allele1, meta$Allele1, meta$Allele2)
    meta$OtherAllele <- ifelse(meta$EA == meta$Allele1, meta$Allele2, meta$Allele1)
    meta$maf_EA.cases <- ifelse(meta$EA == meta$Allele1, meta$Freq1, 1 - meta$Freq1)
    meta$beta <- ifelse(meta$EffectAllele == meta$Allele1, meta$Effect, meta$Effect*-1)
    
    data <- meta %>%
            mutate(CHROM = as.numeric(gsub("chr","",CHROM), POS = as.numeric(POS))) %>%
            filter(POS != is.na(POS)) %>%
            select(CHROM,POS,MarkerName,EffectAllele,OtherAllele,maf_EA.cases,beta,StdErr,P,Direction,HetISq) %>%
            arrange(CHROM,POS)
    
    dim(meta)
    dim(data)
    
    # Get frequency information
    us <- fread(args[2],header=T) %>%
          select(marker,Freq_EffectAllele,Freq_EffectAllele.CASE,Freq_EffectAllele.CTRL,
                 ALLELE_CT,CASE_ALLELE_CT,CTRL_ALLELE_CT,
                 EffectAllele_CT,EffectAllele_CT.CASE,EffectAllele_CT.CTRL)
    colnames(us)[2:10] <- paste(colnames(us)[2:10],"_US",sep="")
    
    itals <- fread(args[3],header=T) %>%
             select(marker,Freq_EffectAllele,Freq_EffectAllele.CASE,Freq_EffectAllele.CTRL,
                    ALLELE_CT,CASE_ALLELE_CT,CTRL_ALLELE_CT,
                    EffectAllele_CT,EffectAllele_CT.CASE,EffectAllele_CT.CTRL)
    colnames(itals)[2:10] <- paste(colnames(itals)[2:10],"_Italy",sep="")
    
    both <- merge(us,itals,by="marker")
    
    dim(us)
    dim(itals)
    dim(both)
    
    both$Freq_EffectAllele <- (both$Freq_EffectAllele_US + both$Freq_EffectAllele_Italy)/2
    both$Freq_EffectAllele.CASE <- (both$Freq_EffectAllele.CASE_US + both$Freq_EffectAllele.CASE_Italy)/2
    both$Freq_EffectAllele.CTRL <- (both$Freq_EffectAllele.CTRL_US + both$Freq_EffectAllele.CTRL_Italy)/2
    
    both$ALLELE_CT <- both$ALLELE_CT_US + both$ALLELE_CT_Italy
    both$ALLELE_CT.CASE <- both$CASE_ALLELE_CT_US + both$CASE_ALLELE_CT_Italy
    both$ALLELE_CT.CTRL <- both$CTRL_ALLELE_CT_US + both$CTRL_ALLELE_CT_Italy
    
    both$EffectAllele_CT <- both$EffectAllele_CT_US + both$EffectAllele_CT_Italy
    both$EffectAllele_CT.CASE <- both$EffectAllele_CT.CASE_US + both$EffectAllele_CT.CASE_Italy
    both$EffectAllele_CT.CTRL <- both$EffectAllele_CT.CTRL_US + both$EffectAllele_CT.CTRL_Italy
    
    both$maf_EA <- both$EffectAllele_CT/both$ALLELE_CT
    both$maf_EA.CASE <- both$EffectAllele_CT.CASE/both$ALLELE_CT.CASE
    both$maf_EA.CTRL <- both$EffectAllele_CT.CTRL/both$ALLELE_CT.CTRL
    write.table(both,"MAF.USmerged.Itals.txt",sep="\t",quote=F,row.names=F,col.names=T)
                
    freq <- both %>%
            select(marker,Freq_EffectAllele,Freq_EffectAllele.CASE,Freq_EffectAllele.CTRL,
                  ALLELE_CT,ALLELE_CT.CASE,ALLELE_CT.CTRL,
                  EffectAllele_CT,EffectAllele_CT.CASE,EffectAllele_CT.CTRL,
                  maf_EA,maf_EA.CASE,maf_EA.CTRL)
    
    
    # merge formatted meta results with frequency file
    tmp <- merge(data,freq,by.x="MarkerName",by.y="marker")
    data2 <- tmp %>% 
             select(CHROM,POS,MarkerName,EffectAllele,OtherAllele,
                    maf_EA.CASE,maf_EA.CTRL,beta,StdErr,P,Direction,HetISq)
    
    write.table(data2,paste(args[4],".TIDY.txt",sep=""),quote=F,row.names=F,col.names=T,sep="\t")



```bash
%%bash
module load R/3.5.2

DIR="/data/NDRS_LNG/MyastheniaGravis/updated.April2020"

Rscript ReformatMetaResults.addMAF.v2.R \
META_USmerged.Itals.FemaleMG.AllControls.UNRELATED.hg38.Rsq03MAF00001.FreqCases.glm1.tbl \
$DIR/US/Analysis.GLM.SexStratified.hg38/US.JointPostImputation/FemalesCases.vs.AllControls/toMeta.USmerged.FemaleMG.AllControls.hg38.glm.MetaMGindexVars.tab \
$DIR/Itals/Analysis.GLM.SexStratified.hg38/FemalesCases.vs.AllControls.rerun/toMeta.Itals.FemaleMG.AllControls.hg38.glm.MetaMGindexVars.tab \
META_USmerged.Itals.FemaleMG.AllControls
```

    [1]  8 23
    [1]  8 11
    [1]  8 10
    [1]  8 10
    [1]  8 19


    [+] Loading gcc  7.3.0  ... 
    [+] Loading GSL 2.4 for GCC 7.2.0 ... 
    [-] Unloading gcc  7.3.0  ... 
    [+] Loading gcc  7.3.0  ... 
    [+] Loading openmpi 3.0.2  for GCC 7.3.0 
    [+] Loading ImageMagick  7.0.8  on cn3305 
    [+] Loading HDF5  1.10.4 
    [+] Loading pandoc  2.11.4  on cn3305 
    [+] Loading R 3.5.2 
    Loading required package: data.table
    Loading required package: tidyverse
    -- Attaching packages --------------------------------------- tidyverse 1.2.1 --
    v ggplot2 3.3.2     v purrr   0.3.4
    v tibble  3.0.3     v dplyr   0.8.5
    v tidyr   0.8.3     v stringr 1.4.0
    v readr   1.3.1     v forcats 0.5.0
    -- Conflicts ------------------------------------------ tidyverse_conflicts() --
    x dplyr::between()   masks data.table::between()
    x dplyr::filter()    masks stats::filter()
    x dplyr::first()     masks data.table::first()
    x dplyr::lag()       masks stats::lag()
    x dplyr::last()      masks data.table::last()
    x purrr::transpose() masks data.table::transpose()



```python
import pandas as pd
import numpy as np

pd.read_csv("META_USmerged.Itals.FemaleMG.AllControls.TIDY.txt",sep="\t").sort_values(by=['CHROM','POS'])
```




<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>CHROM</th>
      <th>POS</th>
      <th>MarkerName</th>
      <th>EffectAllele</th>
      <th>OtherAllele</th>
      <th>maf_EA.CASE</th>
      <th>maf_EA.CTRL</th>
      <th>beta</th>
      <th>StdErr</th>
      <th>P</th>
      <th>Direction</th>
      <th>HetISq</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>3</th>
      <td>1</td>
      <td>113834946</td>
      <td>1:113834946:G:A</td>
      <td>a</td>
      <td>g</td>
      <td>0.089367</td>
      <td>0.093099</td>
      <td>0.2893</td>
      <td>0.0903</td>
      <td>1.349000e-03</td>
      <td>++</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>4</th>
      <td>2</td>
      <td>174764492</td>
      <td>2:174764492:G:A</td>
      <td>a</td>
      <td>g</td>
      <td>0.050905</td>
      <td>0.036239</td>
      <td>0.2961</td>
      <td>0.1215</td>
      <td>1.483000e-02</td>
      <td>++</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>5</th>
      <td>6</td>
      <td>31358836</td>
      <td>6:31358836:G:A</td>
      <td>a</td>
      <td>g</td>
      <td>0.403281</td>
      <td>0.347127</td>
      <td>0.4687</td>
      <td>0.0544</td>
      <td>7.146000e-18</td>
      <td>++</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>6</th>
      <td>6</td>
      <td>32620936</td>
      <td>6:32620936:T:C</td>
      <td>c</td>
      <td>t</td>
      <td>0.027149</td>
      <td>0.061287</td>
      <td>-0.6361</td>
      <td>0.1516</td>
      <td>2.728000e-05</td>
      <td>++</td>
      <td>75.4</td>
    </tr>
    <tr>
      <th>7</th>
      <td>8</td>
      <td>108247378</td>
      <td>8:108247378:T:C</td>
      <td>c</td>
      <td>t</td>
      <td>0.007919</td>
      <td>0.003849</td>
      <td>0.5194</td>
      <td>0.3192</td>
      <td>1.037000e-01</td>
      <td>--</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>0</th>
      <td>10</td>
      <td>7410781</td>
      <td>10:7410781:A:G</td>
      <td>g</td>
      <td>a</td>
      <td>0.275452</td>
      <td>0.213473</td>
      <td>0.2313</td>
      <td>0.0585</td>
      <td>7.729000e-05</td>
      <td>--</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>1</th>
      <td>11</td>
      <td>95578258</td>
      <td>11:95578258:T:C</td>
      <td>c</td>
      <td>t</td>
      <td>0.213235</td>
      <td>0.173797</td>
      <td>0.2359</td>
      <td>0.0637</td>
      <td>2.114000e-04</td>
      <td>--</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>2</th>
      <td>18</td>
      <td>62342581</td>
      <td>18:62342581:T:C</td>
      <td>c</td>
      <td>t</td>
      <td>0.507919</td>
      <td>0.460888</td>
      <td>0.1835</td>
      <td>0.0519</td>
      <td>4.076000e-04</td>
      <td>--</td>
      <td>67.4</td>
    </tr>
  </tbody>
</table>
</div>


